---
title: "02c_rnaseq_ConsistentConflict"
author: "Rayna M Harris"
date: "8/24/17"
output: md_document
---

The figures made from this script were compiled in Adobe.

<img src="../figures/02c_RNAseqConflict/02_RNAseq_Conflict-01.png" width="1370" />


```{r setup, message=F}
library(ggplot2) ## for awesome plots!
library(cowplot) ## for some easy to use themes
library(dplyr) ## for filtering and selecting rows
library(car) ## stats
library(VennDiagram) ## venn diagrams
library(pheatmap) ## awesome heatmaps
library(viridis) # for awesome color pallette
library(reshape2) ## for melting dataframe
library(DESeq2) ## for gene expression analysis
library(edgeR)  ## for basic read counts status
library(magrittr) ## to use the weird pipe
library(genefilter)  ## for PCA fuction
library(ggrepel) ## for labeling volcano plot
library(colorblindr) # simulates color-blind viewing 

## load functions 
source("figureoptions.R")
source("functions_RNAseq.R")

## set output file for figures 
knitr::opts_chunk$set(fig.path = '../figures/02b_RNAseqAll/')

```

## Design
The major comparision here is 
Hippocampal subfield: "DG","CA3", "CA1"
Behavioral Groups: "conflict", "consistent"


```{r data, message=F, warning=F, echo=F}
colData <- read.csv("../data/02a_colData.csv", header = T)
countData <- read.csv("../data/02a_countData.csv", header = T, check.names = F, row.names = 1)
colData %>% select(APA2,Punch)  %>%  summary()
totalCounts=colSums(countData)

dds <- DESeqDataSetFromMatrix(countData = countData,
                              colData = colData,
                              design = ~ Punch + APA2 + Punch*APA2)

dds$Punch <- factor(dds$Punch, levels=c("DG","CA3", "CA1")) ## specify the factor levels
dds$APA2 <- factor(dds$APA2, levels=c("yoked_consistent", "consistent", "yoked_conflict" , "conflict")) ## specify the factor levels

dds # view the DESeq object - note numnber of genes
dds <- dds[ rowSums(counts(dds)) > 1, ]  # Pre-filtering genes with 0 counts
dds # view number of genes afternormalization and the number of samples
dds <- DESeq(dds) # Differential expression analysis
rld <- rlog(dds, blind=FALSE) ## log transformed data

contrast1 <- resvals(contrastvector = c("Punch", "CA1", "DG"), mypval = 0.05) # 1159
contrast2 <- resvals(contrastvector = c("Punch", "CA1", "CA3"), mypval = 0.05) # 771
contrast3 <- resvals(contrastvector = c("Punch", "CA3", "DG"), mypval = 0.05) # 2122

DEGes <- assay(rld)
DEGes <- cbind(DEGes, contrast1, contrast2, contrast3)
DEGes <- as.data.frame(DEGes) # convert matrix to dataframe
DEGes$rownames <- rownames(DEGes)  # add the rownames to the dataframe
DEGes$padjmin <- with(DEGes, pmin(padjPunchCA1DG, padjPunchCA1CA3, padjPunchCA3DG)) 
DEGes <- DEGes %>% filter(padjmin < 0.01)
rownames(DEGes) <- DEGes$rownames
drop.cols <-colnames(DEGes[,grep("padj|pval|rownames", colnames(DEGes))])
DEGes <- DEGes %>% dplyr::select(-one_of(drop.cols))
DEGes <- as.matrix(DEGes)
DEGes <- DEGes - rowMeans(DEGes)


df <- as.data.frame(colData(dds)[,c("APA2", "Punch")]) ## matrix to df
rownames(df) <- names(countData)
ann_colors <- ann_colors4 # see color options 
DEGes <- as.matrix(DEGes) 
paletteLength <- 30
myBreaks <- c(seq(min(DEGes), 0, length.out=ceiling(paletteLength/2) + 1), 
              seq(max(DEGes)/paletteLength, max(DEGes), length.out=floor(paletteLength/2)))

pheatmap(DEGes, show_colnames=T, show_rownames = F,
         annotation_col=df, annotation_colors = ann_colors,
         treeheight_row = 0, treeheight_col = 25,
         annotation_row = NA, 
         annotation_legend = FALSE,
         annotation_names_row = FALSE, annotation_names_col = FALSE,
         fontsize = 8, 
         border_color = "grey60" ,
         color = viridis(30),
         cellwidth = 6, 
         clustering_method="average",
         breaks=myBreaks,
         clustering_distance_cols="correlation" 
         )

pheatmap(DEGes, show_colnames=F, show_rownames = F,
         annotation_col=df, annotation_colors = ann_colors, 
         annotation_row = NA, 
         annotation_legend = FALSE,
         annotation_names_row = FALSE, 
         annotation_names_col = FALSE,
         treeheight_row = 0, treeheight_col = 25,
         fontsize = 8, 
         border_color = "grey60" ,
         color = viridis(30),
         height = 4.75, 
         width = 3.5,
         clustering_method="average",
         breaks=myBreaks,
         clustering_distance_cols="correlation", 
         filename = "../figures/02b_RNAseqALL/pheatmap1.pdf"
         )

```

## Principle component analysis

```{r pca, echo=F, message=F}
# create the dataframe using my function pcadataframe
pcadata <- pcadataframe(rld, intgroup=c("Punch","APA2"), returnData=TRUE)
percentVar <- round(100 * attr(pcadata, "percentVar"))
percentVar

pcadata$PunchAPA <- as.factor(paste(pcadata$Punch, pcadata$APA2, sep="_"))

summary(aov(PC1 ~ Punch, data=pcadata)) 
TukeyHSD((aov(PC1 ~ Punch, data=pcadata)), which = "Punch") 

summary(aov(PC2 ~ Punch, data=pcadata)) 
TukeyHSD((aov(PC2 ~ Punch, data=pcadata)), which = "Punch") 

summary(aov(PC3 ~ APA2, data=pcadata)) 
summary(aov(PC4 ~ APA2, data=pcadata)) 
summary(aov(PC5 ~ APA2, data=pcadata)) 
summary(aov(PC6 ~ APA2, data=pcadata)) 

pcadata$Punch <- factor(pcadata$Punch, levels=c("DG","CA3", "CA1"))
pcadata$APA2 <- factor(pcadata$APA2, levels=c("yoked_consistent","consistent",  "yoked_conflict","conflict"))

pcadata$wrap <- "Principal Component Analyses of Hippocampal Samples"

plotPCs <- function(df, xcol, ycol, aescolor, colorname, colorvalues){
  ggplot(df, aes(df[xcol], df[ycol], color=aescolor)) +
    geom_point(size=3, aes(shape=APA2), alpha = 0.8) +
    xlab(paste0("PC", xcol, ": ", percentVar[xcol],"% variance")) +
    ylab(paste0("PC", ycol, ": ", percentVar[ycol],"% variance")) +
    #stat_ellipse(level = 0.95, (aes(color=Punch)),size=0.25) + 
    scale_colour_manual(name=colorname, values=c(colorvalues))+ 
    theme_cowplot(font_size = 8, line_size = 0.25)  +
    theme(legend.position="none") +
    scale_shape_manual(values=c(16, 1, 15,2), aes(color=APA2)) +
    facet_wrap(~wrap)
}

PCA12 <- plotPCs(pcadata, 1, 2, aescolor = pcadata$Punch, colorname = " ", colorvalues = colorvalPunch)
PCA12

pdf(file="../figures/02b_RNAseqAll/PCA12.pdf", width=3.5, height=2.5)
plot(PCA12)
dev.off()

plotPCs <- function(df, xcol, ycol, aescolor, colorname, colorvalues){
  ggplot(df, aes(df[xcol], df[ycol], color=aescolor, shape=Punch)) +
    geom_point(size=3, aes(shape=Punch), alpha = 0.8) +
    xlab(paste0("PC", xcol, ": ", percentVar[xcol],"% variance")) +
    ylab(paste0("PC", ycol, ": ", percentVar[ycol],"% variance")) +
    #stat_ellipse(level = 0.95, (aes(color=aescolor)),size=0.25) + 
    scale_colour_manual(name=colorname, values=c(colorvalues))+ 
    theme_cowplot(font_size = 8, line_size = 0.25)  +
    theme(legend.position="none") +
    scale_shape_manual(values=c(15, 16,17)) 
}

PCA62 <- plotPCs(pcadata, 6, 2, aescolor = pcadata$APA2, colorname = "APA2",  colorvalues = colorvalAPA00)
PCA62

pdf(file="../figures/02b_RNAseqAll/PCA62.pdf", width=3.5, height=2.5)
plot(PCA62)
dev.off()

```

# Number of differentially expressed genes per two-way contrast

```{r Twowaycontrasts3}
contrast1 <- resvals(contrastvector = c("Punch", "CA1", "DG"), mypval = 0.05) # 2685
contrast2 <- resvals(contrastvector = c("Punch", "CA1", "CA3"), mypval = 0.05) # 1411
contrast3 <- resvals(contrastvector = c("Punch", "CA3", "DG"), mypval = 0.05) # 3902
```

## venn diagrams

```{r venndiagrams, echo=F}
rldpadjs <- assay(rld)
rldpadjs <- cbind(rldpadjs, contrast1, contrast2, contrast3)
rldpadjs <- as.data.frame(rldpadjs)
rldpadjs <- rldpadjs[ , grepl( "padj" , names( rldpadjs ) ) ]

venn1 <- row.names(rldpadjs[rldpadjs[1] <0.05 & !is.na(rldpadjs[1]),]) # CA1 DG
venn2 <- row.names(rldpadjs[rldpadjs[2] <0.05 & !is.na(rldpadjs[2]),]) # CA1 CA3
venn3 <- row.names(rldpadjs[rldpadjs[3] <0.05 & !is.na(rldpadjs[3]),]) # CA3 DG
```


## Volcanos plots and and gene lists


```{r volcanos, include=F}
# gene lists
res <- results(dds, contrast =c("Punch", "CA1", "DG"), independentFiltering = T, alpha = 0.05)
resOrdered <- res[order(res$padj),]
head(resOrdered, 10)
data <- data.frame(gene = row.names(res), pvalue = -log10(res$padj), lfc = res$log2FoldChange)
data <- na.omit(data)
data <- data %>%
  mutate(color = ifelse(data$lfc > 0 & data$pvalue > 1.3, 
                        yes = "CA1", 
                        no = ifelse(data$lfc < 0 & data$pvalue > 1.3, 
                                    yes = "DG", 
                                    no = "none")))
top_labelled <- top_n(data, n = 5, wt = lfc)

# Color corresponds to fold change directionality
colored <- ggplot(data, aes(x = lfc, y = pvalue)) + 
  geom_point(aes(color = factor(color)), size = 1, alpha = 0.5, na.rm = T) + # add gene points
  scale_color_manual(values = c("CA1" = "#7570b3",
                                "DG" = "#d95f02", 
                                "none" = "#d9d9d9")) + 
  theme_cowplot(font_size = 8, line_size = 0.25) +
  geom_hline(yintercept = 1.3,  size = 0.25, linetype = 2) + 
  scale_y_continuous(name="-log10(pvalue") +
  scale_x_continuous(name="log2(CA1/DG)") +
  theme(panel.grid.minor=element_blank(),
        legend.position = "none", # remove legend 
        panel.grid.major=element_blank())
colored

pdf(file="../figures/02b_RNAseq/AllCA1DG.pdf", width=1.5, height=2)
plot(colored)
dev.off()

res <- results(dds, contrast =c("Punch", "CA1", "CA3"), independentFiltering = T, alpha = 0.05)
resOrdered <- res[order(res$padj),]
head(resOrdered, 10)
data <- data.frame(gene = row.names(res), pvalue = -log10(res$padj), lfc = res$log2FoldChange)
data <- na.omit(data)
head(data)
data <- data %>%
  mutate(color = ifelse(data$lfc > 0 & data$pvalue > 1.3, 
                        yes = "CA1", 
                        no = ifelse(data$lfc < 0 & data$pvalue > 1.3, 
                                    yes = "CA3", 
                                    no = "none")))
top_labelled <- top_n(data, n = 5, wt = lfc)
colored <- ggplot(data, aes(x = lfc, y = pvalue)) + 
  geom_point(aes(color = factor(color)), size = 1, alpha = 0.5, na.rm = T) + # add gene points
  theme_bw(base_size = 8) + # clean up theme
  theme(legend.position = "none") + # remove legend 
  scale_color_manual(values = c("CA1" = "#7570b3",
                                "CA3" = "#1b9e77", 
                                "none" = "#d9d9d9")) + theme(panel.grid.minor=element_blank(),
           panel.grid.major=element_blank()) + 
  scale_x_continuous(name="log2 (CA1/CA3)",
                     limits=c(-10, 10)) +
  scale_y_log10(name="-log10 (adjusted p-value")  +
  geom_hline(yintercept = 1.3,  size = 0.25, linetype = 2 ) 
colored
pdf(file="../figures/02b_RNAseqAll/AllCA1CA3.pdf", width=1.5, height=1.75)
plot(colored)
dev.off()



res <- results(dds, contrast =c("Punch", "CA3", "DG"), independentFiltering = T, alpha = 0.05)
resOrdered <- res[order(res$padj),]
head(resOrdered, 10)
data <- data.frame(gene = row.names(res), pvalue = -log10(res$padj), lfc = res$log2FoldChange)
data <- na.omit(data)
head(data)
data <- data %>%
  mutate(color = ifelse(data$lfc > 0 & data$pvalue > 1.3, 
                        yes = "CA3", 
                        no = ifelse(data$lfc < 0 & data$pvalue > 1.3, 
                                    yes = "DG", 
                                    no = "none")))
top_labelled <- top_n(data, n = 5, wt = lfc)
colored <- ggplot(data, aes(x = lfc, y = pvalue)) + 
  geom_point(aes(color = factor(color)), size = 1, alpha = 0.5, na.rm = T) + # add gene points
  theme_bw(base_size = 8) + # clean up theme
  theme(legend.position = "none") + # remove legend 
  scale_color_manual(values = c("CA3" = "#1b9e77",
                                "DG" = "#d95f02", 
                                "none" = "#d9d9d9")) + theme(panel.grid.minor=element_blank(),
           panel.grid.major=element_blank()) + 
  scale_x_continuous(name="log2 (CA3/DG)",
                     limits=c(-10, 10)) +
  scale_y_log10(name="-log10 (adjusted p-value")  +
  geom_hline(yintercept = 1.3,  size = 0.25, linetype = 2 ) 

colored
#cvd_grid(colored)
pdf(file="../figures/02b_RNAseqAll/AllDGCA3.pdf", width=1.5, height=1.75)
plot(colored)
dev.off()



res <- results(dds, contrast =c("APA2",  "conflict", "consistent"), independentFiltering = T, alpha = 0.05)
resOrdered <- res[order(res$padj),]
head(resOrdered, 10)

data <- data.frame(gene = row.names(res),
                   pvalue = -log10(res$padj), 
                   lfc = res$log2FoldChange)
data <- na.omit(data)
data <- data %>%
  mutate(color = ifelse(data$lfc > 0 & data$pvalue > 1.3, 
                        yes = "conflict", 
                        no = ifelse(data$lfc < 0 & data$pvalue > 1.3, 
                                    yes = "consistent", 
                                    no = "none")))
top_labelled <- top_n(data, n = 5, wt = lfc)
# Color corresponds to fold change directionality
colored <- ggplot(data, aes(x = lfc, y = pvalue)) + 
  geom_point(aes(color = factor(color)), size = 1, alpha = 0.5, na.rm = T) + # add gene points
  theme_bw(base_size = 8) + # clean up theme
  theme(legend.position = "none") + # remove legend 
  scale_color_manual(values = volcano2) + 
  theme(panel.grid.minor=element_blank(),
           panel.grid.major=element_blank()) + 
  scale_x_continuous(name="log2 (conflict/consistent)",
                     limits=c(-10, 10)) +
  scale_y_log10(name="-log10 (adjusted p-value")  +
  geom_hline(yintercept = 1.3,  size = 0.25, linetype = 2 ) 
colored


#cvd_grid(colored) # to view plot for color blind 
pdf(file="../figures/02b_RNAseqAll/Allconsistentconflict.pdf", width=1.5, height=1.75)
plot(colored)
dev.off()

```

