---
title: "02b_rnaseq_4by3results"
author: "Rayna M Harris"
date: "8/22/17"
output: md_document
---

The figures made from this script were compiled in Adobe.

<img src="../figures/02_RNAseq/02_RNAseq_4by3results.png" width="1370" />


```{r setup, message=F}
library(ggplot2) ## for awesome plots!
library(cowplot) ## for some easy to use themes
library(dplyr) ## for filtering and selecting rows
library(car) ## stats
library(VennDiagram) ## venn diagrams
library(pheatmap) ## awesome heatmaps
library(viridis) # for awesome color pallette
library(reshape2) ## for melting dataframe
library(DESeq2) ## for gene expression analysis
library(edgeR)  ## for basic read counts status
library(magrittr) ## to use the weird pipe
library(genefilter)  ## for PCA fuction
library(ggrepel) ## for labeling volcano plot
library(colorblindr) # simulates color-blind viewing 

## load functions 
source("figureoptions.R")
source("functions_RNAseq.R")

## set output file for figures 
knitr::opts_chunk$set(fig.path = '../figures/02_RNAseq_4by3/')
```


## Design

The major comparision here is 
Hippocampal subfield: "DG","CA3", "CA1"
Behavioral Groups: "yoked_consistent", "yoked_conflict", "consistent", "conflict"


```{r data, message=F, warning=F}
colData <- read.csv("../data/02a_colData.csv", header = T)
countData <- read.csv("../data/02a_countData.csv", header = T, check.names = F, row.names = 1)
```


```{r DESeq2}
## create DESeq object using the factors Punch and APA
dds <- DESeqDataSetFromMatrix(countData = countData,
                              colData = colData,
                              design = ~ Punch + APA2 + Punch*APA2)

dds$Punch <- factor(dds$Punch, levels=c("DG","CA3", "CA1")) ## specify the factor levels
dds$APA2 <- factor(dds$APA2, levels=c("yoked_consistent", "yoked_conflict", "consistent", "conflict")) ## specify the factor levels

dds # view the DESeq object - note numnber of genes
dds <- dds[ rowSums(counts(dds)) > 1, ]  # Pre-filtering genes with 0 counts
dds # view number of genes afternormalization and the number of samples
dds <- DESeq(dds) # Differential expression analysis
rld <- rlog(dds, blind=FALSE) ## log transformed data
```

# Number of differentially expressed genes per two-way contrast

```{r Twowaycontrasts3}
#calculate significance of all two way comparisions
# see source "functions_RNAseq.R" 

contrast1 <- resvals(contrastvector = c("Punch", "CA1", "DG"), mypval = 0.05) 
contrast2 <- resvals(contrastvector = c("Punch", "CA1", "CA3"), mypval = 0.05) 
contrast3 <- resvals(contrastvector = c("Punch", "CA3", "DG"), mypval = 0.05) 

contrast4 <- resvals(contrastvector = c("APA2", "conflict", "yoked_conflict"), mypval = 0.05)  # conflict yoked
contrast5 <- resvals(contrastvector = c("APA2", "consistent", "yoked_consistent"), mypval = 0.05) #consistent yoked
contrast6 <- resvals(contrastvector = c("APA2", "yoked_conflict", "yoked_consistent"), mypval = 0.05)  # yoked
contrast7 <- resvals(contrastvector = c("APA2", "consistent", "conflict"), mypval = 0.05) # trained

```






## Principle component analysis

```{r pca}

# create the dataframe using my function pcadataframe
pcadata <- pcadataframe(rld, intgroup=c("Punch","APA2"), returnData=TRUE)
percentVar <- round(100 * attr(pcadata, "percentVar"))
percentVar

aov1 <- aov(PC1 ~ Punch, data=pcadata)
summary(aov1) 
TukeyHSD(aov1, which = "Punch") 

aov2 <- aov(PC2 ~ Punch, data=pcadata)
summary(aov2) 
TukeyHSD(aov2, which = "Punch") 

aov3 <- aov(PC3 ~ APA2, data=pcadata)
summary(aov3) 
TukeyHSD(aov3, which = "APA2")

aov4 <- aov(PC4 ~ APA2, data=pcadata)
summary(aov4) 
TukeyHSD(aov4, which = "APA2") 

lm4 <- lm(PC4~APA2*Punch, data=pcadata)
summary(lm4)
anova(lm4) 

lm124 <- lm(PC1+PC2+PC4~APA2*Punch, data=pcadata)
summary(lm124)
anova(lm124)

pcadata$Punch <- factor(pcadata$Punch, levels=c("DG","CA3", "CA1"))
pcadata$APA2 <- factor(pcadata$APA2, levels=c("yoked_consistent", "yoked_conflict", "consistent", "conflict"))


plotPCs(pcadata, 1, 2, aescolor = pcadata$Punch, colorname = " ", aesshape = pcadata$APA2, shapename = " ",  colorvalues = colorvalPunch)

plotPCs(pcadata, 2, 1, aescolor = pcadata$Punch, colorname = " ", aesshape = pcadata$APA2, shapename = "APA",  colorvalues = colorvalPunch)

plotPCs(pcadata, 4, 2, aescolor = pcadata$APA2, colorname = "APA2", aesshape = pcadata$Punch, shapename = "Punch",  colorvalues = colorvalAPA2)


# pdf the same pca plots descripbed above of the above
pdf(file="../figures/02_RNAseq/PCA12.pdf", width=4.5, height=3)
PCA12 <- plotPCs(pcadata, 1, 2, aescolor = pcadata$Punch, colorname = " ", aesshape = pcadata$APA, shapename = " ",  colorvalues = colorvalPunch)
plot(PCA12)
dev.off()

pdf(file="../figures/02_RNAseq/PCA42.pdf", width=4.5, height=3)
PCA42 <- plotPCs(pcadata, 4, 2, aescolor = pcadata$APA, colorname = "APA2", aesshape = pcadata$Punch, shapename = "Punch",  colorvalues = colorvalAPA2)
plot(PCA42)
dev.off()


```


## venn diagrams

```{r venndiagrams}
rldpadjs <- assay(rld)
rldpadjs <- cbind(rldpadjs, contrast1, contrast2, contrast3, contrast4, contrast5, contrast6)
rldpadjs <- as.data.frame(rldpadjs)
rldpadjs <- rldpadjs[ , grepl( "padj" , names( rldpadjs ) ) ]

venn1 <- row.names(rldpadjs[rldpadjs[1] <0.05 & !is.na(rldpadjs[1]),]) # CA1 DG
venn2 <- row.names(rldpadjs[rldpadjs[2] <0.05 & !is.na(rldpadjs[2]),]) # CA1 CA3
venn3 <- row.names(rldpadjs[rldpadjs[3] <0.05 & !is.na(rldpadjs[3]),]) # CA3 DG
venn4 <- row.names(rldpadjs[rldpadjs[4] <0.05 & !is.na(rldpadjs[4]),]) # conflict conflictyoked
venn5 <- row.names(rldpadjs[rldpadjs[5] <0.05 & !is.na(rldpadjs[5]),]) # consistent consistentyoked
venn6 <- row.names(rldpadjs[rldpadjs[6] <0.05 & !is.na(rldpadjs[6]),]) # yoked yoked
```

```{r venndiagram3}
candidates <- list("conflict vs yoked" = venn4, "consistent vs yoked" = venn5, "yoked vs yoked" = venn6 )

prettyvenn <- venn.diagram(
  scaled=T,
  x = candidates, filename=NULL, 
  col = "black",
  fill = c( "white", "white", "white"),
  alpha = 0.5,
  cex = 1, fontfamily = "sans", #fontface = "bold",
  cat.default.pos = "text",
  #cat.dist = c(0.08, 0.08, 0.08), cat.pos = 1,
  cat.cex = 1, cat.fontfamily = "sans")
grid.draw(prettyvenn)

```

```{r venndiagram4}
candidates <- list("conflict vs yoked" = venn4, "CA3 vs DG" = venn3  ,"consistent vs yoked" = venn5, "CA1" = venn1 )

prettyvenn <- venn.diagram(
  scaled=T,
  x = candidates, filename=NULL, 
  col = "black",
  fill = c( "white", "white", "white", "white"),
  alpha = 0.5,
  cex = 1, fontfamily = "sans", #fontface = "bold",
  cat.default.pos = "text",
  #cat.dist = c(0.08, 0.08, 0.08), cat.pos = 1,
  cat.cex = 1, cat.fontfamily = "sans")
grid.draw(prettyvenn)

```


```{r venndiagram3brainregions} 
candidates <- list("DG vs CA1" = venn1, "DG vs CA3" = venn3,  "CA3 vs CA1" = venn2)

prettyvenn <- venn.diagram(
  scaled=T,
  x = candidates, filename=NULL, 
  col = "black",
  fill = c( "white", "white", "white"),
  alpha = 0.5,
  cex = 1, fontfamily = "sans", #fontface = "bold",
  cat.default.pos = "text",
  #cat.dist = c(0.08, 0.08, 0.08), cat.pos = 1,
  cat.cex = 1, cat.fontfamily = "sans")
grid.draw(prettyvenn)
```

## heatmap

```{r heatmap}
DEGes <- assay(rld)
DEGes <- cbind(DEGes, contrast1, contrast2, contrast3, contrast4, contrast5, contrast6)
DEGes <- as.data.frame(DEGes) # convert matrix to dataframe
DEGes$rownames <- rownames(DEGes)  # add the rownames to the dataframe
DEGes$padjmin <- with(DEGes, pmin(padjPunchCA1DG, padjPunchCA1CA3, padjPunchCA3DG, padjAPA2conflictyoked_conflict, padjAPA2consistentyoked_consistent, padjAPA2yoked_conflictyoked_consistent)) 



# create new col with min padj
DEGes <- DEGes %>% filter(padjmin < 0.05)
rownames(DEGes) <- DEGes$rownames
drop.cols <-colnames(DEGes[,grep("padj|pval|rownames", colnames(DEGes))])
DEGes <- DEGes %>% dplyr::select(-one_of(drop.cols))
DEGes <- as.matrix(DEGes)
DEGes <- DEGes - rowMeans(DEGes)
head(DEGes)

## the heatmap annotation file
df <- as.data.frame(colData(dds)[,c("Punch","APA2")]) ## matrix to df
rownames(df) <- names(countData)


ann_colors <- ann_colors4 # see color options 

# make sure the data is a matrix
DEGes <- as.matrix(DEGes) 

# set color breaks
paletteLength <- 30
myBreaks <- c(seq(min(DEGes), 0, length.out=ceiling(paletteLength/2) + 1), 
              seq(max(DEGes)/paletteLength, max(DEGes), length.out=floor(paletteLength/2)))

pheatmap(DEGes, show_colnames=T, show_rownames = F,
         annotation_col=df, annotation_colors = ann_colors,
         treeheight_row = 0, treeheight_col = 25,
         fontsize = 11, 
         #width=4.5, height=3,
         border_color = "grey60" ,
         color = viridis(30),
         cellwidth = 8, 
         clustering_method="average",
         breaks=myBreaks,
         clustering_distance_cols="correlation" 
         )

# for adobe
pheatmap(DEGes, show_colnames=F, show_rownames = F,
         annotation_col=df, annotation_colors = ann_colors,
         treeheight_row = 0, treeheight_col = 50,
         fontsize = 10, 
         #width=4.5, height=3,
         border_color = "grey60" ,
         color = viridis(30),
         cellwidth = 8, 
         clustering_method="average",
         breaks=myBreaks,
         clustering_distance_cols="correlation",
         filename = "../figures/02_RNAseq/pheatmap.pdf"
         )
```


## Volcanos plots and and gene lists


```{r volcanos}
# gene lists


res <- results(dds, contrast =c("APA2", "consistent", "yoked_consistent"), independentFiltering = F)
resOrdered <- res[order(res$padj),]
head(resOrdered, 10)

data <- data.frame(gene = row.names(res),
                   pvalue = -log10(res$padj), 
                   lfc = res$log2FoldChange)
data <- na.omit(data)
data <- data %>%
  mutate(color = ifelse(data$lfc > 0 & data$pvalue > 1.3, 
                        yes = "consistent", 
                        no = ifelse(data$lfc < 0 & data$pvalue > 1.3, 
                                    yes = "yoked_consistent", 
                                    no = "none")))
top_labelled <- top_n(data, n = 3, wt = pvalue)
# Color corresponds to fold change directionality
colored <- ggplot(data, aes(x = lfc, y = pvalue)) + 
  geom_point(aes(color = factor(color)), size = 1.75, alpha = 0.8, na.rm = T) + # add gene points
  theme_bw(base_size = 16) + # clean up theme
  theme(legend.position = "none") + # remove legend 
  #xlim(c(-2.5, 2.5)) +  
  #ylim(c(0, 10)) +  
  scale_color_manual(values = c("consistent" = "#f4a582",
                                "yoked_consistent" = "#bababa", 
                                "none" = "#bdbdbd")) + theme(panel.grid.minor=element_blank(),
           panel.grid.major=element_blank()) + 
  theme(axis.title.x = element_blank())+ 
  theme(axis.title.y = element_blank()) + 
  geom_text_repel(data = top_labelled, 
                          mapping = aes(label = gene), 
                          size = 3,
                          fontface = 'bold', 
                          color = 'black',
                          box.padding = unit(0.5, "lines"),
                          point.padding = unit(0.5, "lines"))

colored
cvd_grid(colored) # to view plot for color blind 


###
res <- results(dds, contrast =c("APA2", "yoked_consistent", "yoked_conflict"), independentFiltering = F)
resOrdered <- res[order(res$padj),]
head(resOrdered, 10)

data <- data.frame(gene = row.names(res),
                   pvalue = -log10(res$padj), 
                   lfc = res$log2FoldChange)
data <- na.omit(data)
data <- data %>%
  mutate(color = ifelse(data$lfc > 0 & data$pvalue > 1.3, 
                        yes = "yoked_conflict", 
                        no = ifelse(data$lfc < 0 & data$pvalue > 1.3, 
                                    yes = "yoked_consistent", 
                                    no = "none")))
top_labelled <- top_n(data, n = 3, wt = pvalue)
# Color corresponds to fold change directionality
colored <- ggplot(data, aes(x = lfc, y = pvalue)) + 
  geom_point(aes(color = factor(color)), size = 1.75, alpha = 0.8, na.rm = T) + # add gene points
  theme_bw(base_size = 16) + # clean up theme
  theme(legend.position = "none") + # remove legend 
  #xlim(c(-2.5, 2.5)) +  
  #ylim(c(0, 10)) +  
  scale_color_manual(values = c("yoked_conflict" = "#404040",
                                "yoked_consistent" = "#bababa", 
                                "none" = "#bdbdbd")) + theme(panel.grid.minor=element_blank(),
           panel.grid.major=element_blank()) + 
  theme(axis.title.x = element_blank())+ 
  theme(axis.title.y = element_blank()) + 
  geom_text_repel(data = top_labelled, 
                          mapping = aes(label = gene), 
                          size = 3,
                          fontface = 'bold', 
                          color = 'black',
                          box.padding = unit(0.5, "lines"),
                          point.padding = unit(0.5, "lines"))

colored
cvd_grid(colored) # to view plot for color blind 


####

res <- results(dds, contrast =c("APA2", "conflict", "yoked_conflict"), independentFiltering = F)
resOrdered <- res[order(res$padj),]
head(resOrdered, 10)
data <- data.frame(gene = row.names(res),
                   pvalue = -log10(res$padj), 
                   lfc = res$log2FoldChange)
data <- na.omit(data)
data <- data %>%
  mutate(color = ifelse(data$lfc > 0 & data$pvalue > 1.3, 
                        yes = "conflict", 
                        no = ifelse(data$lfc < 0 & data$pvalue > 1.3, 
                                    yes = "yoked_conflict", 
                                    no = "none")))
top_labelled <- top_n(data, n = 3, wt = pvalue)
# Color corresponds to fold change directionality
colored <- ggplot(data, aes(x = lfc, y = pvalue)) + 
  geom_point(aes(color = factor(color)), size = 1.75, alpha = 0.8, na.rm = T) + # add gene points
  theme_bw(base_size = 16) + # clean up theme
  theme(legend.position = "none") + # remove legend 
  #xlim(c(-2.5, 2.5)) +  
  #ylim(c(0, 10)) +  
  scale_color_manual(values = c("conflict" = "#ca0020",
                                "yoked_conflict" = "#404040", 
                                "none" = "#bdbdbd")) + theme(panel.grid.minor=element_blank(),
           panel.grid.major=element_blank()) + 
  theme(axis.title.x = element_blank())+ 
  theme(axis.title.y = element_blank()) + 
  geom_text_repel(data = top_labelled, 
                          mapping = aes(label = gene), 
                          size = 3,
                          fontface = 'bold', 
                          color = 'black',
                          box.padding = unit(0.5, "lines"),
                          point.padding = unit(0.5, "lines"))

colored
cvd_grid(colored) # to view plot for color blind 
 

### 
res <- results(dds, contrast = c("APA2", "consistent", "conflict"), independentFiltering = F)
resOrdered <- res[order(res$padj),]
head(resOrdered, 10)

data <- data.frame(gene = row.names(res),
                   pvalue = -log10(res$padj), 
                   lfc = res$log2FoldChange)
data <- na.omit(data)
data <- data %>%
  mutate(color = ifelse(data$lfc > 0 & data$pvalue > 1.3, 
                        yes = "conflict", 
                        no = ifelse(data$lfc < 0 & data$pvalue > 1.3, 
                                    yes = "consistent", 
                                    no = "none")))
top_labelled <- top_n(data, n = 3, wt = pvalue)
# Color corresponds to fold change directionality
colored <- ggplot(data, aes(x = lfc, y = pvalue)) + 
  geom_point(aes(color = factor(color)), size = 1.75, alpha = 0.8, na.rm = T) + # add gene points
  theme_bw(base_size = 16) + # clean up theme
  theme(legend.position = "none") + # remove legend 
  #xlim(c(-2.5, 2.5)) +  
  #ylim(c(0, 10)) +  
  scale_color_manual(values = c("conflict" = "#ca0020",
                                "consistent" = "#f4a582", 
                                "none" = "#bdbdbd")) + theme(panel.grid.minor=element_blank(),
           panel.grid.major=element_blank()) + 
  theme(axis.title.x = element_blank())+ 
  theme(axis.title.y = element_blank()) + 
  geom_text_repel(data = top_labelled, 
                          mapping = aes(label = gene), 
                          size = 3,
                          fontface = 'bold', 
                          color = 'black',
                          box.padding = unit(0.5, "lines"),
                          point.padding = unit(0.5, "lines"))

colored
cvd_grid(colored) # to view plot for color blind 

```

```{r writecsv}
#write.csv(rldpadjs, file = "../data/02b_rldpadjs.csv", row.names = T)
#write.csv(DEGes, file = "../data/02b_DEGes.csv", row.names = T)
#write.csv(df, file = "../data/02b_df.csv", row.names = F)
#write.csv(pcadata, file = "../data/02b_pcadata.csv", row.names = F)
#write.table(percentVar, file = "../data/02b_percentVar.txt")
```


