---
title: "02c_rnaseqSubfield"
author: "Rayna M Harris"
date: "8/22/17"
output: md_document
---

```{r setup, message=F}
library(ggplot2) ## for awesome plots!
library(cowplot) ## for some easy to use themes
library(dplyr) ## for filtering and selecting rows
library(car) ## stats
library(VennDiagram) ## venn diagrams
library(pheatmap) ## awesome heatmaps
library(viridis) # for awesome color pallette
library(reshape2) ## for melting dataframe
library(DESeq2) ## for gene expression analysis
library(edgeR)  ## for basic read counts status
library(magrittr) ## to use the weird pipe
library(genefilter)  ## for PCA fuction

## load functions 
source("figureoptions.R")
source("functions_RNAseq.R")

## set output file for figures 
knitr::opts_chunk$set(fig.path = '../figures/02c_rnaseqSubfield/')
```

## DG

```{r DG, message=F, warning=F}
colData <- read.csv("../data/02a_colData.csv", header = T)
countData <- read.csv("../data/02a_countData.csv", header = T, check.names = F, row.names = 1)

colData <- colData %>% 
  filter(Punch %in% c("DG"))  %>% 
  droplevels()

savecols <- as.character(colData$RNAseqID) 
savecols <- as.vector(savecols) 
countData <- countData %>% dplyr::select(one_of(savecols)) 

colData %>% select(APA2,Punch)  %>%  summary()


## create DESeq object using the factors Punch and APA
dds <- DESeqDataSetFromMatrix(countData = countData,
                              colData = colData,
                              design = ~ APA2)

dds # view the DESeq object - note numnber of genes
dds <- dds[ rowSums(counts(dds)) > 1, ]  # Pre-filtering genes with 0 counts
dds # view number of genes afternormalization and the number of samples
dds <- DESeq(dds) # Differential expression analysis
rld <- rlog(dds, blind=FALSE) ## log transformed data


# create the dataframe using my function pcadataframe
pcadata <- pcadataframe(rld, intgroup=c("Punch","APA2"), returnData=TRUE)
percentVar <- round(100 * attr(pcadata, "percentVar"))
percentVar

 
summary(aov(PC1 ~ APA2, data=pcadata)) 
TukeyHSD((aov(PC1 ~ APA2, data=pcadata)) , which = "APA2") 

summary(aov(PC2 ~ APA2, data=pcadata)) 
TukeyHSD((aov(PC2 ~ APA2, data=pcadata)), which = "APA2") 

summary(aov(PC3 ~ APA2, data=pcadata)) 

summary(aov(PC4 ~ APA2, data=pcadata)) 

pcadata$APA2 <- factor(pcadata$APA2, levels=c("yoked_consistent", "consistent", "yoked_conflict", "conflict"))
pcadata$wrap <- "Principle Compent Analysis"

#calculate significance of all two way comparisions
# see source "functions_RNAseq.R" 
contrast1 <- resvals(contrastvector = c("APA2", "consistent", "yoked_consistent"), mypval = 0.1) # 95
contrast2 <- resvals(contrastvector = c("APA2", "conflict", "yoked_conflict"), mypval = 0.1) # 5
contrast3 <- resvals(contrastvector = c("APA2", "conflict", "consistent"), mypval = 0.1) # 0
contrast4 <- resvals(contrastvector = c("APA2", "yoked_conflict", "yoked_consistent"), mypval = 0.1) # 2


res <- results(dds, contrast =c("APA2", "consistent", "yoked_consistent"), independentFiltering = T, alpha = 0.1)
summary(res)
resOrdered <- res[order(res$padj),]
head(resOrdered, 10)

topGene <- rownames(res)[which.min(res$padj)]
plotCounts(dds, gene = topGene, intgroup=c("APA2"))


data <- data.frame(gene = row.names(res),
                   pvalue = -log10(res$padj), 
                   lfc = res$log2FoldChange)
data <- na.omit(data)
data <- data %>%
  mutate(color = ifelse(data$lfc > 0 & data$pvalue > 1, 
                        yes = "consistent", 
                        no = ifelse(data$lfc < 0 & data$pvalue > 1, 
                                    yes = "yoked_consistent", 
                                    no = "none")))
top_labelled <- top_n(data, n = 5, wt = lfc)

DGvolcano <- ggplot(data, aes(x = lfc, y = pvalue)) + 
  geom_point(aes(color = factor(color)), size = 1, alpha = 0.5, na.rm = T) + # add gene points
  scale_color_manual(values = volcano1)  + 
  scale_x_continuous(name="log2(consistent/yoked)") +
  scale_y_continuous(name=NULL) +
  theme_cowplot(font_size = 8, line_size = 0.25) +
  geom_hline(yintercept = 1,  size = 0.25, linetype = 2 )+ 
  theme(panel.grid.minor=element_blank(),
        legend.position = "none", # remove legend 
        panel.grid.major=element_blank())
DGvolcano

pdf(file="../figures/02c_rnaseqSubfield/DGvolcano.pdf", width=1.25, height=2)
plot(DGvolcano)
dev.off()

plotPCs <- function(df, xcol, ycol, aescolor, colorname, colorvalues){
  ggplot(df, aes(df[xcol], df[ycol], color=aescolor)) +
    geom_point(size=2) +
    xlab(paste0("PC", xcol, ": ", percentVar[xcol],"% variance")) +
    ylab(paste0("PC", ycol, ": ", percentVar[ycol],"% variance")) +
    stat_ellipse(level = 0.95, (aes(color=aescolor)),size=0.25) + 
    scale_colour_manual(name=colorname, values=c(colorvalues))+ 
    theme_cowplot(font_size = 8, line_size = 0.25)  +
    theme(legend.position="none")
}


PCA12 <- plotPCs(pcadata, 1, 2, aescolor = pcadata$APA2, colorname = " ",  colorvalues = colorvalAPA2)
PCA12

pdf(file="../figures/02c_rnaseqSubfield/DGpca12.pdf", width=1.75, height=2)
plot(PCA12)
dev.off()

```

## venn diagrams

```{r DGvenndiagrams, echo=F}
rldpadjs <- assay(rld)
rldpadjs <- cbind(rldpadjs, contrast1, contrast2, contrast4)
rldpadjs <- as.data.frame(rldpadjs)
rldpadjs <- rldpadjs[ , grepl( "padj" , names( rldpadjs ) ) ]

venn1 <- row.names(rldpadjs[rldpadjs[1] <0.05 & !is.na(rldpadjs[1]),]) 
venn2 <- row.names(rldpadjs[rldpadjs[2] <0.05 & !is.na(rldpadjs[2]),]) 
venn3 <- row.names(rldpadjs[rldpadjs[3] <0.05 & !is.na(rldpadjs[3]),]) 

candidates <- list("consistent vs yoked" = venn1 , "conflict vs yoked" = venn2,  "yoked vs yoked" = venn3)

prettyvenn <- venn.diagram(
  scaled=T,
  x = candidates, filename=NULL, 
  col = "black",
  fill = c( "white", "white", "white"),
  alpha = 0.5,
  cex = 1, fontfamily = "sans", #fontface = "bold",
  cat.default.pos = "text",
  cat.dist = c(0.03, 0.03, 0.02), cat.pos = 1,
  cat.cex = 1, cat.fontfamily = "sans")
grid.draw(prettyvenn)
```

## CA3

```{r CA3, message=F, warning=F}
colData <- read.csv("../data/02a_colData.csv", header = T)
countData <- read.csv("../data/02a_countData.csv", header = T, check.names = F, row.names = 1)

colData <- colData %>% 
  filter(Punch %in% c("CA3"))  %>% 
  droplevels()
savecols <- as.character(colData$RNAseqID) 
savecols <- as.vector(savecols) 
countData <- countData %>% dplyr::select(one_of(savecols)) 
colData %>% select(APA2,Punch)  %>%  summary()

## create DESeq object using the factors Punch and APA
dds <- DESeqDataSetFromMatrix(countData = countData,
                              colData = colData,
                              design = ~ APA2)

dds # view the DESeq object - note numnber of genes
dds <- dds[ rowSums(counts(dds)) > 1, ]  # Pre-filtering genes with 0 counts
dds # view number of genes afternormalization and the number of samples
dds <- DESeq(dds) # Differential expression analysis
rld <- rlog(dds, blind=FALSE) ## log transformed data

# create the dataframe using my function pcadataframe
pcadata <- pcadataframe(rld, intgroup=c("Punch","APA2"), returnData=TRUE)
percentVar <- round(100 * attr(pcadata, "percentVar"))
percentVar

TukeyHSD((aov(PC1 ~ APA2, data=pcadata)), which = "APA2") 
TukeyHSD((aov(PC2 ~ APA2, data=pcadata)), which = "APA2") 
TukeyHSD((aov(PC3 ~ APA2, data=pcadata)), which = "APA2") 
TukeyHSD((aov(PC4 ~ APA2, data=pcadata)), which = "APA2") 
TukeyHSD((aov(PC4 ~ APA2, data=pcadata)), which = "APA2") 
summary(aov(PC5 ~ APA2, data=pcadata))
summary(aov(PC6 ~ APA2, data=pcadata))

pcadata$APA2 <- factor(pcadata$APA2, levels=c("yoked_consistent", "consistent", "yoked_conflict", "conflict"))
pcadata$wrap <- "Principle Compent Analysis"

plotPCs <- function(df, xcol, ycol, aescolor, colorname, colorvalues){
  ggplot(df, aes(df[xcol], df[ycol], color=aescolor)) +
    geom_point(size=2, alpha= 0.8) +
    xlab(paste0("PC", xcol, ": ", percentVar[xcol],"% variance")) +
    ylab(paste0("PC", ycol, ": ", percentVar[ycol],"% variance")) +
    stat_ellipse(level = 0.95, (aes(color=aescolor)),size=0.25) + 
    scale_colour_manual(name=colorname, values=c(colorvalues))+ 
    theme_cowplot(font_size = 8, line_size = 0.25)  +
    theme(legend.position="none")
}

PCA14 <- plotPCs(pcadata, 1, 4, aescolor = pcadata$APA2, colorname = " ",  colorvalues = colorvalAPA2)
PCA14

pdf(file="../figures/02c_rnaseqSubfield/CA3pca14.pdf", width=1.75, height=2)
plot(PCA14)
dev.off()
 
#calculate significance of all two way comparisions
# see source "functions_RNAseq.R" 
contrast1 <- resvals(contrastvector = c("APA2", "consistent", "yoked_consistent"), mypval = 0.1) # 0
contrast2 <- resvals(contrastvector = c("APA2", "conflict", "yoked_conflict"), mypval = 0.1) # 0
contrast3 <- resvals(contrastvector = c("APA2", "conflict", "consistent"), mypval = 0.1) # 1
contrast4 <- resvals(contrastvector = c("APA2", "yoked_conflict", "yoked_consistent"), mypval = 0.1) # 0

```



```{r CA3venndiagrams, echo=F}
rldpadjs <- assay(rld)
rldpadjs <- cbind(rldpadjs, contrast1, contrast2, contrast3, contrast4)
rldpadjs <- as.data.frame(rldpadjs)
rldpadjs <- rldpadjs[ , grepl( "padj" , names( rldpadjs ) ) ]

venn1 <- row.names(rldpadjs[rldpadjs[1] <0.05 & !is.na(rldpadjs[1]),]) 
venn2 <- row.names(rldpadjs[rldpadjs[2] <0.05 & !is.na(rldpadjs[2]),]) 
venn3 <- row.names(rldpadjs[rldpadjs[3] <0.05 & !is.na(rldpadjs[3]),]) 
venn4 <- row.names(rldpadjs[rldpadjs[4] <0.05 & !is.na(rldpadjs[4]),]) 


candidates <- list("consistent vs yoked" = venn1 , "conflict vs yoked" = venn2, "conflict vs consistent" = venn3, "yoked vs yoked" = venn4)

prettyvenn <- venn.diagram(
  scaled=T,
  x = candidates, filename=NULL, 
  col = "black",
  fill = c( "white", "white", "white", "white"),
  alpha = 0.5,
  cex = 1, fontfamily = "sans", #fontface = "bold",
  cat.default.pos = "text",
  cat.dist = c(0.03, 0.03, 0.02,0.02), cat.pos = 1,
  cat.cex = 1, cat.fontfamily = "sans")
grid.draw(prettyvenn)
```


## CA1

```{r CA1, message=F, warning=F}
colData <- read.csv("../data/02a_colData.csv", header = T)
countData <- read.csv("../data/02a_countData.csv", header = T, check.names = F, row.names = 1)
colData <- colData %>% 
  filter(Punch %in% c("CA1"))  %>% 
  droplevels()
savecols <- as.character(colData$RNAseqID) 
savecols <- as.vector(savecols) 
countData <- countData %>% dplyr::select(one_of(savecols)) 
colData %>% select(APA2,Punch)  %>%  summary()
dds <- DESeqDataSetFromMatrix(countData = countData,
                              colData = colData,
                              design = ~ APA2)

dds # view the DESeq object - note numnber of genes
dds <- dds[ rowSums(counts(dds)) > 1, ]  # Pre-filtering genes with 0 counts
dds # view number of genes afternormalization and the number of samples
dds <- DESeq(dds) # Differential expression analysis
rld <- rlog(dds, blind=FALSE) ## log transformed data


# PCA analysis
pcadata <- pcadataframe(rld, intgroup=c("Punch","APA2"), returnData=TRUE)
percentVar <- round(100 * attr(pcadata, "percentVar"))
percentVar
summary(aov(PC1 ~ APA2, data=pcadata)) 
TukeyHSD((aov(PC1 ~ APA2, data=pcadata)), which = "APA2") 
summary(aov(PC2 ~ APA2, data=pcadata)) 
TukeyHSD((aov(PC2 ~ APA2, data=pcadata)) , which = "APA2") 
summary(aov(PC3 ~ APA2, data=pcadata))
summary(aov(PC4 ~ APA2, data=pcadata))

pcadata$APA2 <- factor(pcadata$APA2, levels=c("yoked_consistent", "consistent", "yoked_conflict", "conflict"))

 
data <- data.frame(gene = row.names(res),
                   pvalue = -log10(res$padj), 
                   lfc = res$log2FoldChange)
data <- na.omit(data)
data <- data %>%
  mutate(color = ifelse(data$lfc > 0 & data$pvalue > 1, 
                        yes = "consistent", 
                        no = ifelse(data$lfc < 0 & data$pvalue > 1, 
                                    yes = "yoked_consistent", 
                                    no = "none")))
top_labelled <- top_n(data, n = 5, wt = lfc)

pcadata$wrap <- "CA1 PCA"
data$wrap <- "CA1 contrasts"

CA1volcano <- ggplot(data, aes(x = lfc, y = pvalue)) + 
  geom_point(aes(color = factor(color)), size = 1, alpha = 0.5, na.rm = T) + # add gene points
  scale_color_manual(values = volcano1)  + 
  scale_x_continuous(name="log2 (consistent/yoked)") +
  scale_y_continuous(name=NULL) +
  theme_cowplot(font_size = 8, line_size = 0.25) +
  geom_hline(yintercept = 1,  size = 0.25, linetype = 2) + 
  theme(panel.grid.minor=element_blank(),
        legend.position = "none", # remove legend 
        panel.grid.major=element_blank())
CA1volcano


pdf(file="../figures/02c_rnaseqSubfield/CA1volcano.pdf",  width=1.5, height=2)
plot(CA1volcano)
dev.off()


plotPCs <- function(df, xcol, ycol, aescolor, colorname, colorvalues){
  ggplot(df, aes(df[xcol], df[ycol], color=aescolor)) +
    geom_point(size=2, alpha = 0.8 ) +
    xlab(paste0("PC", xcol, ": ", percentVar[xcol],"% variance")) +
    ylab(paste0("PC", ycol, ": ", percentVar[ycol],"% variance")) +
    stat_ellipse(level = 0.95, (aes(color=aescolor)),size=0.25) + 
    scale_colour_manual(name=colorname, values=c(colorvalues))+ 
    theme_cowplot(font_size = 8, line_size = 0.25)  +
    theme(legend.position="none") +
    facet_wrap(~wrap)
}


PCA12 <- plotPCs(pcadata, 1, 2, aescolor = pcadata$APA2, colorname = " ",  colorvalues = colorvalAPA00)
PCA12


pdf(file="../figures/02c_rnaseqSubfield/CA1volcano.pdf", width=1.75, height = 2)
plot(CA1volcano)
dev.off()


PCA14 <- plotPCs(pcadata, 1, 4, aescolor = pcadata$APA2, colorname = " ",  colorvalues = colorvalAPA2)
PCA14

pdf(file="../figures/02c_rnaseqSubfield/CA1pca14.pdf", width=1.75, height=2)
plot(PCA14)
dev.off()
```




```{r CA1venndiagrams, echo=F}
res <- results(dds, contrast =c("APA2", "consistent", "yoked_consistent"), independentFiltering = T, alpha = 0.1)
summary(res)
resOrdered <- res[order(res$padj),]
head(resOrdered, 10)
topGene <- rownames(res)[which.min(res$padj)]
plotCounts(dds, gene = topGene, intgroup=c("APA2"))


rldpadjs <- assay(rld)
rldpadjs <- cbind(rldpadjs, contrast1, contrast2, contrast4)
rldpadjs <- as.data.frame(rldpadjs)
rldpadjs <- rldpadjs[ , grepl( "padj" , names( rldpadjs ) ) ]

venn1 <- row.names(rldpadjs[rldpadjs[1] <0.05 & !is.na(rldpadjs[1]),]) 
venn2 <- row.names(rldpadjs[rldpadjs[2] <0.05 & !is.na(rldpadjs[2]),]) 
venn3 <- row.names(rldpadjs[rldpadjs[3] <0.05 & !is.na(rldpadjs[3]),]) 

candidates <- list("consistent vs yoked" = venn1 , "conflict vs yoked" = venn2,  "yoked vs yoked" = venn3)

prettyvenn <- venn.diagram(
  scaled=T,
  x = candidates, filename=NULL, 
  col = "black",
  fill = c( "white", "white", "white"),
  alpha = 0.5,
  cex = 1, fontfamily = "sans", #fontface = "bold",
  cat.default.pos = "text",
  cat.dist = c(0.03, 0.03, 0.02), cat.pos = 1,
  cat.cex = 1, cat.fontfamily = "sans")
grid.draw(prettyvenn)
```

