---
title: "Brain region specific only"
author: "Rayna M Harris"
date: "May 2, 2017"
output: md_document
---
```{r setup, message=F, warning=F}
library(dplyr) ## for filtering and selecting rows
library(plyr) ## for renmaing factors
library(reshape2) ## for melting dataframe
library(DESeq2) ## for gene expression analysis
library(edgeR)  ## for basic read counts status
library(magrittr) ## to use the weird pipe
library(genefilter)  ## for PCA fuction
library(ggplot2) ## for awesome plots!
library(pheatmap) ## awesome heatmaps
library(VennDiagram)

## Functions
source("functions_RNAseq.R")
source("resvalsfunction.R")
source("figureoptions.R")

## set output file for figures 
knitr::opts_chunk$set(fig.path = '../figures/02_RNAseq/')
```

## CA1 only differential gene expression
Zero genes have padj < 0.05

```{r CA1only}
countData <- read.csv("../data/02a_countData.csv", header = T, check.names = F, row.names = 1)
colData <- read.csv("../data/02a_colData.csv", header = T)
colData$APA <- factor(colData$APA, levels=c("Control", "Consistent", "Conflict"))

colData <- colData %>% 
  filter(Punch == "CA1") 
savecols <- as.character(colData$RNAseqID) 
savecols <- as.vector(savecols) 
countData <- countData %>% dplyr::select(one_of(savecols)) 

dds <- DESeqDataSetFromMatrix(countData = countData,
                              colData = colData,
                              design = ~ APA )
dds$APA <- factor(dds$APA, levels=c("Control", "Consistent", "Conflict"))
dds <- dds[ rowSums(counts(dds)) > 1, ] 
dds # dim: 16467 15  
dds <- DESeq(dds)
rld <- rlog(dds, blind=FALSE)

contrast4 <- resvals(contrastvector = c("APA", "Consistent", "Control"), mypval = 0.05) #0
contrast5 <- resvals(contrastvector = c("APA", "Conflict", "Control"), mypval = 0.05) #0
contrast6 <- resvals(contrastvector = c("APA", "Conflict", "Consistent"), mypval = 0.05) # 0 

res <- results(dds, contrast =c("APA", "Consistent", "Control"), independentFiltering = F)
with(res, plot(log2FoldChange, -log10(pvalue), pch=20, main="CA1 Control - Consistent", xlim=c(-8,8), ylim=c(0,14)))
with(subset(res, log2FoldChange>0), points(log2FoldChange, -log10(pvalue), pch=20, col=c("#f4a582")))
with(subset(res, log2FoldChange<0), points(log2FoldChange, -log10(pvalue), pch=20, col=c("#404040")))
with(subset(res, padj>.05 ), points(log2FoldChange, -log10(pvalue), pch=20, col="grey"))
resOrdered <- res[order(res$padj),]
head(resOrdered)

res <- results(dds, contrast =c("APA", "Conflict", "Control"), independentFiltering = F)

with(res, plot(log2FoldChange, -log10(pvalue), pch=20, main="CA1 Control - Conflict", xlim=c(-8,8), ylim=c(0,14)))
with(subset(res, log2FoldChange>0), points(log2FoldChange, -log10(pvalue), pch=20, col=c("#ca0020")))
with(subset(res, log2FoldChange<0), points(log2FoldChange, -log10(pvalue), pch=20, col=c("#404040")))
with(subset(res, padj>.05 ), points(log2FoldChange, -log10(pvalue), pch=20, col="grey"))

resOrdered <- res[order(res$padj),]
head(resOrdered)

res <- results(dds, contrast =c("APA", "Conflict", "Consistent"), independentFiltering = F)
with(res, plot(log2FoldChange, -log10(pvalue), pch=20, main="CA1 Consistent - Conflict", xlim=c(-8,8), ylim=c(0,14)))
with(subset(res, log2FoldChange>0), points(log2FoldChange, -log10(pvalue), pch=20, col=c("#ca0020")))
with(subset(res, log2FoldChange<0), points(log2FoldChange, -log10(pvalue), pch=20, col=c("#f4a582")))
with(subset(res, padj>.05 ), points(log2FoldChange, -log10(pvalue), pch=20, col="grey"))
resOrdered <- res[order(res$padj),]
head(resOrdered)
```

## CA3 only differential gene expression

```{r CA3only}
countData <- read.csv("../data/02a_countData.csv", header = T, check.names = F, row.names = 1)
colData <- read.csv("../data/02a_colData.csv", header = T)
colData$APA <- factor(colData$APA, levels=c("Control", "Consistent", "Conflict"))

colData <- colData %>% 
  filter(Punch == "CA3") 
savecols <- as.character(colData$RNAseqID) 
savecols <- as.vector(savecols) 
countData <- countData %>% dplyr::select(one_of(savecols)) 

dds <- DESeqDataSetFromMatrix(countData = countData,
                              colData = colData,
                              design = ~ APA )
dds$APA <- factor(dds$APA, levels=c("Control", "Consistent", "Conflict"))
dds <- dds[ rowSums(counts(dds)) > 1, ] 
dds # dim: 16658 16   
dds <- DESeq(dds)
rld <- rlog(dds, blind=FALSE)

contrast4 <- resvals(contrastvector = c("APA", "Consistent", "Control"), mypval = 0.05) #0
contrast5 <- resvals(contrastvector = c("APA", "Conflict", "Control"), mypval = 0.05) #0
contrast6 <- resvals(contrastvector = c("APA", "Conflict", "Consistent"), mypval = 0.05) # 3 

res <- results(dds, contrast = c("APA", "Conflict", "Consistent"), independentFiltering = F)
resOrdered <- res[order(res$padj),]
head(resOrdered)

# Oprd1 = Opioid Receptor Delta 1
# Crnkl1 = Crooked Neck Pre- MRNA Splicing Factor 1
# Slc9a2 = Solute Carrier Family 9 Member A2



res <- results(dds, contrast =c("APA", "Consistent", "Control"), independentFiltering = F)
with(res, plot(log2FoldChange, -log10(pvalue), pch=20, main="CA3 Control - Consistent", xlim=c(-8,8), ylim=c(0,14)))
with(subset(res, log2FoldChange>0), points(log2FoldChange, -log10(pvalue), pch=20, col=c("#f4a582")))
with(subset(res, log2FoldChange<0), points(log2FoldChange, -log10(pvalue), pch=20, col=c("#404040")))
with(subset(res, padj>.05 ), points(log2FoldChange, -log10(pvalue), pch=20, col="grey"))
resOrdered <- res[order(res$padj),]
head(resOrdered)

res <- results(dds, contrast =c("APA", "Conflict", "Control"), independentFiltering = F)
with(res, plot(log2FoldChange, -log10(pvalue), pch=20, main="CA3 Control - Conflict", xlim=c(-8,8), ylim=c(0,14)))
with(subset(res, log2FoldChange>0), points(log2FoldChange, -log10(pvalue), pch=20, col=c("#ca0020")))
with(subset(res, log2FoldChange<0), points(log2FoldChange, -log10(pvalue), pch=20, col=c("#404040")))
with(subset(res, padj>.05 ), points(log2FoldChange, -log10(pvalue), pch=20, col="grey"))
resOrdered <- res[order(res$padj),]
head(resOrdered)

res <- results(dds, contrast =c("APA", "Conflict", "Consistent"), independentFiltering = F)
with(res, plot(log2FoldChange, -log10(pvalue), pch=20, main="CA3 Consistent - Conflict", xlim=c(-8,8), ylim=c(0,14)))
with(subset(res, log2FoldChange>0), points(log2FoldChange, -log10(pvalue), pch=20, col=c("#ca0020")))
with(subset(res, log2FoldChange<0), points(log2FoldChange, -log10(pvalue), pch=20, col=c("#f4a582")))
with(subset(res, padj>.05 ), points(log2FoldChange, -log10(pvalue), pch=20, col="grey"))
resOrdered <- res[order(res$padj),]
head(resOrdered)
```



```{r DGonly}

pcadata <- pcadataframe(rld, intgroup=c("Punch","APA"), returnData=TRUE)
percentVar <- round(100 * attr(pcadata, "percentVar"))
percentVar

pcadata <- pcadataframe(rld, intgroup=c("Punch","APA"), returnData=TRUE)
percentVar <- round(100 * attr(pcadata, "percentVar"))
aov1 <- aov(PC1 ~ APA, data=pcadata)
summary(aov1) 
TukeyHSD(aov1, which = "APA")
pcadata$Punch <- factor(pcadata$APA, levels=c("Control", "Consistent", "Conflict"))
plotPCs(pcadata, 1, 2, aescolor = pcadata$APA, colorname = "APA", aesshape = pcadata$APA, shapename = "APA",  colorvalues = colorvalAPA)

DEGes <- assay(rld)
DEGes <- cbind(DEGes, contrast4, contrast5, contrast6)
DEGes <- as.data.frame(DEGes) 
DEGes$rownames <- rownames(DEGes)  
DEGes$padjmin <- with(DEGes, pmin(padjAPAConsistentControl, padjAPAConflictControl, padjAPAConflictConsistent)) 
DEGes <- DEGes %>% filter(padjmin < 0.05)
rownames(DEGes) <- DEGes$rownames
drop.cols <-colnames(DEGes[,grep("padj|pval|rownames", colnames(DEGes))])
DEGes <- DEGes %>% dplyr::select(-one_of(drop.cols))
DEGes <- as.matrix(DEGes)
DEGes <- DEGes - rowMeans(DEGes)
df <- as.data.frame(colData(dds)[,c("APA", "Punch")]) #
rownames(df) <- names(countData)
ann_colors = ann_colors1 
DEGes <- as.matrix(DEGes) 
paletteLength <- 30
myBreaks <- c(seq(min(DEGes), 0, length.out=ceiling(paletteLength/2) + 1), 
              seq(max(DEGes)/paletteLength, max(DEGes), length.out=floor(paletteLength/2)))
pheatmap(DEGes, show_colnames=T, show_rownames = F,
         annotation_col=df, annotation_colors = ann_colors,
         treeheight_row = 0, treeheight_col = 25,
         fontsize = 11, 
         #width=4.5, height=3,
         border_color = "grey60" ,
         color = colorpalette,
         cellwidth = 8,
         cellheight = 2,
         clustering_method="average",
         breaks=myBreaks,
         clustering_distance_cols="correlation" 
         )
```

```{r DGvenn}

rldpvals <- assay(rld)
rldpvals <- cbind(rldpvals, contrast4, contrast5, contrast6)
rldpvals <- as.data.frame(rldpvals)
rldpvals <- rldpvals[ , grepl( "padj|pval" , names( rldpvals ) ) ]


# venn with padj values
venn1 <- row.names(rldpvals[rldpvals[2] <0.05 & !is.na(rldpvals[2]),])
venn2 <- row.names(rldpvals[rldpvals[4] <0.05 & !is.na(rldpvals[4]),])
venn3 <- row.names(rldpvals[rldpvals[6] <0.05 & !is.na(rldpvals[6]),])
venn12 <- union(venn1,venn2)
venn123 <- union(venn12,venn3)
venn1minus2 <- setdiff(venn1,venn2)
# save files for big venn diagram
write(venn123, "../data/20c_vennDGAll.txt")
write(venn1, "../data/20c_vennDGControlConsistent.txt")
write(venn2, "../data/20c_vennDGControlConflict.txt")
write(venn1minus2, "../data/20c_vennDGControlminusConsistent.txt")



## check order for correctness
candidates <- list("Control-Consistent" = venn1, "Control-Conflict" = venn2,"Consistent-Conflict" = venn3)

prettyvenn <- venn.diagram(
  scaled=T,
  x = candidates, filename=NULL, 
  col = "black",
  fill = c( "white", "white", "white"),
  alpha = 0.5,
  cex = 1, fontfamily = "sans", #fontface = "bold",
  cat.default.pos = "text",
  cat.dist = c(0.06, 0.06, 0.05), cat.pos = 1,
  cat.cex = 1, cat.fontfamily = "sans")
#dev.off()
grid.draw(prettyvenn)



res <- results(dds, contrast =c("APA", "Consistent", "Control"), independentFiltering = F)
table(res$padj<0.05)
table(res$pvalue<0.05)
logs <- data.frame(cbind("gene"=row.names(res),"logP"=round(-log(res$pvalue+1e-10,10),1)))
logs$logP <- as.numeric(as.character(logs$logP))
sign <- rep(1,nrow(logs))
sign[res$log2FoldChange<0]=-1  ##change to correct model
table(sign)
logs$logP <- logs$logP*sign
write.csv(logs, file = "./02d_GO_MWU/padjAPAConsistentControlDG.csv", row.names = F)

res <- results(dds, contrast =c("APA", "Conflict", "Control"), independentFiltering = F)
table(res$padj<0.05)
table(res$pvalue<0.05)
logs <- data.frame(cbind("gene"=row.names(res),"logP"=round(-log(res$pvalue+1e-10,10),1)))
logs$logP <- as.numeric(as.character(logs$logP))
sign <- rep(1,nrow(logs))
sign[res$log2FoldChange<0]=-1  ##change to correct model
table(sign)
logs$logP <- logs$logP*sign
write.csv(logs, file = "./02d_GO_MWU/padjAPAConflictControlDG.csv", row.names = F)
```




```{r ConsistentOnly}
countData <- read.csv("../data/02a_countData.csv", header = T, check.names = F, row.names = 1)
colData <- read.csv("../data/02a_colData.csv", header = T)

colData <- colData %>% 
  filter(APA == "Consistent") 
savecols <- as.character(colData$RNAseqID) 
savecols <- as.vector(savecols) 
countData <- countData %>% dplyr::select(one_of(savecols)) 

dds <- DESeqDataSetFromMatrix(countData = countData,
                              colData = colData,
                              design = ~ Punch )

dds <- dds[ rowSums(counts(dds)) > 1, ] 
dds # dim: 16435 9    
dds <- DESeq(dds)
dds$APA <- factor(dds$Punch, levels=c("DG", "CA3", "CA1"))
rld <- rlog(dds, blind=FALSE)

contrast1 <- resvals(contrastvector = c("Punch", "CA1", "DG"), mypval = 0.05) #1683
contrast2 <- resvals(contrastvector = c("Punch", "CA1", "CA3"), mypval = 0.05) #1212
contrast3 <- resvals(contrastvector = c("Punch", "CA3", "DG"), mypval = 0.05) #2107


res <- results(dds, contrast =c("Punch", "CA1", "DG"), independentFiltering = F)
with(res, plot(log2FoldChange, -log10(pvalue), pch=20, main="Consistent: DG - CA1", xlim=c(-8,8), ylim=c(0,100)))
with(subset(res, log2FoldChange>0), points(log2FoldChange, -log10(pvalue), pch=20, col=c("#7570b3")))
with(subset(res, log2FoldChange<0), points(log2FoldChange, -log10(pvalue), pch=20, col=c("#d95f02")))
with(subset(res, padj>.05 ), points(log2FoldChange, -log10(pvalue), pch=20, col="grey"))
resOrdered <- res[order(res$padj),]
head(resOrdered)


res <- results(dds, contrast =c("Punch", "CA1", "CA3"), independentFiltering = F)
with(res, plot(log2FoldChange, -log10(pvalue), pch=20, main="Consistent: CA3 - CA1", xlim=c(-8,8), ylim=c(0,100)))
with(subset(res, log2FoldChange>0), points(log2FoldChange, -log10(pvalue), pch=20, col=c("#7570b3")))
with(subset(res, log2FoldChange<0), points(log2FoldChange, -log10(pvalue), pch=20, col=c("#1b9e77")))
with(subset(res, padj>.05 ), points(log2FoldChange, -log10(pvalue), pch=20, col="grey"))
resOrdered <- res[order(res$padj),]
head(resOrdered)

res <- results(dds, contrast =c("Punch", "CA3", "DG"), independentFiltering = F)
with(res, plot(log2FoldChange, -log10(pvalue), pch=20, main="Consistent: DG - CA3", xlim=c(-8,8), ylim=c(0,100)))
with(subset(res, log2FoldChange>0), points(log2FoldChange, -log10(pvalue), pch=20, col=c("#1b9e77")))
with(subset(res, log2FoldChange<0), points(log2FoldChange, -log10(pvalue), pch=20, col=c("#d95f02")))
with(subset(res, padj>.05 ), points(log2FoldChange, -log10(pvalue), pch=20, col="grey"))
resOrdered <- res[order(res$padj),]
head(resOrdered)

```


```{r ConflictOnly}
countData <- read.csv("../data/02a_countData.csv", header = T, check.names = F, row.names = 1)
colData <- read.csv("../data/02a_colData.csv", header = T)

colData <- colData %>% 
  filter(APA == "Conflict") 
savecols <- as.character(colData$RNAseqID) 
savecols <- as.vector(savecols) 
countData <- countData %>% dplyr::select(one_of(savecols)) 

dds <- DESeqDataSetFromMatrix(countData = countData,
                              colData = colData,
                              design = ~ Punch )

dds <- dds[ rowSums(counts(dds)) > 1, ] 
dds # dim: 16576 14    
dds <- DESeq(dds)
dds$APA <- factor(dds$Punch, levels=c("DG", "CA3", "CA1"))
rld <- rlog(dds, blind=FALSE)

contrast1 <- resvals(contrastvector = c("Punch", "CA1", "DG"), mypval = 0.05) #2526
contrast2 <- resvals(contrastvector = c("Punch", "CA1", "CA3"), mypval = 0.05) #1318
contrast3 <- resvals(contrastvector = c("Punch", "CA3", "DG"), mypval = 0.05) #3776


res <- results(dds, contrast =c("Punch", "CA1", "DG"), independentFiltering = F)
with(res, plot(log2FoldChange, -log10(pvalue), pch=20, main="Conflict: DG - CA1", xlim=c(-8,8), ylim=c(0,100)))
with(subset(res, log2FoldChange>0), points(log2FoldChange, -log10(pvalue), pch=20, col=c("#7570b3")))
with(subset(res, log2FoldChange<0), points(log2FoldChange, -log10(pvalue), pch=20, col=c("#d95f02")))
with(subset(res, padj>.05 ), points(log2FoldChange, -log10(pvalue), pch=20, col="grey"))
resOrdered <- res[order(res$padj),]
head(resOrdered)

res <- results(dds, contrast =c("Punch", "CA1", "CA3"), independentFiltering = F)
with(res, plot(log2FoldChange, -log10(pvalue), pch=20, main="Conflict: CA3 - CA1", xlim=c(-8,8), ylim=c(0,100)))
with(subset(res, log2FoldChange>0), points(log2FoldChange, -log10(pvalue), pch=20, col=c("#7570b3")))
with(subset(res, log2FoldChange<0), points(log2FoldChange, -log10(pvalue), pch=20, col=c("#1b9e77")))
with(subset(res, padj>.05 ), points(log2FoldChange, -log10(pvalue), pch=20, col="grey"))
resOrdered <- res[order(res$padj),]
head(resOrdered)

res <- results(dds, contrast =c("Punch", "CA3", "DG"), independentFiltering = F)
with(res, plot(log2FoldChange, -log10(pvalue), pch=20, main="Conflict: DG - CA3", xlim=c(-8,8), ylim=c(0,100)))
with(subset(res, log2FoldChange>0), points(log2FoldChange, -log10(pvalue), pch=20, col=c("#1b9e77")))
with(subset(res, log2FoldChange<0), points(log2FoldChange, -log10(pvalue), pch=20, col=c("#d95f02")))
with(subset(res, padj>.05 ), points(log2FoldChange, -log10(pvalue), pch=20, col="grey"))
resOrdered <- res[order(res$padj),]
head(resOrdered)
```

```{r ControlOnly}
countData <- read.csv("../data/02a_countData.csv", header = T, check.names = F, row.names = 1)
colData <- read.csv("../data/02a_colData.csv", header = T)

colData <- colData %>% 
  filter(APA == "Control") 
savecols <- as.character(colData$RNAseqID) 
savecols <- as.vector(savecols) 
countData <- countData %>% dplyr::select(one_of(savecols)) 

dds <- DESeqDataSetFromMatrix(countData = countData,
                              colData = colData,
                              design = ~ Punch )

dds <- dds[ rowSums(counts(dds)) > 1, ] 
dds # dim: 16716 21     
dds <- DESeq(dds)
dds$APA <- factor(dds$Punch, levels=c("DG", "CA3", "CA1"))

rld <- rlog(dds, blind=FALSE)
contrast1 <- resvals(contrastvector = c("Punch", "CA1", "DG"), mypval = 0.05) #2013
contrast2 <- resvals(contrastvector = c("Punch", "CA1", "CA3"), mypval = 0.05) #1391
contrast3 <- resvals(contrastvector = c("Punch", "CA3", "DG"), mypval = 0.05) #2728

res <- results(dds, contrast =c("Punch", "CA1", "DG"), independentFiltering = F)
with(res, plot(log2FoldChange, -log10(pvalue), pch=20, main="Control: DG - CA1", xlim=c(-8,8), ylim=c(0,100)))
with(subset(res, log2FoldChange>0), points(log2FoldChange, -log10(pvalue), pch=20, col=c("#7570b3")))
with(subset(res, log2FoldChange<0), points(log2FoldChange, -log10(pvalue), pch=20, col=c("#d95f02")))
with(subset(res, padj>.05 ), points(log2FoldChange, -log10(pvalue), pch=20, col="grey"))
resOrdered <- res[order(res$padj),]
head(resOrdered)


data <- data.frame(gene = row.names(res), pvalue = -log10(res$padj), lfc = res$log2FoldChange)
data <- na.omit(data)
head(data)

data <- data %>%
  mutate(color = ifelse(data$lfc > 0 & data$pvalue > 1.3, 
                        yes = "CA1", 
                        no = ifelse(data$lfc < 0 & data$pvalue > 1.3, 
                                    yes = "DG", 
                                    no = "none")))

# Color corresponds to fold change directionality
colored <- ggplot(data, aes(x = lfc, y = pvalue)) + 
  geom_point(aes(color = factor(color)), size = 1.75, alpha = 0.8, na.rm = T) + # add gene points
  theme_bw(base_size = 16) + # clean up theme
  theme(legend.position = "none") + # remove legend 
  #ggtitle(label = "Volcano Plot", subtitle = "Colored by directionality") +  # add title
  #xlab(expression(log[2]("Consistent" / "Control"))) + # x-axis label
  #ylab(expression(-log[10]("adjusted p-value"))) + # y-axis label
  geom_vline(xintercept = 0, colour = "black") + # add line at 0
  geom_hline(yintercept = 1.3, colour = "black") + # p(0.05) = 1.3
  xlim(c(-8, 8)) +  
  ylim(c(0, 95)) +   
  scale_color_manual(values = c("CA1" = "#7570b3",
                                "DG" = "#d95f02", 
                                "none" = "#bdbdbd")) + theme(panel.grid.minor=element_blank(),
           panel.grid.major=element_blank()) + 
  theme(axis.title.x = element_blank())+ 
  theme(axis.title.y = element_blank())

colored

pdf(file="../figures/02_RNAseq/ControlDGCA1.pdf", width=3, height=3)
plot(colored)
dev.off()



res <- results(dds, contrast =c("Punch", "CA1", "CA3"), independentFiltering = F)
with(res, plot(log2FoldChange, -log10(pvalue), pch=20, main="Control: CA3 - CA1", xlim=c(-8,8), ylim=c(0,100)))
with(subset(res, log2FoldChange>0), points(log2FoldChange, -log10(pvalue), pch=20, col=c("#7570b3")))
with(subset(res, log2FoldChange<0), points(log2FoldChange, -log10(pvalue), pch=20, col=c("#1b9e77")))
with(subset(res, padj>.05 ), points(log2FoldChange, -log10(pvalue), pch=20, col="grey"))
resOrdered <- res[order(res$padj),]
head(resOrdered)

res <- results(dds, contrast =c("Punch", "CA3", "DG"), independentFiltering = F)
with(res, plot(log2FoldChange, -log10(pvalue), pch=20, main="Control: DG - CA3", xlim=c(-8,8), ylim=c(0,100)))
with(subset(res, log2FoldChange>0), points(log2FoldChange, -log10(pvalue), pch=20, col=c("#1b9e77")))
with(subset(res, log2FoldChange<0), points(log2FoldChange, -log10(pvalue), pch=20, col=c("#d95f02")))
with(subset(res, padj>.05 ), points(log2FoldChange, -log10(pvalue), pch=20, col="grey"))
resOrdered <- res[order(res$padj),]
head(resOrdered)

data <- data.frame(gene = row.names(res), pvalue = -log10(res$padj), lfc = res$log2FoldChange)
data <- na.omit(data)
head(data)

data <- data %>%
  mutate(color = ifelse(data$lfc > 0 & data$pvalue > 1.3, 
                        yes = "CA3", 
                        no = ifelse(data$lfc < 0 & data$pvalue > 1.3, 
                                    yes = "DG", 
                                    no = "none")))

# Color corresponds to fold change directionality
colored <- ggplot(data, aes(x = lfc, y = pvalue)) + 
  geom_point(aes(color = factor(color)), size = 1.75, alpha = 0.8, na.rm = T) + # add gene points
  theme_bw(base_size = 16) + # clean up theme
  theme(legend.position = "none") + # remove legend 
  #ggtitle(label = "Volcano Plot", subtitle = "Colored by directionality") +  # add title
  #xlab(expression(log[2]("Consistent" / "Control"))) + # x-axis label
  #ylab(expression(-log[10]("adjusted p-value"))) + # y-axis label
  geom_vline(xintercept = 0, colour = "black") + # add line at 0
  geom_hline(yintercept = 1.3, colour = "black") + # p(0.05) = 1.3
  xlim(c(-8, 8)) +  
  ylim(c(0, 95)) +  
  scale_color_manual(values = c("CA3" = "#1b9e77",
                                "DG" = "#d95f02", 
                                "none" = "#bdbdbd")) + theme(panel.grid.minor=element_blank(),
           panel.grid.major=element_blank()) + 
  theme(axis.title.x = element_blank())+ 
  theme(axis.title.y = element_blank())

colored

pdf(file="../figures/02_RNAseq/ControlDGCA3.pdf", width=3, height=3)
plot(colored)
dev.off()

```


