---
title: "02d_rnaseqAvoidance"
author: "Rayna M Harris"
date: "8/22/17"
output: md_document
---

```{r setup, message=F}
library(ggplot2) ## for awesome plots!
library(cowplot) ## for some easy to use themes
library(dplyr) ## for filtering and selecting rows
library(car) ## stats
library(VennDiagram) ## venn diagrams
library(pheatmap) ## awesome heatmaps
library(viridis) # for awesome color pallette
library(reshape2) ## for melting dataframe
library(DESeq2) ## for gene expression analysis
library(edgeR)  ## for basic read counts status
library(magrittr) ## to use the weird pipe
library(genefilter)  ## for PCA fuction

## load functions 
source("figureoptions.R")
source("functions_RNAseq.R")

## set output file for figures 
knitr::opts_chunk$set(fig.path = '../figures/02d_rnaseqavoidance/')
```

## DG

```{r DG, message=F, warning=F}
colData <- read.csv("../data/02a_colData.csv", header = T)
countData <- read.csv("../data/02a_countData.csv", header = T, check.names = F, row.names = 1)

colData$avoidance <-  ifelse(grepl("Control", colData$APA), "no", "yes")

colData <- colData %>% 
  filter(Punch %in% c("DG"))  %>% 
  droplevels()

savecols <- as.character(colData$RNAseqID) 
savecols <- as.vector(savecols) 
countData <- countData %>% dplyr::select(one_of(savecols)) 

colData %>% select(avoidance,Punch)  %>%  summary()


## create DESeq object using the factors Punch and APA
dds <- DESeqDataSetFromMatrix(countData = countData,
                              colData = colData,
                              design = ~ avoidance)

dds # view the DESeq object - note numnber of genes
dds <- dds[ rowSums(counts(dds)) > 1, ]  # Pre-filtering genes with 0 counts
dds # view number of genes afternormalization and the number of samples
dds <- DESeq(dds) # Differential expression analysis
rld <- rlog(dds, blind=FALSE) ## log transformed data


# create the dataframe using my function pcadataframe
pcadata <- pcadataframe(rld, intgroup=c("Punch","avoidance"), returnData=TRUE)
percentVar <- round(100 * attr(pcadata, "percentVar"))
percentVar

summary(aov(PC1 ~ avoidance, data=pcadata)) 
summary(aov(PC2 ~ avoidance, data=pcadata)) 
summary(aov(PC3 ~ avoidance, data=pcadata)) 
summary(aov(PC4 ~ avoidance, data=pcadata)) 


res <- results(dds, contrast =c("avoidance", "yes", "no"), independentFiltering = T, alpha = 0.1)
summary(res)
resOrdered <- res[order(res$padj),]
head(resOrdered, 10)

topGene <- rownames(res)[which.min(res$padj)]
plotCounts(dds, gene = topGene, intgroup=c("avoidance"))


data <- data.frame(gene = row.names(res),
                   pvalue = -log10(res$padj), 
                   lfc = res$log2FoldChange)
data <- na.omit(data)
data <- data %>%
  mutate(color = ifelse(data$lfc > 0 & data$pvalue > 1, 
                        yes = "yes", 
                        no = ifelse(data$lfc < 0 & data$pvalue > 1, 
                                    yes = "no", 
                                    no = "none")))

# for faceting
pcadata$wrap <- "DG PCA"
data$wrap <- "DG contrasts"

DGvolcano <- ggplot(data, aes(x = lfc, y = pvalue)) + 
  geom_point(aes(color = factor(color)), size = 1, alpha = 0.5, na.rm = T) + # add gene points
  scale_color_manual(values = volcano5)  + 
  xlab(paste0("All Trained / All Yoked")) +
  ylab(paste0("log10(pvalue)")) +
  theme_cowplot(font_size = 8, line_size = 0.25) +
  geom_hline(yintercept = 1,  size = 0.25, linetype = 2 )+ 
  theme(panel.grid.minor=element_blank(),
        legend.position = "none", # remove legend 
        panel.grid.major=element_blank()) +
  facet_wrap(~wrap)
DGvolcano

pdf(file="../figures/02d_rnaseqavoidance/DGvolcano.pdf", width=1.5, height=2)
plot(DGvolcano)
dev.off()


PCA12 <- plotPCwrap(pcadata, 1, 2, aescolor = pcadata$avoidance, colorname = " ",  colorvalues = colorvalavoidance)
PCA12

pdf(file="../figures/02d_rnaseqavoidance/DGpca12.pdf", width=1.75, height = 2)
plot(PCA12)
dev.off()
```


## CA3

```{r CA3, message=F, warning=F}
colData <- read.csv("../data/02a_colData.csv", header = T)
countData <- read.csv("../data/02a_countData.csv", header = T, check.names = F, row.names = 1)

colData$avoidance <-  ifelse(grepl("Control", colData$APA), "no", "yes")

colData <- colData %>% 
  filter(Punch %in% c("CA3"))  %>% 
  droplevels()
savecols <- as.character(colData$RNAseqID) 
savecols <- as.vector(savecols) 
countData <- countData %>% dplyr::select(one_of(savecols)) 
colData %>% select(avoidance,Punch)  %>%  summary()

## create DESeq object using the factors Punch and APA
dds <- DESeqDataSetFromMatrix(countData = countData,
                              colData = colData,
                              design = ~ avoidance)

dds # view the DESeq object - note numnber of genes
dds <- dds[ rowSums(counts(dds)) > 1, ]  # Pre-filtering genes with 0 counts
dds # view number of genes afternormalization and the number of samples
dds <- DESeq(dds) # Differential expression analysis
rld <- rlog(dds, blind=FALSE) ## log transformed data

# create the dataframe using my function pcadataframe
pcadata <- pcadataframe(rld, intgroup=c("Punch","avoidance"), returnData=TRUE)
percentVar <- round(100 * attr(pcadata, "percentVar"))
percentVar

summary(aov(PC1 ~ avoidance, data=pcadata))
summary(aov(PC2 ~ avoidance, data=pcadata)) 
summary(aov(PC3 ~ avoidance, data=pcadata)) 
summary(aov(PC4 ~ avoidance, data=pcadata))
summary(aov(PC4 ~ avoidance, data=pcadata)) 
summary(aov(PC5 ~ avoidance, data=pcadata))
summary(aov(PC6 ~ avoidance, data=pcadata))


#calculate significance of all two way comparisions
# see source "functions_RNAseq.R" 
contrast1 <- resvals(contrastvector = c("avoidance", "yes", "no"), mypval = 0.1) # 0

```


## CA1

```{r CA1, message=F, warning=F}
colData <- read.csv("../data/02a_colData.csv", header = T)
countData <- read.csv("../data/02a_countData.csv", header = T, check.names = F, row.names = 1)

colData$avoidance <-  ifelse(grepl("Control", colData$APA), "no", "yes")

colData <- colData %>% 
  filter(Punch %in% c("CA1"))  %>% 
  droplevels()
savecols <- as.character(colData$RNAseqID) 
savecols <- as.vector(savecols) 
countData <- countData %>% dplyr::select(one_of(savecols)) 
colData %>% select(avoidance,Punch)  %>%  summary()
dds <- DESeqDataSetFromMatrix(countData = countData,
                              colData = colData,
                              design = ~ avoidance)

dds # view the DESeq object - note numnber of genes
dds <- dds[ rowSums(counts(dds)) > 1, ]  # Pre-filtering genes with 0 counts
dds # view number of genes afternormalization and the number of samples
dds <- DESeq(dds) # Differential expression analysis
rld <- rlog(dds, blind=FALSE) ## log transformed data

pcadata <- pcadataframe(rld, intgroup=c("Punch","avoidance"), returnData=TRUE)
percentVar <- round(100 * attr(pcadata, "percentVar"))
percentVar
summary(aov(PC1 ~ avoidance, data=pcadata)) 
summary(aov(PC2 ~ avoidance, data=pcadata)) 
summary(aov(PC3 ~ avoidance, data=pcadata))
summary(aov(PC4 ~ avoidance, data=pcadata))


# calculate significance of all two way comparisions
contrast1 <- resvals(contrastvector = c("avoidance", "yes", "no"), mypval = 0.1) # 9

res <- results(dds, contrast =c("avoidance", "yes", "no"), independentFiltering = T, alpha = 0.1)
summary(res)
resOrdered <- res[order(res$padj),]
head(resOrdered, 10)
topGene <- rownames(res)[which.min(res$padj)]
plotCounts(dds, gene = topGene, intgroup=c("avoidance"))

data <- data.frame(gene = row.names(res),
                   pvalue = -log10(res$padj), 
                   lfc = res$log2FoldChange)
data <- na.omit(data)

# for faceting
pcadata$wrap <- "CA1 PCA"
data$wrap <- "CA1 contrasts"


data <- data %>%
  mutate(color = ifelse(data$lfc > 0 & data$pvalue > 1, 
                        yes = "yes", 
                        no = ifelse(data$lfc < 0 & data$pvalue > 1, 
                                    yes = "no", 
                                    no = "none")))
top_labelled <- top_n(data, n = 5, wt = lfc)

# Color corresponds to fold change directionality
CA1volcano <- ggplot(data, aes(x = lfc, y = pvalue)) + 
  geom_point(aes(color = factor(color)), size = 1, alpha = 0.5, na.rm = T) + # add gene points
  scale_color_manual(values = volcano5)  + 
  xlab(paste0("All Trained / All Yoked")) +
  ylab(paste0("log10(pvalue)")) +
  theme_cowplot(font_size = 8, line_size = 0.25) +
  geom_hline(yintercept = 1,  size = 0.25, linetype = 2 )+ 
  theme(panel.grid.minor=element_blank(),
        legend.position = "none", # remove legend 
        panel.grid.major=element_blank()) +
  facet_wrap(~wrap)
CA1volcano


pdf(file="../figures/02d_rnaseqavoidance/CA1volcano.pdf",  width=1.5, height=2)
plot(CA1volcano)
dev.off()


PCA12 <- plotPCwrap(pcadata, 1, 2, aescolor = pcadata$avoidance, colorname = " ",  colorvalues = colorvalavoidance)
PCA12

pdf(file="../figures/02d_rnaseqavoidance/CA1pca12.pdf", width=1.75, height = 2)
plot(PCA12)
dev.off()
 
```

## All

```{r}
colData <- read.csv("../data/02a_colData.csv", header = T)
countData <- read.csv("../data/02a_countData.csv", header = T, check.names = F, row.names = 1)

colData$avoidance <-  ifelse(grepl("Control", colData$APA), "no", "yes")
colData$avoidance <- as.factor(colData$avoidance )
colData %>% select(avoidance,Punch)  %>%  summary()


## create DESeq object using the factors Punch and APA
dds <- DESeqDataSetFromMatrix(countData = countData,
                              colData = colData,
                              design = ~ avoidance*Punch)

dds # view the DESeq object - note numnber of genes
dds <- dds[ rowSums(counts(dds)) > 1, ]  # Pre-filtering genes with 0 counts
dds # view number of genes afternormalization and the number of samples
dds <- DESeq(dds) # Differential expression analysis
rld <- rlog(dds, blind=FALSE) ## log transformed data

# create the dataframe using my function pcadataframe
pcadata <- pcadataframe(rld, intgroup=c("Punch","avoidance"), returnData=TRUE)
pcadata$wrap <- "Principle Compent Analysis"
percentVar <- round(100 * attr(pcadata, "percentVar"))
percentVar

pcadata$PunchAPA <- as.factor(paste(pcadata$Punch, pcadata$APA2, sep="_"))
pcadata$Punch <- factor(pcadata$Punch, levels=c("DG","CA3", "CA1"))
pcadata$APA2 <- factor(pcadata$avoidance, levels=c("no","yes"))

summary(aov(PC1 ~ avoidance, data=pcadata)) 
summary(aov(PC2 ~ avoidance, data=pcadata)) 
summary(aov(PC3 ~ avoidance, data=pcadata)) 
summary(aov(PC4 ~ avoidance, data=pcadata)) 
summary(aov(PC5 ~ avoidance, data=pcadata)) 
summary(aov(PC6 ~ avoidance, data=pcadata)) 

summary(aov(PC1 ~ Punch, data=pcadata)) 
summary(aov(PC2 ~ Punch, data=pcadata)) 
summary(aov(PC3 ~ Punch, data=pcadata)) 
summary(aov(PC4 ~ Punch, data=pcadata)) 
summary(aov(PC5 ~ Punch, data=pcadata)) 
summary(aov(PC6 ~ Punch, data=pcadata)) 

TukeyHSD((aov(PC1 ~ Punch, data=pcadata)), which = "Punch") 
TukeyHSD((aov(PC2 ~ Punch, data=pcadata)), which = "Punch") 

res <- results(dds, contrast =c("avoidance", "yes", "no"), independentFiltering = T, alpha = 0.1)
summary(res)
resOrdered <- res[order(res$padj),]
head(resOrdered, 10)

topGene <- rownames(res)[which.min(res$padj)]
plotCounts(dds, gene = topGene, intgroup=c("avoidance"))

data <- data.frame(gene = row.names(res),
                   pvalue = -log10(res$padj), 
                   lfc = res$log2FoldChange)
data <- na.omit(data)
data <- data %>%
  mutate(color = ifelse(data$lfc > 0 & data$pvalue > 1, 
                        yes = "yes", 
                        no = ifelse(data$lfc < 0 & data$pvalue > 1, 
                                    yes = "no", 
                                    no = "none")))
top_labelled <- top_n(data, n = 5, wt = lfc)

contrast1 <- resvals(contrastvector = c("Punch", "CA1", "DG"), mypval = 0.05) # 2556
contrast2 <- resvals(contrastvector = c("Punch", "CA1", "CA3"), mypval = 0.05) # 1851
contrast3 <- resvals(contrastvector = c("Punch", "CA3", "DG"), mypval = 0.05) # 23541
contrast4 <- resvals(contrastvector = c("avoidance", "yes", "no"), mypval = 0.1) # 8

DEGes <- assay(rld)
DEGes <- cbind(DEGes, contrast1, contrast2, contrast3, contrast4)
DEGes <- as.data.frame(DEGes) # convert matrix to dataframe
DEGes$rownames <- rownames(DEGes)  # add the rownames to the dataframe
DEGes$padjmin <- with(DEGes, pmin(padjPunchCA1DG, padjPunchCA1CA3, padjPunchCA3DG, padjavoidanceyesno)) 
DEGes <- DEGes %>% filter(padjmin < 0.01)
rownames(DEGes) <- DEGes$rownames
drop.cols <-colnames(DEGes[,grep("padj|pval|rownames", colnames(DEGes))])
DEGes <- DEGes %>% dplyr::select(-one_of(drop.cols))
DEGes <- as.matrix(DEGes)
DEGes <- DEGes - rowMeans(DEGes)
dim(DEGes)

df <- as.data.frame(colData(dds)[,c("avoidance", "Punch")]) ## matrix to df
rownames(df) <- names(countData)
ann_colors <- ann_colors8 # see color options 
DEGes <- as.matrix(DEGes) 
paletteLength <- 30
myBreaks <- c(seq(min(DEGes), 0, length.out=ceiling(paletteLength/2) + 1), 
              seq(max(DEGes)/paletteLength, max(DEGes), length.out=floor(paletteLength/2)))

pheatmap(DEGes, show_colnames=T, show_rownames = F,
         annotation_col=df, annotation_colors = ann_colors,
         treeheight_row = 0, treeheight_col = 25,
         annotation_row = NA, 
         annotation_legend = FALSE,
         annotation_names_row = FALSE, annotation_names_col = FALSE,
         fontsize = 8, 
         border_color = "grey60" ,
         color = viridis(30),
         cellwidth = 6, 
         clustering_method="average",
         breaks=myBreaks,
         clustering_distance_cols="correlation" 
         )

pheatmap(DEGes, show_colnames=F, show_rownames = F,
         annotation_col=df, annotation_colors = ann_colors, 
         annotation_row = NA, 
         annotation_legend = FALSE,
         annotation_names_row = FALSE, annotation_names_col = FALSE,
         treeheight_row = 0, treeheight_col = 25,
         fontsize = 8, 
         border_color = "grey60" ,
         color = viridis(30),
         height = 3, 
         width = 3,
         clustering_method="average",
         breaks=myBreaks,
         clustering_distance_cols="correlation", 
         filename = "../figures/02d_rnaseqavoidance/pheatmap1.pdf"
         )


PCA34 <- plotPCwrap(pcadata, 3, 4, aescolor = pcadata$avoidance, colorname = " ", colorvalues = colorvalavoidance)
PCA34

pdf(file="../figures/02d_rnaseqavoidance/allPCA34.pdf", width=1.5, height=1.5)
plot(PCA34)
dev.off()
```

## venn diagrams

```{r venndiagrams, echo=F}
rldpadjs <- assay(rld)
rldpadjs <- cbind(rldpadjs, contrast1, contrast2, contrast3, contrast4)
rldpadjs <- as.data.frame(rldpadjs)
rldpadjs <- rldpadjs[ , grepl( "padj" , names( rldpadjs ) ) ]

venn1 <- row.names(rldpadjs[rldpadjs[1] <0.1 & !is.na(rldpadjs[1]),]) # CA1 DG
venn2 <- row.names(rldpadjs[rldpadjs[2] <0.1 & !is.na(rldpadjs[2]),]) # CA1 CA3
venn3 <- row.names(rldpadjs[rldpadjs[3] <0.1 & !is.na(rldpadjs[3]),]) # CA3 DG
venn4 <- row.names(rldpadjs[rldpadjs[4] <0.1 & !is.na(rldpadjs[3]),]) # yes no

## check order for correctness
candidates <- list("CA1DG" = venn1,  "avoidance" = venn4,  "CA3DG" = venn3, "CA1CA3" = venn2)

prettyvenn <- venn.diagram(
  scaled=T,
  x = candidates, filename=NULL, 
  col = "black",
  #fill = c( "white", "white"),
  alpha = 0.5,
  cex = 1, fontfamily = "sans", #fontface = "bold",
  cat.default.pos = "text",
  #cat.dist = c(0.08, 0.08), cat.pos = 1,
  cat.cex = 1, cat.fontfamily = "sans")
#dev.off()
grid.draw(prettyvenn)
```


```{r pcadf}
write.csv(pcadata, file = "../data/02a_pcadata.csv", row.names = T)
```
