---
title: "Qualtity Control of RNAseq Data"
output: md_document
---


```{r setup, message=F, warning=F}
library(tidyverse)
library(cowplot) ## for some easy to use themes
library(DESeq2) ## for gene expression analysis
library(Rtsne) # for tSNE

library(BiocParallel)
register(MulticoreParam(6))

## load functions 
source("figureoptions.R")
source("functions_RNAseq.R")

## set output file for figures 
knitr::opts_chunk$set(fig.path = '../figures/02_rnaseqQC/', cache= T)
```

# Quality control

## MultiQC


```{r multiqc, fig.width = 6.69, fig.height = 2.4}
# read meta data for plotting
colData <- read.csv("../data/00_colData.csv")
sampletreatment <- colData %>% select(RNAseqID, treatment)  

# stats from fastqc 
fastqc <- read.csv(file = "../data/multiqc/multiqc_fastqc.csv")
fastqc$read <- ifelse(grepl("R1", fastqc$Sample.Name), "R1", "R2")  # make read columns
fastqc$mouse <- sapply(strsplit(as.character(fastqc$Sample.Name),"\\_"), "[", 1)
fastqc$subfield <- sapply(strsplit(as.character(fastqc$Sample.Name),"\\_"), "[", 2)
fastqc$section <- sapply(strsplit(as.character(fastqc$Sample.Name),"\\_"), "[", 3)
fastqc$RNAseqID <- paste(fastqc$mouse, fastqc$subfield, fastqc$section, sep = "-")
fastqc <- left_join(sampletreatment,fastqc) %>% 
  select(RNAseqID, treatment, subfield, read, QualityFiltered, Dups, GC, Length, MillionReads)  %>% 
  filter(QualityFiltered == "No")

fastqc$treatment <- factor(fastqc$treatment, levels = levelstreatment)
fastqc$subfield <- factor(fastqc$subfield, levels = levelssubfield)
head(fastqc)

# stats from kallisto
kallisto <- read.csv(file = "../data/multiqc/multiqc_kallisto.csv")
kallisto$mouse <- sapply(strsplit(as.character(kallisto$sample),"\\_"), "[", 1)
kallisto$subfield <- sapply(strsplit(as.character(kallisto$sample),"\\_"), "[", 2)
kallisto$section <- sapply(strsplit(as.character(kallisto$sample),"\\_"), "[", 3)
kallisto$RNAseqID <- paste(kallisto$mouse, kallisto$subfield, kallisto$section, sep = "-")
kallisto <- left_join(sampletreatment,kallisto) %>% 
  select(RNAseqID, treatment, subfield, QC, bp, fracalign, millalign) %>% 
  filter(QC == "raw")
kallisto$treatment <- factor(kallisto$treatment, levels = levelstreatment)
kallisto$subfield <- factor(kallisto$subfield, levels = levelssubfield)
head(kallisto)

multiqc <- left_join(fastqc, kallisto) 

summary(multiqc)

mean(multiqc$MillionReads)
sd(multiqc$MillionReads)

mean(multiqc$millalign)
sd(multiqc$millalign)

mean(multiqc$fracalign)
sd(multiqc$fracalign)

summary(aov(MillionReads ~ treatment, multiqc))
summary(aov(millalign ~ treatment, multiqc))
summary(aov(fracalign ~ treatment, multiqc))


a <- ggplot(multiqc, aes(x = subfield, y = MillionReads, color = subfield)) +
  geom_boxplot(outlier.size = 0.75) + geom_point(size = 0.75) +
  scale_color_manual(values = colorvalsubfield) +  
  #facet_wrap(~subfield, nrow = 3) +
  labs(y = "Total Reads (millions)", x = NULL, subtitle = " ") +
  theme_ms() + theme(legend.position = "none")


b <- ggplot(multiqc, aes(x = subfield, y = millalign, color = subfield)) +
  geom_boxplot() + geom_point(size = 0.5) +
  scale_color_manual(values = colorvalsubfield) +  
  labs(y = "Aligned Reads (millions)", x = NULL, subtitle = " ") +
  theme_ms() + theme(legend.position = "none")

c <- ggplot(multiqc, aes(x = subfield, y = fracalign, color = subfield)) +
         geom_boxplot() + geom_point(size = 0.5) +
  scale_color_manual(values = colorvalsubfield) +  
  labs(y = "Fraction aligned (millions)", x = NULL, subtitle = " ") +
  theme_ms() + theme(legend.position = "none")


multiqcplots <- plot_grid( a, b , c , 
          labels = c( "(a)", "(b)", "(c)"),
          label_size = 8, nrow = 1)
multiqcplots

summary(aov(MillionReads ~ subfield, multiqc))
summary(aov(millalign ~ subfield, multiqc))
summary(aov(fracalign ~ subfield, multiqc))
```

## DESeq2

The two two catagorical variables are

* Hippocampal subfield: DG, CA3, CA1
* Treatment: standard yoked, standard trained, conflict yoked, conflict trained

```{r data, message=F, warning=F}
colData <- read.csv("../data/00_colData.csv", header = T)
countData <- read.csv("../data/00_countData.csv", header = T, check.names = F, row.names = 1)
colData %>% select(treatment, subfield)  %>%  summary()

dds <- DESeqDataSetFromMatrix(countData = countData,
                              colData = colData,
                              design = ~ subfield * treatment )

dds$subfield <- factor(dds$subfield, levels=c("DG","CA3", "CA1")) ## specify the factor levels

dds$treatment <- factor(dds$treatment, levels=c("standard.yoked" ,"standard.trained", "conflict.yoked", "conflict.trained")) ## specify the factor levels

dds # view the DESeq object - note numnber of genes
dds <- dds[ rowSums(counts(dds)) > 10, ]  # Pre-filtering genes
dds # view number of genes afternormalization and the number of samples
dds <- DESeq(dds, parallel = TRUE) # Differential expression analysis
#rld <- rlog(dds, blind=FALSE) ## log transformed data
vsd <- vst(dds, blind=FALSE) ## variance stabilized
head(assay(vsd),3)
```


### Summary DEGs

This first function shows the total number of up and down regulated genes and the top 3 most significant genes.

```{r twowaycontrasts}
res_summary(c("subfield", "CA1", "DG"))
res_summary(c("subfield", "CA1", "CA3"))
res_summary(c("subfield", "CA3", "DG"))

res_summary(c("treatment", "standard.trained", "standard.yoked"))
res_summary(c("treatment", "conflict.trained", "conflict.yoked"))
res_summary(c("treatment", "conflict.trained", "standard.trained"))
res_summary(c("treatment", "conflict.yoked", "standard.yoked"))
```


## Principle component analysis

```{r pca}
# create the dataframe using my function pcadataframe
pcadata <- pcadataframe(vsd, intgroup=c("subfield","treatment"), returnData=TRUE)
percentVar <- round(100 * attr(pcadata, "percentVar"))
pcadata$subfieldAPA <- as.factor(paste(pcadata$subfield, pcadata$treatment, sep="_"))
pcadata$subfield <- factor(pcadata$subfield, levels=c("DG","CA3", "CA1"))
pcadata$treatment <- factor(pcadata$treatment, levels=c("standard.yoked","standard.trained",  "conflict.yoked","conflict.trained"))

levels(pcadata$treatment) <- c("standard yoked","standard trained",  "conflict yoked", "conflict trained")

summary(aov(PC1 ~ subfield * treatment, data=pcadata)) 
TukeyHSD((aov(PC1 ~ subfield, data=pcadata)), which = "subfield") 

summary(aov(PC2 ~ subfield * treatment, data=pcadata)) 
TukeyHSD((aov(PC2 ~ subfield, data=pcadata)), which = "subfield") 

subfields <- png::readPNG("../figures/00_schematics/figure_hippocampus.png")
subfields <-  grid::rasterGrob(subfields, interpolate=TRUE)


PCA12 <- ggplot(pcadata, aes(pcadata$PC1, pcadata$PC2, colour=subfield)) +
    geom_point(size=2, aes(shape=treatment), alpha = 0.8) +
    xlab(paste0("PC1: ", percentVar[1],"%")) +
    ylab(paste0("PC2: ", percentVar[2],"%")) +
    scale_colour_manual(values=c(colorvalsubfield))+ 
   theme_ms()  +
      theme(legend.position= "none") +
    scale_shape_manual(values=c(1, 16, 0, 15), aes(color=colorvalsubfield)) +
  labs(color = "subfield", shape = "treatment", subtitle = " ") +
  guides(color = FALSE) + 
  annotation_custom(subfields, ymin = 5, ymax = 15, xmin = 10) 
PCA12
```

## tSNE

```{r tSNE, fig.width=1, fig.height = 1}
vsddf <- as.data.frame(assay(vsd))
vsddf <- as.data.frame(t(vsddf))

euclidist <- dist(vsddf) # euclidean distances between the rows

mytsneplot <- plot_tSNE(10, " ")
mytsneplot + theme(legend.position = "none",
                   axis.text = element_text(size = 8)) 
```

## pca + tsne

```{r PCAtSNE}
mylegend <- get_legend(mytsneplot)
top <- plot_grid(PCA12, mytsneplot + theme(legend.position = "none"), nrow = 1,
          labels = c( "(d)", "(e)"),
          label_size = 8)
PCAtSNE <- plot_grid(top, mylegend, nrow = 2, rel_heights = c(1,0.1))
PCAtSNE
```



```{r supplfig, fig.width=5.1, fig.height=4}
# 3 box plots, 1 pca plot, 1 tsne plot with hippocampus image
supplfig1 <- plot_grid(multiqcplots, PCAtSNE, nrow = 2, rel_heights = c(1,1.25))
supplfig1

pdf(file="../figures/supplfig1.pdf", width=5.1, height=4)
plot(supplfig1)
dev.off()
```



```{r markergenes}

# import cembrowski markers
# Genes from Cembrowski sublement file 1 <https://elifesciences.org/articles/14997/figures>
cembrowskisupp <- read.table("../data/elife-14997-supp1-v1.txt", sep="\t", header = T)

# select just columns with gene symbol and enriched column
# then rename gene column and convert to uppercase
cembrowskisupp <- cembrowskisupp %>% dplyr::select(gene_short_name, enriched) %>%
  mutate(tissue = fct_collapse(enriched,
                               "DG" = c("dg_d", "dg_d-dg_v"),
                               "CA1" = c("ca1_d", "ca1_d-ca1_v"),
                               "CA3" = c("ca3_d",  "ca3_d-ca3_v-ca2-ca1_d-ca1_v")))
colnames(cembrowskisupp)[1] <- "gene"
cembrowskisupp$gene <- str_to_upper(cembrowskisupp$gene)
head(cembrowskisupp)

# subset my maker

cembrowksimarkers <- function(subfields){
  mylist <- cembrowskisupp %>% 
    dplyr::filter(tissue %in% subfields) %>% 
    pull(gene)
  print(subfields)
  print(mylist)
  return(mylist)
}

# dorsal markers 
DGmarkers <- cembrowksimarkers("DG")
CA1markers <- cembrowksimarkers("CA1")
CA3markers <- cembrowksimarkers("CA3")

## calculate DEGs
returnDEGs <-  function(whichfactor, up, down){
  
  # calculate DEG results
  res <- results(dds, contrast = c(whichfactor, up, down),
                 independentFiltering = T, alpha = 0.1)
  
  # create dataframe with pvalues and lfc
  data <- data.frame(gene = row.names(res),
                     padj = res$padj, 
                     logpadj = -log10(res$padj),
                     lfc = res$log2FoldChange)
  data <- na.omit(data)
  data <- data %>%
    dplyr::mutate(direction = ifelse(data$lfc > 0 & data$padj < 0.1, 
                                     yes = up, no = ifelse(data$lfc < 0 & data$padj < 0.1, 
                                                           yes = down, no = "NS")),
                  gene = str_to_upper(gene)) 
  data$direction <- factor(data$direction, levels = c(down, "NS", up))
  data$comparison <- paste0(down, " vs. ", up, sep = "" )
  data <- data %>% select(gene, lfc, padj, logpadj, comparison, direction)  
  return(data)
}  


CA1DG <- returnDEGs("subfield", "CA1", "DG")
CA1CA3 <- returnDEGs("subfield", "CA1", "CA3")
CA3DG <- returnDEGs("subfield", "CA3", "DG")

allDEGs <- rbind(CA1DG, CA1CA3) %>%
  rbind(., CA3DG)  %>%
  filter(direction != "NS")
head(allDEGs)

DEGsigmarkers <- allDEGs %>%
  filter(gene %in% DGmarkers, comparison %in% c("DG vs. CA1", "DG vs. CA3")) %>%
  pivot_wider(gene, names_from = comparison, values_from = direction)
DEGsigmarkers
58/78

# 58 of 78 (or 74%) of the dorsal DG marker genes enrighted in the cembrowski study are increased in the DG relative the CA1 and/or CA3.  


CA1sigmarkers <- allDEGs %>%
  filter(gene %in% CA1markers, comparison %in% c("DG vs. CA1", "CA3 vs. CA1")) %>%
  pivot_wider(gene, names_from = comparison, values_from = direction)
CA1sigmarkers
17/22


# 17 of 22 (or 77%) of the dorsal CA1 marker genes enrighted in the cembrowski study are increased in the CA1 relative the DG and/or CA3. 


CA3sigmarkers <- allDEGs %>%
  filter(gene %in% CA3markers, comparison %in% c("DG vs. CA3", "CA3 vs. CA1")) %>%
  pivot_wider(gene, names_from = comparison, values_from = direction)
CA3sigmarkers
9/10

markeranalysis <- full_join(DEGsigmarkers, CA1sigmarkers) %>%
  full_join(., CA3sigmarkers) %>%
  arrange(gene)
markeranalysis

write.csv(markeranalysis, "../data/suppltable-7.csv", row.names = F)

# 9 of 10 (or 10%) of the dorsal CA3 marker genes enrighted in the cembrowski study are increased in the CA3 relative the DG and/or CA1. 
```


```{r save}
write.csv(assay(vsd), file = "../data/02_vsd.csv", row.names = T)
```

## Session Info

```{r SessionInfo}
sessionInfo()
```

```{r citations}
citation("tidyverse") 
citation("cowplot")  
citation("DESeq2")  
citation("Rtsne")  
citation("BiocParallel")
```