---
title: "DESeq2 Data Analysis"
author: "Rayna M Harris"
date: "Last updated December 20, 2018"
output: md_document
---

## RNAseq gene expression analysis with DESeq2 

This workflow was modified from the DESeq2 tutorial found at: https://www.bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.pdf

First I load a handful of packages for data wrangling, gene expression analysis, data visualization, and statistics.

```{r setup, message=F, warning=F}
library(dplyr) ## for filtering and selecting rows
library(plyr) ## for renmaing factors
library(reshape2) ## for melting dataframe
library(DESeq2) ## for gene expression analysis
library(edgeR)  ## for basic read counts status
library(magrittr) ## to use the weird pipe
library(genefilter)  ## for PCA fuction
library(ggplot2)

## Functions
source("functions_RNAseq.R")

## set output file for figures 
knitr::opts_chunk$set(fig.path = '../figures/02_RNAseq/')
```

Now, I create data frames from three csv files
- count: Contains counts for all transcripts generated from the program Kallisto. This data can be reproducibed from the file kallisto.Rmd
- geneids: Contains the ensemble ids and gene names for all the transcripts in the counts data frame. This file will be used to convert transcipt counts to gene counts. This file was also created via kallisto.Rmd file
- colData: This file contains all the information I collected for each sample that was sequenced. Not all columns will be needed, so some are removed later. 

```{r getdata, message=F}
countData <- read.csv("../data/00_CountData.csv", row.names=1, check.names=FALSE )
colData <- read.csv("../data/IntegrativeWT2015ColData.csv")
```

In this next section, I tidy the trait data for each sample so that I can calculate differential gene expression for the colData of interest. I also remove some samples for reasons described within the code blocks.

```{r colData}
rownames(colData) <- colData$RNAseqID    # set $genoAPAsessionInd as rownames
colData <- colData[c(1,2,5,7:11)]  #keeping informative volumns
colData <- colData %>% dplyr::filter(!grepl("147-|148-", RNAseqID))  # remove 147, and 148:  homecage animals 
colData$ID <- gsub("[[:punct:]]", "", colData$Mouse) #make a column that thas id without the dash
colData$APA <- NULL ## delete old APA column
names(colData)[names(colData)=="APA_Conflict"] <- "APA3" #rename  APA3 to match color scheme
names(colData)[names(colData)=="Region"] <- "Punch" #rename  region to punch

# rename factors & group all Control animals into 1 group
colData$APA2 <- colData$APA3 
colData$APA2 <- revalue(colData$APA2, c("Trained_Conflict" = "conflict")) 
colData$APA2 <- revalue(colData$APA2, c("Trained_NoConflict" = "consistent")) 
colData$APA2 <- revalue(colData$APA2, c("Yoked_Conflict" = "yoked_conflict")) 
colData$APA2 <- revalue(colData$APA2, c("Yoked_NoConflict" = "yoked_consistent")) 

# reorder 
colData <- colData[c(1:5,7:9)]
```

Now, we are ready to calculate differential gene expression using the DESeq package. For simplicity, I will use the standard nameing of "countData" and "colData" for the gene counts and gene information, respectively.

```{r prepforDESeq2}
colData <- colData %>% arrange(RNAseqID) %>% droplevels() #set the coldata to be the countbygene df

## colData and countData must contain the exact same sample. I'll use the next three lines to make that happen
savecols <- as.character(colData$RNAseqID) #select the sample name column that corresponds to row names
savecols <- as.vector(savecols) # make it a vector
countData <- countData %>% dplyr::select(one_of(savecols)) # select just the columns that match the samples in colData

# colData must be factors
cols = c(1:8)
colData[,cols] %<>% lapply(function(x) as.factor(as.character(x)))

# summary data
colData %>% select(APA2,Punch)  %>%  summary()
dim(countData)
```

## Write the two files

```{r writecsv}
write.csv(colData, file = "../data/02a_colData.csv", row.names = F)
write.csv(countData, file = "../data/02a_countData.csv", row.names = T)
```

## Total Gene Counts Per Sample 
this could say something about data before normalization

```{r totalRNAseqcounts}
## stats
counts <- countData
dim( counts )
colSums( counts ) / 1e06  # in millions of reads
table( rowSums( counts ) )[ 1:30 ] # Number of genes with low counts

rowsum <- as.data.frame(colSums( counts ) / 1e06 )
names(rowsum)[1] <- "millioncounts"
rowsum$sample <- row.names(rowsum)

ggplot(rowsum, aes(x=millioncounts)) + 
  geom_histogram(binwidth = 1, colour = "black", fill = "darkgrey") +
  theme_classic() +
  scale_x_continuous(name = "Millions of Gene Counts per Sample") +
  scale_y_continuous(name = "Number of Samples")
```
