---
title: "02b_KallistoGather"
author: "Rayna M Harris"
date: "March 6, 2017"
output: md_document
---
  
```{r install packages load data, warning=FALSE, message=FALSE}
library("dplyr") ## for filtering and selecting rows
library("plyr")  ## for renmaing factors
library("reshape2") ##  for melting dataframe
knitr::opts_chunk$set(fig.path = '../figures/02a_KallistoGather/') # set output file for figures 
```

## The Data

Here is a summary of the data so far.

```{r}

multiqc <- read.csv(file = "../data/multiqc_report_0407.csv")
summary(multiqc)
levels(multiqc$QC) <- c("raw", "filtrim")
# mean 2.2 million reads alligned per sample. max 3.3 min 0.4
hist(multiqc$millalign)

boxplot(multiqc$millalign ~ multiqc$QC,
        xlab = "All samples",
        names = c("Raw", "Trimmed and Filtered"),
        ylab = "Million reads aligned per sample")

boxplot(multiqc$fracalign ~ multiqc$QC,
        xlab = "All samples",
        names = c("Raw", "Trimmed and Filtered"),
        ylab = "Fraction aligned")

boxplot(multiqc$bp ~ multiqc$QC,
        xlab = "All samples",
        names = c("Raw", "Trimmed and Filtered"),
        ylab = "Lenght (bp)")
```

## Kallisto Gather

The kallisto output gives you read counts for sample in an abundance file for every single sample. This portion of the code goes through and finds each samples' abundance.tsv file, extracts the data, and combines it all into a dataframe. The `counts` file is unnormalized, but the `tpm` is the data after being normalized by transcripts per million.  
(P.S. Unfortunately, I have no idea how to do this next part without changing directories. This script was developed with assistance from Anna Batthenhouse and Dennis Whylie.)

```{r kallisto gather, warning=FALSE}
setwd("../data/rnaseq/04_kallistoquant/")
## this will create lists of all the samples
kallistoDirs = dir(".")
kallistoDirs = kallistoDirs[!grepl("\\.(R|py|pl|sh|xlsx?|txt|tsv|csv|org|md|obo|png|jpg|pdf)$",
kallistoDirs, ignore.case=TRUE)]

kallistoFiles = paste0(kallistoDirs, "/abundance.tsv")
names(kallistoFiles) = kallistoDirs
if(file.exists(kallistoFiles))
  kallistoData = lapply(
  kallistoFiles,
  read.table,
  sep = "\t",
  row.names = 1,
  header = TRUE
)

## this for loop uses the reduce function to make two data frame with counts or tpm from all the samples
ids = Reduce(f=union, x=lapply(kallistoData, rownames))
if (all(sapply(kallistoData, function(x) {all(rownames(x)==ids)}))) {
  count = data.frame(
    id = ids,
    sapply(kallistoData, function(x) {x$est_counts}),
    check.names = FALSE,
    stringsAsFactors = FALSE
)
  tpm = data.frame(
    id = ids,
    sapply(kallistoData, function(x) {x$tpm}),
    check.names = FALSE,
    stringsAsFactors = FALSE
)
}
```

## Tranform transcipts-level measurements to gene-level measurements

Kallisto is cool because it does give you transcript level stuff, but right now I think I have more power if I just look at gene level stuff. I'll go back and look at transcripts if I want to. I use some string splitting to take the very long transcript identifying and create a `geneids` file that has all the database identifiers for each transcript.


```{r bygene, warning=FALSE, }
## make a dataframe with the parts of the gene id as columns
geneids <- count[c(1)] 
geneids$ENSMUST <- sapply(strsplit(as.character(geneids$id),'\\|'), "[", 1)
geneids$ENSMUSG <- sapply(strsplit(as.character(geneids$id),'\\|'), "[", 2)
geneids$OTTMUSG <- sapply(strsplit(as.character(geneids$id),'\\|'), "[", 3)
geneids$OTTMUST <- sapply(strsplit(as.character(geneids$id),'\\|'), "[", 4)
geneids$transcript <- sapply(strsplit(as.character(geneids$id),'\\|'), "[", 5)
geneids$gene <- sapply(strsplit(as.character(geneids$id),'\\|'), "[", 6)
geneids$length <- sapply(strsplit(as.character(geneids$id),'\\|'), "[", 7)
geneids$structure1 <- sapply(strsplit(as.character(geneids$id),'\\|'), "[", 8)
geneids$structure2 <- sapply(strsplit(as.character(geneids$id),'\\|'), "[", 9)
geneids$structure3 <- sapply(strsplit(as.character(geneids$id),'\\|'), "[", 10)
geneids$transcript_lenght <- as.factor(paste(geneids$transcript, geneids$length, sep="_"))

# tpm to tpmbygene
tpmbygene <-  full_join(geneids, tpm) # merge tpm and genids
tpmbygene <- tpmbygene[-c(1:6,8:12)]   ## keep gene name and tpm for samples)
tpmbygene <- melt(tpmbygene, id=c("gene")) ## lenghten 
tpmbygene <- dcast(tpmbygene, gene ~ variable, value.var= "value", fun.aggregate=mean) #then widen by sum
row.names(tpmbygene) <- tpmbygene$gene ## make gene the row name
tpmbygene[1] <- NULL ## make gene the row name
tpmbygene <- round(tpmbygene) #round all value to nearest 1s place


# count to countbygene
countbygene <- full_join(geneids, count) # merge count and genids
countbygene <- countbygene[-c(1:6,8:12)]   ## rkeep gene name and counts for samples)
countbygene <- melt(countbygene, id=c("gene")) ## lenghten 
countbygene  <- dcast(countbygene, gene ~ variable, value.var= "value", fun.aggregate=sum) #then widen by sum
row.names(countbygene) <- countbygene$gene ## make gene the row name
countbygene[1] <- NULL ## make gene the row name
countbygene <- round(countbygene) #round all value to nearest 1s place
```

## Final samples removal
At this point, we need to remove a few more samples that either didn't have an abundance tsv file or were deemed outliers.

```{r removingsamples}
## remove samples that never produced quality reads
savesamples <- colnames(countbygene)
kallistosamples <- tidysamples %>%
  filter(RNAseqID %in% savesamples) %>% droplevels()

## remove outlier samples
kallistosamples <- kallistosamples %>%
  filter(RNAseqID != "146C-CA3-4", RNAseqID != "16-116D") # two outliters
savecols <- as.character(kallistosamples$RNAseqID) #selects all good samples
savecols <- as.vector(savecols) # make it a vector

countbygene <- countbygene %>% select(one_of(savecols)) # keep good samples
count <- count %>% select(one_of(savecols)) # keep good samples
tpm <- tpm %>% select(one_of(savecols)) # keep good samples
tpmbygene <- tpmbygene %>% select(one_of(savecols)) # keep good samples

summary(kallistosamples)
```



## Write files 
Now, let's save the output files for future use
```{r writefiles}
write.csv(countbygene, "../data/rnaseq/countbygene.csv", row.names=T)
write.csv(tpmbygene, "../data/rnaseq/tpmbygene.csv", row.names=T)
write.csv(geneids, "../data/rnaseq/geneids.csv", row.names=FALSE)
write.csv(kallistosamples, "../data/rnaseq/kallistosamples.csv", row.names=FALSE)
```

## Session Info
```{r SessionInfo}
sessionInfo()
```