---
title: "02b_rnaseq_All"
output: md_document
---

The figures made from this script were compiled in Adobe.

<img src="../figures/figures-02.png" width="1370" />

```{r setup, message=F, warning=F}
library(tidyverse)
library(cowplot) ## for some easy to use themes
library(pheatmap) ## awesome heatmaps
library(viridis) # for awesome color pallette
library(reshape2) ## for melting dataframe
library(DESeq2) ## for gene expression analysis
library(edgeR)  ## for basic read counts status
library(magrittr) ## to use the weird pipe
library(ggrepel) ## for labeling volcano plot
library(stringr) ## for uppercase gene names
library(car) # stats
library(RColorBrewer)  # colors
library(Rtsne) # for tSNE

library(BiocParallel)
register(MulticoreParam(6))

## load functions 
source("figureoptions.R")
source("functions_RNAseq.R")

## set output file for figures 
knitr::opts_chunk$set(fig.path = '../figures/02b_RNAseqAll/', cache = T)
```

## Design

The two two catagorical variables are

* Hippocampal subfield: DG, CA3, CA1
* Treatment: standard yoked, standard trained, conflict yoked, conflict trained

```{r data, message=F, warning=F}
colData <- read.csv("../data/02a_colData.csv", header = T)
countData <- read.csv("../data/02a_countData.csv", header = T, check.names = F, row.names = 1)
colData <- colData %>% dplyr::rename(treatment = APA2, subfield = Punch)
colData %>% select(treatment, subfield)  %>%  summary()

dds <- DESeqDataSetFromMatrix(countData = countData,
                              colData = colData,
                              design = ~ subfield + treatment + subfield*treatment)

dds$subfield <- factor(dds$subfield, levels=c("DG","CA3", "CA1")) ## specify the factor levels

dds$treatment <- factor(dds$treatment, levels=c("standard.yoked" ,"standard.trained", "conflict.yoked", "conflict.trained")) ## specify the factor levels

dds # view the DESeq object - note numnber of genes
dds <- dds[ rowSums(counts(dds)) > 10, ]  # Pre-filtering genes
dds # view number of genes afternormalization and the number of samples
dds <- DESeq(dds, parallel = TRUE) # Differential expression analysis
#rld <- rlog(dds, blind=FALSE) ## log transformed data
vsd <- vst(dds, blind=FALSE) ## variance stabilized
head(assay(vsd),3)
write.csv(assay(vsd), file = "../data/02b_vsd.csv", row.names = T)
```

## check for outliers


```{r outliers}
boxplot(log10(assays(dds)[["cooks"]]), range=0, las=2)
plotDispEsts(dds)

sampleDists <- dist(t(assay(vsd)))
sampleDistMatrix <- as.matrix(sampleDists)
rownames(sampleDistMatrix) <- paste(vsd$subfield, vsd$treatment, sep="-")
colnames(sampleDistMatrix) <- NULL
colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)
pheatmap(sampleDistMatrix,
         clustering_distance_rows=sampleDists,
         clustering_distance_cols=sampleDists,
         col=colors, fontsize = 6)
```

## Summary 2 way contrasts

This first function shows the total number of up and down regulated genes and the top 3 most significant genes.

```{r twowaycontrasts}
res_summary <- function(mycontrast){
  res <- results(dds, contrast = mycontrast, independentFiltering = T)
  print(mycontrast)
  print(sum(res$padj < 0.1, na.rm=TRUE))
  print(summary(res))
  print(head((res[order(res$padj),]), 5))
  cat("\n")
}

res_summary(c("subfield", "CA1", "DG"))
res_summary(c("subfield", "CA1", "CA3"))
res_summary(c("subfield", "CA3", "DG"))

res_summary(c("treatment", "standard.trained", "standard.yoked"))
res_summary(c("treatment", "conflict.trained", "conflict.yoked"))
res_summary(c("treatment", "conflict.trained", "standard.trained"))
res_summary(c("treatment", "conflict.yoked", "standard.yoked"))
```

This second function only prints the total number of DEGs, but it saves lots of useful info to a df for downstream dataviz.

```{r heatmap_prep}
# note: see resvals fucntion in `functions_RNAseq.R`

contrast1 <- resvals(contrastvector = c("subfield", "CA1", "DG"), mypval = 0.1) # 3060
contrast2 <- resvals(contrastvector = c("subfield", "CA1", "CA3"), mypval = 0.1) # 2388
contrast3 <- resvals(contrastvector = c("subfield", "CA3", "DG"), mypval = 0.1) # 3078
contrast4 <- resvals(contrastvector = c("treatment", "standard.trained", "standard.yoked"), mypval = 0.1) #  80
contrast5 <- resvals(contrastvector = c("treatment", "conflict.trained", "conflict.yoked"), mypval = 0.1) # 23
contrast6 <- resvals(contrastvector = c("treatment", "conflict.trained", "standard.trained"), mypval = 0.1) #  1
contrast7 <- resvals(contrastvector = c("treatment", "conflict.yoked", "standard.yoked"), mypval = 0.1) # 37
```

```{r pheatmap}
# heatmap with all DEGs
DEGes <- assay(vsd)
DEGes <- cbind(DEGes, contrast1, contrast2, contrast3, contrast4, contrast5, contrast6, contrast7)
DEGes <- as.data.frame(DEGes) # convert matrix to dataframe
DEGes$rownames <- rownames(DEGes)  # add the rownames to the dataframe
DEGes$rownames <- str_to_upper(DEGes$rownames) ## uppercase gene names

names(DEGes)
DEGes$padjmin <- with(DEGes, pmin(# padjsubfieldCA1DG, padjsubfieldCA1CA3, padjsubfieldCA3DG,
                                  padjtreatmentstandard.trainedstandard.yoked, padjtreatmentconflict.trainedconflict.yoked,
                                  padjtreatmentconflict.trainedstandard.trained,padjtreatmentconflict.yokedstandard.yoked)) 
DEGes <- DEGes %>% filter(padjmin < 0.1)

rownames(DEGes) <- DEGes$rownames
drop.cols <-colnames(DEGes[,grep("padj|pval|rownames", colnames(DEGes))])
DEGes <- DEGes %>% dplyr::select(-one_of(drop.cols))
DEGes <- as.matrix(DEGes)
DEGes <- DEGes - rowMeans(DEGes)

df <- as.data.frame(colData(dds)[,c("treatment", "subfield")]) ## matrix to df
rownames(df) <- names(countData)
levels(df$treatment) <- c("standard yoked", "standard trained",  "conflict yoked","conflict trained")


DEGes <- as.matrix(DEGes) 
paletteLength <- 80
myBreaks <- c(seq(min(DEGes), 0, length.out=ceiling(paletteLength/2) + 1), 
              seq(max(DEGes)/paletteLength, max(DEGes), length.out=floor(paletteLength/2)))

pheatmap(DEGes, show_colnames=F, show_rownames = F,
         annotation_col=df, 
         annotation_colors = pheatmapcolors,
         treeheight_row = 0, treeheight_col = 25,
         annotation_row = NA, 
         annotation_legend = TRUE,
         annotation_names_row = FALSE, annotation_names_col = TRUE,
         fontsize = 8, 
         border_color = NA ,
         color = viridis(80),
         cellwidth = 6, 
         clustering_method="average",
         breaks=myBreaks,
         clustering_distance_cols="correlation" 
         )

pheatmap(DEGes, show_colnames=F, show_rownames = F,
         annotation_col=df, annotation_colors = pheatmapcolors, 
         annotation_row = NA, 
         annotation_legend = F,
         annotation_names_row = FALSE, 
         annotation_names_col = FALSE,
         treeheight_row = 0, treeheight_col = 10,
         fontsize = 6, 
         border_color = NA ,
         color = viridis(80),
         height = 2.5, 
         width = 2.5,
         clustering_method="average",
         breaks=myBreaks,
         clustering_distance_cols="correlation", 
         filename = "../figures/02b_RNAseqALL/pheatmap1.pdf"
         )

pheatmap(DEGes, show_colnames=F, show_rownames = T,
         annotation_col=df, 
         annotation_colors = pheatmapcolors,
         treeheight_row = 0, treeheight_col = 25,
         annotation_row = NA, 
         annotation_legend = TRUE,
         annotation_names_row = FALSE, annotation_names_col = TRUE,
         fontsize = 8, 
         border_color = NA ,
         color = viridis(80),
         cellwidth = 6, 
         clustering_method="average",
         breaks=myBreaks,
         clustering_distance_cols="correlation" 
         )

pheatmap(DEGes, show_colnames=F, show_rownames = T,
         annotation_col=df, annotation_colors = pheatmapcolors, 
         annotation_row = NA, 
         annotation_legend = F,
         annotation_names_row = FALSE, 
         annotation_names_col = FALSE,
         treeheight_row = 0, treeheight_col = 10,
         fontsize = 4, 
         border_color = NA ,
         color = viridis(80),
         height = 5, 
         width = 2.5,
         clustering_method="average",
         breaks=myBreaks,
         clustering_distance_cols="correlation", 
         filename = "../figures/02b_RNAseqALL/pheatmap2.pdf"
         )
```

## Principle component analysis

```{r pca}
# create the dataframe using my function pcadataframe
pcadata <- pcadataframe(vsd, intgroup=c("subfield","treatment"), returnData=TRUE)
percentVar <- round(100 * attr(pcadata, "percentVar"))
pcadata$subfieldAPA <- as.factor(paste(pcadata$subfield, pcadata$treatment, sep="_"))
pcadata$subfield <- factor(pcadata$subfield, levels=c("DG","CA3", "CA1"))
pcadata$treatment <- factor(pcadata$treatment, levels=c("standard.yoked","standard.trained",  "conflict.yoked","conflict.trained"))

levels(pcadata$treatment) <- c("standard yoked","standard trained",  "conflict yoked", "conflict trained")

summary(aov(PC1 ~ subfield * treatment, data=pcadata)) 
TukeyHSD((aov(PC1 ~ subfield, data=pcadata)), which = "subfield") 

summary(aov(PC2 ~ subfield * treatment, data=pcadata)) 
TukeyHSD((aov(PC2 ~ subfield, data=pcadata)), which = "subfield") 

PCA12 <- ggplot(pcadata, aes(pcadata$PC1, pcadata$PC2, colour=subfield)) +
    geom_point(size=2, aes(shape=treatment), alpha = 0.8) +
    xlab(paste0("PC1: ", percentVar[1],"%")) +
    ylab(paste0("PC2: ", percentVar[2],"%")) +
    scale_colour_manual(values=c(colorvalsubfield))+ 
   theme_ms()  +
      theme(legend.position= "none") +
    scale_shape_manual(values=c(1, 16, 0, 15), aes(color=colorvalsubfield)) +
  labs(color = "subfield", shape = "treatment") +
  guides(color = FALSE)
PCA12

```

## tSNE

```{r tSNE}
vsddf <- as.data.frame(assay(vsd))
vsddf <- as.data.frame(t(vsddf))

euclidist <- dist(vsddf) # euclidean distances between the rows
tsne_model <- Rtsne(euclidist, check_duplicates=FALSE, pca=TRUE, perplexity=2, theta=0.5, dims=2)

tsne_df = as.data.frame(tsne_model$Y) 
tsne_df <- cbind(colData, tsne_df)
tsne_df$subfield <- factor(tsne_df$subfield, levels = c("DG", "CA3", "CA1"))
tsne_df$treatment <- factor(tsne_df$treatment, levels = c("standard.yoked" ,"standard.trained", "conflict.yoked", "conflict.trained"))

tnseplot  <- ggplot(tsne_df, aes(x = V1, y = V2, shape = treatment, color = subfield)) +
  geom_point(size = 2) +
  scale_color_manual(guide = FALSE, values = c("#d95f02","#1b9e77", "#7570b3")) +
  theme_ms()  +
  theme(legend.position = "bottom",
        legend.title = element_blank(),
        legend.spacing.x = unit(0.01, 'cm'),
        legend.spacing.y = unit(0.01, 'cm')) +
  labs(x = "tSNE 1", y = "tSNE 2") +
  scale_shape_manual(aes(colour=colorvalsubfield), values=c(1, 16, 0, 15)) 
tnseplot
```

## pca + tsne

```{r PCA-tSNE}
mylegend <- get_legend(tnseplot)

top <- plot_grid(PCA12, tnseplot + theme(legend.position = "none"), nrow = 1)
topl <- plot_grid(top, mylegend, nrow = 2, rel_heights = c(1,0.1))
topl

pdf(file="../figures/02b_RNAseqALL/PCA-tSNE.pdf", width=4, height=2.5)
plot(topl)
dev.off()
```

## Volcanos plots and and gene lists

```{r volcanos}
makevolcanodf <- function(mycontrast, myup, mydown, filename){
  res <- results(dds, contrast = mycontrast, independentFiltering = T)

  data <- data.frame(gene = row.names(res), pvalue = (res$padj), 
                     lfc = res$log2FoldChange)
  data <- na.omit(data)

  data <- data %>%
  mutate(direction = ifelse(data$lfc > 0 & data$pvalue < 0.05, 
                        yes = myup, 
                        no = ifelse(data$lfc < 0 & data$pvalue < 0.05, 
                                    yes = mydown, 
                                    no = "NS")))
  data$logp <- -log10(data$pvalue)
  data <- dplyr::arrange(data, logp)
  write.csv(data, filename, row.names = F) # save for downstream analysis
  return(data)
}

DGvCA3 <- makevolcanodf(c("subfield", "CA3", "DG"), "CA3", "DG", "../data/DGvCA3.csv")
head(DGvCA3)
DGvCA1 <- makevolcanodf(c("subfield", "CA1", "DG"),"CA1", "DG", "../data/DGvCA1.csv")
CA3vCA1 <- makevolcanodf(c("subfield", "CA1", "CA3"),"CA1", "CA3", "../data/CA3vCA1.csv")

volcanoplot <- function(mydata, mycolors, mybreaks){
  
  myvolcano <- mydata %>%
    #dplyr::filter(direction != "NS") %>%
    ggplot(aes(x = lfc, y = logp)) + 
  geom_point(aes(color = direction), alpha = 0.5, na.rm = T) + 
  scale_color_manual(values = mycolors,
                     breaks = mybreaks,
                     name = "higher in") + 
  theme_ms() +
  #geom_hline(yintercept = 1.3,  size = 0.25, linetype = 2) + 
  scale_y_continuous(limits=c(0, 60)) +
  scale_x_continuous(limits=c(-10, 10)) +
  xlab(paste0("log fold difference")) +
  ylab(paste0("log10 p-value")) +       
  theme(panel.grid.minor=element_blank(),
        #legend.title = element_blank(),
        legend.position = "bottom",
        legend.spacing.x = unit(-0.1, 'cm'),
        panel.grid.major=element_blank(),
        legend.margin=margin(t=-0.25, r=0, b=0, l=0, unit="cm")) 

return(myvolcano)
}
  
d <- volcanoplot(DGvCA3, volcanoDGvCA3, c("DG", "CA3"))
e <- volcanoplot(DGvCA1, volcanoDGvCA1, c("DG", "CA1"))  
f <- volcanoplot(CA3vCA1, volcanoCA3vCA1, c("CA3", "CA1")) 

myvolcanoplots <- plot_grid(d,e,f,nrow = 1)
myvolcanoplots
```


## mareker gene analysis

Genes from Cembrowski sublement file 1 <https://elifesciences.org/articles/14997/figures>

```{r markergenes}
# import cembrowski markers
cembrowskisupp <- read.table("../data/cembrowksi_markers/elife-14997-supp1-v1_dendrogram.txt", sep="\t", header = T)

# select just columns with gene symbol and enriched column
# then rename gene column and convert to uppercase
cembrowskisupp <- cembrowskisupp %>% dplyr::select(gene_short_name, enriched) 
colnames(cembrowskisupp)[1] <- "gene"
cembrowskisupp$gene <- str_to_upper(cembrowskisupp$gene)
head(cembrowskisupp)

# subset my maker

cembrowksimarkers <- function(subfields){
  mydf <- cembrowskisupp %>% 
  dplyr::filter(enriched %in% subfields) %>% 
  dplyr::select(gene)
  names(mydf) <- NULL
  mylist <- as.list(mydf[,1])
  return(mylist)
}

levels(cembrowskisupp$enriched)

# dorsal markers 
CA1_markers <- cembrowksimarkers(c("ca1_d", "ca1_d-ca1_v"))
DG_markers <- cembrowksimarkers(c("dg_d", "dg_d-dg_v"))
CA3_markers <- cembrowksimarkers(c("ca3_d",  "ca3_d-ca3_v-ca2-ca1_d-ca1_v"))
```


```{r subfieldpvals}
# import subfield specific data
wrangledata <- function(filename, mycomparison){
  mydata <- read.csv(filename, header = T)
  mydata$gene <- str_to_upper(mydata$gene)
  mydata$comparison <- mycomparison
  return(mydata)
}

CA1DG <- wrangledata("../data/DGvCA1.csv", "CA1-DG")
CA1CA3 <- wrangledata("../data/CA3vCA1.csv", "CA1-CA3")
CA3DG <- wrangledata("../data/DGvCA3.csv", "CA3-DG")
```

```{r enrichementvalidation}
#look for markers in each supfield
# make data frames of genes expression results for markers 

marker_summary <- function(mydf, markers){
    df <- mydf %>%
    dplyr::filter(gene %in% c(markers)) 
    #return((df))
    return(summary(df$direction))
}

# comparing all lists with a given brain region
for(i in list(CA1DG, CA1CA3)){
  j <- marker_summary(i, CA1_markers)
  print(i[1, 6])
  print(j)
}

15/(15+3)
15/(15+3)

for(i in list(CA1CA3, CA3DG)){
  j <- marker_summary(i, CA3_markers)
  print(i[1, 6])
  print(j)
}

6/10
7/10

for(i in list(CA1DG, CA3DG)){
  j <- marker_summary(i, DG_markers)
  print(i[1, 6])
  print(j)
}

48/(48+23)
49/(49+22)

```

Then, I checked to see how many of the markers that Cembrowski found to be enriched in discrete dorsal cell populations were also enriched in my comparisons. The enriched coloumn is what percent were confirmed, the depleted column means that the marker was experssed in the opposite direction, and neither means that the Cembrowski marker was not significantly different in expression between the two cell types. Here are the results:

| Maker | Comparison | Enriched | Depleted | Neither |
|-------|------------|----------|----------|---------|
| CA1 | CA1 v DG | 0.83 | 0.00 | 0.17 |
| CA1 | CA1 v CA3 | 0.83 | 0.06 | 0.11 |
| CA3 | CA3 v DG | 0.60 | 0.00 | 0.40 |
| CA3 | CA3 v CA1 | 0.70 | 0.00 | 0.30 |
| DG | DG v CA1 | 0.68 | 0.00 | 0.32 |
| DG | DG v CA3 | 0.69 | 0.00 | 0.31 |

```{r barplot}
mybarplot <- read.csv("../data/02h_markers.csv")
head(mybarplot)

p <- ggplot(data=mybarplot, aes(x=comparison, y=correct, fill = marker)) +
  geom_bar(stat="identity", position=position_dodge()) +
  theme_ms() +
  scale_fill_manual(values = c( "#7570b3","#1b9e77",  "#d95f02"),
                    name = "markers") +
  ylab("% marker genes recovered") +
  xlab("subfields compared") +
  theme(panel.grid.minor=element_blank(),
        panel.grid.major=element_blank(),
        legend.key.width =  unit(0.2, "cm"),
        legend.key.height =  unit(0.1, "cm"),
        axis.text.x = element_text(angle = 45, hjust = 1),
        legend.position = "bottom") +
  scale_y_continuous(labels = scales::percent,
                     limits = c(0,1)) 
p
```


```{r volcanobar}

myvolcanoplots <- plot_grid(d,e + labs(y = NULL),f + labs(y = NULL),p, nrow = 1)
myvolcanoplots

pdf(file="../figures/02b_RNAseqALL/volcanoplots.pdf", width=6.69, height=2.5)
plot(myvolcanoplots)
dev.off()
```