---
title: "02b_rnaseq_All"
output: md_document
---

The figures made from this script were compiled in Adobe.

<img src="../figures/figures-02.png" width="1370" />

```{r setup, message=F}
library(ggplot2) ## for awesome plots!
library(cowplot) ## for some easy to use themes
library(dplyr) ## for filtering and selecting rows
library(car) ## stats
library(VennDiagram) ## venn diagrams
library(pheatmap) ## awesome heatmaps
library(viridis) # for awesome color pallette
library(reshape2) ## for melting dataframe
library(DESeq2) ## for gene expression analysis
library(edgeR)  ## for basic read counts status
library(magrittr) ## to use the weird pipe
library(genefilter)  ## for PCA fuction
library(ggrepel) ## for labeling volcano plot
library(stringr) ## for uppercase gene names

## load functions 
source("figureoptions.R")
source("functions_RNAseq.R")

## set output file for figures 
knitr::opts_chunk$set(fig.path = '../figures/02b_RNAseqAll/',
                      cache = T)
```

## Design

The two two catagorical variables are

* Hippocampal subfield: DG, CA3, CA1
* Treatment: yoked_consistent, consistent, yoked_conflict, conflict


```{r data, message=F, warning=F}
colData <- read.csv("../data/02a_colData.csv", header = T)
countData <- read.csv("../data/02a_countData.csv", header = T, check.names = F, row.names = 1)
colData <- colData %>% dplyr::rename(treatment = APA2, subfield = Punch)
colData %>% select(treatment, subfield)  %>%  summary()

head(colData)

totalCounts=colSums(countData)
### on average 1 million gene counts per sample 
summary((colSums(countData)/1000000))

dds <- DESeqDataSetFromMatrix(countData = countData,
                              colData = colData,
                              design = ~ subfield + treatment + subfield*treatment)

dds$subfield <- factor(dds$subfield, levels=c("DG","CA3", "CA1")) ## specify the factor levels
dds$treatment <- factor(dds$treatment, levels=c("yoked_consistent", "consistent", "yoked_conflict" , "conflict")) ## specify the factor levels


dds # view the DESeq object - note numnber of genes
dds <- dds[ rowSums(counts(dds)) > 1, ]  # Pre-filtering genes with 0 counts
dds # view number of genes afternormalization and the number of samples
dds <- DESeq(dds) # Differential expression analysis
#rld <- rlog(dds, blind=FALSE) ## log transformed data
vsd <- vst(dds, blind=FALSE) ## variance stabilized
head(assay(vsd),3)
write.csv(assay(vsd), file = "../data/02b_vsd.csv", row.names = T)
```

## Summary 2 way contrasts

This first function shows the total number of up and down regulated genes and the top 3 most significant genes.

```{r twowaycontrasts}
res_summary <- function(mycontrast){
  res <- results(dds, contrast = mycontrast, independentFiltering = T)
  print(mycontrast)
  print(summary(res))
  print(head((res[order(res$padj),]), 3))
  cat("\n")
}

res_summary(c("subfield", "CA1", "DG"))
res_summary(c("subfield", "CA1", "CA3"))
res_summary(c("subfield", "CA3", "DG"))

res_summary(c("treatment", "consistent", "yoked_consistent"))
res_summary(c("treatment", "yoked_conflict", "yoked_consistent"))
res_summary(c("treatment", "conflict", "yoked_conflict"))
res_summary(c("treatment", "conflict", "consistent"))

```

This second function only prints the total number of DEGs, but it saves lots of useful info to a df for downstream dataviz.

```{r heatmap_prep}
# note: see resvals fucntion in `functions_RNAseq.R`

contrast1 <- resvals(contrastvector = c("subfield", "CA1", "DG"), mypval = 0.1) # 2765
contrast2 <- resvals(contrastvector = c("subfield", "CA1", "CA3"), mypval = 0.1) # 2165
contrast3 <- resvals(contrastvector = c("subfield", "CA3", "DG"), mypval = 0.1) # 2948
contrast4 <- resvals(contrastvector = c("treatment", "consistent", "yoked_consistent"), mypval = 0.1) #  114
contrast5 <- resvals(contrastvector = c("treatment", "conflict", "yoked_conflict"), mypval = 0.1) # 61
contrast6 <- resvals(contrastvector = c("treatment", "conflict", "consistent"), mypval = 0.1) #  0
contrast7 <- resvals(contrastvector = c("treatment", "yoked_conflict", "yoked_consistent"), mypval = 0.1) # 40
```

```{r heatmap}
# heatmap with all DEGs
DEGes <- assay(vsd)
DEGes <- cbind(DEGes, contrast1, contrast2, contrast3, contrast4, contrast5, contrast6, contrast7)
DEGes <- as.data.frame(DEGes) # convert matrix to dataframe
DEGes$rownames <- rownames(DEGes)  # add the rownames to the dataframe
DEGes$rownames <- str_to_upper(DEGes$rownames) ## uppercase gene names
DEGes$padjmin <- with(DEGes, pmin(padjsubfieldCA1DG, padjsubfieldCA1CA3, padjsubfieldCA3DG, padjtreatmentconflictconsistent, padjtreatmentyoked_conflictyoked_consistent,padjtreatmentconflictyoked_conflict, padjtreatmentconsistentyoked_consistent)) 
DEGes <- DEGes %>% filter(padjmin < 0.00000000000000000001)
rownames(DEGes) <- DEGes$rownames
drop.cols <-colnames(DEGes[,grep("padj|pval|rownames", colnames(DEGes))])
DEGes <- DEGes %>% dplyr::select(-one_of(drop.cols))
DEGes <- as.matrix(DEGes)
DEGes <- DEGes - rowMeans(DEGes)

df <- as.data.frame(colData(dds)[,c("treatment", "subfield")]) ## matrix to df
rownames(df) <- names(countData)
levels(df$treatment) <- c("yoked consistent","consistent",  "yoked conflict","conflict")


DEGes <- as.matrix(DEGes) 
paletteLength <- 30
myBreaks <- c(seq(min(DEGes), 0, length.out=ceiling(paletteLength/2) + 1), 
              seq(max(DEGes)/paletteLength, max(DEGes), length.out=floor(paletteLength/2)))

pheatmap(DEGes, show_colnames=F, show_rownames = T,
         annotation_col=df, annotation_colors = pheatmapcolors,
         treeheight_row = 0, treeheight_col = 25,
         annotation_row = NA, 
         annotation_legend = TRUE,
         annotation_names_row = FALSE, annotation_names_col = FALSE,
         fontsize = 8, 
         border_color = NA ,
         color = viridis(30),
         cellwidth = 6, 
         clustering_method="average",
         breaks=myBreaks,
         clustering_distance_cols="correlation" 
         )

pheatmap(DEGes, show_colnames=F, show_rownames = T,
         annotation_col=df, annotation_colors = pheatmapcolors, 
         annotation_row = NA, 
         annotation_legend = TRUE,
         annotation_names_row = FALSE, 
         annotation_names_col = FALSE,
         treeheight_row = 10, treeheight_col = 10,
         fontsize = 6, 
         border_color = NA ,
         color = viridis(30),
         height = 2.5, 
         width = 3.3,
         clustering_method="average",
         breaks=myBreaks,
         clustering_distance_cols="correlation", 
         filename = "../figures/02b_RNAseqALL/pheatmap1.pdf"
         )

### heatmap with only treatment DEGs

# heatmap with all DEGs
DEGes <- assay(vsd)
DEGes <- cbind(DEGes, contrast4, contrast5, contrast7)
DEGes <- as.data.frame(DEGes) # convert matrix to dataframe
DEGes$rownames <- rownames(DEGes)  # add the rownames to the dataframe
DEGes$rownames <- str_to_upper(DEGes$rownames) ## uppercase gene names
DEGes$padjmin <- with(DEGes, pmin(padjtreatmentyoked_conflictyoked_consistent,padjtreatmentconflictyoked_conflict, padjtreatmentconsistentyoked_consistent)) 
DEGes <- DEGes %>% filter(padjmin < 0.001)
rownames(DEGes) <- DEGes$rownames
drop.cols <-colnames(DEGes[,grep("padj|pval|rownames", colnames(DEGes))])
DEGes <- DEGes %>% dplyr::select(-one_of(drop.cols))
DEGes <- as.matrix(DEGes)
DEGes <- DEGes - rowMeans(DEGes)

df <- as.data.frame(colData(dds)[,c("treatment", "subfield")]) ## matrix to df
rownames(df) <- names(countData)
levels(df$treatment) <- c("yoked consistent","consistent",  "yoked conflict","conflict")

DEGes <- as.matrix(DEGes) 
paletteLength <- 30
myBreaks <- c(seq(min(DEGes), 0, length.out=ceiling(paletteLength/2) + 1), 
              seq(max(DEGes)/paletteLength, max(DEGes), length.out=floor(paletteLength/2)))

pheatmap(DEGes, show_colnames=F, show_rownames = T,
         annotation_col=df, annotation_colors = pheatmapcolors,
         treeheight_row = 0, treeheight_col = 25,
         annotation_row = NA, 
         annotation_legend = TRUE,
         annotation_names_row = FALSE, annotation_names_col = FALSE,
         fontsize = 7, 
         border_color = NA ,
         color = viridis(30),
         cellwidth = 6, 
         clustering_method="average",
         breaks=myBreaks,
         clustering_distance_cols="correlation" 
         )

pheatmap(DEGes, show_colnames=F, show_rownames = T,
         annotation_col=df, annotation_colors = pheatmapcolors, 
         annotation_row = NA, 
         annotation_legend = TRUE,
         annotation_names_row = FALSE, 
         annotation_names_col = FALSE,
         treeheight_row = 10, treeheight_col = 10,
         fontsize = 6, 
         border_color = NA ,
         color = viridis(30),
         height = 5, 
         width = 4,
         clustering_method="average",
         breaks=myBreaks,
         clustering_distance_cols="correlation", 
         filename = "../figures/02b_RNAseqALL/pheatmap2.pdf"
         )

```

## Principle component analysis

```{r pca, echo=F, message=F}
# create the dataframe using my function pcadataframe
pcadata <- pcadataframe(vsd, intgroup=c("subfield","treatment"), returnData=TRUE)
percentVar <- round(100 * attr(pcadata, "percentVar"))
pcadata$subfieldAPA <- as.factor(paste(pcadata$subfield, pcadata$treatment, sep="_"))
pcadata$subfield <- factor(pcadata$subfield, levels=c("DG","CA3", "CA1"))
pcadata$treatment <- factor(pcadata$treatment, levels=c("yoked_consistent","consistent",  "yoked_conflict","conflict"))

levels(pcadata$treatment) <- c("yoked consistent","consistent",  "yoked conflict","conflict")

summary(aov(PC1 ~ subfield * treatment, data=pcadata)) 
TukeyHSD((aov(PC1 ~ subfield, data=pcadata)), which = "subfield") 

summary(aov(PC2 ~ subfield * treatment, data=pcadata)) 
TukeyHSD((aov(PC2 ~ subfield, data=pcadata)), which = "subfield") 

summary(aov(PC3 ~ subfield * treatment, data=pcadata)) 
TukeyHSD((aov(PC3 ~ subfield, data=pcadata)), which = "subfield") 

summary(aov(PC4 ~ subfield * treatment, data=pcadata)) 
summary(aov(PC5 ~ subfield * treatment, data=pcadata)) 
summary(aov(PC6 ~ subfield * treatment, data=pcadata)) 


PCA12 <- ggplot(pcadata, aes(pcadata$PC1, pcadata$PC2, color=subfield)) +
    geom_point(size=1.5, aes(shape=treatment), alpha = 0.8) +
    xlab(paste0("PC1: ", percentVar[1],"%")) +
    ylab(paste0("PC2: ", percentVar[2],"%")) +
    #stat_ellipse(level = 0.95, (aes(color=subfield)),size=0.25) + 
    scale_colour_manual(values=c(colorvalsubfield))+ 
    theme_cowplot(font_size = 7, line_size = 0.25)  +
    #theme(legend.title=element_blank()) +
    scale_x_continuous(limits=c(-15, 25)) +
      scale_y_continuous(limits=c(-15, 15)) +
    scale_shape_manual(values=c(1, 16, 0, 15), aes(color=colorvalsubfield))
PCA12


PCA42 <- ggplot(pcadata, aes(pcadata$PC5, pcadata$PC2)) +
    geom_point(size=1.5, aes(color=subfield, shape=treatment), alpha = 0.8) +
    xlab(paste0("PC4: ", percentVar[4],"%")) +
    ylab(paste0("PC2: ", percentVar[2],"%")) +
    #stat_ellipse(level = 0.95, (aes(color=subfield)),size=0.25) + 
    scale_colour_manual(values=c(colorvalsubfield))+ 
    theme_cowplot(font_size = 7, line_size = 0.25)  +
    theme(legend.title=element_blank()) +
  scale_x_continuous(limits=c(-10, 15)) +
        scale_y_continuous(limits=c(-15, 15)) +
    scale_shape_manual(aes(color=colorvalsubfield), values=c(1, 16, 0, 15)) +
  guides(color = guide_legend(order=1),
         shape = guide_legend(order=2)) 
PCA42


plotnolegend <- plot_grid(PCA12 + theme(legend.position="none"),
           PCA42 + theme(legend.position="none"),
           labels = c("B", "C"),
           nrow = 2,
           label_size = 7
           )

plotnolegend
legend <- get_legend(PCA42)

pcaplots <- plot_grid(plotnolegend, legend, ncol = 2, rel_widths  = c(1, .4))
pcaplots

pdf(file="../figures/02b_RNAseqALL/pcaplots.pdf", width=3.3, height=2.5)
plot(pcaplots)
dev.off()

```

## Volcanos plots and and gene lists

```{r volcanos}

makevolcanodf <- function(mycontrast, myup, mydown, filename){
  res <- results(dds, contrast = mycontrast, independentFiltering = T)

  data <- data.frame(gene = row.names(res), pvalue = (res$padj), 
                     lfc = res$log2FoldChange)
  data <- na.omit(data)

  data <- data %>%
  mutate(direction = ifelse(data$lfc > 0 & data$pvalue < 0.05, 
                        yes = myup, 
                        no = ifelse(data$lfc < 0 & data$pvalue < 0.05, 
                                    yes = mydown, 
                                    no = "neither")))
  data$logp <- -log10(data$pvalue)
  data <- dplyr::arrange(data, logp)
  write.csv(data, filename, row.names = F)
  return(data)
}

DGvCA3 <- makevolcanodf(c("subfield", "CA3", "DG"), "CA3", "DG", "../data/DGvCA3.csv")
DGvCA1 <- makevolcanodf(c("subfield", "CA1", "DG"),"CA1", "DG", "../data/DGvCA1.csv")
CA3vCA1 <- makevolcanodf(c("subfield", "CA1", "CA3"),"CA1", "CA3", "../data/CA3vCA1.csv")
CsYcs <- makevolcanodf(c("treatment", "consistent", "yoked_consistent"),"consistent", "yoked_consistent", "../data/CsYcs.csv")

volcanoplot <- function(mydata, mycolors, mybreaks){
  
  myvolcano <- mydata %>%
    dplyr::filter(direction != "neither") %>%
    ggplot(aes(x = lfc, y = logp)) + 
  geom_point(aes(color = direction), size = 0.5, alpha = 0.5, na.rm = T) + 
  scale_color_manual(values = mycolors,
                     breaks = mybreaks) + 
  theme_cowplot(font_size = 7, line_size = 0.25) +
  geom_hline(yintercept = 1.3,  size = 0.25, linetype = 2) + 
  scale_y_continuous(limits=c(0, 60)) +
  scale_x_continuous( limits=c(-10, 10)) +
  xlab(paste0("log fold difference")) +
  ylab(paste0("log10 p-value")) +       
  theme(panel.grid.minor=element_blank(),
        legend.title = element_blank(),
        legend.position = "bottom", # remove legend 
        panel.grid.major=element_blank(),
        legend.margin=margin(t=-0.25, r=0, b=0, l=0, unit="cm"))

return(myvolcano)
}
  
d <- volcanoplot(DGvCA3, volcanoDGvCA3, c("DG", "CA3"))  
e <- volcanoplot(DGvCA1, volcanoDGvCA1, c("DG", "CA1"))  
f <- volcanoplot(CA3vCA1, volcanoCA3vCA1, c("CA3", "CA1")) 
g <- volcanoplot(CsYcs, volcano1, c("yoked_consistent", "consistent")) 

myvolcanoplots <- plot_grid(d,e,f,g,
           labels = c("D", "E", "F", "G"),
           nrow = 1,
           label_size = 7
           )
myvolcanoplots

pdf(file="../figures/02b_RNAseqALL/myvolcanoplots.pdf", width=6.6, height=2)
plot(myvolcanoplots)
dev.off()
```


## plot single gene counts

```{r Prkcz}
plotCounts(dds, "Prkcz", intgroup = "subfield", normalized = TRUE)
plotCounts(dds, "Prkcz", intgroup = "treatment", normalized = TRUE)
```

## Observed versus expected ration of DEGs

```{r stats}
# chisq.test equal expression of increased versus decreased expression
chisq.test(c(1099,	1427), p = c(0.45, 0.55))$expected
chisq.test(c(1099,	1427), p = c(0.45, 0.55))
prop.table(c(1099,	1427))

chisq.test(c(850,	1172), p = c(0.4, 0.6))$expected
chisq.test(c(850,	1172), p = c(0.4, 0.6))
prop.table(c(850,	1172))

chisq.test(c(1585,	1560), p = c(0.5, 0.5))$expected
chisq.test(c(1585,	1560), p = c(0.5, 0.5))
prop.table(c(1585,	1560))

chisq.test(c(1,	0), p = c(0.5, 0.5))$expected
chisq.test(c(1,	0), p = c(0.5, 0.5))
prop.table(c(1,	0))
```