---
title: "02b_rnaseq_All"
output: md_document
---

The figures made from this script were compiled in Adobe.

<img src="../figures/figures-02.png" width="1370" />

```{r setup, message=F, warning=F}
library(tidyverse)
library(cowplot) ## for some easy to use themes
library(pheatmap) ## awesome heatmaps
library(viridis) # for awesome color pallette
library(reshape2) ## for melting dataframe
library(DESeq2) ## for gene expression analysis
library(edgeR)  ## for basic read counts status
library(magrittr) ## to use the weird pipe
library(ggrepel) ## for labeling volcano plot
library(stringr) ## for uppercase gene names
library(car) # stats
library(RColorBrewer)  # colors
library(Rtsne) # for tSNE

library(BiocParallel)
register(MulticoreParam(6))

## load functions 
source("figureoptions.R")
source("functions_RNAseq.R")

## set output file for figures 
knitr::opts_chunk$set(fig.path = '../figures/02b_RNAseqAll/', cache = F)
```

## Design

The two two catagorical variables are

* Hippocampal subfield: DG, CA3, CA1
* Treatment: standard yoked, standard trained, conflict yoked, conflict trained

```{r data, message=F, warning=F}
colData <- read.csv("../data/02a_colData.csv", header = T)
countData <- read.csv("../data/02a_countData.csv", header = T, check.names = F, row.names = 1)
colData %>% select(treatment, subfield)  %>%  summary()

dds <- DESeqDataSetFromMatrix(countData = countData,
                              colData = colData,
                              design = ~ subfield * treatment )

dds$subfield <- factor(dds$subfield, levels=c("DG","CA3", "CA1")) ## specify the factor levels

dds$treatment <- factor(dds$treatment, levels=c("standard.yoked" ,"standard.trained", "conflict.yoked", "conflict.trained")) ## specify the factor levels

dds # view the DESeq object - note numnber of genes
dds <- dds[ rowSums(counts(dds)) > 10, ]  # Pre-filtering genes
dds # view number of genes afternormalization and the number of samples
dds <- DESeq(dds, parallel = TRUE) # Differential expression analysis
#rld <- rlog(dds, blind=FALSE) ## log transformed data
vsd <- vst(dds, blind=FALSE) ## variance stabilized
head(assay(vsd),3)
```

## check for outliers


```{r outliers, fig.height=6}
boxplot(log10(assays(dds)[["cooks"]]), range=0, las=2)
plotDispEsts(dds)

sampleDists <- dist(t(assay(vsd)))
sampleDistMatrix <- as.matrix(sampleDists)

rownames(sampleDistMatrix) <- paste(vsd$subfield, vsd$treatment, sep="-")
colnames(sampleDistMatrix) <- NULL
colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)

pheatmap(sampleDistMatrix,
         clustering_distance_rows=sampleDists,
         clustering_distance_cols=sampleDists,
         col=colors, fontsize = 8)

rownames(sampleDistMatrix) <- (vsd$RNAseqID)
pheatmap(sampleDistMatrix,
         clustering_distance_rows=sampleDists,
         clustering_distance_cols=sampleDists,
         col=colors, fontsize = 8)

```

## Summary 2 way contrasts

This first function shows the total number of up and down regulated genes and the top 3 most significant genes.

```{r twowaycontrasts}
res_summary <- function(mycontrast){
  res <- results(dds, contrast = mycontrast, independentFiltering = T)
  print(mycontrast)
  print(sum(res$padj < 0.1, na.rm=TRUE))
  print(summary(res))
  print(head((res[order(res$padj),]), 5))
  cat("\n")
}

res_summary(c("subfield", "CA1", "DG"))
res_summary(c("subfield", "CA1", "CA3"))
res_summary(c("subfield", "CA3", "DG"))

res_summary(c("treatment", "standard.trained", "standard.yoked"))
res_summary(c("treatment", "conflict.trained", "conflict.yoked"))
res_summary(c("treatment", "conflict.trained", "standard.trained"))
res_summary(c("treatment", "conflict.yoked", "standard.yoked"))
```

This second function only prints the total number of DEGs, but it saves lots of useful info to a df for downstream dataviz.

```{r heatmap_prep}
# note: see resvals fucntion in `functions_RNAseq.R`

contrast1 <- resvals(contrastvector = c("subfield", "CA1", "DG"), mypval = 0.1)  
contrast2 <- resvals(contrastvector = c("subfield", "CA1", "CA3"), mypval = 0.1)  
contrast3 <- resvals(contrastvector = c("subfield", "CA3", "DG"), mypval = 0.1)  
contrast4 <- resvals(contrastvector = c("treatment", "standard.trained", "standard.yoked"), mypval = 0.1)  
contrast5 <- resvals(contrastvector = c("treatment", "conflict.trained", "conflict.yoked"), mypval = 0.1)  
contrast6 <- resvals(contrastvector = c("treatment", "conflict.trained", "standard.trained"), mypval = 0.1)  
contrast7 <- resvals(contrastvector = c("treatment", "conflict.yoked", "standard.yoked"), mypval = 0.1)  
```

```{r pheatmap}
# heatmap with all DEGs
DEGes <- assay(vsd)
DEGes <- cbind(DEGes, contrast1, contrast2, contrast3, contrast4, contrast5, contrast6, contrast7)
DEGes <- as.data.frame(DEGes) # convert matrix to dataframe
DEGes$rownames <- rownames(DEGes)  # add the rownames to the dataframe
DEGes$rownames <- str_to_upper(DEGes$rownames) ## uppercase gene names

names(DEGes)
DEGes$padjmin <- with(DEGes, pmin(# padjsubfieldCA1DG, padjsubfieldCA1CA3, padjsubfieldCA3DG,
                                  padjtreatmentstandard.trainedstandard.yoked, padjtreatmentconflict.trainedconflict.yoked,
                                  padjtreatmentconflict.trainedstandard.trained,padjtreatmentconflict.yokedstandard.yoked)) 
DEGes <- DEGes %>% filter(padjmin < 0.1)

rownames(DEGes) <- DEGes$rownames
drop.cols <-colnames(DEGes[,grep("padj|pval|rownames", colnames(DEGes))])
DEGes <- DEGes %>% dplyr::select(-one_of(drop.cols))
DEGes <- as.matrix(DEGes)
DEGes <- DEGes - rowMeans(DEGes)

df <- as.data.frame(colData(dds)[,c("treatment", "subfield")]) ## matrix to df
rownames(df) <- names(countData)
levels(df$treatment) <- c("standard yoked", "standard trained",  "conflict yoked","conflict trained")


DEGes <- as.matrix(DEGes) 
paletteLength <- 80
myBreaks <- c(seq(min(DEGes), 0, length.out=ceiling(paletteLength/2) + 1), 
              seq(max(DEGes)/paletteLength, max(DEGes), length.out=floor(paletteLength/2)))

pheatmap(DEGes, show_colnames=F, show_rownames = F,
         annotation_col=df, 
         annotation_colors = pheatmapcolors,
         treeheight_row = 0, treeheight_col = 25,
         annotation_row = NA, 
         annotation_legend = TRUE,
         annotation_names_row = FALSE, annotation_names_col = TRUE,
         fontsize = 8, 
         border_color = NA ,
         color = viridis(80),
         cellwidth = 6, 
         clustering_method="average",
         breaks=myBreaks,
         clustering_distance_cols="correlation" 
         )

pheatmap(DEGes, show_colnames=F, show_rownames = F,
         annotation_col=df, annotation_colors = pheatmapcolors, 
         annotation_row = NA, 
         annotation_legend = F,
         annotation_names_row = FALSE, 
         annotation_names_col = FALSE,
         treeheight_row = 0, treeheight_col = 10,
         fontsize = 6, 
         border_color = NA ,
         color = viridis(80),
         height = 2.5, 
         width = 2.5,
         clustering_method="average",
         breaks=myBreaks,
         clustering_distance_cols="correlation", 
         filename = "../figures/02b_RNAseqALL/pheatmap1.pdf"
         )
```

## Principle component analysis

```{r pca}
# create the dataframe using my function pcadataframe
pcadata <- pcadataframe(vsd, intgroup=c("subfield","treatment"), returnData=TRUE)
percentVar <- round(100 * attr(pcadata, "percentVar"))
pcadata$subfieldAPA <- as.factor(paste(pcadata$subfield, pcadata$treatment, sep="_"))
pcadata$subfield <- factor(pcadata$subfield, levels=c("DG","CA3", "CA1"))
pcadata$treatment <- factor(pcadata$treatment, levels=c("standard.yoked","standard.trained",  "conflict.yoked","conflict.trained"))

levels(pcadata$treatment) <- c("standard yoked","standard trained",  "conflict yoked", "conflict trained")

summary(aov(PC1 ~ subfield * treatment, data=pcadata)) 
TukeyHSD((aov(PC1 ~ subfield, data=pcadata)), which = "subfield") 

summary(aov(PC2 ~ subfield * treatment, data=pcadata)) 
TukeyHSD((aov(PC2 ~ subfield, data=pcadata)), which = "subfield") 

PCA12 <- ggplot(pcadata, aes(pcadata$PC1, pcadata$PC2, colour=subfield)) +
    geom_point(size=2, aes(shape=treatment), alpha = 0.8) +
    xlab(paste0("PC1: ", percentVar[1],"%")) +
    ylab(paste0("PC2: ", percentVar[2],"%")) +
    scale_colour_manual(values=c(colorvalsubfield))+ 
   theme_ms()  +
      theme(legend.position= "none") +
    scale_shape_manual(values=c(1, 16, 0, 15), aes(color=colorvalsubfield)) +
  labs(color = "subfield", shape = "treatment") +
  guides(color = FALSE)
PCA12

head(pcadata)
```

## tSNE

```{r tSNE}
vsddf <- as.data.frame(assay(vsd))
vsddf <- as.data.frame(t(vsddf))

euclidist <- dist(vsddf) # euclidean distances between the rows

plot_tSNE <- function(myperplecity, mysubtitle){
  
  print(myperplecity)
  tsne_model <- Rtsne(euclidist, check_duplicates=FALSE, pca=TRUE, perplexity=myperplecity)
  tsne_df = as.data.frame(tsne_model$Y) 
  tsne_df <- cbind(colData, tsne_df)
  tsne_df$subfield <- factor(tsne_df$subfield, levels = c("DG", "CA3", "CA1"))
  tsne_df$treatment <- factor(tsne_df$treatment, 
                              levels = c("standard.yoked" ,"standard.trained", 
                                         "conflict.yoked", "conflict.trained"))

  tnseplot  <- tsne_df %>%
    ggplot(aes(x = V1, y = V2, shape = treatment, color = subfield, label = ID)) +
    geom_point(size = 2) +
    scale_color_manual(values = colorvalsubfield) +
    theme_ms()  +
    theme(legend.position = "bottom",
          legend.title = element_blank(),
          legend.spacing.x = unit(0.01, 'cm'),
          legend.spacing.y = unit(0.01, 'cm')) +
    labs(x = "tSNE 1", y = "tSNE 2", subtitle = mysubtitle) +
    scale_shape_manual(aes(colour=colorvalsubfield), values=c(1, 16, 0, 15)) 
  return(tnseplot)
  
}


p2 <- plot_tSNE(4, "perplexity = 4") + theme(legend.position = "none")
p4 <- plot_tSNE(8, "perplexity = 8") + theme(legend.position = "none")
p6 <- plot_tSNE(10, "perplexity = 10")

mylegend <- get_legend(p6)

top <- plot_grid(p2,p4,p6 + theme(legend.position = "none"), nrow = 1)
topl <- plot_grid(top, mylegend, nrow = 2, rel_heights = c(1,0.1))
topl

```

## pca + tsne

```{r PCA-tSNE}
mylegend <- get_legend(p6)

top <- plot_grid(PCA12, p6 + theme(legend.position = "none"), nrow = 1)
topl <- plot_grid(top, mylegend, nrow = 2, rel_heights = c(1,0.1))
topl

pdf(file="../figures/02b_RNAseqALL/PCA-tSNE.pdf", width=4, height=2.5)
plot(topl)
dev.off()
```



```{r}
write.csv(assay(vsd), file = "../data/02b_vsd.csv", row.names = T)
```