---
title: "02c_rnaseqSubfield"
output: md_document
---

## Subfield analysis

This script is used to identify treatement differences within each subfield, generate volcano plots, venn diagrams, and tables for subsequent GO analyses. The final mutlipanel figures for the manuscript have been inserted just below the subheadings. 

```{r setup, message=F, warning=F}
library(tidyverse)
library(cowplot) ## for some easy to use themes
library(VennDiagram) ## venn diagrams
library(DESeq2) ## for gene expression analysis
library(UpSetR)

## load functions 
source("figureoptions.R")
source("functions_RNAseq.R")

## set output file for figures 
knitr::opts_chunk$set(fig.path = '../figures/02c_rnaseqSubfield/', cache = T)
```


## Get varience stabilized gene expression for each tissue

```{r DGdeseq}
a.colData <- read.csv("../data/02a_colData.csv", header = T)
a.countData <- read.csv("../data/02a_countData.csv", header = T, check.names = F, row.names = 1)

returndds <- function(mytissue){
  print(mytissue)
  colData <- a.colData %>% 
    filter(Punch %in% c(mytissue))  %>% 
  droplevels()
  
  savecols <- as.character(colData$RNAseqID) 
  savecols <- as.vector(savecols) 
  countData <- a.countData %>% dplyr::select(one_of(savecols)) 

  ## create DESeq object using the factors Punch and APA
  dds <- DESeqDataSetFromMatrix(countData = countData,
                              colData = colData,
                              design = ~ APA2)

  dds <- dds[ rowSums(counts(dds)) > 1, ]  # Pre-filtering genes with 0 counts
   # Differential expression analysis
  return(DESeq(dds))
}

DGdds <- returndds("DG") 
CA1dds <- returndds("CA1") 
CA3dds <- returndds("CA3") 
```

## Consistent versus yoked-consistent

```{r consyokcons}
plot.cons.yokcons <- function(mydds, mytissue){
  print(mytissue)
  
  res <- results(mydds, contrast =c("APA2", "consistent", "yoked_consistent"),
                 independentFiltering = T, alpha = 0.1)
  print(summary(res))

  data <- data.frame(gene = row.names(res),
                   padj = res$padj, 
                   logpadj = -log10(res$padj),
                   lfc = res$log2FoldChange)
  data <- na.omit(data)
  data <- data %>%
    dplyr::mutate(direction = ifelse(data$lfc > 1 & data$padj < 0.1, 
                        yes = "consistent", 
                        no = ifelse(data$lfc < -1 & data$padj < 0.1, 
                                    yes = "yoked_consistent", 
                                    no = "NS")))

  write.csv(data, file = paste0("../data/02c_", mytissue, "_consyokcons.csv", sep = ""))
  
  volcano <- ggplot(data, aes(x = lfc, y = logpadj)) + 
    geom_point(aes(color = factor(direction)), size = 0.5, alpha = 0.5, na.rm = T) + 
    theme_cowplot(font_size = 7, line_size = 0.25) +
    geom_hline(yintercept = 1,  size = 0.25, linetype = 2) + 
    scale_color_manual(values = volcano1,
                      name = "Higher expression in",
                      breaks = c("yoked_consistent", "NS", "consistent"))  + 
    scale_y_continuous(limits=c(0, 12.5)) +
    scale_x_continuous(limits=c(-10, 10),
                        name="Log fold difference")+
    ylab(paste0("log10 p-value")) +  
    labs(subtitle = mytissue) +
    theme(legend.position = "bottom",
          legend.margin=margin(t=-0.25, r=0, b=0, l=0, unit="cm")) 
  plot(volcano)
}


DGconsyokcons <-  plot.cons.yokcons(DGdds, "DG") + theme(legend.position = "none")
CA3consyokcons <-  plot.cons.yokcons(CA3dds, "CA3") + theme(legend.position = "none")
CA1consyokcons <-  plot.cons.yokcons(CA1dds, "CA1") + theme(legend.position = "none")

training <- plot_grid(DGconsyokcons, CA3consyokcons, CA1consyokcons, nrow = 1,
                      labels = "AUTO",
                      label_size = 7)
legend <- get_legend(plot.cons.yokcons(DGdds, "DG"))

p <- plot_grid(training, legend, nrow = 2, rel_heights = c(3, .3))
p

pdf(file="../figures/02c_rnaseqSubfield/volcano-consyokcons.pdf", width=5, height=2)
plot(p)
dev.off()
```

## Confict versus Consistent

```{r confcons}
plot.conf.cons <- function(mydds, mytissue){
  
  print(mytissue)
  
  res <- results(mydds, contrast =c("APA2", "conflict", "consistent"),
                 independentFiltering = T, alpha = 0.1)
  print(summary(res))

  data <- data.frame(gene = row.names(res),
                   padj = res$padj, 
                   logpadj = -log10(res$padj),
                   lfc = res$log2FoldChange)
  data <- na.omit(data)
  
  data <- data %>%
    dplyr::mutate(direction = ifelse(data$lfc > 1 & data$padj < 0.1, 
                        yes = "conflict", 
                        no = ifelse(data$lfc < -1 & data$padj < 0.1, 
                                    yes = "consistent", 
                                    no = "NS")))
  
  write.csv(data, file = paste0("../data/02c_", mytissue, "_confcons.csv", sep = ""))
  
  volcano <- ggplot(data, aes(x = lfc, y = logpadj)) + 
    geom_point(aes(color = factor(direction)), size = 0.5, alpha = 0.5, na.rm = T) + 
    theme_cowplot(font_size = 7, line_size = 0.25) +
    geom_hline(yintercept = 1,  size = 0.25, linetype = 2) + 
    scale_color_manual(values = volcano2,
                       breaks = c("consistent", "NS", "conflict"),
                      name = NULL)  + 
    scale_y_continuous(limits=c(0, 12.5)) +
    scale_x_continuous(limits=c(-22, 22),
                        name="Log fold difference")+
    ylab(paste0("log10 p-value")) +  
    labs(subtitle = mytissue) +
    theme(panel.grid.minor=element_blank(),
          legend.position = "bottom",
          legend.spacing.x = unit(-0.1, 'cm'),
          panel.grid.major=element_blank(),
          legend.margin=margin(t=-0.25, r=0, b=0, l=0, unit="cm")) 
  plot(volcano)
}


DGconflict <-  plot.conf.cons(DGdds, "DG")
CA3conflict <-  plot.conf.cons(CA3dds, "CA3")
CA1conflict <-  plot.conf.cons(CA1dds, "CA1")

plot_grid(DGconflict, CA3conflict, CA1conflict, nrow = 1)
```


## Yoked confict versus yoked consistent

```{r yokeconfyokcons}
plot.yokconf.yokcons <- function(mydds, mytissue){
  
  print(mytissue)
  
  res <- results(mydds, contrast =c("APA2", "yoked_conflict", "yoked_consistent"),
                 independentFiltering = T, alpha = 0.1)
  print(summary(res))

  data <- data.frame(gene = row.names(res),
                   padj = res$padj, 
                   logpadj = -log10(res$padj),
                   lfc = res$log2FoldChange)
  data <- na.omit(data)
  data <- data %>%
    dplyr::mutate(direction = ifelse(data$lfc > 1 & data$padj < 0.1, 
                        yes = "yoked_conflict", 
                        no = ifelse(data$lfc < -1 & data$padj < 0.1, 
                                    yes = "yoked_consistent", 
                                    no = "NS")))
  
  write.csv(data, file = paste0("../data/02c_", mytissue, "_yokeconfyokcons.csv", sep = ""))
  
  volcano <- ggplot(data, aes(x = lfc, y = logpadj)) + 
    geom_point(aes(color = factor(direction)), size = 0.5, alpha = 0.5, na.rm = T) + 
    theme_cowplot(font_size = 7, line_size = 0.25) +
    geom_hline(yintercept = 1,  size = 0.25, linetype = 2) + 
    scale_color_manual(values = volcano3,
                       breaks = c("yoked_consistent", "NS", "yoked_conflict"),
                      name = NULL)  + 
    scale_y_continuous(limits=c(0, 12.5)) +
    scale_x_continuous(limits=c(-22, 22),
                        name="Log fold difference")+
    ylab(paste0("log10 p-value")) +  
    labs(subtitle = mytissue) +
    theme(panel.grid.minor=element_blank(),
          legend.position = "bottom",
          legend.spacing.x = unit(1.0, 'cm'),
          panel.grid.major=element_blank(),
          legend.margin=margin(t=-0.25, r=2, b=0, l=0, unit="cm")) 
  plot(volcano)
}


DGyoked <-  plot.yokconf.yokcons(DGdds, "DG")
CA3yoked <-  plot.yokconf.yokcons(CA3dds, "CA3")
CA1yoked <-  plot.yokconf.yokcons(CA1dds, "CA1")

plot_grid(DGyoked, CA3yoked, CA1yoked, nrow = 1)
```

# pkmzz


```{r pkmz}
plotCounts(DGdds, "Prkcz", intgroup = "APA2", normalized = TRUE, main="Prkcz in DG")
plotCounts(CA3dds, "Prkcz", intgroup = "APA2", normalized = TRUE, main="Prkcz in CA3")
plotCounts(CA1dds, "Prkcz", intgroup = "APA2", normalized = TRUE, main="Prkcz in CA1")
```





## Upset plots

What genes overlap within cetain comparisons?

```{r upsetplotprep}
a.colData <- read.csv("../data/02a_colData.csv", header = T)
a.countData <- read.csv("../data/02a_countData.csv", header = T, check.names = F, row.names = 1)

eachsubfield <- levels(a.colData$Punch)

listofDEGs <- function(group1, group2){
  res <- results(dds, contrast = c("APA2", group1, group2), independentFiltering = T)
  data <- data.frame(gene = row.names(res),
                     lfc = res$log2FoldChange,
                     padj = res$padj,
                     tissue = i,
                     comparison = paste(group1, group2, sep = "-"))
  data <- data %>% dplyr::filter(padj < 0.1) %>% droplevels()
  return(data)
}

for(i in eachsubfield){
  
  colData <- a.colData %>% 
    dplyr::filter(Punch == i)  %>%
    droplevels()
  print(i)
  
  savecols <- as.character(colData$RNAseqID) 
  savecols <- as.vector(savecols) 
  countData <- a.countData %>% dplyr::select(one_of(savecols)) 

## create DESeq object using the factors Punch and APA
dds <- DESeqDataSetFromMatrix(countData = countData,
                              colData = colData,
                              design = ~ APA2)

dds # view the DESeq object - note numnber of genes
dds <- dds[ rowSums(counts(dds)) > 1, ]  # Pre-filtering genes with 0 counts
dds <- DESeq(dds) # Differential expression analysis

A <- listofDEGs("consistent","yoked_consistent")
B <- listofDEGs("conflict","consistent")
C <- listofDEGs("conflict","yoked_conflict")
D <- listofDEGs("yoked_conflict","yoked_consistent")


all <- rbind(A,B,C,D)

write.csv(all, file = paste("../data/02c_",i,"forupset.csv", sep = ""), row.names = F)
}

DG <- read.csv("../data/02c_DGforupset.csv")  
CA1<- read.csv("../data/02c_CA1forupset.csv")  
CA3 <- read.csv("../data/02c_CA3forupset.csv") 

all <- rbind(DG,CA1,CA3)

levels(all$comparison) <- c("conf-cons",
                            "conf-yconf", 
                            "cons-ycons", 
                            "yconf-ycons")

all$significant <- paste(all$tissue, all$comparison, sep = "-")

all <- all %>%
  select(gene,significant)

myupsetdf <- all %>%
  mutate(yesno = 1) %>%
  distinct %>%
  spread(significant, yesno, fill = 0)

write.csv(myupsetdf, "../data/02c_upsetdf.csv")

```

```{r upsetplot}

upset(myupsetdf, keep.order = T)

names(myupsetdf)

trained <- myupsetdf %>%
  select(gene, 'DG-cons-ycons', 'CA3-cons-ycons', 'CA1-cons-ycons')

colnames(trained) <- c("gene", "DG", "CA3", "CA1")


pdf(file="../figures/02c_rnaseqSubfield/upsettraining.pdf",  onefile=FALSE) # or other device
upset(trained, keep.order = T,
      sets = c("CA1", "CA3", "DG"),
      sets.bar.color=c("#7570b3","#1b9e77", "#d95f02"),
      queries = list(list(query = intersects, params = list("CA1"), color = "#ca0020", active = T),
                     list(query = intersects, params = list("DG"), color = "#ca0020", active = T),
                     list(query = intersects, params = list("CA3"), color = "#ca0020", active = T),
                     list(query = intersects, params = list("CA1", "DG"), 
                          color = "#ca0020", active = T)),
      text.scale = 2,
      sets.x.label = NULL,
      #point.size = 1, line.size = 1,
      mb.ratio = c(0.6, 0.4)
)
dev.off()


shared <- myupsetdf %>%
  select(gene, "CA1-yconf-ycons", "CA1-cons-ycons", "DG-cons-ycons")
colnames(shared) <- c("gene", "CA1stress", "CA1learn", "DGlearn")

# ca1 DG learning
shared %>% filter(CA1learn == 1 & DGlearn == 1)


# CA1 and DG learning only (none in stress)
shared %>% filter(CA1learn == 1 & DGlearn == 1 & CA1stress == 0)

# learning only CA1
shared %>% filter(CA1learn == 1 & CA1stress == 0)

# stress only CA1
shared %>% filter(CA1learn == 0 & CA1stress == 1)

# stress only CA1 and DG
shared %>% filter(CA1learn == 0 & CA1stress == 1 & DGlearn == 0)
```
