---
title: "02c_rnaseqSubfield"
output: md_document
---

## Subfield analysis

This script is used to identify treatement differences within each subfield, generate volcano plots, venn diagrams, and tables for subsequent GO analyses. The final mutlipanel figures for the manuscript have been inserted just below the subheadings. 

```{r setup, message=F, warning=F}
library(tidyverse)
library(cowplot) ## for some easy to use themes
library(DESeq2) ## for gene expression analysis
library(UpSetR)
#devtools::install_github("clauswilke/ggtextures")
library(ggtextures)
library(magick)

## load functions 
source("figureoptions.R")
source("functions_RNAseq.R")

## set output file for figures 
knitr::opts_chunk$set(fig.path = '../figures/02c_rnaseqSubfield/', cache = T)
```


## Get varience stabilized gene expression for each tissue

```{r DGdeseq}
a.colData <- read.csv("../data/02a_colData.csv", header = T)
a.countData <- read.csv("../data/02a_countData.csv", header = T, check.names = F, row.names = 1)

returndds <- function(mytissue){
  print(mytissue)
  colData <- a.colData %>% 
    filter(Punch %in% c(mytissue))  %>% 
  droplevels()
  
  savecols <- as.character(colData$RNAseqID) 
  savecols <- as.vector(savecols) 
  countData <- a.countData %>% dplyr::select(one_of(savecols)) 

  ## create DESeq object using the factors Punch and APA
  dds <- DESeqDataSetFromMatrix(countData = countData,
                              colData = colData,
                              design = ~ APA2)

  dds <- dds[ rowSums(counts(dds)) > 1, ]  # Pre-filtering genes with 0 counts
   # Differential expression analysis
  return(DESeq(dds))
}

DGdds <- returndds("DG") 
CA1dds <- returndds("CA1") 
CA3dds <- returndds("CA3") 


returnvsds <- function(mydds, vsdfilename){
  dds <- mydds
  vsd <- vst(dds, blind=FALSE) ## variance stabilized
  print(head(assay(vsd),3))
  return(write.csv(assay(vsd), file = vsdfilename, row.names = T))
}

returnvsds(DGdds, "../data/02c_DGvsd.csv")
returnvsds(CA1dds, "../data/02c_CA1vsd.csv")
returnvsds(CA3dds, "../data/02c_CA3vsd.csv")
```

## Consistent versus yoked-consistent

```{r consyokcons}
plot.cons.yokcons <- function(mydds, mytissue, mytitle){
  print(mytissue)
  
  res <- results(mydds, contrast =c("APA2", "consistent", "yoked_consistent"),
                 independentFiltering = T, alpha = 0.1)
  print(summary(res))

  data <- data.frame(gene = row.names(res),
                   padj = res$padj, 
                   logpadj = -log10(res$padj),
                   lfc = res$log2FoldChange)
  data <- na.omit(data)
  data <- data %>%
    dplyr::mutate(direction = ifelse(data$lfc > 1 & data$padj < 0.1, 
                        yes = "consistent", 
                        no = ifelse(data$lfc < -1 & data$padj < 0.1, 
                                    yes = "yoked\nconsistent", 
                                    no = "NS")))


  write.csv(data, file = paste0("../data/02c_", mytissue, "_consyokcons.csv", sep = ""), row.names = F)
  
  volcano <- ggplot(data, aes(x = lfc, y = logpadj)) + 
    geom_point(aes(color = factor(direction)), size = 0.5, alpha = 0.75, na.rm = T) + 
    theme_cowplot(font_size = 7, line_size = 0.25) +
    geom_hline(yintercept = 1,  size = 0.25, linetype = 2) + 
    scale_color_manual(values = volcano1,
                      name = "higher in",
                      breaks = c("yoked\nconsistent", "NS", "consistent"))  + 
    scale_y_continuous(limits=c(0, 12.5)) +
    scale_x_continuous(limits=c(-10, 10),
                        name="Log fold difference")+
    ylab(paste0("log10 p-value")) +  
    labs(subtitle = mytitle) +
    theme(legend.position = "bottom",
          legend.spacing.x = unit(0.1, 'cm'),
          #legend.text=element_text(size=4),
          legend.title = element_text(size=6),
          legend.key.size = unit(0.2, "cm"),
          legend.margin=margin(t=-0.1, r=0, b=0, l=-0.1, unit="cm")) 
  plot(volcano)
}


DGconsyokcons <-  plot.cons.yokcons(DGdds, "DG", "DG train") + theme(legend.position = "none")  
CA3consyokcons <-  plot.cons.yokcons(CA3dds, "CA3", "CA3 train")  
CA1consyokcons <-  plot.cons.yokcons(CA1dds, "CA1", "CA1 train")  + theme(legend.position = c(0.6,0.9))  

training <- plot_grid(DGconsyokcons,CA1consyokcons, nrow = 1,
                      labels = "AUTO",
                      label_size = 7,
                      rel_widths = c(0.5,0.5))
training
```

## Confict versus Consistent

```{r confcons}
plot.conf.cons <- function(mydds, mytissue){
  
  print(mytissue)
  
  res <- results(mydds, contrast =c("APA2", "conflict", "consistent"),
                 independentFiltering = T, alpha = 0.1)
  print(summary(res))

  data <- data.frame(gene = row.names(res),
                   padj = res$padj, 
                   logpadj = -log10(res$padj),
                   lfc = res$log2FoldChange)
  data <- na.omit(data)
  
  data <- data %>%
    dplyr::mutate(direction = ifelse(data$lfc > 1 & data$padj < 0.1, 
                        yes = "conflict", 
                        no = ifelse(data$lfc < -1 & data$padj < 0.1, 
                                    yes = "consistent", 
                                    no = "NS")))
  
  write.csv(data, file = paste0("../data/02c_", mytissue, "_confcons.csv", sep = ""), row.names = F)
  
  volcano <- ggplot(data, aes(x = lfc, y = logpadj)) + 
    geom_point(aes(color = factor(direction)), size = 0.5, alpha = 0.75, na.rm = T) + 
    theme_cowplot(font_size = 7, line_size = 0.25) +
    geom_hline(yintercept = 1,  size = 0.25, linetype = 2) + 
    scale_color_manual(values = volcano2,
                       breaks = c("consistent", "NS", "conflict"),
                      name = "higher in")  + 
    scale_y_continuous(limits=c(0, 12.5)) +
    scale_x_continuous(limits=c(-10, 10),
                        name="Log fold difference")+
    ylab(paste0("log10 p-value")) +  
    labs(subtitle = mytissue) +
    theme(panel.grid.minor=element_blank(),
          legend.position = "bottom",
          legend.spacing.x = unit(-0.1, 'cm'),
          panel.grid.major=element_blank(),
          legend.margin=margin(t=-0.25, r=0, b=0, l=0, unit="cm")) 
  plot(volcano)
}


DGconflict <-  plot.conf.cons(DGdds, "DG")
CA3conflict <-  plot.conf.cons(CA3dds, "CA3")
CA1conflict <-  plot.conf.cons(CA1dds, "CA1")

plot_grid(DGconflict, CA3conflict, CA1conflict, nrow = 1)
```


## Yoked confict versus yoked consistent

```{r yokeconfyokcons}
plot.yokconf.yokcons <- function(mydds, mytissue, mytitle){
  
  print(mytissue)
  
  res <- results(mydds, contrast =c("APA2", "yoked_conflict", "yoked_consistent"),
                 independentFiltering = T, alpha = 0.1)
  print(summary(res))

  data <- data.frame(gene = row.names(res),
                   padj = res$padj, 
                   logpadj = -log10(res$padj),
                   lfc = res$log2FoldChange)
  data <- na.omit(data)
  data <- data %>%
    dplyr::mutate(direction = ifelse(data$lfc > 1 & data$padj < 0.1, 
                        yes = "yoked\nconsistent", 
                        no = ifelse(data$lfc < -1 & data$padj < 0.1, 
                                    yes = "yoked\nconflict", 
                                    no = "NS")))
  
  data$direction <- factor(data$direction, levels = c("yoked\nconsistent", "NS", "yoked\nconflict"))
  
  write.csv(data, file = paste0("../data/02c_", mytissue, "_yokeconfyokcons.csv", sep = ""), row.names = F)
  
  volcano <- ggplot(data, aes(x = lfc, y = logpadj, color = direction)) + 
    geom_point(size = 0.5, alpha = 0.75, na.rm = T) + 
    theme_cowplot(font_size = 7, line_size = 0.25) +
    geom_hline(yintercept = 1,  size = 0.25, linetype = 2) + 
    scale_color_manual(values = volcano3,
                       name = "higher in")  + 
    scale_y_continuous(limits=c(0, 12.5)) +
    scale_x_continuous(limits=c(-10, 10),
                        name="Log fold difference")+
    ylab(paste0("log10 p-value")) +  
    labs(subtitle = mytitle) +
    theme(legend.position = "bottom",
          legend.spacing.x = unit(0.1, 'cm'),
          #legend.text=element_text(size=4),
          legend.key.size = unit(0.2, "cm"),
          legend.margin=margin(t=-0.1, r=0, b=0, l=-0.1, unit="cm")) 
  plot(volcano)  
}

DGyoked <-  plot.yokconf.yokcons(DGdds, "DG", "DG stress")
CA3yoked <-  plot.yokconf.yokcons(CA3dds, "CA3", "CA3 stress")
CA1yoked <-  plot.yokconf.yokcons(CA1dds, "CA1", "CA1 stress")  

plot_grid(DGyoked, CA3yoked, CA1yoked, nrow = 1)
```


# pkmz

```{r pkmz}
plotCounts(DGdds, "Prkcz", intgroup = "APA2", normalized = TRUE, main="Prkcz in DG")
plotCounts(CA3dds, "Prkcz", intgroup = "APA2", normalized = TRUE, main="Prkcz in CA3")
plotCounts(CA1dds, "Prkcz", intgroup = "APA2", normalized = TRUE, main="Prkcz in CA1")

```

## genes that are correlated with number of entrances 

Requires anlaysis of `04_integration.Rmd` first.

```{r DGcorrelations}
plotCounts(DGdds, "Acan", intgroup = "APA2", normalized = TRUE, main="Acan in DG")
plotCounts(DGdds, "Amigo2", intgroup = "APA2", normalized = TRUE, main="Amigo2 in DG")
plotCounts(DGdds, "Armcx5", intgroup = "APA2", normalized = TRUE, main="Armcx5 in DG")
plotCounts(DGdds, "Ptgs2", intgroup = "APA2", normalized = TRUE, main="Ptgs2 in DG")
plotCounts(DGdds, "Rgs2", intgroup = "APA2", normalized = TRUE, main="Rgs2 in DG")
plotCounts(DGdds, "Syt4", intgroup = "APA2", normalized = TRUE, main="Syt4 in DG")
```


## Upset plots

What genes overlap within cetain comparisons?

```{r upsetplotprep}
a.colData <- read.csv("../data/02a_colData.csv", header = T)
a.countData <- read.csv("../data/02a_countData.csv", header = T, check.names = F, row.names = 1)

eachsubfield <- levels(a.colData$Punch)

listofDEGs <- function(group1, group2){
  res <- results(dds, contrast = c("APA2", group1, group2), independentFiltering = T)
  data <- data.frame(gene = row.names(res),
                     lfc = res$log2FoldChange,
                     padj = res$padj,
                     tissue = i,
                     comparison = paste(group1, group2, sep = "-"))
  data <- data %>% dplyr::filter(padj < 0.1) %>% droplevels()
  return(data)
}

for(i in eachsubfield){
  
  colData <- a.colData %>% 
    dplyr::filter(Punch == i)  %>%
    droplevels()
  print(i)
  
  savecols <- as.character(colData$RNAseqID) 
  savecols <- as.vector(savecols) 
  countData <- a.countData %>% dplyr::select(one_of(savecols)) 

## create DESeq object using the factors Punch and APA
dds <- DESeqDataSetFromMatrix(countData = countData,
                              colData = colData,
                              design = ~ APA2)

dds # view the DESeq object - note numnber of genes
dds <- dds[ rowSums(counts(dds)) > 1, ]  # Pre-filtering genes with 0 counts
dds <- DESeq(dds) # Differential expression analysis

A <- listofDEGs("consistent","yoked_consistent")
B <- listofDEGs("conflict","consistent")
C <- listofDEGs("conflict","yoked_conflict")
D <- listofDEGs("yoked_conflict","yoked_consistent")


all <- rbind(A,B,C,D)

write.csv(all, file = paste("../data/02c_",i,"forupset.csv", sep = ""), row.names = F)
}

DG <- read.csv("../data/02c_DGforupset.csv")  
CA1<- read.csv("../data/02c_CA1forupset.csv")  
CA3 <- read.csv("../data/02c_CA3forupset.csv") 


# upset plot without direction
all <- rbind(DG,CA1,CA3)

levels(all$comparison) <- c("conf-cons",
                            "conf-yconf", 
                            "cons-ycons", 
                            "yconf-ycons")

all$significant <- paste(all$tissue, all$comparison, sep = "-")

myupsetdf <- all %>%
  select(gene,significant) %>%
  mutate(yesno = 1) %>%
  distinct %>%
  spread(significant, yesno, fill = 0)
head(myupsetdf)

write.csv(myupsetdf, "../data/02c_upsetdf.csv")


# upset plot with direction, only CA1learn, CA1 stress, and DGlearn
head(all) 
all$direction <- ifelse(all$lfc > 0, "up", "down")
all$sigdir <- paste(all$significant, all$direction, sep = "-")

myupsetslim  <- all %>%
  filter(significant %in% c("DG-cons-ycons", "CA1-cons-ycons", "CA1-yconf-ycons")) %>%
  select(gene,sigdir) %>%
  mutate(yesno = 1) %>%
  distinct %>%
  spread(sigdir, yesno, fill = 0)
head(myupsetslim)
row.names(myupsetslim) <- myupsetslim$gene
myupsetslim$gene <- NULL
colSums(myupsetslim)
myupsetslim$gene <- row.names(myupsetslim)

```

```{r upsetplot}
upset(myupsetdf, keep.order = T)

trained <- myupsetdf %>%
  select(gene, 'DG-cons-ycons', 'CA3-cons-ycons', 'CA1-cons-ycons')

colnames(trained) <- c("gene", "DG", "CA3", "CA1")


pdf(file="../figures/02c_rnaseqSubfield/upsettraining.pdf",  onefile=FALSE) # or other device
upset(trained, keep.order = T,
      sets = c("CA1", "CA3", "DG"),
      sets.bar.color=c("#7570b3","#1b9e77", "#d95f02"),
      queries = list(list(query = intersects, params = list("CA1"), color = "#ca0020", active = T),
                     list(query = intersects, params = list("DG"), color = "#ca0020", active = T),
                     list(query = intersects, params = list("CA3"), color = "#ca0020", active = T),
                     list(query = intersects, params = list("CA1", "DG"), 
                          color = "#ca0020", active = T)),
      text.scale = 2,
      sets.x.label = NULL,
      #point.size = 1, line.size = 1,
      mb.ratio = c(0.6, 0.4)
)
dev.off()
```

Create a list of genes afected by stress and learning. Save. THen use to filter out nonspecific gene expression responses

```{r listofDEGs}
shared <- myupsetdf %>%
  select(gene, "CA1-yconf-ycons", "CA1-cons-ycons", "DG-cons-ycons")
colnames(shared) <- c("gene", "CA1stress", "CA1learn", "DGlearn")


# CA1 and DG learning only (none in stress)
shared %>% filter(CA1learn == 1 & DGlearn == 1 & CA1stress == 0) %>%
  select(gene) 

# learning and stress CA1
CA1learnstress <- shared %>% filter(CA1learn == 1 & CA1stress == 1) %>%
  select(gene)
head(CA1learnstress)

write.csv(CA1learnstress, "../data/02_CA1learningstressgenes.csv", row.names = F)
```


```{r myupset}
# All degs
updown <- read.csv("../data/02c_setsize_updown.csv") 
updown$direction <- factor(updown$direction,  levels = c("yoked_consistent", "yoked_conflict", "consistent"))
updown$set <- factor(updown$set,  levels = c("DGtrain", "CA1train", "CA1stress"))
updown$status <- factor(updown$status,  levels = c("train", "stress"))

levels(updown$set)[levels(updown$set)=="DGtrain"] <- "DG train"
levels(updown$set)[levels(updown$set)=="CA1train"]   <- "CA1 train"
levels(updown$set)[levels(updown$set)=="CA1stress"]   <- "CA1 stress"
levels(updown$direction)[levels(updown$direction)=="yoked_consistent"] <- "yoked\nconsistent"
levels(updown$direction)[levels(updown$direction)=="yoked_conflict"]   <- "yoked\nconflict"

# unique and shared
shared2 <- read.csv("../data/02c_intersect_updown_shared2.csv") 
shared2$group <- paste(shared2$set, shared2$direction, sep = " ")
shared2$set <- factor(shared2$set,  levels = c("DGtrain", "CA1train", "CA1stress"))
shared2$direction <- factor(shared2$direction,  
                            levels = c("yoked_consistent", "yoked_conflict", "consistent"))
shared2$group <-factor(shared2$group,  
                       levels = c("DGtrain yoked_consistent", "DGtrain consistent", 
                                  "CA1train yoked_consistent","CA1train consistent",
                                  "CA1stress yoked_consistent", "CA1stress yoked_conflict" ))
shared2$status <- factor(shared2$status,  levels = c(  "stress","train","unique"))

levels(shared2$direction)[levels(shared2$direction)=="yoked_consistent"] <- "yoked\nconsistent"
levels(shared2$direction)[levels(shared2$direction)=="yoked_conflict"]   <- "yoked\nconflict"
levels(shared2$status)[levels(shared2$status)=="stress"]   <- "shared"
levels(shared2$status)[levels(shared2$status)=="train"]   <- "shared"
```

```{r newbarplot}

d1 <- ggplot(updown, aes(x=direction, y=setsize, fill = direction)) +
  geom_bar(stat="identity", position=position_dodge()) +
  theme_cowplot(font_size = 6, line_size = 0.25) +
  scale_fill_manual(values = c("#404040", "#bababa", "#ca0020"),
                    name = NULL)  +
  labs(x = NULL, y = "Total DEGs") +
  scale_y_continuous(limits = c(0, 560),
                     breaks = c(0,125,250,375,500)) +
    facet_wrap(~set, scales = "free_x") +
    theme(axis.text.x = element_blank(),
          legend.position = "none",
          legend.text=element_text(size=4),
          legend.key.size = unit(0.2, "cm"),
          #legend.margin=margin(t=-0.25, r=0, b=0, l=0, unit="cm"),
          strip.text.x = element_text(size = 5),
          strip.background = element_rect(colour=NA, fill=NA),
        panel.border = element_rect(fill = NA, color = "black"),
        legend.margin=margin(t=-0.1, r=0, b=-0.1, l=0, unit="cm"))



images = c(
  unique = "../figures/00_schematics/patterns_blank.png",
  shared = "../figures/00_schematics/patterns_crosed-lines.png")
images

head(shared2)

d2 <- ggplot(shared2, aes(x=direction, y=setsize, image = status)) +
  geom_textured_bar(stat = "identity") +
  theme_cowplot(font_size = 6, line_size = 0.25) +
  scale_image_manual(values = images,
                     name = NULL) +
  labs(x = "subfield * treatment", y = "Shared DEGs") +
    scale_y_continuous(limits = c(0, 560),
                     breaks = c(0,125,250,375,500)) +
  theme(axis.text.x=element_text(angle=60, vjust = 1, hjust = 1),
        legend.position = c(0.05, 0.9),
        legend.text=element_text(size=4),
        legend.key.size = unit(0.2, "cm"),
        strip.text.x = element_text(size = 0),
        #axis.text=element_text(size=7),
        strip.background = element_rect(colour=NA, fill=NA),
        panel.border = element_rect(fill = NA, color = "black"),
        legend.margin=margin(t=-0.1, r=0, b=-0.1, l=-0.1, unit="cm")) +
  facet_wrap(~set, scales = "free_x") 
d2

newbarplots <- plot_grid(d1,d2, nrow = 2, rel_heights =c(0.4,0.6))
newbarplots

pdf(file="../figures/02c_rnaseqSubfield/barplots.pdf", width=1.9, height=2.15)
plot(newbarplots)    
dev.off() 

```

```{r combo, cache=FALSE}
bottomplots <- plot_grid(CA1yoked, newbarplots,
                    labels = c("C", "D"),
                    label_size = 7)

volcanos <- plot_grid(training, bottomplots, nrow = 2,
                      rel_heights = c(0.475,0.525)) 
volcanos

#althought this technically now contains volcanos and bar plots

pdf(file="../figures/02c_rnaseqSubfield/volcanos.pdf", width=3.15, height=4)
plot(volcanos)    
dev.off()     
```

