---
title: "02c_rnaseqSubfield"
author: "Rayna M Harris"
date: "02/13/19"
output: md_document
---

## Subfield analysis

This script is used to identify treatement differences within each subfield, generate volcano plots, venn diagrams, and tables for subsequent GO analyses. The final mutlipanel figures for the manuscript have been inserted just below the subheadings. 

```{r setup, message=F, warning=F}
library(tidyverse)
library(cowplot) ## for some easy to use themes
library(car) ## stats
library(VennDiagram) ## venn diagrams
library(pheatmap) ## awesome heatmaps
library(viridis) # for awesome color pallette
library(reshape2) ## for melting dataframe
library(DESeq2) ## for gene expression analysis
library(edgeR)  ## for basic read counts status
library(magrittr) ## to use the weird pipe
library(genefilter)  ## for PCA fuction
library(xtable) # for latex or html tables
library(ggrepel) # for labeling points

## load functions 
source("figureoptions.R")
source("functions_RNAseq.R")

## set output file for figures 
knitr::opts_chunk$set(fig.path = '../figures/02c_rnaseqSubfield/', cache = T)
```

## DG

The most notable comparison within DG is the consistent verses yoked-consistent.


```{r DGdeseq}
colData <- read.csv("../data/02a_colData.csv", header = T)
countData <- read.csv("../data/02a_countData.csv", header = T, check.names = F, row.names = 1)

colData <- colData %>% 
  filter(Punch %in% c("DG"))  %>% 
  droplevels()

savecols <- as.character(colData$RNAseqID) 
savecols <- as.vector(savecols) 
countData <- countData %>% dplyr::select(one_of(savecols)) 

colData %>% select(APA2,Punch)  %>%  summary()


## create DESeq object using the factors Punch and APA
dds <- DESeqDataSetFromMatrix(countData = countData,
                              colData = colData,
                              design = ~ APA2)

dds # view the DESeq object - note numnber of genes
dds <- dds[ rowSums(counts(dds)) > 1, ]  # Pre-filtering genes with 0 counts
dds # view number of genes afternormalization and the number of samples
dds <- DESeq(dds) # Differential expression analysis
vsd <- getVarianceStabilizedData(dds)

write.csv(colData, file = "../data/02c_DGcolData.csv", row.names = T)
write.csv(vsd, file = "../data/02c_DGvsd.csv", row.names = T)
```


```{r DG}
###  "consistent", "yoked_consistent"
res <- results(dds, contrast =c("APA2", "consistent", "yoked_consistent"), independentFiltering = T, alpha = 0.1)
summary(res)

resOrdered <- res[order(res$padj),]
head(resOrdered, 10)

data <- data.frame(gene = row.names(res),
                   padj = res$padj, 
                   logpadj = -log10(res$padj),
                   lfc = res$log2FoldChange)
data <- na.omit(data)
data <- data %>%
  mutate(direction = ifelse(data$lfc > 1 & data$padj < 0.1, 
                        yes = "consistent", 
                        no = ifelse(data$lfc < -1 & data$padj < 0.1, 
                                    yes = "yoked_consistent", 
                                    no = "neither")))
DGvolcano <- ggplot(data, aes(x = lfc, y = logpadj)) + 
  geom_point(aes(color = factor(direction)), size = 1, alpha = 0.5, na.rm = T) + # add gene points
  theme_cowplot(font_size = 8, line_size = 0.25) +
  geom_hline(yintercept = 1,  size = 0.25, linetype = 2) + 
  scale_color_manual(values = volcano1)  + 
  scale_y_continuous(limits=c(0, 8)) +
  scale_x_continuous( limits=c(-3, 3),
                      name="Log fold change")+
  ylab(paste0("log10 p-value")) +       
  theme(panel.grid.minor=element_blank(),
        legend.position = "none", # remove legend 
        panel.grid.major=element_blank()) 
DGvolcano

pdf(file="../figures/02c_rnaseqSubfield/DGvolcano.pdf", width=1.5, height=2)
plot(DGvolcano)
dev.off()


# save DEGs
DGvolcanoDEGs <- data %>%
  filter(direction != "neither") %>%
  arrange(desc(lfc))
head(DGvolcanoDEGs,10)
write.csv(DGvolcanoDEGs, "../data/DG-consistent-yokedconsistent.csv", row.names = F)

# are any protein kinases differentially expressed?
pkcs <- data[grep("Prkc", data$gene), ]
pkcs # no pkcs are differentially expressed

## go setup
table(res$padj<0.1)
logs <- data.frame(cbind("gene"=row.names(res),"logP"=round(-log(res$pvalue+1e-10,10),1)))
logs$logP=as.numeric(as.character(logs$logP))
sign <- rep(1,nrow(logs))
sign[res$log2FoldChange<0]=-1  ##change to correct model
table(sign)
logs$logP <- logs$logP*sign
write.csv(logs, file = "./02e_GO_MWU/DGconsistentyoked.csv", row.names = F)

## yoked yoked
res <- results(dds, contrast =c("APA2", "yoked_conflict", "yoked_consistent"), independentFiltering = T, alpha = 0.1)
summary(res)
resOrdered <- res[order(res$padj),]
head(resOrdered, 10)


## go setup
table(res$padj<0.1)
logs <- data.frame(cbind("gene"=row.names(res),"logP"=round(-log(res$pvalue+1e-10,10),1)))
logs$logP=as.numeric(as.character(logs$logP))
sign <- rep(1,nrow(logs))
sign[res$log2FoldChange<0]=-1  ##change to correct model
table(sign)
logs$logP <- logs$logP*sign
write.csv(logs, file = "./02e_GO_MWU/DGyokedyoked.csv", row.names = F)


##  "conflict", "yoked_conflict"
res <- results(dds, contrast =c("APA2", "conflict", "yoked_conflict"), independentFiltering = T, alpha = 0.1)
summary(res)
resOrdered <- res[order(res$padj),]
head(resOrdered, 10)


## go setup
table(res$padj<0.1)
logs <- data.frame(cbind("gene"=row.names(res),"logP"=round(-log(res$pvalue+1e-10,10),1)))
logs$logP=as.numeric(as.character(logs$logP))
sign <- rep(1,nrow(logs))
sign[res$log2FoldChange<0]=-1  ##change to correct model
table(sign)
logs$logP <- logs$logP*sign
write.csv(logs, file = "./02e_GO_MWU/DGconflictyoked.csv", row.names = F)

#### "conflict", "consistent"
res <- results(dds, contrast =c("APA2", "conflict", "consistent"), independentFiltering = T, alpha = 0.1)
summary(res)
resOrdered <- res[order(res$padj),]
head(resOrdered, 10)

## go setup
table(res$padj<0.1)
logs <- data.frame(cbind("gene"=row.names(res),"logP"=round(-log(res$pvalue+1e-10,10),1)))
logs$logP=as.numeric(as.character(logs$logP))
sign <- rep(1,nrow(logs))
sign[res$log2FoldChange<0]=-1  ##change to correct model
table(sign)
logs$logP <- logs$logP*sign
write.csv(logs, file = "./02e_GO_MWU/DGconflictconsistent.csv", row.names = F)

## plot of pkmz
plotCounts(dds, "Prkcz", intgroup = "APA2", normalized = TRUE, main="Prkcz in DG")

# order results table by the smallest adjusted p value:
res <- res[order(res$padj),]

results = as.data.frame(dplyr::mutate(as.data.frame(res), sig=ifelse(res$padj<0.05, "FDR<0.05", "Not Sig")), row.names=rownames(res))
head(results)
```


```{r heatmapsDG}
numDEGs <- function(group1, group2){
  res <- results(dds, contrast = c("APA2", group1, group2), independentFiltering = T)
  sumpadj <- sum(res$padj < 0.1, na.rm = TRUE)
  return(sumpadj)
}

#create list of groups
a <- levels(colData$APA2)
b <- levels(colData$APA2)

# comapre all contrasts, save to datafrmes
dat=data.frame()
for (i in a){
  for (j in b){
    if (i != j) {
      k <- paste(i,j, sep = "") #assigns usique rownames
      dat[k,1]<-i               
      dat[k,2]<-j
      dat[k,3]<- numDEGs(i,j) #caluculates number of DEGs
    }
  }
}

head(dat)

rownames(dat) <- NULL #remove row names
data_wide <- spread(dat, V2, V3)
data_wide

dat$V1 <- factor(dat$V1, levels = 
                              c("yoked_consistent", "consistent", "yoked_conflict", "conflict"))
dat$V2 <- factor(dat$V2, levels = 
                              c("yoked_consistent", "consistent", "yoked_conflict", "conflict"))

# rename levels
levels(dat$V1) <- c("yoked\nconsistent","consistent",  "yoked\nconflict","conflict")
levels(dat$V2) <- c("yoked\nconsistent","consistent",  "yoked\nconflict","conflict")


DGheat <- ggplot(dat, aes(V1, V2)) +
  geom_tile(aes(fill = V3)) +
  scale_fill_viridis(na.value="#FFFFFF00",
                     limits = c(0,1000)) + 
  ylab(" ") + xlab("Active Place Avoidance") +
  labs(fill = "# of DEGs",
       subtitle = "DG") +
  guides(fill=FALSE) + 
  theme_cowplot(font_size = 8, line_size = 0.25) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
DGheat
```

## CA3

There are so few differences in the CA3 that I don't make any figures for the manuscript.

```{r CA3, message=F, warning=F}
colData <- read.csv("../data/02a_colData.csv", header = T)
countData <- read.csv("../data/02a_countData.csv", header = T, check.names = F, row.names = 1)

colData <- colData %>% 
  filter(Punch %in% c("CA3"))  %>% 
  droplevels()
savecols <- as.character(colData$RNAseqID) 
savecols <- as.vector(savecols) 
countData <- countData %>% dplyr::select(one_of(savecols)) 
colData %>% select(APA2,Punch)  %>%  summary()

## create DESeq object using the factors Punch and APA
dds <- DESeqDataSetFromMatrix(countData = countData,
                              colData = colData,
                              design = ~ APA2)

dds # view the DESeq object - note numnber of genes
dds <- dds[ rowSums(counts(dds)) > 1, ]  # Pre-filtering genes with 0 counts
dds # view number of genes afternormalization and the number of samples
dds <- DESeq(dds) # Differential expression analysis
vsd <- getVarianceStabilizedData(dds)

res <- results(dds, contrast =c("APA2", "consistent", "yoked_consistent"), independentFiltering = T, alpha = 0.1)
summary(res)

res <- results(dds, contrast =c("APA2", "conflict", "yoked_conflict"), independentFiltering = T, alpha = 0.1)
summary(res)

res <- results(dds, contrast =c("APA2", "yoked_conflict", "yoked_consistent"), independentFiltering = T, alpha = 0.1)
summary(res)

res <- results(dds, contrast =c("APA2", "conflict", "consistent"), independentFiltering = T, alpha = 0.1)
summary(res)

## go setup
table(res$padj<0.1)
logs <- data.frame(cbind("gene"=row.names(res),"logP"=round(-log(res$pvalue+1e-10,10),1)))
logs$logP=as.numeric(as.character(logs$logP))
sign <- rep(1,nrow(logs))
sign[res$log2FoldChange<0]=-1  ##change to correct model
table(sign)
logs$logP <- logs$logP*sign
write.csv(logs, file = "./02e_GO_MWU/CA3conflictconsistent.csv", row.names = F)

plotCounts(dds, "Prkcz", intgroup = "APA2", normalized = TRUE, main="Prkcz in CA3")
```


```{r heatmapsCA3}
numDEGs <- function(group1, group2){
  res <- results(dds, contrast = c("APA2", group1, group2), independentFiltering = T)
  sumpadj <- sum(res$padj < 0.1, na.rm = TRUE)
  return(sumpadj)
}

#create list of groups
a <- levels(colData$APA2)
b <- levels(colData$APA2)

# comapre all contrasts, save to datafrmes
dat=data.frame()
for (i in a){
  for (j in b){
    if (i != j) {
      k <- paste(i,j, sep = "") #assigns usique rownames
      dat[k,1]<-i               
      dat[k,2]<-j
      dat[k,3]<- numDEGs(i,j) #caluculates number of DEGs
    }
  }
}

head(dat)

rownames(dat) <- NULL #remove row names
data_wide <- spread(dat, V2, V3)
data_wide

dat$V1 <- factor(dat$V1, levels = 
                              c("yoked_consistent", "consistent", "yoked_conflict", "conflict"))
dat$V2 <- factor(dat$V2, levels = 
                              c("yoked_consistent", "consistent", "yoked_conflict", "conflict"))

# rename levels
levels(dat$V1) <- c("yoked\nconsistent","consistent",  "yoked\nconflict","conflict")
levels(dat$V2) <- c("yoked\nconsistent","consistent",  "yoked\nconflict","conflict")


CA3heat <- ggplot(dat, aes(V1, V2)) +
  geom_tile(aes(fill = V3)) +
  scale_fill_viridis(na.value="#FFFFFF00",
                     limits = c(0,1000)) + 
  ylab(" ") + xlab("Active Place Avoidance") +
  labs(fill = "# of DEGs",
       subtitle = "CA3") +
  guides(fill=FALSE) + 
  theme_cowplot(font_size = 8, line_size = 0.25)  +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
CA3heat

```


## CA1

```{r CA1, message=F, warning=F}
colData <- read.csv("../data/02a_colData.csv", header = T)
countData <- read.csv("../data/02a_countData.csv", header = T, check.names = F, row.names = 1)
colData <- colData %>% 
  filter(Punch %in% c("CA1"))  %>% 
  droplevels()
savecols <- as.character(colData$RNAseqID) 
savecols <- as.vector(savecols) 
countData <- countData %>% dplyr::select(one_of(savecols)) 
colData %>% select(APA2,Punch)  %>%  summary()
dds <- DESeqDataSetFromMatrix(countData = countData,
                              colData = colData,
                              design = ~ APA2)

dds # view the DESeq object - note numnber of genes
dds <- dds[ rowSums(counts(dds)) > 1, ]  # Pre-filtering genes with 0 counts
dds # view number of genes afternormalization and the number of samples
dds <- DESeq(dds) # Differential expression analysis
vsd <- getVarianceStabilizedData(dds)

res <- results(dds, contrast =c("APA2", "consistent", "yoked_consistent"), independentFiltering = T, alpha = 0.1)
summary(res)
resOrdered <- res[order(res$padj),]
head(resOrdered, 10)

topGene <- rownames(res)[which.min(res$padj)]
plotCounts(dds, gene = topGene, intgroup=c("APA2"))

data <- data.frame(gene = row.names(res),
                   padj = res$padj, 
                   logpadj = -log10(res$padj),
                   lfc = res$log2FoldChange)
data <- na.omit(data)
data <- data %>%
  mutate(direction = ifelse(data$lfc > 1 & data$padj < 0.1, 
                        yes = "consistent", 
                        no = ifelse(data$lfc < -1 & data$padj < 0.1, 
                                    yes = "yoked_consistent", 
                                    no = "neither")))

CA1volcano <- ggplot(data, aes(x = lfc, y = logpadj)) + 
  geom_point(aes(color = factor(direction)), size = 1, alpha = 0.5, na.rm = T) + # add gene points
    theme_cowplot(font_size = 8, line_size = 0.25) +
  scale_color_manual(values = volcano1)  + 
  geom_hline(yintercept = 1,  size = 0.25, linetype = 2) + 
  scale_y_continuous(limits=c(0, 8)) +
  scale_x_continuous( limits=c(-3, 3),
                      name="Log fold change")+
  ylab(paste0("log10 p-value")) +       
  theme(panel.grid.minor=element_blank(),
        legend.position = "none", # remove legend 
        panel.grid.major=element_blank()) + 
  geom_text_repel(data = subset(data, logpadj > 3.5), aes(label = gene),
                  colour = "black", min.segment.length = 0,
                  box.padding = 0.5, size = 2)
CA1volcano

pdf(file="../figures/02c_rnaseqSubfield/CA1volcano.pdf",  width=1.5, height=2)
plot(CA1volcano)
dev.off()

# save DEGs
CA1volcanoDEGs <- data %>%
  filter(direction != "neither") %>%
  arrange(desc(lfc))
head(CA1volcanoDEGs,10)
write.csv(CA1volcanoDEGs, "../data/CA1-consistent-yokedconsistent.csv", row.names = F)

## go setup
table(res$padj<0.1)
logs <- data.frame(cbind("gene"=row.names(res),"logP"=round(-log(res$pvalue+1e-10,10),1)))
logs$logP=as.numeric(as.character(logs$logP))
sign <- rep(1,nrow(logs))
sign[res$log2FoldChange<0]=-1  ##change to correct model
table(sign)
logs$logP <- logs$logP*sign
write.csv(logs, file = "./02e_GO_MWU/CA1consistentyoked.csv", row.names = F)

#conflict yoked conflict

res <- results(dds, contrast =c("APA2", "conflict", "yoked_conflict"), independentFiltering = T, alpha = 0.1)
summary(res)
resOrdered <- res[order(res$padj),]
head(resOrdered, 3)

# volcano plots

res <- results(dds, contrast =c("APA2", "yoked_conflict", "yoked_consistent"), independentFiltering = T, alpha = 0.1)
summary(res)
resOrdered <- res[order(res$padj),]
head(resOrdered, 10)

topGene <- rownames(res)[which.min(res$padj)]
plotCounts(dds, gene = topGene, intgroup=c("APA2"))

data <- data.frame(gene = row.names(res),
                   padj = res$padj, 
                   logpadj = -log10(res$padj),
                   lfc = res$log2FoldChange)
data <- na.omit(data)
data <- data %>%
  mutate(direction = ifelse(data$lfc > 1 & data$padj < 0.1, 
                        yes = "yoked_conflict", 
                        no = ifelse(data$lfc < -1 & data$padj < 0.1, 
                                    yes = "yoked_consistent", 
                                    no = "neither")))

CA1volcano2 <- ggplot(data, aes(x = lfc, y = logpadj)) + 
  geom_point(aes(color = factor(direction)), size = 1, alpha = 0.5, na.rm = T) + # add gene points
  theme_cowplot(font_size = 8, line_size = 0.25) +
  scale_color_manual(values = volcano3)  + 
  geom_hline(yintercept = 1,  size = 0.25, linetype = 2) + 
  scale_y_continuous(limits=c(0, 8)) +
  scale_x_continuous( limits=c(-3, 3),
                      name="Log fold change")+
  ylab(paste0("log10 p-value")) +       
  theme(panel.grid.minor=element_blank(),
        legend.position = "none", # remove legend 
        panel.grid.major=element_blank()) +
  geom_text_repel(data = subset(data, logpadj > 3.5), aes(label = gene),
                  colour = "black", min.segment.length = 0,
                  box.padding = 0.5, size = 2)
CA1volcano2

pdf(file="../figures/02c_rnaseqSubfield/CA1volcano2.pdf",  width=1.5, height=2)
plot(CA1volcano2)
dev.off()

# save DEGs
CA1volcano2DEGs <- data %>%
  filter(direction != "neither") %>%
  arrange(desc(lfc))
head(CA1volcano2DEGs,10)
write.csv(CA1volcano2DEGs, "../data/CA1-yokedconflict-yokedconsistent.csv", row.names = F)

res <- results(dds, contrast =c("APA2", "conflict", "consistent"), independentFiltering = T, alpha = 0.1)
summary(res)

plotCounts(dds, "Prkcz", intgroup = "APA2", normalized = TRUE, main="Prkcz in CA1")
```


```{r heatmapsCA1}

# order levels
colData$APA2 <- factor(colData$APA2, 
                          levels = c("yoked_consistent", "consistent", "yoked_conflict", "conflict"))

#create list of groups
a <- levels(colData$APA2)
b <- levels(colData$APA2)

# comapre all contrasts, save to datafrmes
dat=data.frame()
for (i in a){
  for (j in b){
    if (i != j) {
      k <- paste(i,j, sep = "") #assigns usique rownames
      dat[k,1]<-i               
      dat[k,2]<-j
      dat[k,3]<- numDEGs(i,j) #caluculates number of DEGs
    }
  }
}

head(dat)

rownames(dat) <- NULL #remove row names
data_wide <- spread(dat, V2, V3)
data_wide

dat$V1 <- factor(dat$V1, levels = 
                              c("yoked_consistent", "consistent", "yoked_conflict", "conflict"))
dat$V2 <- factor(dat$V2, levels = 
                              c("yoked_consistent", "consistent", "yoked_conflict", "conflict"))

# rename levels
levels(dat$V1) <- c("yoked\nconsistent","consistent",  "yoked\nconflict","conflict")
levels(dat$V2) <- c("yoked\nconsistent","consistent",  "yoked\nconflict","conflict")


CA1heat <- ggplot(dat, aes(V1, V2)) +
  geom_tile(aes(fill = V3)) +
  scale_fill_viridis(na.value="#FFFFFF00",
                     limits = c(0,1000)) + 
  ylab(" ") + xlab("Active Place Avoidance") +
  labs(fill = "# of DEGs",
       subtitle = "CA1") +
  guides(fill=FALSE) + 
  theme_cowplot(font_size = 7, line_size = 0.25) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
CA1heat

CA1heat2 <- ggplot(dat, aes(V1, V2)) +
  geom_tile(aes(fill = V3)) +
  scale_fill_viridis(na.value="#FFFFFF00",
                     limits = c(0,1000)) + 
  ylab(" ") + xlab("Active Place Avoidance") +
  labs(fill = "# of DEGs",
       subtitle = "CA1") +
  theme_cowplot(font_size = 8, line_size = 0.25) +
  theme(legend.position="bottom")


```

```{r totalDEGs}
# three plots no lengend
allplots <- plot_grid(DGheat, CA3heat, CA1heat, nrow = 1)
 
legend <- get_legend(CA1heat2 + theme(legend.position="bottom"))
p <- plot_grid(allplots, legend, ncol = 1, rel_heights = c(1, .2))
p 

pdf(file="../figures/02c_rnaseqSubfield/totalDEGs.pdf", width=6, height=2.5)
plot(p)
dev.off()

```