---
title: "02c_rnaseqSubfield"
output: md_document
---

## Subfield analysis

This script is used to identify treatement differences within each subfield, generate volcano plots, venn diagrams, and tables for subsequent GO analyses. The final mutlipanel figures for the manuscript have been inserted just below the subheadings. 

```{r setup, message=F, warning=F}
library(tidyverse)
library(cowplot) ## for some easy to use themes
library(DESeq2) ## for gene expression analysis
library(UpSetR)
#devtools::install_github("clauswilke/ggtextures")
library(ggtextures)
library(magick)
library(ggtext) # for markdown in plots

library(BiocParallel)
register(MulticoreParam(6))

## load functions 
source("figureoptions.R")
source("functions_RNAseq.R")

## set output file for figures 
knitr::opts_chunk$set(fig.path = '../figures/02c_rnaseqSubfield/', cache = F)
```

## Get varience stabilized gene expression for each tissue

```{r DGdeseq}
a.colData <- read.csv("../data/02a_colData.csv", header = T)
a.countData <- read.csv("../data/02a_countData.csv", header = T, check.names = F, row.names = 1)

returndds <- function(mytissue){
  print(mytissue)
  colData <- a.colData %>% 
    filter(Punch %in% c(mytissue))  %>% 
  droplevels()
  
  savecols <- as.character(colData$RNAseqID) 
  savecols <- as.vector(savecols) 
  countData <- a.countData %>% dplyr::select(one_of(savecols)) 

  ## create DESeq object using the factors Punch and APA
  dds <- DESeqDataSetFromMatrix(countData = countData,
                              colData = colData,
                              design = ~ APA2)

  dds <- dds[ rowSums(counts(dds)) > 1, ]  # Pre-filtering genes with 0 counts
  dds <- DESeq(dds, parallel = TRUE)
  return(dds)
}

DGdds <- returndds("DG") 
CA1dds <- returndds("CA1") 
CA3dds <- returndds("CA3") 

returnvsds <- function(mydds, vsdfilename){
  dds <- mydds
  vsd <- vst(dds, blind=FALSE) ## variance stabilized
  print(head(assay(vsd),3))
  return(write.csv(assay(vsd), file = vsdfilename, row.names = T))
}

returnvsds(DGdds, "../data/02c_DGvsd.csv")
returnvsds(CA1dds, "../data/02c_CA1vsd.csv")
returnvsds(CA3dds, "../data/02c_CA3vsd.csv")
```

## Res summary

```{r twowaycontrasts}
res_summary_subfield <- function(mydds, mycontrast){
  res <- results(mydds, contrast = mycontrast, independentFiltering = T)
  print(mycontrast)
  print(sum(res$padj < 0.1, na.rm=TRUE))
  print(summary(res))
  print(head((res[order(res$padj),]), 5))
  cat("\n")
}

print("DG")
res_summary_subfield(DGdds, c("APA2", "standard.trained", "standard.yoked"))
res_summary_subfield(DGdds, c("APA2", "conflict.trained", "conflict.yoked"))
res_summary_subfield(DGdds, c("APA2", "conflict.trained", "standard.trained"))
res_summary_subfield(DGdds, c("APA2", "conflict.yoked", "standard.yoked"))

print("CA3")
res_summary_subfield(CA3dds, c("APA2", "standard.trained", "standard.yoked"))
res_summary_subfield(CA3dds, c("APA2", "conflict.trained", "conflict.yoked"))
res_summary_subfield(CA3dds, c("APA2", "conflict.trained", "standard.trained"))
res_summary_subfield(CA3dds, c("APA2", "conflict.yoked", "standard.yoked"))

print("CA1")
res_summary_subfield(CA1dds, c("APA2", "standard.trained", "standard.yoked"))
res_summary_subfield(CA1dds, c("APA2", "conflict.trained", "conflict.yoked"))
res_summary_subfield(CA1dds, c("APA2", "conflict.trained", "standard.trained"))
res_summary_subfield(CA1dds, c("APA2", "conflict.yoked", "standard.yoked"))

```

## Volcano plots

```{r volcanos}
DGconsyokcons <-  plot.cons.yokcons(DGdds, "DG", "DG: standard yoked v. trained") 
CA1consyokcons <-  plot.cons.yokcons(CA1dds, "CA1", "CA1: standard yoked v. trained")  
CA1yoked <-  plot.yokconf.yokcons(CA1dds, "CA1", "CA1: standard v. conflict yoked")  

volcanos <- plot_grid(DGconsyokcons, CA1consyokcons, CA1yoked , nrow = 1,
                      labels = c("(d)", "(e)", "(f)"), label_size = 8) 
volcanos

pdf(file="../figures/02c_rnaseqSubfield/volcanos.pdf", width=6.65, height=2)
plot(volcanos)    
dev.off()
```

# candidate gnees

```{r DG_boxplots}
betterPlotCounts <- function(mygene, mydds, mysubfield){
  df <- plotCounts(mydds, mygene, intgroup = "APA2",  transform = F, replaced = F, returnData = T)
  names(df) <- c("count", "treatment")
  df$treatment <- factor(df$treatment, levels = c("standard.yoked","standard.trained",
                                                  "conflict.yoked", "conflict.trained"))
  
  #df <- df %>% filter(treatment %in% c("home.cage","standard.trained"))
  
  ggplot(df, aes(x = treatment, y = count)) +
    geom_boxplot(aes(fill = treatment)) + 
    geom_point() +
    labs(subtitle = paste(mysubfield, " *", mygene, "*",  sep = "")) +
    theme_classic() +
    theme(plot.subtitle  = element_markdown(),
          legend.position = "none", 
          panel.grid.major  = element_blank(),  # remove major gridlines
          panel.grid.minor  = element_blank()) + # remove minor gridlines) +
    scale_fill_manual(values = fourgroups) 

}

betterPlotCounts("Prkcz", DGdds, "DG")
betterPlotCounts("Dusp16", DGdds, "DG")
betterPlotCounts("Thbs1", DGdds, "DG")
betterPlotCounts("Slc16a1", DGdds, "DG")
betterPlotCounts("Hes5", DGdds, "DG")
betterPlotCounts("Rtl1", DGdds, "DG")

betterPlotCounts("Nlrp3", DGdds, "DG")
betterPlotCounts("Kcnc2", DGdds, "DG")
betterPlotCounts("1110008F13Rik", DGdds, "DG")

a <- betterPlotCounts("Irs1", DGdds, "DG") + labs(title = "wald = -6, p = 1.577782e-06")
b <- betterPlotCounts("Dab2ip", DGdds, "DG")  + labs(title = "wald =  5, p = 3.227358e-05")
plot_grid(a,b)

```

```{r CA1_boxplots}
betterPlotCounts("Prkcz", CA1dds, "CA1")
betterPlotCounts("Grm1", CA1dds, "CA1")
betterPlotCounts("Grin2b", CA1dds, "CA1")
betterPlotCounts("Foxj3", CA1dds, "CA1")
betterPlotCounts("Grik3", CA1dds, "CA1")
betterPlotCounts("0610010K14Rik", CA1dds, "CA1")
```

```{r pkrcs}
a <- betterPlotCounts("Prkcz", CA1dds, "CA1")
b <- betterPlotCounts("Prkcz", CA3dds, "CA3")
c <- betterPlotCounts("Prkcz", DGdds, "DG")

d <- betterPlotCounts("Prkci", CA1dds, "CA1")
e <- betterPlotCounts("Prkci", CA3dds, "CA3")
f <- betterPlotCounts("Prkci", DGdds, "DG")

plot_grid(a,b,c,d,e,f)
```

## genes that are correlated with number of entrances 

Requires anlaysis of `04_integration.Rmd` first.

```{r DGcorrelations}
plotCounts(DGdds, "Acan", intgroup = "APA2", normalized = TRUE, main="Acan in DG")
plotCounts(DGdds, "Amigo2", intgroup = "APA2", normalized = TRUE, main="Amigo2 in DG")
plotCounts(DGdds, "Armcx5", intgroup = "APA2", normalized = TRUE, main="Armcx5 in DG")
plotCounts(DGdds, "Ptgs2", intgroup = "APA2", normalized = TRUE, main="Ptgs2 in DG")
plotCounts(DGdds, "Rgs2", intgroup = "APA2", normalized = TRUE, main="Rgs2 in DG")
plotCounts(DGdds, "Syt4", intgroup = "APA2", normalized = TRUE, main="Syt4 in DG")
```

## Upset plots

What genes overlap within cetain comparisons?

```{r upsetplotprep}
a.colData <- read.csv("../data/02a_colData.csv", header = T)
a.countData <- read.csv("../data/02a_countData.csv", header = T, check.names = F, row.names = 1)

eachsubfield <- levels(a.colData$Punch)

listofDEGs <- function(group1, group2){
  res <- results(dds, contrast = c("APA2", group1, group2), independentFiltering = T)
  
  print(paste(group1,group2, sep = " vs "))
  print(summary(res))
  
  data <- data.frame(gene = row.names(res),
                     lfc = res$log2FoldChange,
                     padj = res$padj,
                     tissue = i,
                     comparison = paste(group1, group2, sep = "-"))
  data <- data %>% dplyr::filter(padj < 0.1) %>% droplevels()
  return(data)
}

for(i in eachsubfield){
  
  colData <- a.colData %>% 
    dplyr::filter(Punch == i)  %>%
    droplevels()
  print(i)
  
  savecols <- as.character(colData$RNAseqID) 
  savecols <- as.vector(savecols) 
  countData <- a.countData %>% dplyr::select(one_of(savecols)) 

## create DESeq object using the factors Punch and APA
dds <- DESeqDataSetFromMatrix(countData = countData,
                              colData = colData,
                              design = ~ APA2)

dds # view the DESeq object - note numnber of genes
dds <- dds[ rowSums(counts(dds)) > 1, ]  # Pre-filtering genes with 0 counts
dds <- DESeq(dds, parallel = TRUE) # Differential expression analysis

A <- listofDEGs("standard.trained","standard.yoked")
B <- listofDEGs("conflict.trained","standard.trained")
C <- listofDEGs("conflict.trained","conflict.yoked")
D <- listofDEGs("conflict.yoked","standard.yoked")

all <- rbind(A,B,C,D)

write.csv(all, file = paste("../data/02c_",i,"forupset.csv", sep = ""), row.names = F)
}
```

