---
title: "02c_rnaseqSubfield"
output: md_document
---

## Subfield analysis

This script is used to identify treatement differences within each subfield, generate volcano plots, venn diagrams, and tables for subsequent GO analyses. The final mutlipanel figures for the manuscript have been inserted just below the subheadings. 

```{r setup, message=F, warning=F}
library(tidyverse)
library(cowplot) ## for some easy to use themes
library(DESeq2) ## for gene expression analysis
library(UpSetR)
#devtools::install_github("clauswilke/ggtextures")
library(ggtextures)
library(magick)
library(ggtext) # for markdown in plots

library(BiocParallel)
register(MulticoreParam(6))

## load functions 
source("figureoptions.R")
source("functions_RNAseq.R")

## set output file for figures 
knitr::opts_chunk$set(fig.path = '../figures/02c_rnaseqSubfield/', cache = T)
```

## Get varience stabilized gene expression for each tissue

```{r DGdeseq}
a.colData <- read.csv("../data/02a_colData.csv", header = T)
a.countData <- read.csv("../data/02a_countData.csv", header = T, check.names = F, row.names = 1)

a.colData <- a.colData %>%
  mutate(combinedgroups = fct_collapse(Treatment,
                                       trained = c("conflict", "trained"),
                                       yoked = c("shocked", "yoked")))
a.colData$combinedgroups <- factor(a.colData$combinedgroups, levels = c("yoked", "trained"))
a.colData$APA2 <- factor(a.colData$APA2, levels = c("standard.yoked","standard.trained",
                                                  "conflict.yoked", "conflict.trained"))

DGdds <- returnddsAPA2("DG") 
CA1dds <- returnddsAPA2("CA1") 
CA3dds <- returnddsAPA2("CA3") 

DGdds2 <- returnddscombinedgroups("DG") 
CA1dds2 <- returnddscombinedgroups("CA1") 
CA3dds2 <- returnddscombinedgroups("CA3") 

returnvsds(DGdds, "../data/02c_DGvsd.csv")
returnvsds(CA1dds, "../data/02c_CA1vsd.csv")
returnvsds(CA3dds, "../data/02c_CA3vsd.csv")
```

## Res summary

```{r twowaycontrasts}
print("DG")
res_summary_subfield(DGdds, c("APA2", "standard.trained", "standard.yoked"))
res_summary_subfield(DGdds, c("APA2", "conflict.trained", "conflict.yoked"))
res_summary_subfield(DGdds, c("APA2", "conflict.trained", "standard.trained"))
res_summary_subfield(DGdds, c("APA2", "conflict.yoked", "standard.yoked"))

print("CA3")
res_summary_subfield(CA3dds, c("APA2", "standard.trained", "standard.yoked"))
res_summary_subfield(CA3dds, c("APA2", "conflict.trained", "conflict.yoked"))
res_summary_subfield(CA3dds, c("APA2", "conflict.trained", "standard.trained"))
res_summary_subfield(CA3dds, c("APA2", "conflict.yoked", "standard.yoked"))

print("CA1")
res_summary_subfield(CA1dds, c("APA2", "standard.trained", "standard.yoked"))
res_summary_subfield(CA1dds, c("APA2", "conflict.trained", "conflict.yoked"))
res_summary_subfield(CA1dds, c("APA2", "conflict.trained", "standard.trained"))
res_summary_subfield(CA1dds, c("APA2", "conflict.yoked", "standard.yoked"))

```

## Volcano plots

```{r volcanos, fig.width=6.65, fig.height=4}
plot.volcano <- function(mydds, whichfactor, up, down, mycolors, mysubfield){
  res <- results(mydds, contrast =c(whichfactor, up, down),
                 independentFiltering = T, alpha = 0.1)
   data <- data.frame(gene = row.names(res),
                     padj = res$padj, 
                     logpadj = -log10(res$padj),
                     lfc = res$log2FoldChange)
  data <- na.omit(data)
  data <- data %>%
    dplyr::mutate(direction = ifelse(data$lfc > 0 & data$padj < 0.1, 
                                     yes = up, 
                                     no = ifelse(data$lfc < 0 & data$padj < 0.1, 
                                                 yes = down, 
                                                 no = "NS")))
  volcano <- data %>%
    ggplot(aes(x = lfc, y = logpadj)) + 
    geom_point(aes(color = factor(direction)), size = 1, alpha = 0.75, na.rm = T) + 
     theme_ms() +
    scale_color_manual(values = mycolors,
                       name = "higher in",
                       breaks=c(down, up))  + 
   # ylim(c(0,7)) +
    #xlim(c(-10,10)) +
    labs(y = "-log10(p)", x = "log fold change", title = mysubfield)  +
    theme(legend.position = "bottom",
          legend.spacing.x = unit(-0.1, 'cm'),
          legend.margin=margin(t=-0.25, r=0, b=0, l=0, unit="cm")) 
  plot(volcano)
}



DGdds2 <- returnddscombinedgroups("DG") 
CA1dds2 <- returnddscombinedgroups("CA1") 
CA3dds2 <- returnddscombinedgroups("CA3") 


DGconsyokcons <-  plot.volcano(DGdds, "APA2", "standard.trained", "standard.yoked", volcano1, "DG") 
DGconflicttrained <-  plot.volcano(DGdds, "APA2", "conflict.trained", "conflict.yoked", volcano5, "DG") 
DGtrained <- plot.volcano(DGdds2, "combinedgroups", "trained", "yoked", volcano7, "DG") 

CA1consyokcons <-  plot.volcano(CA1dds, "APA2", "standard.trained", "standard.yoked", volcano1,  "CA1")  
CA1yoked <-  plot.volcano(CA1dds, "APA2","conflict.yoked", "standard.yoked", volcano6,  "CA1") 
CA1trained <- plot.volcano(CA1dds2, "combinedgroups", "trained", "yoked", volcano7, "CA1") 


volcanos <- plot_grid(DGconsyokcons, DGconflicttrained, DGtrained,
                      CA1consyokcons, CA1yoked , CA1trained, nrow = 2,
                      labels = c("(a)", "(b)","(c)" ,"(d)", "(e)", "(f)"), label_size = 8) 
volcanos

pdf(file="../figures/02c_rnaseqSubfield/volcanos.pdf", width=6.65, height=4)
plot(volcanos)    
dev.off()
```

# candidate gnees

```{r DG_boxplots}
betterPlotCounts <- function(mygene, mydds, mysubfield){
  df <- plotCounts(mydds, mygene, intgroup = "APA2",  transform = F, replaced = F, returnData = T)
  names(df) <- c("count", "treatment")
  df$treatment <- factor(df$treatment, levels = c("standard.yoked","standard.trained",
                                                  "conflict.yoked", "conflict.trained"))
  
  #df <- df %>% filter(treatment %in% c("home.cage","standard.trained"))
  
  ggplot(df, aes(x = treatment, y = count)) +
    geom_boxplot(aes(fill = treatment)) + 
    geom_point() +
    labs(subtitle = paste(mysubfield, " *", mygene, "*",  sep = "")) +
    theme_classic() +
    theme(plot.subtitle  = element_markdown(),
          legend.position = "none", 
          panel.grid.major  = element_blank(),  # remove major gridlines
          panel.grid.minor  = element_blank()) + # remove minor gridlines) +
    scale_fill_manual(values = fourgroups) 

}

betterPlotCounts("Prkcz", DGdds, "DG")
betterPlotCounts("Dusp16", DGdds, "DG")
betterPlotCounts("Thbs1", DGdds, "DG")
betterPlotCounts("Slc16a1", DGdds, "DG")
betterPlotCounts("Hes5", DGdds, "DG")
betterPlotCounts("Rtl1", DGdds, "DG")

betterPlotCounts("Nlrp3", DGdds, "DG")
betterPlotCounts("Kcnc2", DGdds, "DG")
betterPlotCounts("1110008F13Rik", DGdds, "DG")

a <- betterPlotCounts("Irs1", DGdds, "DG") + labs(title = "wald = -6, p = 1.577782e-06")
b <- betterPlotCounts("Dab2ip", DGdds, "DG")  + labs(title = "wald =  5, p = 3.227358e-05")
plot_grid(a,b)

```

```{r CA1_boxplots}
betterPlotCounts("Prkcz", CA1dds, "CA1")
betterPlotCounts("Grm1", CA1dds, "CA1")
betterPlotCounts("Grin2b", CA1dds, "CA1")
betterPlotCounts("Foxj3", CA1dds, "CA1")
betterPlotCounts("Grik3", CA1dds, "CA1")
betterPlotCounts("0610010K14Rik", CA1dds, "CA1")
```

```{r pkrcs}
a <- betterPlotCounts("Prkcz", CA1dds, "CA1")
b <- betterPlotCounts("Prkcz", CA3dds, "CA3")
c <- betterPlotCounts("Prkcz", DGdds, "DG")

d <- betterPlotCounts("Prkci", CA1dds, "CA1")
e <- betterPlotCounts("Prkci", CA3dds, "CA3")
f <- betterPlotCounts("Prkci", DGdds, "DG")

plot_grid(a,b,c,d,e,f)
```

## genes that are correlated with number of entrances 

Requires anlaysis of `04_integration.Rmd` first.

```{r DGcorrelations}
plotCounts(DGdds, "Acan", intgroup = "APA2", normalized = TRUE, main="Acan in DG")
plotCounts(DGdds, "Amigo2", intgroup = "APA2", normalized = TRUE, main="Amigo2 in DG")
plotCounts(DGdds, "Armcx5", intgroup = "APA2", normalized = TRUE, main="Armcx5 in DG")
plotCounts(DGdds, "Ptgs2", intgroup = "APA2", normalized = TRUE, main="Ptgs2 in DG")
plotCounts(DGdds, "Rgs2", intgroup = "APA2", normalized = TRUE, main="Rgs2 in DG")
plotCounts(DGdds, "Syt4", intgroup = "APA2", normalized = TRUE, main="Syt4 in DG")
```

## Upset plots

What genes overlap within cetain comparisons?

```{r upsetplotprep}
a.colData <- read.csv("../data/02a_colData.csv", header = T)
a.countData <- read.csv("../data/02a_countData.csv", header = T, check.names = F, row.names = 1)

eachsubfield <- levels(a.colData$Punch)

listofDEGs <- function(group1, group2){
  res <- results(dds, contrast = c("APA2", group1, group2), independentFiltering = T)
  
  print(paste(group1,group2, sep = " vs "))
  print(summary(res))
  
  data <- data.frame(gene = row.names(res),
                     lfc = res$log2FoldChange,
                     padj = res$padj,
                     tissue = i,
                     comparison = paste(group1, group2, sep = "-"))
  data <- data %>% dplyr::filter(padj < 0.1) %>% droplevels()
  return(data)
}

for(i in eachsubfield){
  
  colData <- a.colData %>% 
    dplyr::filter(Punch == i)  %>%
    droplevels()
  print(i)
  
  savecols <- as.character(colData$RNAseqID) 
  savecols <- as.vector(savecols) 
  countData <- a.countData %>% dplyr::select(one_of(savecols)) 

## create DESeq object using the factors Punch and APA
dds <- DESeqDataSetFromMatrix(countData = countData,
                              colData = colData,
                              design = ~ APA2)

dds # view the DESeq object - note numnber of genes
dds <- dds[ rowSums(counts(dds)) > 1, ]  # Pre-filtering genes with 0 counts
dds <- DESeq(dds, parallel = TRUE) # Differential expression analysis

A <- listofDEGs("standard.trained","standard.yoked")
B <- listofDEGs("conflict.trained","standard.trained")
C <- listofDEGs("conflict.trained","conflict.yoked")
D <- listofDEGs("conflict.yoked","standard.yoked")

all <- rbind(A,B,C,D)

write.csv(all, file = paste("../data/02c_",i,"forupset.csv", sep = ""), row.names = F)
}
```

