---
title: "02d_rnaseqSubfieldUnique"
output: md_document
---

```{r setup, include=FALSE}
library(tidyverse)
library(cowplot) ## for some easy to use themes
library(DESeq2) ## for gene expression analysis
library(UpSetR)
#devtools::install_github("clauswilke/ggtextures")
library(ggtextures)
library(magick)


knitr::opts_chunk$set(fig.path = '../figures/02c_rnaseqSubfield/', cache = T)
```


```{r beginupset}

DG <- read.csv("../data/02c_DGforupset.csv")  
CA1<- read.csv("../data/02c_CA1forupset.csv")  
CA3 <- read.csv("../data/02c_CA3forupset.csv") 


# upset plot without direction
all <- rbind(DG,CA1,CA3)

levels(all$comparison) <- c("conf-cons",
                            "conf-yconf", 
                            "cons-ycons", 
                            "yconf-ycons")

all$significant <- paste(all$tissue, all$comparison, sep = "-")

myupsetdf <- all %>%
  select(gene,significant) %>%
  mutate(yesno = 1) %>%
  distinct %>%
  spread(significant, yesno, fill = 0)
head(myupsetdf)

write.csv(myupsetdf, "../data/02c_upsetdf.csv")


# upset plot with direction, only CA1learn, CA1 stress, and DGlearn
head(all) 
all$direction <- ifelse(all$lfc > 0, "up", "down")
all$sigdir <- paste(all$significant, all$direction, sep = "-")

myupsetslim  <- all %>%
  filter(significant %in% c("DG-cons-ycons", "CA1-cons-ycons", "CA1-yconf-ycons")) %>%
  select(gene,sigdir) %>%
  mutate(yesno = 1) %>%
  distinct %>%
  spread(sigdir, yesno, fill = 0)
head(myupsetslim)
row.names(myupsetslim) <- myupsetslim$gene
myupsetslim$gene <- NULL
colSums(myupsetslim)
myupsetslim$gene <- row.names(myupsetslim)

```

```{r upsetplot}
upset(myupsetdf, keep.order = T)

trained <- myupsetdf %>%
  select(gene, 'DG-cons-ycons', 'CA3-cons-ycons', 'CA1-cons-ycons')

colnames(trained) <- c("gene", "DG", "CA3", "CA1")


pdf(file="../figures/02c_rnaseqSubfield/upsettraining.pdf",  onefile=FALSE) # or other device
upset(trained, keep.order = T,
      sets = c("CA1", "CA3", "DG"),
      sets.bar.color=c("#7570b3","#1b9e77", "#d95f02"),
      queries = list(list(query = intersects, params = list("CA1"), color = "#ca0020", active = T),
                     list(query = intersects, params = list("DG"), color = "#ca0020", active = T),
                     list(query = intersects, params = list("CA3"), color = "#ca0020", active = T),
                     list(query = intersects, params = list("CA1", "DG"), 
                          color = "#ca0020", active = T)),
      text.scale = 2,
      sets.x.label = NULL,
      #point.size = 1, line.size = 1,
      mb.ratio = c(0.6, 0.4)
)
dev.off()
```

Create a list of genes afected by stress and learning. Save. THen use to filter out nonspecific gene expression responses

```{r listofDEGs}
shared <- myupsetdf %>%
  select(gene, "CA1-yconf-ycons", "CA1-cons-ycons", "DG-cons-ycons")
colnames(shared) <- c("gene", "CA1stress", "CA1learn", "DGlearn")


# CA1 and DG learning only (none in stress)
shared %>% filter(CA1learn == 1 & DGlearn == 1 & CA1stress == 0) %>%
  select(gene) 

# learning and stress CA1
CA1learnstress <- shared %>% filter(CA1learn == 1 & CA1stress == 1) %>%
  select(gene)
head(CA1learnstress)

# CA1 learning but not CA1 stress 
CA1learn <- shared %>% filter(CA1learn == 1 & CA1stress == 0) %>%
  select(gene)

# CA1 learning but not CA1 stress 
CA1stress <- shared %>% filter(CA1learn == 0 & CA1stress == 1) %>%
  select(gene)

write.csv(CA1learnstress, "../data/02_CA1learningstressgenes.csv", row.names = F)
write.csv(CA1learn, "../data/02_CA1learninggenes.csv", row.names = F)
write.csv(CA1stress, "../data/02_CA1lstressgenes.csv", row.names = F)

```


```{r myupset}
# All degs
updown <- read.csv("../data/02c_setsize_updown.csv") 
updown$direction <- factor(updown$direction,  levels = c("yoked_consistent", "yoked_conflict", "consistent"))
updown$set <- factor(updown$set,  levels = c("DGtrain", "CA1train", "CA1stress"))
updown$status <- factor(updown$status,  levels = c("train", "stress"))

levels(updown$set)[levels(updown$set)=="DGtrain"] <- "DG train"
levels(updown$set)[levels(updown$set)=="CA1train"]   <- "CA1 train"
levels(updown$set)[levels(updown$set)=="CA1stress"]   <- "CA1 stress"
levels(updown$direction)[levels(updown$direction)=="yoked_consistent"] <- "yoked\nconsistent"
levels(updown$direction)[levels(updown$direction)=="yoked_conflict"]   <- "yoked\nconflict"

# unique and shared
shared2 <- read.csv("../data/02c_intersect_updown_shared2.csv") 
str(shared2)

shared2$group <- paste(shared2$set, shared2$direction, sep = " ")
shared2$set <- factor(shared2$set,  levels = c("DGtrain", "CA1train", "CA1stress"))
shared2$direction <- factor(shared2$direction,  
                            levels = c("yoked_consistent", "yoked_conflict", "consistent"))
shared2$group <-factor(shared2$group,  
                       levels = c("DGtrain yoked_consistent", "DGtrain consistent", 
                                  "CA1train yoked_consistent","CA1train consistent",
                                  "CA1stress yoked_consistent", "CA1stress yoked_conflict" ))
shared2$status <- factor(shared2$status,  levels = c(  "stress","train","unique"))

levels(shared2$direction)[levels(shared2$direction)=="yoked_consistent"] <- "yoked\nconsistent"
levels(shared2$direction)[levels(shared2$direction)=="yoked_conflict"]   <- "yoked\nconflict"
levels(shared2$status)[levels(shared2$status)=="stress"]   <- "shared"
levels(shared2$status)[levels(shared2$status)=="train"]   <- "shared"

levels(shared2$set) <- c("DG", "CA1", " CA1 ")
str(shared2)
```

```{r newbarplot}

d1 <- ggplot(updown, aes(x=direction, y=setsize, fill = direction)) +
  geom_bar(stat="identity", position=position_dodge()) +
  theme_cowplot(font_size = 7, line_size = 0.25) +
  scale_fill_manual(values = c("#404040", "#bababa", "#ca0020"),
                    name = NULL)  +
  labs(x = NULL, y = "Total DEGs") +
  scale_y_continuous(limits = c(0, 560),
                     breaks = c(0,125,250,375,500)) +
    facet_wrap(~set, scales = "free_x") +
    theme(axis.text.x = element_blank(),
          legend.position = "none",
          legend.text=element_text(size=4),
          legend.key.size = unit(0.2, "cm"),
          #legend.margin=margin(t=-0.25, r=0, b=0, l=0, unit="cm"),
          #strip.text.x = element_text(size = 5),
          strip.background = element_rect(colour=NA, fill=NA),
        panel.border = element_rect(fill = NA, color = "black"),
        legend.margin=margin(t=-0.1, r=0, b=-0.1, l=0, unit="cm"))



images = c(
  unique = "../figures/00_schematics/patterns_blank.png",
  shared = "../figures/00_schematics/patterns_crosed-lines.png")
images

head(shared2)

d2 <- ggplot(shared2, aes(x=direction, y=setsize, image = status)) +
  geom_textured_bar(stat = "identity") +
  theme_cowplot(font_size = 7, line_size = 0.25) +
  scale_image_manual(values = images,
                     name = NULL) +
  labs(x = "subfield * treatment", y = "Unique DEGs") +
    scale_y_continuous(limits = c(0, 560),
                     breaks = c(0,125,250,375,500)) +
  theme(axis.text.x=element_text(angle=60, vjust = 1, hjust = 1),
        legend.position = c(0.05, 0.8),
        #legend.text=element_text(size=4),
        legend.key.size = unit(0.2, "cm"),
        #strip.text.x = element_text(size = 5),
        strip.background = element_rect(colour=NA, fill=NA),
        panel.border = element_rect(fill = NA, color = "black"),
        legend.margin=margin(t=-0.1, r=0, b=-0.1, l=-0.1, unit="cm")) +
  facet_wrap(~set, scales = "free_x") 
d2

newbarplots <- plot_grid(d1,d2, nrow = 2, rel_heights =c(0.4,0.6),
                         labels = c("D","E"), label_size = 7)
newbarplots

pdf(file="../figures/02c_rnaseqSubfield/barplots.pdf", width=1.9, height=2.15)
plot(newbarplots)    
dev.off() 

forgraphicalsummary <- shared2 %>%
  filter(status != "shared") %>%
  ggplot(aes(x=direction, y=setsize, fill = direction)) +
  geom_bar(stat = "identity", position=position_dodge()) +
  theme_cowplot(font_size = 7, line_size = 0.25) +
  labs(x = NULL, y = "Genes for GO") +
  theme(axis.text.x=element_text(angle=60, vjust = 1, hjust = 1),
        legend.position = "none",
        legend.key.size = unit(0.2, "cm"),
        strip.text.x = element_text(size = 7),
        strip.background = element_rect(colour=NA, fill=NA),
        panel.border = element_rect(fill = NA, color = "black"),
        legend.margin=margin(t=-0.1, r=0, b=-0.1, l=-0.1, unit="cm")) +
  facet_wrap(~set, scales = "free_x") +
    scale_fill_manual(values = c("#404040", "#bababa", "#ca0020"),
                    name = NULL) +
    scale_y_continuous(limits = c(0, 560),
                     breaks = c(0,125,250,375,500)) 
forgraphicalsummary




pdf(file="../figures/02c_rnaseqSubfield/forgraphicalsummary.pdf", width=1.5, height=1)
plot(forgraphicalsummary)    
dev.off() 

```

```{r combo, cache=FALSE}

# 3 volcanos and a bar plots in a grid
bottomplots <- plot_grid(CA1yoked, newbarplots,
                    labels = c("C", NULL),
                    label_size = 7)

volcanos <- plot_grid(training, bottomplots, nrow = 2,
                      rel_heights = c(0.475,0.525)) 
volcanos



pdf(file="../figures/02c_rnaseqSubfield/volcanos.pdf", width=3.15, height=4)
plot(volcanos)    
dev.off()

# 3 volcanos, 2 bar plots, space for 3 GOs
spaceforgo <- plot_grid(d1 + labs(x = NULL) + theme(axis.text.x=element_text(angle=60, vjust = 1, hjust = 1)), 
                        d2 + labs(x = NULL),
          DGconsyokcons + theme(legend.position = c(0.8,0.9),
                                axis.text.x=element_blank(),
                               axis.title.x=element_blank()), NULL,
          CA1consyokcons + theme(legend.position = c(0.8,0.9),
                                axis.text.x=element_blank(),
                               axis.title.x=element_blank()), NULL,
          CA1yoked + theme(legend.position = c(0.8,0.9)), NULL,
          labels= c("A1", "A2", "B1", "B2", "C1", "C2", "D1", "D2"),
          label_size = 7,
          ncol = 2,
          rel_heights = c(0.29, 0.19, 0.25, 0.27))
spaceforgo

pdf(file="../figures/02c_rnaseqSubfield/spaceforgo.pdf", width=5, height=5)
plot(spaceforgo)    
dev.off()

```
