---
title: "02d_rnaseqSubfieldUnique"
output: md_document
---

```{r setup, include=FALSE}
library(tidyverse)
library(cowplot) ## for some easy to use themes
library(DESeq2) ## for gene expression analysis
library(UpSetR)
#devtools::install_github("clauswilke/ggtextures")
library(ggtextures)
library(magick)

knitr::opts_chunk$set(fig.path = '../figures/02c_rnaseqSubfield/')
```


```{r importdata}
DG <- read.csv("../data/02c_DGforupset.csv", stringsAsFactors = F)  
CA1 <- read.csv("../data/02c_CA1forupset.csv", stringsAsFactors = F)  
CA3 <- read.csv("../data/02c_CA3forupset.csv", stringsAsFactors = F) 

# make df for upset plots without direction
myupsetdf <- rbind(DG,CA1,CA3)

myupsetdf <- myupsetdf %>%
  mutate(comparison = fct_recode(comparison,
    "ConfT.ConfY"    = "conflict.trained-conflict.yoked",
    "ConfY.StdY"      = "conflict.yoked-standard.yoked",
    "StdT.StdY" = "standard.trained-standard.yoked")) %>%
  mutate(tissue.contrast = paste(tissue, comparison, sep = ".")) %>%
  select(gene,tissue.contrast) %>%
  mutate(yesno = 1) %>%
  distinct %>%
  spread(tissue.contrast, yesno, fill = 0)

write.csv(myupsetdf, "../data/02d_upsetdf.csv")
```

```{r upsetplot}
upset1 <- upset(myupsetdf, keep.order = F,
      nsets = 14,
      order.by = "freq",
      sets.x.label = NULL,
      point.size = 1.5, 
      mb.ratio = c(0.5, 0.5))
upset1

pdf(file="../figures/02c_rnaseqSubfield/upsetall.pdf",  onefile=FALSE, width=6.69, height=5.1) # or other device
upset1
dev.off()


DGupset <- myupsetdf %>% select(gene, starts_with("DG")) 
CA1upset <- myupsetdf %>% select(gene, starts_with("CA1")) 
CA3upset <- myupsetdf %>% select(gene, starts_with("CA3")) 

upset2 <- upset(DGupset, keep.order = F,
      order.by = "freq",
      sets.bar.color=c("#d95f02"),
      sets.x.label = NULL,
      point.size = 1.5, 
      mb.ratio = c(0.5, 0.5),
      nsets = 6)
upset2

upset3 <- upset(CA1upset, keep.order = F,
      order.by = "freq",
      sets.bar.color=c("#7570b3"),
      sets.x.label = NULL,
      point.size = 1.5, 
      mb.ratio = c(0.5, 0.5),
      nsets = 6)
upset3

upset4 <- upset(CA3upset, keep.order = F,
      order.by = "freq",
      sets.bar.color=c("#1b9e77"),
      sets.x.label = NULL,
      point.size = 1.5, 
      mb.ratio = c(0.5, 0.5),
      nsets = 5)
upset4
```



```{r}
# to use GORILLA, I need a list of all genes used
genesforgo <- read_csv("../data/00_geneids.csv") %>% 
  select(gene) %>% arrange(gene)
write.csv(genesforgo, "../data/02d_genesforgo.csv")
```


```{r wide}
DGwide <- DG %>%
  select(-padj,-tissue) %>%
  spread(key = comparison, value = lfc)  %>%
  select(gene, "standard.trained-standard.yoked", "conflict.yoked-standard.yoked", "conflict.trained-conflict.yoked")
head(DGwide)


CA1wide <- CA1 %>%
  select(-padj,-tissue) %>%
  spread(key = comparison, value = lfc)  %>%
  select(gene,  "standard.trained-standard.yoked", "conflict.yoked-standard.yoked", "conflict.trained-conflict.yoked")
head(CA1wide)
```