---
title: "02d_rnaseqSubfieldUnique"
output: md_document
---

```{r setup, include=FALSE}
library(tidyverse)
library(cowplot) ## for some easy to use themes
library(DESeq2) ## for gene expression analysis
library(UpSetR)
#devtools::install_github("clauswilke/ggtextures")
library(ggtextures)
library(magick)

knitr::opts_chunk$set(fig.path = '../figures/02c_rnaseqSubfield/')
```


```{r importdata}
DG <- read.csv("../data/02c_DGforupset.csv", stringsAsFactors = F)  
CA1 <- read.csv("../data/02c_CA1forupset.csv", stringsAsFactors = F)  
CA3 <- read.csv("../data/02c_CA3forupset.csv", stringsAsFactors = F) 

# make df for upset plots without direction
myupsetdf <- rbind(DG,CA1,CA3)

myupsetdf <- myupsetdf %>%
  mutate(comparison = fct_recode(comparison,
    "ConfT.ConfY"    = "conflict.trained-conflict.yoked",
    "ConfY.StdY"      = "conflict.yoked-standard.yoked",
    "StdT.StdY" = "standard.trained-standard.yoked",
    "HC.ConfT" = "home.cage-conflict.trained" ,
    "HC.ConfY" =  "home.cage-conflict.yoked" ,
    "HC.StdT" = "home.cage-standard.trained" ,
    "HC.StdY" =  "home.cage-standard.yoked")) %>%
  mutate(tissue.contrast = paste(tissue, comparison, sep = ".")) %>%
  select(gene,tissue.contrast) %>%
  mutate(yesno = 1) %>%
  distinct %>%
  spread(tissue.contrast, yesno, fill = 0)

write.csv(myupsetdf, "../data/02d_upsetdf.csv")
```

```{r upsetplot}
upset1 <- upset(myupsetdf, keep.order = F,
     sets.bar.color=c(
       "#d95f02","#d95f02","#d95f02","#d95f02",
       "#7570b3","#7570b3","#7570b3","#7570b3","#7570b3",
       "#1b9e77",
       "#7570b3",
       "#1b9e77",
       "#d95f02",
       "#1b9e77"),
      nsets = 14,
      order.by = "freq",
      sets.x.label = NULL,
      point.size = 1.5, 
      mb.ratio = c(0.5, 0.5))
upset1

pdf(file="../figures/02c_rnaseqSubfield/upsetall.pdf",  onefile=FALSE, width=6.69, height=5.1) # or other device
upset1
dev.off()


DGupset <- myupsetdf %>% select(gene, starts_with("DG")) 
CA1upset <- myupsetdf %>% select(gene, starts_with("CA1")) 
CA3upset <- myupsetdf %>% select(gene, starts_with("CA3")) 

upset2 <- upset(DGupset, keep.order = F,
      order.by = "freq",
      sets.bar.color=c("#d95f02"),
      sets.x.label = NULL,
      point.size = 1.5, 
      mb.ratio = c(0.5, 0.5),
      nsets = 6)
upset2

upset3 <- upset(CA1upset, keep.order = F,
      order.by = "freq",
      sets.bar.color=c("#7570b3"),
      sets.x.label = NULL,
      point.size = 1.5, 
      mb.ratio = c(0.5, 0.5),
      nsets = 6)
upset3

upset4 <- upset(CA3upset, keep.order = F,
      order.by = "freq",
      sets.bar.color=c("#1b9e77"),
      sets.x.label = NULL,
      point.size = 1.5, 
      mb.ratio = c(0.5, 0.5),
      nsets = 5)
upset4
```

Create a list of genes afected by stress and learning. Save. THen use to filter out nonspecific gene expression responses

```{r listofDEGs}
# DG learning but not homecage or yoked yoked
DGtrainingonlygenes <- DGupset %>% 
  filter(DG.StdT.StdY == 1 &  DG.ConfY.StdY == 0 &DG.HC.ConfT == 0 & 
           DG.HC.ConfY == 0 & DG.HC.StdT == 0 & DG.HC.StdY == 0 ) %>% 
  select(gene)  %>% arrange(gene) %>% droplevels()

print("DG training but not homecage or yoked")
as.list(DGtrainingonlygenes)

# CA1 learning  but not homecage or yoked yoked
CA1trainingonlygenes <- CA1upset %>% 
  filter(CA1.StdT.StdY == 1 &  CA1.ConfY.StdY == 0 & CA1.HC.ConfT == 0 & 
           CA1.HC.ConfY == 0 & CA1.HC.StdT == 0 & CA1.HC.StdY == 0 ) %>% 
  select(gene) %>% arrange(gene) %>% droplevels()

print("CA1 training but not homecage or yoked")
as.list(CA1trainingonlygenes)
str(CA1trainingonlygenes)

# CA3 learning  but not homecage or yoked yoked
CA3trainingonlygenes <- CA3upset %>% 
  filter(CA3.StdT.StdY == 1  ) %>% 
  select(gene)  %>% droplevels()

print("CA3 training but not homecage or yoked")
as.list(CA3trainingonlygenes)

write.csv(CA1trainingonlygenes, "../data/02d_CA1_trainingonlygenes.csv", row.names = F)
write.csv(DGtrainingonlygenes, "../data/02d_DG_trainingonlygenes.csv", row.names = F)
```

```{r}
# to use GORILLA, I need a list of all genes used
genesforgo <- read_csv("../data/00_geneids.csv") %>% 
  select(gene) %>% arrange(gene)
write.csv(genesforgo, "../data/02d_genesforgo.csv")
```
