---
title: "02d_rnaseqSubfieldUnique"
output: md_document
---

```{r setup, include=FALSE}
library(tidyverse)
library(cowplot) ## for some easy to use themes
library(DESeq2) ## for gene expression analysis
library(UpSetR)
#devtools::install_github("clauswilke/ggtextures")
library(ggtextures)
library(magick)


knitr::opts_chunk$set(fig.path = '../figures/02c_rnaseqSubfield/')
```


```{r importdata}
DG <- read.csv("../data/02c_DGforupset.csv")  
CA1 <- read.csv("../data/02c_CA1forupset.csv")  
CA3 <- read.csv("../data/02c_CA3forupset.csv") 

# make df for upset plots without direction
myupsetdf <- rbind(DG,CA1,CA3)

myupsetdf <- myupsetdf %>%
  mutate(comparison = fct_recode(comparison,
    "ConfT.ConfY"    = "conflict.trained-conflict.yoked",
    "ConfY.StdY"      = "conflict.yoked-standard.yoked",
    "StdT.StdY" = "standard.trained-standard.yoked",
    "HC.ConfT" = "home.cage-conflict.trained" ,
    "HC.ConfY" =  "home.cage-conflict.yoked" ,
    "HC.StdT" = "home.cage-standard.trained" ,
    "HC.StdY" =  "home.cage-standard.yoked")) %>%
  mutate(tissue.contrast = paste(tissue, comparison, sep = ".")) %>%
  select(gene,tissue.contrast) %>%
  mutate(yesno = 1) %>%
  distinct %>%
  spread(tissue.contrast, yesno, fill = 0)

write.csv(myupsetdf, "../data/02d_upsetdf.csv")
```

```{r upsetplot}
upset1 <- upset(myupsetdf, keep.order = F,
     sets.bar.color=c(
       "#d95f02","#d95f02","#d95f02","#d95f02",
       "#7570b3","#7570b3","#7570b3","#7570b3","#7570b3",
       "#1b9e77",
       "#7570b3",
       "#1b9e77",
       "#d95f02",
       "#1b9e77"),
      nsets = 14,
      order.by = "freq",
      sets.x.label = NULL,
      point.size = 1.5, 
      mb.ratio = c(0.5, 0.5))
upset1

pdf(file="../figures/02c_rnaseqSubfield/upsetall.pdf",  onefile=FALSE, width=6.69, height=5.1) # or other device
upset1
dev.off()


DGupset <- myupsetdf %>% select(starts_with("DG")) 
CA1upset <- myupsetdf %>% select(starts_with("CA1")) 
CA3upset <- myupsetdf %>% select(starts_with("CA3")) 

upset2 <- upset(DGupset, keep.order = F,
      order.by = "freq",
      sets.bar.color=c("#d95f02"),
      sets.x.label = NULL,
      point.size = 1.5, 
      mb.ratio = c(0.5, 0.5),
      nsets = 6)
upset2

upset3 <- upset(CA1upset, keep.order = F,
      order.by = "freq",
      sets.bar.color=c("#7570b3"),
      sets.x.label = NULL,
      point.size = 1.5, 
      mb.ratio = c(0.5, 0.5),
      nsets = 6)
upset3

upset4 <- upset(CA3upset, keep.order = F,
      order.by = "freq",
      sets.bar.color=c("#1b9e77"),
      sets.x.label = NULL,
      point.size = 1.5, 
      mb.ratio = c(0.5, 0.5),
      nsets = 5)
upset4
```

Create a list of genes afected by stress and learning. Save. THen use to filter out nonspecific gene expression responses

```{r listofDEGs}
shared <- myupsetdf %>%
  select(gene, "CA1.ConfY.StdY", "CA1.StdT.StdY",  "DG.StdT.StdY")
colnames(shared) <- c("gene", "CA1stress", "CA1learn", "DGlearn")


# CA1 and DG learning only (none in stress)
shared %>% filter(CA1learn == 1 & DGlearn == 1 & CA1stress == 0) %>%
  select(gene) 

# learning and stress CA1
CA1learnstress <- shared %>% filter(CA1learn == 1 & CA1stress == 1) %>%
  select(gene)
head(CA1learnstress)

# CA1 learning but not CA1 stress 
CA1learn <- shared %>% filter(CA1learn == 1 & CA1stress == 0) %>%
  select(gene)

# CA1 learning but not CA1 stress 
CA1stress <- shared %>% filter(CA1learn == 0 & CA1stress == 1) %>%
  select(gene)


```


```{r newbarplot}

d1 <- ggplot(updown, aes(x=direction, y=setsize, fill = direction)) +
  geom_bar(stat="identity", position=position_dodge()) +
  theme_cowplot(font_size = 7, line_size = 0.25) +
  scale_fill_manual(values = c("#404040", "#bababa", "#ca0020"),
                    name = NULL)  +
  labs(x = NULL, y = "Total DEGs") +
  scale_y_continuous(limits = c(0, 560),
                     breaks = c(0,125,250,375,500)) +
    facet_wrap(~set, scales = "free_x") +
    theme(axis.text.x = element_blank(),
          legend.position = "none",
          legend.text=element_text(size=4),
          legend.key.size = unit(0.2, "cm"),
          #legend.margin=margin(t=-0.25, r=0, b=0, l=0, unit="cm"),
          #strip.text.x = element_text(size = 5),
          strip.background = element_rect(colour=NA, fill=NA),
        panel.border = element_rect(fill = NA, color = "black"),
        legend.margin=margin(t=-0.1, r=0, b=-0.1, l=0, unit="cm"))
d1


images = c(
  unique = "../figures/00_schematics/patterns_blank.png",
  shared = "../figures/00_schematics/patterns_crosed-lines.png")
images

head(shared2)

d2 <- ggplot(shared2, aes(x=direction, y=setsize, image = status)) +
  geom_textured_bar(stat = "identity") +
  theme_cowplot(font_size = 7, line_size = 0.25) +
  scale_image_manual(values = images,
                     name = NULL) +
  labs(x = "subfield * treatment", y = "Unique DEGs") +
    scale_y_continuous(limits = c(0, 560),
                     breaks = c(0,125,250,375,500)) +
  theme(axis.text.x=element_text(angle=60, vjust = 1, hjust = 1),
        legend.position = c(0.05, 0.8),
        #legend.text=element_text(size=4),
        legend.key.size = unit(0.2, "cm"),
        #strip.text.x = element_text(size = 5),
        strip.background = element_rect(colour=NA, fill=NA),
        panel.border = element_rect(fill = NA, color = "black"),
        legend.margin=margin(t=-0.1, r=0, b=-0.1, l=-0.1, unit="cm")) +
  facet_wrap(~set, scales = "free_x") 
d2

newbarplots <- plot_grid(d1,d2, nrow = 2, rel_heights =c(0.4,0.6),
                         labels = c("D","E"), label_size = 7)
newbarplots

pdf(file="../figures/02c_rnaseqSubfield/barplots.pdf", width=1.9, height=2.15)
plot(newbarplots)    
dev.off() 

```

```{r combo}
# 3 volcanos, 2 bar plots, space for 3 GOs
fig4barplots <- plot_grid(d1 + labs(x = NULL) + theme(axis.text.x=element_text(angle=60, vjust = 1, hjust = 1)), 
                        d2 + labs(x = NULL),
          labels= c("B1", "B2"),
          label_size = 7,
          ncol = 2)
fig4barplots

pdf(file="../figures/02c_rnaseqSubfield/fig4barplots.pdf", width=5, height=1.5)
plot(fig4barplots)    
dev.off()

```
