---
title: "trained v yoked"
output: md_document
---

```{r setup, message=F, warning=F}
library(tidyverse)
library(forcats)
library(cowplot) ## for some easy to use themes
library(DESeq2) ## for gene expression analysis
library(pheatmap)
library(viridis)
library(Rtsne) # for tSNE
library(scales)


library(BiocParallel)
register(MulticoreParam(6))

## load functions 
source("figureoptions.R")
source("functions_RNAseq.R")

## set output file for figures 
knitr::opts_chunk$set(fig.path = '../figures/02f_trainedvyoked/', cache = T)
```

```{r DESeq}
a.countData <- read.csv("../data/02a_countData.csv", header = T, check.names = F, row.names = 1)

a.colData <- read.csv("../data/02a_colData.csv", header = T)
a.colData <- a.colData 
a.colData$training <- factor(a.colData$training, levels = c("yoked", "trained"))

DGdds2 <- returndds2("DG") 
CA3dds2 <- returndds2("CA3") 
CA1dds2 <- returndds2("CA1") 
```

```{r vsd}
DGvsd <- returnvsds2(DGdds2)
CA3vsd <- returnvsds2(CA3dds2)
CA1vsd <- returnvsds2(CA1dds2)
```

```{r res}
print("DG")
res_summary_subfield(DGdds2, c("training", "trained", "yoked"))

print("CA3")
res_summary_subfield(CA3dds2, c("training", "trained", "yoked"))

print("CA1")
res_summary_subfield(CA1dds2, c("training", "trained", "yoked"))

contrast1 <- resvals2(DGdds2, contrastvector = c("training", "trained", "yoked"), mypval = 0.1) # 206
contrast2 <- resvals2(CA1dds2, contrastvector = c("training", "trained", "yoked"), mypval = 0.1) # 16

```

```{r listofDEGs}
options(digits = 3)

listofDEGstrainedvyoked <- function(mydds, myitssue){
  res <- results(mydds, 
                 contrast = c("training", "trained", "yoked"), 
                 independentFiltering = T)
  
  print(paste(myitssue, "trained vs yoked", sep = " "))
  
  data <- data.frame(gene = row.names(res),
                     lfc = round(res$log2FoldChange,2),
                     padj = res$padj,
                     tissue = myitssue)
  data <- data %>% 
    dplyr::mutate(gene = toupper(gene)) %>% 
    dplyr::filter(padj < 0.1) %>% 
    dplyr::mutate(direction = ifelse(lfc >0, "increased", "decreased"),
                  comparison = paste("trained", "yoked", sep = "-")) %>% 
    dplyr::select(gene, lfc,padj, direction, tissue, comparison )  %>% 
    dplyr::arrange(desc(direction), gene) %>% 
    droplevels()
  print(head(data))
  return(data)
}

DGDEGs <- listofDEGstrainedvyoked(DGdds2, "DG")
CA1DEGs <- listofDEGstrainedvyoked(CA1dds2, "CA1")


write.csv(DGDEGs, "../data/02f_DG_DEGs.csv", row.names = F)
write.csv(CA1DEGs, "../data/02f_CA1_DEGs.csv", row.names = F)

upDG <- DGDEGs %>% filter(lfc > 0)
upDG <- as.vector(upDG$gene)
upDG

downDG <- DGDEGs %>% filter(lfc < 0)
downDG <- as.vector(downDG$gene)
downDG

upCA1 <- CA1DEGs %>% filter(lfc > 0)
upCA1 <- as.vector(upCA1$gene)
upCA1

downCA1 <- CA1DEGs %>% filter(lfc < 0)
downCA1 <- as.vector(downCA1$gene)
downCA1

```




```{r pheatmap-DG}
## DG 
DEGes <- assay(vst(DGdds2))
DEGes <- cbind(DEGes, contrast1)
DEGes <- as.data.frame(DEGes) # convert matrix to dataframe

DEGes$rownames <- rownames(DEGes)  # add the rownames to the dataframe
DEGes$rownames <- str_to_upper(DEGes$rownames) ## uppercase gene names
DEGes$padjmin <- with(DEGes, pmin(padjtrainingtrainedyoked)) 
DEGes <- DEGes %>% filter(padjmin < 0.1)
rownames(DEGes) <- DEGes$rownames
drop.cols <-colnames(DEGes[,grep("padj|pval|rownames", colnames(DEGes))])
DEGes <- DEGes %>% dplyr::select(-one_of(drop.cols))
DEGes <- as.matrix(DEGes)
DEGes <- DEGes - rowMeans(DEGes)
DEGes <- as.matrix(DEGes)


write.csv(DEGes, "../data/02f_DG_DEGs_vsd.csv")

paletteLength <- 10
myBreaks <- c(seq(min(DEGes), 0, length.out=ceiling(paletteLength/2) + 1), 
              seq(max(DEGes)/paletteLength, max(DEGes), length.out=floor(paletteLength/2)))
df <- as.data.frame(colData(DGdds2)[,c("training")]) ## matrix to df
rownames(df) <- row.names(colData(DGdds2))
names(df) <- c("training")
df$subfield <- "DG"
levels(df$training) <- c("yoked", "trained")

pheatmap(DEGes, show_colnames=F, show_rownames = T,
         annotation_col=df, 
         annotation_colors = pheatmapcolors2,
         treeheight_row = 25, treeheight_col = 25,
         annotation_row = NA, 
         annotation_legend = TRUE,
         annotation_names_row = FALSE, annotation_names_col = FALSE,
         fontsize = 8, 
         border_color = NA ,
         color = viridis(10),
         cellwidth = 6, 
         clustering_method="average",
         breaks=myBreaks,
         clustering_distance_cols="correlation" ,
         clustering_distance_rows="correlation" 
         )

pheatmap(DEGes, show_colnames=F, show_rownames = T,
         annotation_col=df, 
         annotation_colors = pheatmapcolors2,
         treeheight_row = 25, treeheight_col = 12.5,
         annotation_row = NA, 
         annotation_legend = TRUE,
         annotation_names_row = FALSE, annotation_names_col = FALSE,
         fontsize = 3, 
         border_color = NA ,
         color = viridis(10),
         cellwidth = 6, 
         cluster_rows = FALSE,
         clustering_method="average",
         breaks=myBreaks,
         clustering_distance_cols="correlation", 
         height = 7, 
         width = 2.5,
         filename = "../figures/02f_trainedvyoked/pheatmap-DG.pdf"
         )
```

```{r pheatmap-CA1}

### CA1

DEGes <- assay(vst(CA1dds2))
DEGes <- cbind(DEGes, contrast2)
DEGes <- as.data.frame(DEGes) # convert matrix to dataframe
DEGes$rownames <- rownames(DEGes)  # add the rownames to the dataframe
DEGes$rownames <- str_to_upper(DEGes$rownames) ## uppercase gene names
DEGes$padjmin <- with(DEGes, pmin(padjtrainingtrainedyoked)) 
DEGes <- DEGes %>% filter(padjmin < 0.1)
rownames(DEGes) <- DEGes$rownames
drop.cols <-colnames(DEGes[,grep("padj|pval|rownames", colnames(DEGes))])
DEGes <- DEGes %>% dplyr::select(-one_of(drop.cols))
DEGes <- as.matrix(DEGes)
DEGes <- DEGes - rowMeans(DEGes)
DEGes <- as.matrix(DEGes) 
paletteLength <- 10
myBreaks <- c(seq(min(DEGes), 0, length.out=ceiling(paletteLength/2) + 1), 
              seq(max(DEGes)/paletteLength, max(DEGes), length.out=floor(paletteLength/2)))
df <- as.data.frame(colData(CA1dds2)[,c("training")]) ## matrix to df
rownames(df) <- row.names(colData(CA1dds2))
names(df) <- c("training")
df$subfield <- "CA1"
levels(df$training) <- c("yoked", "trained")

pheatmap(DEGes, show_colnames=F, show_rownames = T,
         annotation_col=df, 
         annotation_colors = pheatmapcolors3,
         treeheight_row = 0, treeheight_col = 25,
         annotation_row = NA, 
         annotation_legend = TRUE,
         annotation_names_row = FALSE, annotation_names_col = FALSE,
         fontsize = 8, 
         border_color = NA ,
         color = viridis(10),
         cellwidth = 6, 
         clustering_method="average",
         breaks=myBreaks,
         clustering_distance_cols="correlation" 
         )

pheatmap(DEGes, show_colnames=F, show_rownames = T,
         annotation_col=df, 
         annotation_colors = pheatmapcolors3,
         treeheight_row = 25, treeheight_col = 12.5,
         annotation_row = NA, 
         annotation_legend = TRUE,
         annotation_names_row = FALSE, annotation_names_col = FALSE,
         fontsize = 3, 
         border_color = NA ,
         color = viridis(10),
         cellwidth = 6, 
         cluster_rows = FALSE,
         clustering_method="average",
         breaks=myBreaks,
         clustering_distance_cols="correlation", 
         height = 2.5, 
         width = 2.5,
         filename = "../figures/02f_trainedvyoked/pheatmap-CA1.pdf"
         )
```

## tSNE

```{r tSNE}
a <- plot.tSNE.trained(DGdds2, 2, "DG")
b <- plot.tSNE.trained(CA1dds2, 2, "CA1")

plot_grid(a,b)
a
```

## Go terms

```{r}
GO_response <- read.table("../data/goterms/GO_term_summary_20191121_150656.txt", sep = "\t", row.names = NULL,  fill=TRUE)
GO_response$GO <- "1. response to stimulus"


GO_translation <- read.table("../data/goterms/GO_term_summary_20191121_141252.txt", sep = "\t", row.names = NULL)
GO_translation$GO <- "2. translation"

GO_synapse <- read.table("../data/goterms/GO_term_summary_20191121_145034.txt", sep = "\t", row.names = NULL)
GO_synapse$GO <- "3. synapse organization"

GO_learningormemory <- read.table("../data/goterms/GO_term_summary_20191121_142500.txt", sep = "\t", row.names = NULL)
GO_learningormemory$GO <- "4. learning or memory"


GO_negregcellprocess <- read.table("../data/goterms/GO_term_summary_20191121_171649.txt", sep = "\t", row.names = NULL,  fill=TRUE)
GO_negregcellprocess$GO <- "5. negative regulation of cellular process"


GOterms <- rbind(GO_learningormemory, GO_response, GO_translation, GO_synapse, GO_negregcellprocess)

GOterms <- GOterms %>%
  dplyr::mutate(gene = toupper(MGI.Gene.Marker.ID)) %>% 
  dplyr::select(gene, GO) %>% 
  dplyr::distinct(gene, GO) %>% 
 group_by(gene) 

head(GOterms)

write.csv(GOterms, "../data/goterms/GOterms.csv", row.names = F)

GOtermsDEGs <- inner_join(GOterms, DGDEGs) %>%
  arrange(GO,gene) %>%
  dplyr::select(gene, GO) %>% 
 group_by(GO) %>%
 summarize(genes = str_c(gene, collapse = ", "))
GOtermsDEGs

write.csv(GOtermsDEGs, "../data/goterms/GOtermsDEGs2.csv", row.names = F)

# what to konw which of our favorite genes were in those list but perhaps not captured by analysis


sanesLichtman <- read.csv("../data/02i_sanesLichtman.csv", header = T, stringsAsFactors = F)
head(sanesLichtman)

GOtermsSanesLichtman <- GOterms %>%
  filter(gene %in% sanesLichtman$genes) %>%
  dplyr::select(gene, GO) %>% 
 group_by(GO) %>%
 summarize(genes = str_c(gene, collapse = ", "))

GOtermsSanesLichtman
write.csv(GOtermsSanesLichtman, "../data/goterms/GOtermsSanesLichtman.csv", row.names = F)

inner_join(GOterms, DGDEGs) %>%
  dplyr::select(gene, GO) %>% 
 group_by(GO) %>%
 summarize(genes = str_c(gene, collapse = ", "))

inner_join(GOterms, DGDEGs) %>%
  filter(gene %in% sanesLichtman$genes)%>%
  dplyr::select(gene, GO) %>% 
 group_by(GO) %>%
 summarize(genes = str_c(gene, collapse = ", "))

# ADRB1 Adrenoceptor Beta 1
# HOMER1
# EGR1 Early Growth Response 1


```