---
title: "trained v yoked"
output: md_document
---

```{r setup, message=F, warning=F}
library(tidyverse)
library(forcats)
library(cowplot) ## for some easy to use themes
library(DESeq2) ## for gene expression analysis
library(pheatmap)
library(viridis)

library(BiocParallel)
register(MulticoreParam(6))

## load functions 
source("figureoptions.R")
source("functions_RNAseq.R")

## set output file for figures 
knitr::opts_chunk$set(fig.path = '../figures/02f_trainedvyoked/', cache = F)
```

```{r DESeq}
a.colData <- read.csv("../data/02a_colData.csv", header = T)
a.countData <- read.csv("../data/02a_countData.csv", header = T, check.names = F, row.names = 1)

a.colData <- a.colData %>%
  mutate(combinedgroups = fct_collapse(Treatment,
                                       trained = c("conflict", "trained"),
                                       yoked = c("shocked", "yoked")))
a.colData$combinedgroups <- factor(a.colData$combinedgroups, levels = c("yoked", "trained"))

returndds2 <- function(mytissue){
  print(mytissue)
  colData <- a.colData %>% 
    filter(Punch %in% c(mytissue))  %>% 
  droplevels()
  
  savecols <- as.character(colData$RNAseqID) 
  savecols <- as.vector(savecols) 
  countData <- a.countData %>% dplyr::select(one_of(savecols)) 

  ## create DESeq object using the factors Punch and APA
  dds <- DESeqDataSetFromMatrix(countData = countData,
                              colData = colData,
                              design = ~ combinedgroups)

  dds <- dds[ rowSums(counts(dds)) > 1, ]  # Pre-filtering genes with 0 counts
  dds <- DESeq(dds, parallel = TRUE)
  return(dds)
}

DGdds <- returndds2("DG") 
CA3dds <- returndds2("CA3") 
CA1dds <- returndds2("CA1") 
```

```{r vsd}
returnvsds2 <- function(mydds, vsdfilename){
  dds <- mydds
  vsd <- vst(dds, blind=FALSE) ## variance stabilized
  print(head(assay(vsd),3))
  myvsd <- assay(vsd)
  myvsd <- as.data.frame(myvsd)
  return(myvsd)
}

DGvsd <- returnvsds2(DGdds)
CA3vsd <- returnvsds2(CA3dds)
CA1vsd <- returnvsds2(CA1dds)
```

```{r res}
print("DG")
res_summary_subfield(DGdds, c("combinedgroups", "trained", "yoked"))

print("CA3")
res_summary_subfield(CA3dds, c("combinedgroups", "trained", "yoked"))

print("CA1")
res_summary_subfield(CA1dds, c("combinedgroups", "trained", "yoked"))
```

```{r listofDEGs}
listofDEGstrainedvyoked <- function(mydds, myitssue){
  res <- results(mydds, 
                 contrast = c("combinedgroups", "trained", "yoked"), 
                 independentFiltering = T)
  
  print(paste(myitssue, "trained vs yoked", sep = " "))
  
  data <- data.frame(gene = row.names(res),
                     lfc = res$log2FoldChange,
                     padj = res$padj,
                     tissue = myitssue,
                     comparison = paste("trained", "yoked", sep = "-"))
  data <- data %>% dplyr::filter(padj < 0.1) %>% droplevels()
  print(head(data))
  return(data)
}

DGDEGs <- listofDEGstrainedvyoked(DGdds, "DG")
CA1DEGs <- listofDEGstrainedvyoked(CA1dds, "CA1")
```


```{r volcanos}
plot.volcano.trainedyoked <- function(mydds){
  res <- results(mydds, contrast =c("combinedgroups", "trained", "yoked"),
                 independentFiltering = T, alpha = 0.1)
   data <- data.frame(gene = row.names(res),
                     padj = res$padj, 
                     logpadj = -log10(res$padj),
                     lfc = res$log2FoldChange)
  data <- na.omit(data)
  data <- data %>%
    dplyr::mutate(direction = ifelse(data$lfc > 0 & data$padj < 0.1, 
                                     yes = "trained", 
                                     no = ifelse(data$lfc < 0 & data$padj < 0.1, 
                                                 yes = "yoked", 
                                                 no = "NS")))
  volcano <- data %>%
    ggplot(aes(x = lfc, y = logpadj)) + 
    geom_point(aes(color = factor(direction)), size = 0.5, alpha = 0.75, na.rm = T) + 
      theme_ms() +
    scale_color_manual(values = c("grey", "red", "black"))  + 
    ylim(c(0,7)) +
    xlim(c(-10,10)) +
    labs(x = "lfc", y = "-log10(p)")  +
    theme(panel.grid.minor=element_blank(),
                panel.grid.major=element_blank(),
                legend.position = "none") 
  plot(volcano)
}

a <- plot.volcano.trainedyoked(DGdds)
#plot.volcano.trainedyoked(CA3dds)
b <- plot.volcano.trainedyoked(CA1dds)

plot_grid(a,b, labels = c("DG", "CA1"))
```

```{r pheatmap}
resvals2 <- function(mydds, contrastvector, mypval){
  res <- results(mydds, contrast = c(contrastvector[1],contrastvector[2],contrastvector[3]), independentFiltering = T)
  sumpvalue <- sum(res$pvalue < mypval, na.rm = TRUE)
  #print(sumpvalue)
  sumpadj <- sum(res$padj < mypval, na.rm = TRUE)
  print(sumpadj)
  vals <- cbind(res$pvalue, res$padj)
  pvalcolname <- as.character(paste("pval",contrastvector[1],contrastvector[2],contrastvector[3], sep=""))
  padjcolname <- as.character(paste("padj",contrastvector[1],contrastvector[2],contrastvector[3], sep=""))
  colnames(vals) <- c(pvalcolname, padjcolname)
  return(vals)
}

contrast1 <- resvals2(DGdds, contrastvector = c("combinedgroups", "trained", "yoked"), mypval = 0.1) # 3060
contrast2 <- resvals2(CA1dds, contrastvector = c("combinedgroups", "trained", "yoked"), mypval = 0.1) # 2388


## DG 
DEGes <- assay(vst(DGdds))
DEGes <- cbind(DEGes, contrast1)
DEGes <- as.data.frame(DEGes) # convert matrix to dataframe
DEGes$rownames <- rownames(DEGes)  # add the rownames to the dataframe
DEGes$rownames <- str_to_upper(DEGes$rownames) ## uppercase gene names
DEGes$padjmin <- with(DEGes, pmin(padjcombinedgroupstrainedyoked)) 
DEGes <- DEGes %>% filter(padjmin < 0.05)
rownames(DEGes) <- DEGes$rownames
drop.cols <-colnames(DEGes[,grep("padj|pval|rownames", colnames(DEGes))])
DEGes <- DEGes %>% dplyr::select(-one_of(drop.cols))
DEGes <- as.matrix(DEGes)
DEGes <- DEGes - rowMeans(DEGes)
DEGes <- as.matrix(DEGes)
paletteLength <- 10
myBreaks <- c(seq(min(DEGes), 0, length.out=ceiling(paletteLength/2) + 1), 
              seq(max(DEGes)/paletteLength, max(DEGes), length.out=floor(paletteLength/2)))
df <- as.data.frame(colData(DGdds)[,c("combinedgroups")]) ## matrix to df
rownames(df) <- row.names(colData(DGdds))
names(df) <- c("combinedgroups")
df$subfield <- "DG"
levels(df$combinedgroups) <- c("yoked", "trained")

pheatmap(DEGes, show_colnames=F, show_rownames = T,
         annotation_col=df, 
         annotation_colors = pheatmapcolors2,
         treeheight_row = 0, treeheight_col = 25,
         annotation_row = NA, 
         annotation_legend = TRUE,
         annotation_names_row = FALSE, annotation_names_col = FALSE,
         fontsize = 8, 
         border_color = NA ,
         color = viridis(10),
         cellwidth = 6, 
         clustering_method="average",
         breaks=myBreaks,
         clustering_distance_cols="correlation" 
         )

pheatmap(DEGes, show_colnames=F, show_rownames = T,
         annotation_col=df, 
         annotation_colors = pheatmapcolors2,
         treeheight_row = 25, treeheight_col = 12.5,
         annotation_row = NA, 
         annotation_legend = TRUE,
         annotation_names_row = FALSE, annotation_names_col = FALSE,
         fontsize = 3, 
         border_color = NA ,
         color = viridis(10),
         cellwidth = 6, 
         cluster_rows = FALSE,
         clustering_method="average",
         breaks=myBreaks,
         clustering_distance_cols="correlation", 
         height = 7, 
         width = 2.5,
         filename = "../figures/02f_trainedvyoked/pheatmap-DG.pdf"
         )


### CA1

DEGes <- assay(vst(CA1dds))
DEGes <- cbind(DEGes, contrast2)
DEGes <- as.data.frame(DEGes) # convert matrix to dataframe
DEGes$rownames <- rownames(DEGes)  # add the rownames to the dataframe
DEGes$rownames <- str_to_upper(DEGes$rownames) ## uppercase gene names
DEGes$padjmin <- with(DEGes, pmin(padjcombinedgroupstrainedyoked)) 
DEGes <- DEGes %>% filter(padjmin < 0.1)
rownames(DEGes) <- DEGes$rownames
drop.cols <-colnames(DEGes[,grep("padj|pval|rownames", colnames(DEGes))])
DEGes <- DEGes %>% dplyr::select(-one_of(drop.cols))
DEGes <- as.matrix(DEGes)
DEGes <- DEGes - rowMeans(DEGes)
DEGes <- as.matrix(DEGes) 
paletteLength <- 10
myBreaks <- c(seq(min(DEGes), 0, length.out=ceiling(paletteLength/2) + 1), 
              seq(max(DEGes)/paletteLength, max(DEGes), length.out=floor(paletteLength/2)))
df <- as.data.frame(colData(CA1dds)[,c("combinedgroups")]) ## matrix to df
rownames(df) <- row.names(colData(CA1dds))
names(df) <- c("combinedgroups")
df$subfield <- "CA1"
levels(df$combinedgroups) <- c("yoked", "trained")

pheatmap(DEGes, show_colnames=F, show_rownames = T,
         annotation_col=df, 
         annotation_colors = pheatmapcolors3,
         treeheight_row = 0, treeheight_col = 25,
         annotation_row = NA, 
         annotation_legend = TRUE,
         annotation_names_row = FALSE, annotation_names_col = FALSE,
         fontsize = 8, 
         border_color = NA ,
         color = viridis(10),
         cellwidth = 6, 
         clustering_method="average",
         breaks=myBreaks,
         clustering_distance_cols="correlation" 
         )

pheatmap(DEGes, show_colnames=F, show_rownames = T,
         annotation_col=df, 
         annotation_colors = pheatmapcolors3,
         treeheight_row = 25, treeheight_col = 12.5,
         annotation_row = NA, 
         annotation_legend = TRUE,
         annotation_names_row = FALSE, annotation_names_col = FALSE,
         fontsize = 3, 
         border_color = NA ,
         color = viridis(10),
         cellwidth = 6, 
         cluster_rows = FALSE,
         clustering_method="average",
         breaks=myBreaks,
         clustering_distance_cols="correlation", 
         height = 2.5, 
         width = 2.5,
         filename = "../figures/02f_trainedvyoked/pheatmap-CA1.pdf"
         )
```