---
title: "candidate genes"
output: md_document
---

```{r setup, include=FALSE}
library(tidyverse)
library(stringr) # for sentence to upper case conversion
library(UpSetR) # for upset plots

knitr::opts_chunk$set(echo = TRUE)

knitr::opts_chunk$set(fig.path = '../figures/02i_candidategenes/')

```

I want to examine the extent to which know learning and memory genes are found to be differentially expressed in this analysis. Do ask that question, I need to do a little data wrangling first. To address this questions, I will use a list of 237 candidate genes from Sanes and Lichtman 1999.

```{r sanes}
# prep sanes data
sanesLichtman <- read.csv("../data/02i_sanesLichtman.csv", header = T, stringsAsFactors = F)
sanes_forupset <- sanesLichtman
sanesLichtman <- as.list(sanesLichtman$genes)
length(sanesLichtman)

# prep region-specifi gene expression data
wrangledata <- function(filename, mycomparison){
  mydata <- read.csv(filename, header = T)
  mydata$gene <- str_to_upper(mydata$gene)
  mydata$comparison <- mycomparison
  return(mydata)
}

DG_Cons_YCons <- wrangledata("../data/DG-consistent-yokedconsistent.csv", "DG_Learning")
CA1_YConf_YCons <- wrangledata("../data/CA1-yokedconflict-yokedconsistent.csv", "CA1_Stress")
CA1_Cons_YCons <- wrangledata("../data/CA1-consistent-yokedconsistent.csv", "CA1_Learning")
CA1DG <- wrangledata("../data/DGvCA1.csv", "CA1_DG")
CA1CA3 <- wrangledata("../data/CA3vCA1.csv", "CA1_CA3")
CA3DG <- wrangledata("../data/DGvCA3.csv", "CA3_DG")

# print sanesLichtman DEGS
candidate_summary <- function(mydf){
    df <- mydf %>%
      dplyr::filter(gene %in% sanesLichtman) %>%
      dplyr::arrange(gene) %>%
      dplyr::filter(direction != "neither")
    print(df)
}
```

Which genes from the Sanes and Licthemn 1999 essay are differntially expressed in **the DG and CA1 response to active place avoidance training?**

```{r APAtraining}
candidate_summary(DG_Cons_YCons)
candidate_summary(CA1_Cons_YCons)
```

Which genes from the Sanes and Licthemn 1999 essay are differntially expressed in **the CA1 response to increased punishment?**

```{r stress}
candidate_summary(CA1_YConf_YCons)
```

Which of the genes are differentially expressed **between subfields, ignoring treatment effects?**

```{r regionalspecificity}
candidate_summary(CA1DG)
candidate_summary(CA1CA3)
candidate_summary(CA3DG)
```

Hmm.  I wonder what the overlap is. Let's examine it with Upset. But first, a little data wranling.

```{r upsetprep}
makeupsetlist <- function(mydf){
    df <- mydf %>%
      dplyr::filter(gene %in% sanesLichtman) %>%
      dplyr::arrange(gene) %>%
      dplyr::filter(direction != "neither") %>%
      dplyr::select(gene,direction,comparison)
    return(df)
}

a <- makeupsetlist(DG_Cons_YCons)
b <- makeupsetlist(CA1_Cons_YCons)
c <- makeupsetlist(CA1_YConf_YCons)
d <- makeupsetlist(CA1DG)
e <- makeupsetlist(CA1CA3)
f <- makeupsetlist(CA3DG)
str(a)

names(sanes_forupset)[1] <- "gene"
sanes_forupset$direction <- "1"
sanes_forupset$comparison <- "candidate_genes"

mydfs <- rbind(a,b,c,d,e,f, sanes_forupset)

# replace direcational values with 1 for present and make it numeric
mydfs$direction <- 1
mydfs$direction <- as.numeric(mydfs$direction)

data_wide <- spread(mydfs, comparison, direction)
data_wide[is.na(data_wide)] <- 0
str(data_wide)
```

These two upset plots show that relatively few of the Sanes Lichtman genes are differntially expresed between consistnely trained and their yoked controls in the CA1 and DG. 

The first plot showsthat 11 genes are differentailly epxressed in the CA1 in both the stress and the leaning paradigms. 4 more are only implicated in stress response while 5 and 3 are specific to learning-induced gene expression in the CA1 and DG, respectively.

The second plot shows that about 133 of the of the Sanes Lichman were not differentially expressed in any of my two way comparisons, but > 50 show some sort of spacital specificity, being differentially expressed between the three subfields of the hippocampus. 

```{r upset}
# small upset plot, only regoin specific stress and learning
upset(data_wide,mainbar.y.label = "No Shared DE Candidate Genes", 
      nsets = 3, keep.order = T,  order.by = "freq",
      sets = c("DG_Learning","CA1_Learning", "CA1_Stress"),
      scale.intersections = "log2",
      sets.x.label = "No. DE Candidate Genes")

# big upset plot with stress, learning, and spatial specifity
upset(data_wide,mainbar.y.label = "No Shared DE Candidate Genes", 
      nsets = 7, keep.order = T,  order.by = "freq",
      sets = c("DG_Learning","CA1_Learning", "CA1_Stress","CA1_CA3", "CA1_DG",  "CA3_DG",   "candidate_genes"),
      scale.intersections = "log2",
      sets.x.label = "No. DE Candidate Genes", 
      sets.bar.color=c("#ca0020","#ca0020",  "#bababa", "black", "black", "black", "black"),
      queries = list(# learning related
                     list(query = intersects, params = list("candidate_genes", "DG_Learning"), color = "#ca0020", active = T),
                     list(query = intersects, params = list("candidate_genes", "DG_Learning", "CA1_CA3",  "CA1_DG"), color = "#ca0020", active = T),
                     list(query = intersects, params = list("candidate_genes", "CA1_Learning"), color = "#ca0020", active = T),
                     list(query = intersects, params = list("candidate_genes", "CA1_Learning", "CA1_CA3",  "CA1_DG"), color = "#ca0020", active = T),
                     list(query = intersects, params = list("candidate_genes", "CA1_Learning", "CA1_Stress"), color = "#ca0020", active = T),
                     list(query = intersects, params = list("candidate_genes", "CA1_Learning", "CA1_DG"), color = "#ca0020", active = T),
                     list(query = intersects, params = list("candidate_genes", "CA1_Learning", "CA1_Stress", "CA1_CA3" ), color = "#ca0020", active = T),
                     list(query = intersects, params = list("candidate_genes", "CA1_Learning", "CA1_Stress", "CA1_DG" ), color = "#ca0020", active = T),
                     list(query = intersects, params = list("candidate_genes", "CA1_Learning", "CA1_Stress", "CA1_CA3", "CA3_DG" ), color = "#ca0020", active = T),
                     list(query = intersects, params = list("candidate_genes", "CA1_Learning", "CA1_Stress", "CA1_DG", "CA3_DG" ), color = "#ca0020", active = T),
                     list(query = intersects, params = list("candidate_genes", "CA1_Learning", "CA1_Stress", "CA1_CA3", "CA1_DG" ), color = "#ca0020", active = T),
                     # stress only
                     list(query = intersects, params = list("candidate_genes","CA1_Stress" ), color = "#bababa", active = T),
                      list(query = intersects, params = list("candidate_genes","CA1_Stress", "CA1_CA3", "CA1_DG",  "CA3_DG" ), color = "#bababa", active = T)))


```
