---
title: "Differential Gene Expression Analysis of Subfields"
output: md_document
---

## Subfield analysis

This script is used to identify treatement differences within each subfield, generate volcano plots, venn diagrams, and tables for subsequent GO analyses. The final mutlipanel figures for the manuscript have been inserted just below the subheadings. 

```{r setup, message=F, warning=F}
library(tidyverse)
library(cowplot) ## for some easy to use themes
library(DESeq2) ## for gene expression analysis
library(png)
library(grid)
library(scales)
library(apaTables) #  for ANOVA tables

library(BiocParallel)
register(MulticoreParam(6))

## load functions 
source("figureoptions.R")
source("functions_RNAseq.R")

## set output file for figures 
knitr::opts_chunk$set(fig.path = '../figures/03_rnaseqSubfield/', cache = T)
```

## Wrangle data

```{r DGdeseq}



# prep col data,
outliers <- c("146D-DG-3", "145A-CA3-2", "146B-DG-2", "146D-CA1-3", "148B-CA1-4")

colData <- read.csv("../data/00_colData.csv", header = T) %>%
  filter(!RNAseqID %in% outliers)
colData$training <- factor(colData$training, levels = levelstraining)
colData$treatment <- factor(colData$treatment, levels = levelstreatment)

# remove outliers
savecols <- as.character(colData$RNAseqID) #select the rowsname 
savecols <- as.vector(savecols) # make it a vector

countData <- read.csv("../data/00_countData.csv", 
                      header = T, check.names = F, row.names = 1) %>%
  dplyr::select(one_of(savecols)) # select just the columns 
head(countData)
```

## Get varience stabilized gene expression for each tissue

```{r}
# DEGs with looking at all four treatments individually
DGdds <- returnddstreatment("DG") 
CA3dds <- returnddstreatment("CA3") 
CA1dds <- returnddstreatment("CA1") 

# DEGs with looking at all grouped trained and yoked
DGdds2 <- returnddstraining("DG") 
CA3dds2 <- returnddstraining("CA3") 
CA1dds2 <- returnddstraining("CA1") 

savevsds(DGdds2, "../data/03_DG_vsdtraining.csv")
savevsds(CA3dds2, "../data/03_CA3_vsdtraining.csv")
savevsds(CA1dds2, "../data/03_CA1_vsdtraining.csv")
```

## Results to compare with volcano plots

```{r twowaycontrasts}
print("DG")
res_summary_subfield(DGdds2, c("training", "trained", "yoked"))
res_summary_subfield(DGdds, c("treatment", "conflict.trained", "standard.trained"))
res_summary_subfield(DGdds, c("treatment", "conflict.yoked", "standard.yoked"))

print("CA3")
res_summary_subfield(CA3dds2, c("training", "trained", "yoked"))
res_summary_subfield(CA3dds, c("treatment", "conflict.trained", "standard.trained"))
res_summary_subfield(CA3dds, c("treatment", "conflict.yoked", "standard.yoked"))

print("CA1")
res_summary_subfield(CA1dds2, c("training", "trained", "yoked"))
res_summary_subfield(CA1dds, c("treatment", "conflict.trained", "standard.trained"))
res_summary_subfield(CA1dds, c("treatment", "conflict.yoked", "standard.yoked"))
```

## Volcano plots

```{r DEGs}

# create data frame for making volcanos plots
DGa <- calculateDEGs(DGdds2, "DG", "training", "trained", "yoked") 
DGb <-  calculateDEGs(DGdds, "DG", "treatment", "conflict.trained", "standard.trained") 
DGc <-  calculateDEGs(DGdds, "DG", "treatment", "conflict.yoked", "standard.yoked") 

CA3a <- calculateDEGs(CA3dds2, "CA3", "training", "trained", "yoked") 
CA3b <-  calculateDEGs(CA3dds, "CA3", "treatment", "conflict.trained", "standard.trained") 
CA3c <-  calculateDEGs(CA3dds, "CA3", "treatment", "conflict.yoked", "standard.yoked") 

CA1a <- calculateDEGs(CA1dds2, "CA1", "training", "trained", "yoked") 
CA1b <-  calculateDEGs(CA1dds, "CA1", "treatment", "conflict.trained", "standard.trained") 
CA1c <-  calculateDEGs(CA1dds, "CA1", "treatment", "conflict.yoked", "standard.yoked") 

# save df with DEGs

allDEG <- rbind(DGa, DGb, DGc, 
                       CA3a, CA3b, CA3c, 
                       CA1a, CA1b, CA1c) %>% 
  dplyr::filter(direction != "NS") %>%
    dplyr::mutate(lfc = round(lfc, 2),
                  padj = scientific(padj, digits = 3),
                  logpadj = round(logpadj, 2)) %>%
  arrange(tissue, comparison, gene)

suppltable4 <- allDEG %>% filter(tissue == "DG"  & comparison == "yoked vs. trained")
head(suppltable4)

suppltable5 <- allDEG %>% filter(tissue != "DG"  & comparison != "yoked vs. trained")
head(suppltable5)

write_csv(suppltable4, "../data/suppltable-4.csv")
write_csv(suppltable5, "../data/suppltable-5.csv")

```


## bar plots


```{r}

DEGbargraph <- function(whichtissue, whichcomparison){

      p <- allDEG %>%
    filter(tissue == whichtissue,
           comparison == whichcomparison) %>%
    ggplot(aes(x = comparison,  fill = direction)) +
    geom_bar(position = "dodge", drop = FALSE) +
    theme_ms() +
    theme(legend.position = "none")  +
    guides(fill = guide_legend(nrow = 1)) +
    labs( y = "DEGs w/ + LFC", x = NULL, 
          subtitle = paste(whichtissue, whichcomparison, sep = ", ")) +
    geom_text(stat='count', aes(label=..count..), vjust =-0.5, 
              position = position_dodge(width = 1),
              size = 2, color = "black")  + 
    #ylim(lowlim, higherlim) +
    scale_fill_manual(values = allcolors, name = "higher in")    
  return(p)
}

DEGbargraph("DG", "standard.yoked vs. conflict.yoked")
DEGbargraph("CA3", "standard.yoked vs. conflict.yoked")
DEGbargraph("CA1", "standard.yoked vs. conflict.yoked")

DEGbargraph("DG", "yoked vs. trained")
DEGbargraph("CA3", "yoked vs. trained")
DEGbargraph("CA1", "yoked vs. trained")

```


```{r}
makenewbargraph <- function(whichtissue, whichsex,  whichcomparison, lowlim, higherlim){
  p <- allDEG %>%
    filter(tissue == whichtissue,
           comparison == whichcomparison,
           sex == whichsex) %>%
    ggplot(aes(x = comparison,  fill = direction)) +
    geom_bar(position = "dodge", drop = FALSE) +
    theme_B3() +
    theme(legend.position = "none")  +
    guides(fill = guide_legend(nrow = 1)) +
    labs( y = "DEGs w/ + LFC", x = NULL) +
    geom_text(stat='count', aes(label=..count..), vjust =-0.5, 
              position = position_dodge(width = 1),
              size = 2, color = "black")  + 
    ylim(lowlim, higherlim) +
    scale_fill_manual(values = allcolors, name = "higher in")    
  return(p)
}

#makenewbargraph()
```



```{r volcanos, fig.width=6.69, fig.height=6}
# volcano plots

p1 <- plot.volcano(DGa, volcano1, "214 DEGs") 
p2 <-  plot.volcano(DGb,volcano2, "0 DEGs") 
p3 <-  plot.volcano(DGc, volcano3, "3 DEGs") 

p4 <- plot.volcano(CA3a,  volcano1, "0 DEGs") 
p5 <-  plot.volcano(CA3b, volcano2, "0 DEGs") 
p6 <-  plot.volcano(CA3c, volcano3, "2 DEGs") 

p7 <- plot.volcano(CA1a,  volcano1, "16 DEGs") 
p8 <-  plot.volcano(CA1b, volcano2, "0 DEGs") 
p9 <-  plot.volcano(CA1c, volcano3, "917 DEGs") 


# plot grid
volcanos <- plot_grid(p1 + theme(legend.position = "none") + labs(y = "DG \n -log10(p-value)"),
                      p2 + theme(legend.position = "none"), 
                      p3 + theme(legend.position = "none"),
                      
                      p4 + theme(legend.position = "none") + labs(y = "CA3 \n -log10(p-value)"),
                      p5 + theme(legend.position = "none"), 
                      p6 + theme(legend.position = "none"), 
                      
                      p7  + labs( y = "CA1 \n -log10(p-value)", x = "log fold change") +
                             guides(color=guide_legend(title="increased in")), 
                      p8 + labs(x = "log fold change"), 
                      p9 + labs(x = "log fold change"), 
                      
                      nrow = 3, 
                      rel_heights =  c(1,1,1.25) ,
                      rel_widths = c(1.2,1,1) ,
                      labels =  c("a", "d", "g",
                                  "b", "e", "h",
                                  "c", "f", "i"),
                      label_size = 8) 
volcanos

comparisons <- png::readPNG("../figures/00_schematics/figure_DEGcomparisons.png")

comparisons <- ggdraw() +  draw_image(comparisons, scale = 1)

circuit <- png::readPNG("../figures/00_schematics/figure_hippocircuit.png")
circuit <- ggdraw() +  draw_image(circuit, scale = 1)

volcanoscomparisons <- plot_grid(volcanos, comparisons, nrow = 2, rel_heights = c(1, 0.1))
volcanoscomparisonscircuit <- plot_grid(circuit, volcanoscomparisons, nrow = 1, rel_widths = c(0.1,1))
volcanoscomparisonscircuit

pdf(file="../figures/03_rnaseqSubfield/volcanos.pdf", width=6.69, height=6)
plot(volcanoscomparisonscircuit)    
dev.off()

pdf(file="../figures/figure_3.pdf", width=6.69, height=6)
plot(volcanoscomparisonscircuit)    
dev.off()
```




## pca analysis 


```{r pca, fig.width=6.69, fig.height=6}
# create the dataframe using my function pcadataframe

plotPCs <- function(mydds, mysubtitle, mytitle){
  
  vsd <-  vst(mydds, blind=FALSE)
  pcadata <- pcadataframe(vsd, intgroup=c("treatment"), returnData=TRUE)
  percentVar <- round(100 * attr(pcadata, "percentVar"))

  #print(summary(aov(PC1 ~ treatment, data=pcadata))) 
  #print(TukeyHSD((aov(PC1 ~ treatment, data=pcadata)), which = "treatment"))

  #summary(aov(PC2 ~ treatment, data=pcadata)) 
  #TukeyHSD((aov(PC2 ~ treatment, data=pcadata)), which = "treatment") 
  
  apa1 <- apa.aov.table(aov(PC1 ~ treatment, data=pcadata))
  apa1 <- as.data.frame(apa1$table_body) 
  errodf <- apa1 %>% filter(Predictor == "Error") %>% pull(df)
  pvalue <- apa1 %>% filter(Predictor == "treatment") %>% pull(p)
  Fstat <- apa1 %>% filter(Predictor == "treatment") %>% pull(F)
  treatmentdf <- apa1 %>% filter(Predictor == "treatment")  %>% pull(df)
  
  mynewsubtitle <- paste(mysubtitle, "\nF(",treatmentdf, ",",errodf, ") = ",
                         Fstat, ", p=", pvalue, sep = "")

  PCA12 <- ggplot(pcadata, aes(pcadata$PC1, pcadata$PC2, color=treatment)) +
    geom_point(size=2, alpha = 0.8) +
    xlab(paste0("PC1: ", percentVar[1],"%")) +
    ylab(paste0("PC2: ", percentVar[2],"%")) +
    scale_color_manual(values = treatmentcolors) +
    labs(subtitle = mynewsubtitle) +
    theme_ms() +
    theme(legend.position = "none")
  PCA12
}

a <- plotPCs(DGdds, "DG")
b <- plotPCs(CA3dds, "CA3")
c <- plotPCs(CA1dds, "CA1")


plotPCs2 <- function(mydds, mysubtitle){
  
  vsd <-  vst(mydds, blind=FALSE)
  pcadata <- pcadataframe(vsd, intgroup=c("training"), returnData=TRUE)
  percentVar <- round(100 * attr(pcadata, "percentVar"))

  apa1 <- apa.aov.table(aov(PC1 ~ training, data=pcadata))
  apa1 <- as.data.frame(apa1$table_body) 
  errodf <- apa1 %>% filter(Predictor == "Error") %>% pull(df)
  pvalue <- apa1 %>% filter(Predictor == "training") %>% pull(p)
  Fstat <- apa1 %>% filter(Predictor == "training") %>% pull(F)
  treatmentdf <- apa1 %>% filter(Predictor == "training")  %>% pull(df)
  
  mynewsubtitle <- paste(mysubtitle, "\nF(",treatmentdf, ",",errodf, ") = ",
                         Fstat, ", p=", pvalue, sep = "")
  
  
  PCA12 <- ggplot(pcadata, aes(pcadata$PC1, pcadata$PC2, color=training)) +
    geom_point(size=2, alpha = 0.8) +
    xlab(paste0("PC1: ", percentVar[1],"%")) +
    ylab(paste0("PC2: ", percentVar[2],"%")) +
    scale_color_manual(values = volcano1) +
    labs(subtitle = mynewsubtitle) +
    theme_ms() +
    theme(legend.position = "none")
  PCA12
}

d <- plotPCs2(DGdds2, "DG")
e <- plotPCs2(CA3dds2, "CA3")
f <- plotPCs2(CA1dds2, "CA1")

abc <- plot_grid(a,b,c, nrow = 1)
def <- plot_grid(d,e,f, nrow = 1)

legend1 <- get_legend(a + theme(legend.position = "bottom", legend.title = element_blank()))
legend2 <- get_legend(d + theme(legend.position = "bottom", legend.title = element_blank()))

abcdefg <- plot_grid(abc,legend1, def, legend2, ncol = 1, rel_heights = c(4,1,4,1))
abcdefg

```


```{r citations}
citation("DESeq2") 
citation("png")
citation("grid")
citation("BiocParallel")
````