---
title: "Differential Gene Expression Analysis of Subfields"
output: md_document
---

## Subfield analysis

This script is used to identify treatement differences within each subfield, generate volcano plots, venn diagrams, and tables for subsequent GO analyses. The final mutlipanel figures for the manuscript have been inserted just below the subheadings. 

```{r setup, message=F, warning=F}
library(tidyverse)
library(cowplot) ## for some easy to use themes
library(DESeq2) ## for gene expression analysis
library(png)
library(grid)
library(scales)
library(apaTables) #  for ANOVA tables

library(BiocParallel)
register(MulticoreParam(6))

## load functions 
source("figureoptions.R")
source("functions_RNAseq.R")

## set output file for figures 
knitr::opts_chunk$set(fig.path = '../figures/03_rnaseqSubfield/', cache = T)
```

## Wrangle data

```{r DGdeseq}

# prep col data,
outliers <- c("146D-DG-3", "145A-CA3-2", "146B-DG-2", "146D-CA1-3", "148B-CA1-4")

colData <- read.csv("../data/00_colData.csv", header = T) %>%
  filter(!RNAseqID %in% outliers)
colData$training <- factor(colData$training, levels = levelstraining)
colData$treatment <- factor(colData$treatment, levels = levelstreatment)

# remove outliers
savecols <- as.character(colData$RNAseqID) #select the rowsname 
savecols <- as.vector(savecols) # make it a vector

countData <- read.csv("../data/00_countData.csv", 
                      header = T, check.names = F, row.names = 1) %>%
  dplyr::select(one_of(savecols)) # select just the columns 
head(countData)
```

## Get varience stabilized gene expression for each tissue

```{r}
# DEGs with looking at all four treatments individually
DGdds <- returnddstreatment("DG") 
CA3dds <- returnddstreatment("CA3") 
CA1dds <- returnddstreatment("CA1") 

# DEGs with looking at all grouped trained and yoked
DGdds2 <- returnddstraining("DG") 
CA3dds2 <- returnddstraining("CA3") 
CA1dds2 <- returnddstraining("CA1") 

savevsds(DGdds2, "../data/03_DG_vsdtraining.csv")
savevsds(CA3dds2, "../data/03_CA3_vsdtraining.csv")
savevsds(CA1dds2, "../data/03_CA1_vsdtraining.csv")
```

## Results to compare with volcano plots

```{r twowaycontrasts}
print("DG")
res_summary_subfield(DGdds2, c("training", "trained", "yoked"))
res_summary_subfield(DGdds, c("treatment", "conflict.trained", "standard.trained"))
res_summary_subfield(DGdds, c("treatment", "conflict.yoked", "standard.yoked"))

print("CA3")
res_summary_subfield(CA3dds2, c("training", "trained", "yoked"))
res_summary_subfield(CA3dds, c("treatment", "conflict.trained", "standard.trained"))
res_summary_subfield(CA3dds, c("treatment", "conflict.yoked", "standard.yoked"))

print("CA1")
res_summary_subfield(CA1dds2, c("training", "trained", "yoked"))
res_summary_subfield(CA1dds, c("treatment", "conflict.trained", "standard.trained"))
res_summary_subfield(CA1dds, c("treatment", "conflict.yoked", "standard.yoked"))
```

## Volcano plots

```{r DEGs}

# create data frame for making volcanos plots
DGa <-  calculateDEGs(DGdds, "DG", "treatment", "standard.trained", "standard.yoked") 
DGb <-  calculateDEGs(DGdds, "DG", "treatment", "conflict.trained", "conflict.yoked") 
DGc <-  calculateDEGs(DGdds, "DG", "treatment", "conflict.trained", "standard.trained") 
DGd <-  calculateDEGs(DGdds, "DG", "treatment", "conflict.yoked", "standard.yoked") 
DGe <- calculateDEGs(DGdds2, "DG", "training", "trained", "yoked") 

CA3a <-  calculateDEGs(CA3dds, "CA3", "treatment", "standard.trained", "standard.yoked") 
CA3b <-  calculateDEGs(CA3dds, "CA3", "treatment", "conflict.trained", "conflict.yoked") 
CA3c <-  calculateDEGs(CA3dds, "CA3", "treatment", "conflict.trained", "standard.trained") 
CA3d <-  calculateDEGs(CA3dds, "CA3", "treatment", "conflict.yoked", "standard.yoked") 
CA3e <- calculateDEGs(CA3dds2, "CA3", "training", "trained", "yoked") 

CA1a <-  calculateDEGs(CA1dds, "CA1", "treatment", "standard.trained", "standard.yoked") 
CA1b <-  calculateDEGs(CA1dds, "CA1", "treatment", "conflict.trained", "conflict.yoked") 
CA1c <-  calculateDEGs(CA1dds, "CA1", "treatment", "conflict.trained", "standard.trained") 
CA1d <-  calculateDEGs(CA1dds, "CA1", "treatment", "conflict.yoked", "standard.yoked") 
CA1e <- calculateDEGs(CA1dds2, "CA1", "training", "trained", "yoked") 

# save df with DEGs

allDEG <- rbind(DGa, DGb, DGc, DGd, DGe,
               CA3a, CA3b, CA3c, CA3d, CA3e, 
               CA1a, CA1b, CA1c, CA1d, CA1e) %>% 
  dplyr::filter(direction != "NS") %>%
    dplyr::mutate(lfc = round(lfc, 2),
                  padj = scientific(padj, digits = 3),
                  logpadj = round(logpadj, 2)) %>%
  arrange(tissue, comparison, gene)

```

## pca analysis and bar plots functions

```{r pca, fig.width=6.69, fig.height=6}


a <- plotPCs(DGdds, "DG") 
b <- plot.volcano(DGa, "std. trained vs.\nstd. yoked") + labs(y = "-log10(p-value)")
c <- plot.volcano(DGb, "con. trained vs.\ncon. yoked")  
d <- plot.volcano(DGc, "std. trained vs. \ncon. trained")  
e <- plot.volcano(DGd, "std. yoked vs. \ncon. yoked")  
f <- plot.volcano(DGe, "all yoked vs. \nall trained")  

g <- plotPCs(CA3dds, "CA3")
h <- plot.volcano(CA3a, " ") + labs(y = "-log10(p-value)")
i <- plot.volcano(CA3b, " ")  
j <- plot.volcano(CA3c, " ")  
k <- plot.volcano(CA3d, " ")  
l <- plot.volcano(CA3e, " ")  


m <- plotPCs(CA1dds, "CA1")
n <- plot.volcano(CA1a, " ") + labs(y = "-log10(p-value)")
o <- plot.volcano(CA1b, " ")  
p <- plot.volcano(CA1c, " ")   
q <- plot.volcano(CA1d, " ")  
r <- plot.volcano(CA1e, " ")  

legend <- get_legend(a + theme(legend.position = "bottom", 
                               legend.title = element_blank()) +
                       guides(color = guide_legend(nrow = 2),
                              shape = guide_legend(nrow = 2)))
legend2 <- get_legend(f + theme(legend.position = "bottom", 
                               legend.title = element_blank()) +
                       guides(shape = guide_legend(nrow = 2)))

legends <- plot_grid(legend, legend2, rel_widths = c(2,1))

mainplot <- plot_grid(a,b,c,d,e,f,g,h,i,j,k,l,m,n,o,p,q,r, 
                      nrow = 3, rel_widths = c(1,1,0.8,0.8,0.8,0.8),
                      labels = c("DG", "", "", "", "", "", 
                                 "CA3", "", "", "", "", "", 
                                 "CA1", "", "", "", "", ""),
                      label_size = 8)

circuit <- png::readPNG("../figures/00_schematics/figure_hippocircuit.png")
circuit <- ggdraw() +  draw_image(circuit, scale = 1)
circuitmainplot <- plot_grid(circuit, mainplot, ncol = 2, rel_widths = c(0.11,1))

fig3 <- plot_grid(circuitmainplot, legends, ncol = 1, rel_heights = c(1, 0.1))
fig3
```

## save files 


```{r save}
pdf(file="../figures/03_rnaseqSubfield/volcanos.pdf", width=6.69, height=6)
plot(fig3)    
dev.off()

pdf(file="../figures/fig-3.pdf", width=6.69, height=6)
plot(fig3)    
dev.off()

suppltable4 <- allDEG %>% filter(tissue == "DG"  & comparison == "yoked vs. trained")
head(suppltable4)

suppltable5 <- allDEG %>% filter(tissue != "DG"  & comparison != "yoked vs. trained")
head(suppltable5)


write_csv(suppltable4, "../data/suppltable-4.csv")
write_csv(suppltable5, "../data/suppltable-5.csv")

```





```{r citations}
citation("DESeq2") 
citation("png")
citation("grid")
citation("BiocParallel")
````