---
title: "Integration"
output: md_document
---

## Setup

```{r setup, message=F}
library(plyr)
library(dplyr)
library(reshape2)
library(Hmisc)
library(corrplot)
library(viridis)
library(cowplot)
library(pheatmap)

source("figureoptions.R")

knitr::opts_chunk$set(fig.path = '../figures/04_integration/')
```

## Import Data

```{r importdata}
# behavior
behavior <- read.csv("../data/01a_behavior.csv", header = T)

# physiology
ephys3 <- read.csv("../data/03_ephys.csv", header = T)
ephys3 <- ephys3[,(c(1:6))]
names(ephys3)[3] <- "Pre_Potentiation"
names(ephys3)[4] <- "Early_Potentiation"
names(ephys3)[5] <- "Late_Potentiation"
names(ephys3)[6] <- "Max_fEPSP"

# clearnup the rosetts data and filter extraneous samples
rossetta <- read.csv("../data/00_rossettastone.csv", header = F)
names(rossetta)[1] <- "organism"
names(rossetta)[2] <- "ID"
names(rossetta)[3] <- "Region"
names(rossetta)[4] <- "RNAseqID"
names(rossetta)[5] <- "R1filename"
rossetta$R2filename <- rossetta$R1filename
rossetta$R2filename <- gsub("R1", "R2", rossetta$R2filename)
head(rossetta)

colData <- read.csv("../data/02a_colData.csv", header = T) 
metadata <- full_join(colData, rossetta)
names(metadata)[1] <- "samplename"
metadata$title <- as.factor(paste(metadata$ID,metadata$Region, metadata$Group, sep=" "))
names(metadata)[6] <- "sourcename"
metadata$char1 <- "Mus musculus"
metadata$char2 <- "C57BL/6"
metadata$mol <- "RNA"
metadata$des <- " "
metadata$processes <- as.factor(paste(metadata$samplename,"/abundance.txt"))
metadata$process2 <- "IntegrativeWT2015ColData.csv"
str(metadata)

metadata <- dplyr::select(metadata, samplename, title, sourcename, organism, char1, char2, mol, des, processes, R1filename, R2filename, process2)

#write.csv(metadata, "../data/00_metadata.csv")

# clearnup the rosetts data and filter extraneous samples
names(rossetta)[1] <- "Mouse"
names(rossetta)[2] <- "ID"
names(rossetta)[3] <- "Region"
names(rossetta)[4] <- "RNAseqID"
names(rossetta)[5] <- "R1filename"
rossetta$R1filename <- NULL
rossetta <- rossetta %>% dplyr::filter(Mouse != "15-100", Mouse != "15-101", Mouse != "15-147")
rossetta <- unique(rossetta[ , c(1:2,4) ]) # for joining keop only

## slim behavior ephy to top 5 pcs and rename the columsn
behaviorpca <- read.csv("../data/01a_scoresdf.csv", header = T)
behaviorpca <- behaviorpca[(c(1:4,6,35:36))]
names(behaviorpca)[names(behaviorpca)=="PC1"] <- "Behavior_PC1"
names(behaviorpca)[names(behaviorpca)=="PC2"] <- "Behavior_PC2"
names(behaviorpca)[names(behaviorpca)=="PC3"] <- "Behavior_PC3"
names(behaviorpca)[names(behaviorpca)=="PC4"] <- "Behavior_PC4"
names(behaviorpca)[names(behaviorpca)=="PC6"] <- "Behavior_PC6"

behaviorpca <- behaviorpca %>% dplyr::filter(ID != "15148", ID !=  "15140A", ID !=  "15140B", ID !=  "15140C", ID !=  "15140D", ID !=  "15141C", ID !=  "15141D", ID !=  "15142C", ID !=  "15142D", ID !=  "15142A", ID !=  "15142B", ID !=  "15145C", ID !=  "15145C", ID !=  "15145D", ID !=  "15147A", ID !=  "15147B", ID !=  "15148C", ID !=  "15148D")


pcadata <- read.csv("../data/02a_pcadata.csv", header = T)
pcadata$Punch <- ifelse(grepl("DG", pcadata$group), "DG", 
                                        ifelse(grepl("CA3", pcadata$group), "CA3","CA1"))
names(pcadata)[names(pcadata)=="name"] <- "RNAseqID"
pcadata <- pcadata[c(11:17,2:10)]
```


```{r integratedata}
head(rossetta)

#widen then length RNAseq data so each row is an animals
pcadatabyregion <- dplyr::left_join(pcadata, rossetta, by = "RNAseqID")
names(pcadatabyregion)
pcadatabyregion <- pcadatabyregion[(c(1:7,17:18,8:16))]

pcadatabyregion <- melt(pcadatabyregion, id = c(1:9))
pcadatabyregion$RegionPC <- as.factor(paste(pcadatabyregion$Punch, pcadatabyregion$variable, sep="_"))
pcadatabyregion <- dcast(pcadatabyregion, Mouse ~ RegionPC)

alldata <- left_join(pcadatabyregion, rossetta , by="Mouse")
alldata <- left_join(alldata, behaviorpca, by="ID")
alldata <- left_join(alldata, ephys3, by="Mouse")
```


```{r correlationsAvoidance, warning=F}
alldataslim <- dplyr::filter(alldata, APA2 %in% c("conflict","consistent"))

names(alldataslim)

alldataslim <- alldataslim[,-c(1,5:10,14:19,23:28,29:30,36:37)]
alldataslim <- sapply( alldataslim, as.numeric )

cormat <- rcorr(as.matrix(alldataslim))
res2 <- rcorr(as.matrix(alldataslim))

corrplot(res2$r, type = "lower", order = "hclust", 
        tl.col = "black", tl.srt = 45)

corrplot(res2$r, type="lower", order="hclust",  tl.col = "black", 
        p.mat = res2$P, sig.level = 0.05, insig = "blank")

DEGes <- as.data.frame(cormat$r)
DEGes$group <- c("CA1","CA1","CA1","CA3","CA3","CA3","DG","DG","DG","Behavior","Behavior","Behavior","Behavior","Behavior","Physiology","Physiology","Physiology","Physiology")
df <- as.data.frame(DEGes$group)
names(df)[1] <- "Level"
row.names(df) <- row.names(DEGes)
DEGes$group <- NULL

paletteLength <- 30
myBreaks <- c(seq(min(cormat$r), 0, length.out=ceiling(paletteLength/2) + 1),
              seq(max(cormat$r)/paletteLength, max(cormat$r), length.out=floor(paletteLength/2)))

pheatmap(DEGes, show_colnames=T, show_rownames = T,
         annotation_col = df, annotation_colors = ann_colorsLevel,
         annotation_row = df, 
         #annotation_legend = FALSE,
         annotation_names_row = FALSE, annotation_names_col = FALSE,
         treeheight_row = 10, treeheight_col = 10,
         fontsize = 8, 
         width=2.5, height=2,
         border_color = "grey60" ,
         color = viridis(30),
         #cellwidth = 10, 
         clustering_method="average",
         breaks=myBreaks,
         clustering_distance_cols="correlation",
         clustering_distance_rows="correlation" ,
         main="Cognitively Trainied"
         )

pheatmap(cormat$r, show_colnames=F, show_rownames = T,
         annotation_col = df, annotation_colors = ann_colorsLevel,
         annotation_row = df, 
         #annotation_legend = FALSE,
         annotation_names_row = FALSE, annotation_names_col = FALSE,
         treeheight_row = 10, treeheight_col = 10,
         fontsize = 8, 
         width=4, height=2.25,
         border_color = "grey60" ,
         color = viridis(30),
         clustering_method="average",
         breaks=myBreaks,
         clustering_distance_cols="correlation",
         clustering_distance_rows="correlation",
         filename = "../figures/04_integration/corrTrained.pdf"
         )

head(alldata)
alldata$APA2 <- factor(alldata$APA2, levels = c("yoked-consistent", "consistent",  "yoked-conflict", "conflict"))

alldata$avoidance <-  ifelse(grepl("yoked", alldata$APA), "no", "yes")

alldata$wrap <- "All Cognitively Trained" 

behavDG <- alldata %>%
  filter(avoidance == "yes") %>%
  ggplot(aes(Behavior_PC1,as.numeric(DG_PC2))) + 
  geom_point(size = 2, alpha = 0.75, aes(color=APA2)) +
  scale_color_manual(values = colorvalAPA6) +
    theme_cowplot(font_size = 8, line_size = 0.25)  +
    theme(legend.position="none") +
    stat_smooth(method = "lm", color="red") + 
    ylab(paste0("DG PC1")) +
    xlab(paste0("Behavior PC1"))  +
  facet_wrap(~wrap)
behavDG

pdf(file="../figures/04_integration/behavDG.pdf", width=1.75, height=2)
plot(behavDG)
dev.off()
```



```{r correlationsYoked, warning=F}
alldata <- left_join(pcadatabyregion, rossetta , by="Mouse")
alldata <- left_join(alldata, behaviorpca, by="ID")
alldata <- left_join(alldata, ephys3, by="Mouse")

alldataslim <- dplyr::filter(alldata, APA2 %in% c("yoked-conflict","yoked-consistent"))

alldataslim <- alldataslim[,-c(1,5:10,14:19,23:28,29:30,36:37)]
alldataslim <- sapply( alldataslim, as.numeric )

cormat <- rcorr(as.matrix(alldataslim))
res2 <- rcorr(as.matrix(alldataslim))

corrplot(res2$r, type = "lower", order = "hclust", 
        tl.col = "black", tl.srt = 45)

corrplot(res2$r, type="lower", order="hclust",  tl.col = "black", 
        p.mat = res2$P, sig.level = 0.05, insig = "blank")

DEGes <- as.data.frame(cormat$r)
DEGes$group <- c("CA1","CA1","CA1","CA3","CA3","CA3","DG","DG","DG","Behavior","Behavior","Behavior","Behavior","Behavior","Physiology","Physiology","Physiology","Physiology")
df <- as.data.frame(DEGes$group)
names(df)[1] <- "Level"
row.names(df) <- row.names(DEGes)
DEGes$group <- NULL

paletteLength <- 30
myBreaks <- c(seq(min(cormat$r), 0, length.out=ceiling(paletteLength/2) + 1),
              seq(max(cormat$r)/paletteLength, max(cormat$r), length.out=floor(paletteLength/2)))
pheatmap(DEGes, show_colnames=T, show_rownames = T,
         annotation_col = df, annotation_colors = ann_colorsLevel,
         annotation_row = df, 
         #annotation_legend = FALSE,
         annotation_names_row = FALSE, annotation_names_col = FALSE,
         treeheight_row = 10, treeheight_col = 10,
         fontsize = 8, 
         width=2.5, height=2,
         border_color = "grey60" ,
         color = viridis(30),
         #cellwidth = 10, 
         clustering_method="average",
         breaks=myBreaks,
         clustering_distance_cols="correlation",
         clustering_distance_rows="correlation" ,
         main="All Yoked"
         )

pheatmap(cormat$r, show_colnames=F, show_rownames = T,
         annotation_col = df, annotation_colors = ann_colorsLevel,
         annotation_row = df, 
         #annotation_legend = FALSE,
         annotation_names_row = FALSE, annotation_names_col = FALSE,
         treeheight_row = 10, treeheight_col = 10,
         fontsize = 8, 
         width=4, height=2.25,
         border_color = "grey60" ,
         color = viridis(30),
         clustering_method="average",
         breaks=myBreaks,
         clustering_distance_cols="correlation",
         clustering_distance_rows="correlation",
         filename = "../figures/04_integration/corrYoked.pdf",
         main="All Yoked"
         )
```

Significant genes

```{r significantgenes}

# read files with pvalues
CA1trainpvals <- read.csv("../data/02c_CA1_consyokcons.csv", stringsAsFactors = F, row.names = NULL) 
CA1stresspvals <- read.csv("../data/02c_CA1_yokeconfyokcons.csv", stringsAsFactors = F, row.names = NULL)
DGtrainpvals <- read.csv("../data/02c_DG_consyokcons.csv", stringsAsFactors = F)

# remove non-significant genes and genes affected by learning and stresss
CA1learnstress <- read.csv("../data/02_CA1learningstressgenes.csv", stringsAsFactors = F)
CA1learnstress <- as.vector(CA1learnstress$gene)

CA1trainpvals <- CA1trainpvals %>% filter(!gene %in% CA1learnstress,
                                          direction != "NS")
CA1stresspvals <- CA1stresspvals %>% filter(!gene %in% CA1learnstress,
                                          direction != "NS")
DGtrainpvals <- DGtrainpvals %>% filter(!gene %in% CA1learnstress,
                                          direction != "NS")

# read variance stabilized count data
DGvsd <- read.csv("../data/02c_DGvsd.csv", stringsAsFactors = F, check.names = F) 
CA1vsd <- read.csv("../data/02c_CA1vsd.csv", stringsAsFactors = F, check.names = F)

# function to calculate and plot correlations between gene expression and behaivor
correlationdf <- function(pvalues, vsds){
  
  # add gene to df for joining
  colnames(vsds)[1] <- "gene"
  
  mydf <- left_join(pvalues, vsds)
  row.names(mydf) <- mydf$gene
  mydf <- mydf[-c(1:6)]
  mydf <- as.data.frame(t(mydf))
  
  # create an ID column to join lists of DEGs with their experssion values and the behavior of the animal 
  mydf$sample <-  row.names(mydf)
  mydf$mouse <- sapply(strsplit(as.character(mydf$sample),'-'), "[", 1)
  mydf$ID <- paste("15", mydf$mouse, sep = "")
  mydf$sample <-  NULL
  mydf$mouse <- NULL
  row.names(mydf) <- mydf$ID
  
  # randomely select 15 genes and add ID column back for joining
  mydfrandom <- mydf
  #mydfrandom <- mydfrandom[sample(1:ncol(mydfrandom), 15,replace=FALSE)]
  
  mydfrandom$ID <- row.names(mydfrandom)
  
  # subset the beahvior data to the last time point and only values shows in figure 2
  behavior <- read.csv("../data/01a_behavior.csv", header = T)
  favbehav <- behavior %>% 
    filter(TrainSession == "Retention") %>%
    select(ID, NumEntrances, Time1stEntr)
    
  favbehavgenes <- left_join(mydfrandom, favbehav)
  favbehavgenes$ID <- NULL
  
  mycordf = cor(favbehavgenes)
  #corrplot(mycordf, method = "circle", title = mytitle, tl.col = "black", tl.cex = 0.8)
  mycordf <- as.data.frame(mycordf)
  mycordf$gene <- row.names(mycordf)
  return(mycordf)
}  

corCA1stress <- correlationdf(CA1stresspvals, CA1vsd)
corCA1train <- correlationdf(CA1trainpvals, CA1vsd)
corDGtrain <- correlationdf(DGtrainpvals, DGvsd) 

corpointplot <- function(mycordf, mytitle){

  # remove self corelations
  mycordf <- mycordf %>%
    filter(!gene %in% c("NumEntrances", "Time1stEntr"))
  
  p1 <- mycordf %>%
    select(-Time1stEntr) %>% # remove corelations with other behavior
    ggplot(aes(x = reorder(gene, -NumEntrances), y = NumEntrances, color = NumEntrances)) +
    geom_point(stat = "identity") +
    theme_minimal(base_size = 7) +
    theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 2),
          panel.grid.major.x = element_blank(),
          panel.grid.minor.x = element_blank(),
          legend.position = "none")   +
    labs(subtitle = mytitle, x = "signficant genes", y = "Correlation to\nNumber of Entrances")  +
    ylim(-1, 1) +
    scale_color_viridis(limits=c(-1,1))
  
  p2 <- mycordf %>%
    select(-NumEntrances) %>% # remove corelations with other behavior
    ggplot(aes(x =  reorder(gene, Time1stEntr), y = Time1stEntr, color = Time1stEntr)) +
    geom_point(stat = "identity") +
    theme_minimal(base_size = 7) +
    theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 2),
          panel.grid.major.x = element_blank(),
          panel.grid.minor.x = element_blank(),
          legend.position = "none",
          legend.title = element_blank())   +
    labs(subtitle = mytitle, x = "signficant genes", y = "Correlation to\nTime to 1st Entrances")  +
    ylim(-1, 1) +
    scale_color_viridis(limits=c(-1,1)) 
    
  
  p12 <- plot_grid(p2,p1, nrow = 2)
    
  print(p12)
}

a <- corpointplot(corCA1train, "CA1 training-responsive genes") 
b <- corpointplot(corDGtrain, "DG training-responsive genes")

ab <- plot_grid(b,a, nrow = 1)
ab 

pdf(file="../figures/04_integration/correlationpointplots.pdf", width=5, height=4)
plot(ab)    
dev.off() 

corDGtrain %>%
  filter(Time1stEntr > 0.75 | NumEntrances < -.75 |  Time1stEntr < -.75 | NumEntrances > 0.75,
         !gene %in% c("Time1stEntr", "NumEntrances")) %>%
  select(gene, Time1stEntr, NumEntrances)

corCA1train %>%
  filter(Time1stEntr > 0.5 | NumEntrances < -.5 |  Time1stEntr < -.5 | NumEntrances > 0.5,
         !gene %in% c("Time1stEntr", "NumEntrances")) %>%
  select(gene, Time1stEntr, NumEntrances)

```

```{r}

CA1learn <- read.csv( "../data/02_CA1learninggenes.csv", stringsAsFactors = F)
CA1learnGO <- left_join(CA1trainpvals,CA1learn)
CA1learnGO


CA1stress <- read.csv( "../data/02_CA1lstressgenes.csv")
CA1stressGO <- left_join(CA1stresspvals,CA1stress)
CA1stressGO

write.csv(CA1learnGO, "../data/04_CA1learnGO.csv", row.names = F)
write.csv(CA1stressGO, "../data/04_CA1stressGO.csv", row.names = F)
```
