---
title: "GO Analysis"
output: md_document
---

```{r setup, include=FALSE}
library(tidyverse)
library(knitr)
library(kableExtra)

knitr::opts_chunk$set(echo = TRUE)
```


## Hypothesised GO Terms

```{r}
GO_response <- read.table("../data/goterms/GO_term_summary_20191121_150656.txt", sep = "\t", row.names = NULL,  fill=TRUE)
GO_response$GO <- "1. Response to Stimulus (GO:0050896)"

GO_translation <- read.table("../data/goterms/GO_term_summary_20191121_141252.txt", sep = "\t", row.names = NULL)
GO_translation$GO <- "2. Translation (GO:0006412)"

GO_synapse <- read.table("../data/goterms/GO_term_summary_20191121_145034.txt", sep = "\t", row.names = NULL)
GO_synapse$GO <- "3. Synapse Organization (GO:0050808)"

GO_learningormemory <- read.table("../data/goterms/GO_term_summary_20191121_142500.txt", sep = "\t", row.names = NULL)
GO_learningormemory$GO <- "4. Learning or Memory (GO:0007611)"

GOterms <- rbind(GO_learningormemory, GO_response, GO_translation, GO_synapse)
GOterms <- GOterms %>%
  dplyr::mutate(gene = toupper(MGI.Gene.Marker.ID)) %>% 
  dplyr::select(gene, GO) %>% 
  dplyr::distinct(gene, GO) %>% 
 group_by(gene) 
GOterms

```

## Upregulated DG DEGS


```{r}
DGDEGs <- read_csv("../data/03_DG_DEGs_yokedtrained.csv") %>% filter(direction == "trained")
DGDEGs$gene <- toupper(DGDEGs$gene)
DGDEGs$gene

GOtermsDEGs <- inner_join(GOterms, DGDEGs) %>%
  arrange(GO,gene) %>%
  dplyr::select(gene, GO) %>% 
  group_by(GO) %>%
  summarize(genes = str_c(gene, collapse = ", "))
```


## sanes

```{r}
sanesLichtman <- read_csv("../data/05_sanesLichtman.csv")  %>% rename(gene = genes)
sanesLichtman

GOtermssanesLichtman <- inner_join(GOterms, sanesLichtman) %>%
  arrange(GO,gene) %>%
  dplyr::select(gene, GO) %>% 
 group_by(GO) %>%
 summarize(genes = str_c(gene, collapse = ", "))

```

## DEGs and Sanes


```{r}
GO_DEGsSanesLichtman <- full_join(GOtermsDEGs, GOtermssanesLichtman, by = "GO") %>%
  rename("Differentially expressed in DG" = genes.x,
         "Molecules associated with LTP"  = genes.y)
kable(GO_DEGsSanesLichtman)
```


## PC1 correlations

```{r}
corrsPC1sig <- read_csv("../data/04_corrsPC1sig.csv")
corrsPC1sig$gene

## these 58 genes were used for a go analysis using shinygo http://bioinformatics.sdstate.edu/go/
## results are stored as 
  #data/05_GO_PC1enrichmentBP.csv
	#data/05_GO_PC1enrichmentCC.csv
	#data/05_GO_PC1enrichmentMF.csv

BP <- read_csv("../data/05_GO_PC1enrichmentBP.csv", n_max = 5) %>% mutate(Domain = "BP") 
CC <- read_csv("../data/05_GO_PC1enrichmentCC.csv", n_max = 5) %>% mutate(Domain = "CC") 
MF <- read_csv("../data/05_GO_PC1enrichmentMF.csv", n_max = 5) %>% mutate(Domain = "MF") 

GO_PC1 <- rbind(BP, CC, MF) %>% select(Domain, `Functional Category`, `Total genes`, `Genes in list`, Genes)

# alphabetize
GO_PC1$Genes <- GO_PC1$Genes %>% str_split(., ' ') %>% lapply(., 'sort') %>%  lapply(., 'paste', collapse=' ') %>% unlist(.)

kable(GO_PC1) 
```
```{r write}
write.csv(GO_DEGsSanesLichtman, "../data/05_GO_DEGsSanesLichtman.csv", row.names = F)
write.csv(GO_PC1, "../data/05_GO_PC1.csv", row.names = F)
```