Hypothesised and random GO Terms
--------------------------------

Do DEGs fall into one of four hypothesized GO catagoreis? Do they do
this more often that 4 randomly selected GO terms that are similar in
size to the hypothesized?

GO terms and genes collected from
<a href="http://www.informatics.jax.org/vocab/gene_ontology" class="uri">http://www.informatics.jax.org/vocab/gene_ontology</a>

    # list of random and hypothesized GO terms and associated gens 
    # https://serialmentor.com/blog/2016/6/13/reading-and-combining-many-tidy-data-files-in-R

    data_path <- "../data/goterms/"   # path to the data
    files <- dir(data_path, pattern = "*.txt") # get file names

    filesdf <- as.data.frame(files)
    filesdf$source <- c(1:8)
    filesdf

    ##                          files source
    ## 1    01_responsetostimulus.txt      1
    ## 2           02_translation.txt      2
    ## 3   03_synapseorganization.txt      3
    ## 4      04_learningormemory.txt      4
    ## 5  05_random_436_GO0050673.txt      5
    ## 6 06_random_1513_GO0022414.txt      6
    ## 7  07_random_283_GO0048511.txt      7
    ## 8 08_random_2720_GO0051704.txt      8

    GOterms <- files %>%
      # read in all the files, appending the path before the filename
      map_df(~ read.table(file.path(data_path, .),  sep = "\t", 
                          row.names = NULL,  fill=TRUE), .id = "source") %>%
      mutate(source = as.numeric(source)) %>%
      select(source, MGI.Gene.Marker.ID) %>%
      left_join(filesdf, .)  %>% 
      mutate_at(.vars = "files",  .funs = gsub,
                pattern = ".txt", replacement = "") %>%
      dplyr::rename("GO" = "files",
             "gene" = "MGI.Gene.Marker.ID") %>%
      dplyr::select(GO, gene)  %>%
      dplyr::distinct(GO, gene) 
    head(GOterms)

    ##                      GO          gene
    ## 1 01_responsetostimulus 1500002F19Rik
    ## 2 01_responsetostimulus 1600014C10Rik
    ## 3 01_responsetostimulus 1700007K13Rik
    ## 4 01_responsetostimulus 1700123M08Rik
    ## 5 01_responsetostimulus 1810014B01Rik
    ## 6 01_responsetostimulus 1810024B03Rik

    GOtermCounts <- GOterms %>%
      group_by(GO) %>%
      summarize("total" = n())
    head(GOtermCounts)

    ## # A tibble: 6 x 2
    ##   GO                       total
    ##   <chr>                    <int>
    ## 1 01_responsetostimulus     6271
    ## 2 02_translation             128
    ## 3 03_synapseorganization     465
    ## 4 04_learningormemory        292
    ## 5 05_random_436_GO0050673    436
    ## 6 06_random_1513_GO0022414  1432

    # DEGs
    DGDEGs <- read_csv("../data/suppltable-4.csv") %>% 
      filter(tissue == "DG", comparison == "yoked vs. trained") %>%
      select(gene) 
    head(DGDEGs)

    ## # A tibble: 6 x 1
    ##   gene         
    ##   <chr>        
    ## 1 1190002N15Rik
    ## 2 A830010M20Rik
    ## 3 Abhd2        
    ## 4 Acan         
    ## 5 Adamts1      
    ## 6 Adrb1

    # candidate genes
    candidategenes <- c("Camk2a" ,  "Gria2",  "Igf2" , "Nsf" , "Pick1",
                            "Prkcb" ,  "Prkci",  "Prkcz", "Wwc1" , "Fmr1", "Mtor")
    candidategenes <- as.data.frame(candidategenes) %>%
      dplyr::rename("gene" = "candidategenes")
    head(candidategenes)

    ##     gene
    ## 1 Camk2a
    ## 2  Gria2
    ## 3   Igf2
    ## 4    Nsf
    ## 5  Pick1
    ## 6  Prkcb

    # Sanes & Lichtman genes
    sanesLichtman <- read_csv("../data/05_sanesLichtman.csv")  %>% 
      dplyr::rename("gene" = "genes") %>%
      dplyr::mutate(gene = str_to_title(gene))  
    head(sanesLichtman)

    ## # A tibble: 6 x 1
    ##   gene  
    ##   <chr> 
    ## 1 Ache  
    ## 2 Adcy1 
    ## 3 Adra2a
    ## 4 Adra2b
    ## 5 Adra2c
    ## 6 Adrb1

    joinGOnGenes <- function(whichgenes){
      GOtermsgenes <- inner_join(GOterms, whichgenes) %>%
        arrange(GO,gene) %>%
        dplyr::select(gene, GO) %>% 
        group_by(GO) %>%
        summarize(genes = str_c(gene, collapse = ", ")) 
      return(GOtermsgenes)
    }

    GO_DEGs <- joinGOnGenes(DGDEGs)
    GO_candidates <- joinGOnGenes(candidategenes)
    GO_sanesLichtman <- joinGOnGenes(sanesLichtman)

    table2 <- left_join(GO_DEGs, GO_candidates, by = "GO") %>%
      left_join(., GO_sanesLichtman, by = "GO") %>% 
      rename("DG_DEGs" = "genes.x",
            "candidates"  = "genes.y", 
            "Sanes_Lichtman" = "genes") 
    table2$GO <- factor(table2$GO)
    table2 <- table2 %>%
      mutate(GO = fct_recode(GO,
                             "Response to stimulus (GO:0050896)" = "01_responsetostimulus",
                              "Translation (GO:0006412)"   =   "02_translation" ,
                        "Synapse organization (GO:0050808)" =    "03_synapseorganization" ,
                            "Learning or memory (GO:0007611)" =   "04_learningormemory" ))  %>%
      filter(!GO %in% c("05_random_436_GO0050673", "06_random_1513_GO0022414",
                      "07_random_283_GO0048511" ,  "08_random_2720_GO0051704" ) ) %>%
      rename("GO terms" = "GO",
             "DG DEGs" = "DG_DEGs",
             "Candidate genes" = "candidates",
             "LTP genes" = "Sanes_Lichtman")
    table2

    ## # A tibble: 4 x 4
    ##   `GO terms`     `DG DEGs`           `Candidate genes`  `LTP genes`        
    ##   <fct>          <chr>               <chr>              <chr>              
    ## 1 Response to s… Abhd2, Adrb1, Ahr,… Camk2a, Fmr1, Gri… Adcy1, Adra2a, Adr…
    ## 2 Translation (… Cpeb4, Eif5         Fmr1, Mtor         <NA>               
    ## 3 Synapse organ… Amigo2, Arc, Bdnf,… Fmr1, Pick1        Ache, Bdnf, Cacna1…
    ## 4 Learning or m… Adrb1, Arc, Bdnf, … Igf2, Mtor, Prkcz  Adcy1, Adrb1, Adrb…

    DGDEGs$candidategene <- DGDEGs$gene

    Allgogenes <- full_join(GOterms, DGDEGs, by = "gene") %>%
        arrange(GO, candidategene,gene) %>%
        dplyr::select( GO, gene)  %>%
      mutate(category = ifelse(grepl("random", GO), "random", "hypothesis")) %>%
      drop_na() %>%
      group_by(GO) %>%
        summarize(total = n()) 
    Allgogenes

    ## # A tibble: 8 x 2
    ##   GO                       total
    ##   <chr>                    <int>
    ## 1 01_responsetostimulus     6271
    ## 2 02_translation             128
    ## 3 03_synapseorganization     465
    ## 4 04_learningormemory        292
    ## 5 05_random_436_GO0050673    436
    ## 6 06_random_1513_GO0022414  1432
    ## 7 07_random_283_GO0048511    283
    ## 8 08_random_2720_GO0051704  2222

    DEGGOgenes <- full_join(GOterms, DGDEGs, by = "gene") %>%
        arrange(GO, candidategene, gene) %>%
        dplyr::select( GO, gene, candidategene)  %>%
      mutate(category = ifelse(grepl("random", GO), "random", "hypothesis")) %>%
      drop_na() %>%
      group_by(GO) %>%
        summarize(DEGs = n(),
                  genes = str_c(gene, collapse = ", ")) 
    DEGGOgenes

    ## # A tibble: 8 x 3
    ##   GO                  DEGs genes                                           
    ##   <chr>              <int> <chr>                                           
    ## 1 01_responsetostim…    79 Abhd2, Adrb1, Ahr, Ankrd27, Apaf1, Arc, Arid5b,…
    ## 2 02_translation         2 Cpeb4, Eif5                                     
    ## 3 03_synapseorganiz…    10 Amigo2, Arc, Bdnf, Flrt3, Fzd5, Homer1, Lrrtm2,…
    ## 4 04_learningormemo…    15 Adrb1, Arc, Bdnf, Btg2, Egr1, Hmgcr, Jun, Npas4…
    ## 5 05_random_436_GO0…    10 Cxadr, Egr3, Errfi1, Jun, Myc, Nr4a1, Nr4a3, Sc…
    ## 6 06_random_1513_GO…    21 Abhd2, Adamts1, Ahr, Arid5b, Cited2, Ctcfl, Dna…
    ## 7 07_random_283_GO0…    11 Adamts1, Adrb1, Ahr, Ciart, Egr1, Fbxw7, Fzd4, …
    ## 8 08_random_2720_GO…    20 Abhd2, Adamts1, Ccnk, Cited2, Ctcfl, Cxadr, Dna…

    propALLgoDEGs <- inner_join(Allgogenes, DEGGOgenes) %>%  
      select(-genes) %>%  
       mutate(prop = DEGs / total * 100)
    propALLgoDEGs

    ## # A tibble: 8 x 4
    ##   GO                       total  DEGs  prop
    ##   <chr>                    <int> <int> <dbl>
    ## 1 01_responsetostimulus     6271    79 1.26 
    ## 2 02_translation             128     2 1.56 
    ## 3 03_synapseorganization     465    10 2.15 
    ## 4 04_learningormemory        292    15 5.14 
    ## 5 05_random_436_GO0050673    436    10 2.29 
    ## 6 06_random_1513_GO0022414  1432    21 1.47 
    ## 7 07_random_283_GO0048511    283    11 3.89 
    ## 8 08_random_2720_GO0051704  2222    20 0.900

    ALLgoDEGs <- inner_join(Allgogenes, DEGGOgenes) %>%  
      mutate(category = ifelse(grepl("random", GO), "random", "hypothesis")) %>%
       group_by(category) %>%
        summarize(totals = sum(total),
                  DEGss = sum(DEGs)) %>%
      mutate(prop = DEGss / totals * 100)
    ALLgoDEGs

    ## # A tibble: 2 x 4
    ##   category   totals DEGss  prop
    ##   <chr>       <int> <int> <dbl>
    ## 1 hypothesis   7156   106  1.48
    ## 2 random       4373    62  1.42

    res <- prop.test(x = c(106, 62), n = c(7156, 4373), alternative = "greater")
    res

    ## 
    ##  2-sample test for equality of proportions with continuity
    ##  correction
    ## 
    ## data:  c(106, 62) out of c(7156, 4373)
    ## X-squared = 0.038384, df = 1, p-value = 0.4223
    ## alternative hypothesis: greater
    ## 95 percent confidence interval:
    ##  -0.003312998  1.000000000
    ## sample estimates:
    ##     prop 1     prop 2 
    ## 0.01481274 0.01417791

    res <- prop.test(x = c(15, 62), n = c(292, 4373), alternative = "greater") #LM
    # Printing the results
    res 

    ## 
    ##  2-sample test for equality of proportions with continuity
    ##  correction
    ## 
    ## data:  c(15, 62) out of c(292, 4373)
    ## X-squared = 21.089, df = 1, p-value = 2.192e-06
    ## alternative hypothesis: greater
    ## 95 percent confidence interval:
    ##  0.01391377 1.00000000
    ## sample estimates:
    ##     prop 1     prop 2 
    ## 0.05136986 0.01417791

    res <- prop.test(x = c(10, 62), n = c(465, 4373), alternative = "greater") #LM
    # Printing the results
    res 

    ## 
    ##  2-sample test for equality of proportions with continuity
    ##  correction
    ## 
    ## data:  c(10, 62) out of c(465, 4373)
    ## X-squared = 1.0801, df = 1, p-value = 0.1493
    ## alternative hypothesis: greater
    ## 95 percent confidence interval:
    ##  -0.005311281  1.000000000
    ## sample estimates:
    ##     prop 1     prop 2 
    ## 0.02150538 0.01417791

PC1 correlations
----------------

    ## these 58 genes that are differentially expressed in DG and correlated with PC1 were use to query GO http://bioinformatics.sdstate.edu/go/
    ## results are stored as 
      #data/05_GO_PC1enrichmentBP.csv
        #data/05_GO_PC1enrichmentCC.csv
        #data/05_GO_PC1enrichmentMF.csv

    BP <- read_csv("../data/05_GO_PC1enrichmentBP.csv", n_max = 5) %>% mutate(Domain = "BP") 
    CC <- read_csv("../data/05_GO_PC1enrichmentCC.csv", n_max = 5) %>% mutate(Domain = "CC") 
    MF <- read_csv("../data/05_GO_PC1enrichmentMF.csv", n_max = 5) %>% mutate(Domain = "MF") 

    table3 <- rbind(BP, CC, MF) %>% 
      select(Domain, `Functional Category`,  Genes) %>%
      rename(gene = Genes) %>%
      dplyr::mutate(gene = str_to_title(gene))

    # alphabetize
    table3$gene <- table3$gene %>% str_split(., ' ') %>% 
      lapply(., 'sort') %>%  
      lapply(., 'paste', collapse=' ') %>% 
      unlist(.)

    head(table3) 

    ## # A tibble: 6 x 3
    ##   Domain `Functional Categor… gene                                         
    ##   <chr>  <chr>                <chr>                                        
    ## 1 BP     Memory               Arc Bdnf Egr1 Kcnk10 Npas4 Plk2 Ptgs2 Sgk1 S…
    ## 2 BP     Learning or memory   Arc Bdnf Btg2 Egr1 Kcnk10 Npas4 Plk2 Ptgs2 S…
    ## 3 BP     Tissue development   Acan Arc Areg Atf3 Bdnf Btg2 Col10a1 Egr1 Er…
    ## 4 BP     Cognition            Arc Bdnf Btg2 Egr1 Kcnk10 Npas4 Plk2 Ptgs2 S…
    ## 5 BP     Behavior             Arc Bdnf Btg2 Egr1 Homer1 Kcnk10 Npas4 Nr4a3…
    ## 6 CC     Neuron projection    Acan Arc Bdnf Cpeb4 Fzd5 Homer1 Nexn Pcdh8 P…

    write.csv(table2, "../data/table-2.csv", row.names = F)
    write.csv(table3, "../data/table-3.csv", row.names = F)
