Hypothesised GO Terms
---------------------

    GO_response <- read.table("../data/goterms/01_responsetostimulus.txt", sep = "\t", row.names = NULL,  fill=TRUE)
    GO_response$GO <- "1. Response to Stimulus (GO:0050896)"

    GO_translation <- read.table("../data/goterms/02_translation.txt", sep = "\t", row.names = NULL)
    GO_translation$GO <- "2. Translation (GO:0006412)"

    GO_synapse <- read.table("../data/goterms/03_synapseorganization.txt", sep = "\t", row.names = NULL)
    GO_synapse$GO <- "3. Synapse Organization (GO:0050808)"

    GO_learningormemory <- read.table("../data/goterms/04_learningormemory.txt", sep = "\t", row.names = NULL)
    GO_learningormemory$GO <- "4. Learning or Memory (GO:0007611)"

    GOterms <- rbind(GO_learningormemory, GO_response, GO_translation, GO_synapse)
    GOterms <- GOterms %>%
      dplyr::mutate(gene = str_to_title(MGI.Gene.Marker.ID)) %>% 
      dplyr::select(gene, GO) %>% 
      dplyr::distinct(gene, GO) %>% 
     group_by(gene) 
    GOterms

    ## # A tibble: 7,156 x 2
    ## # Groups:   gene [6,717]
    ##    gene  GO                                
    ##    <chr> <chr>                             
    ##  1 Aaas  4. Learning or Memory (GO:0007611)
    ##  2 Aal   4. Learning or Memory (GO:0007611)
    ##  3 Abca7 4. Learning or Memory (GO:0007611)
    ##  4 Abcc8 4. Learning or Memory (GO:0007611)
    ##  5 Abi2  4. Learning or Memory (GO:0007611)
    ##  6 Abl2  4. Learning or Memory (GO:0007611)
    ##  7 Adam2 4. Learning or Memory (GO:0007611)
    ##  8 Adcy1 4. Learning or Memory (GO:0007611)
    ##  9 Adcy3 4. Learning or Memory (GO:0007611)
    ## 10 Adcy8 4. Learning or Memory (GO:0007611)
    ## # … with 7,146 more rows

Upregulated DG DEGS
-------------------

    DGDEGs <- read_csv("../data/suppltable-4.csv") %>% 
      filter(tissue == "DG", comparison == "yoked vs. trained")

    ## Parsed with column specification:
    ## cols(
    ##   tissue = col_character(),
    ##   gene = col_character(),
    ##   lfc = col_double(),
    ##   padj = col_double(),
    ##   logpadj = col_double(),
    ##   comparison = col_character(),
    ##   direction = col_character()
    ## )

    DGDEGs$gene

    ##   [1] "Hs6st1"        "Pou3f3"        "Zdbf2"         "Fzd5"         
    ##   [5] "Scg2"          "Cul3"          "Irs1"          "Per2"         
    ##   [9] "Btg2"          "Rgs2"          "Ptgs2"         "Rgs4"         
    ##  [13] "Olfml2b"       "Eprs"          "Atf3"          "Fam107b"      
    ##  [17] "Arl5b"         "Otud1"         "Slc25a25"      "Sh2d3c"       
    ##  [21] "Nr4a2"         "Gad1"          "Ctnnd1"        "Gm13889"      
    ##  [25] "Kcna4"         "Bdnf"          "Thbs1"         "Pak6"         
    ##  [29] "Ccdc32"        "Lcmt2"         "Flrt3"         "Tspyl3"       
    ##  [33] "Ctcfl"         "Gmeb2"         "Gyg"           "Foxo1"        
    ##  [37] "Tsc22d2"       "Siah2"         "Tiparp"        "Fbxw7"        
    ##  [41] "Lmna"          "Ciart"         "Slc16a1"       "Prpf38b"      
    ##  [45] "Dnajb4"        "Dnaja1"        "Nr4a3"         "Jun"          
    ##  [49] "Plk3"          "Errfi1"        "Plch2"         "Cyp51"        
    ##  [53] "Cldn12"        "Fosl2"         "Tnip2"         "Rbm47"        
    ##  [57] "Rasl11b"       "Coq2"          "A830010M20Rik" "Mn1"          
    ##  [61] "Ankrd13a"      "Pxn"           "Ubc"           "Nptx2"        
    ##  [65] "Rasl11a"       "Uspl1"         "Hsph1"         "Peg10"        
    ##  [69] "Bmt2"          "Mest"          "Kdm7a"         "Neurod6"      
    ##  [73] "Nap1l5"        "Egr4"          "Cecr6"         "Slc2a3"       
    ##  [77] "Dusp16"        "Gpr19"         "Fosb"          "Erf"          
    ##  [81] "Gpi1"          "Ankrd27"       "Ppp1r15a"      "Spty2d1"      
    ##  [85] "Lysmd4"        "Sv2b"          "Acan"          "Abhd2"        
    ##  [89] "Il16"          "Fzd4"          "Zfp668"        "Dusp8"        
    ##  [93] "Irs2"          "Dusp4"         "Lonrf1"        "Klkb1"        
    ##  [97] "Ing2"          "Mfap3l"        "Naf1"          "Zfp869"       
    ## [101] "Jund"          "Slc5a5"        "Klf2"          "Scoc"         
    ## [105] "Dnajb1"        "Junb"          "Tmem170"       "Cpne7"        
    ## [109] "Mc1r"          "Irf2bp2"       "Senp8"         "C2cd4b"       
    ## [113] "1190002N15Rik" "Arpp21"        "Plagl1"        "Cited2"       
    ## [117] "Heca"          "Sgk1"          "Ranbp2"        "Sowahc"       
    ## [121] "Jmjd1c"        "Arid5b"        "Gnaz"          "Nuak1"        
    ## [125] "Apaf1"         "Dusp6"         "Kitl"          "Phlda1"       
    ## [129] "Dyrk2"         "Lemd3"         "BC048403"      "Srgap1"       
    ## [133] "Pgam2"         "Peli1"         "B3gnt2"        "Cpeb4"        
    ## [137] "Med7"          "Rasd1"         "Per1"          "Kdm6b"        
    ## [141] "Zfp207"        "Dusp14"        "Cwc25"         "Stac2"        
    ## [145] "Arl4d"         "Kcnj2"         "Sox9"          "Lrrc45"       
    ## [149] "Kcnf1"         "Odc1"          "Ahr"           "Arl4a"        
    ## [153] "Foxg1"         "Fbxo33"        "Frmd6"         "Dbpht2"       
    ## [157] "Fos"           "Jdp2"          "Ccnk"          "Eif5"         
    ## [161] "Klf6"          "Eef1e1"        "Nedd9"         "Gadd45g"      
    ## [165] "Nfil3"         "Homer1"        "Hmgcr"         "Plk2"         
    ## [169] "Snx18"         "Ankrd28"       "Fermt2"        "Stmn4"        
    ## [173] "Dpysl2"        "Nefm"          "Egr3"          "Pcdh8"        
    ## [177] "Slitrk5"       "March11"       "Ankrd33b"      "Trib1"        
    ## [181] "Myc"           "Slc45a4"       "Arc"           "Fam118a"      
    ## [185] "Amigo2"        "Nr4a1"         "Igf2bp2"       "Tra2b"        
    ## [189] "Arl13b"        "Zfp654"        "Cxadr"         "Adamts1"      
    ## [193] "Bach1"         "Rgmb"          "Hspa1a"        "Srf"          
    ## [197] "Rfx2"          "Tubb4a"        "Lbh"           "Syt4"         
    ## [201] "Egr1"          "Lrrtm2"        "Pde6a"         "Smad7"        
    ## [205] "Npas4"         "Nxf1"          "Prune2"        "Zfand5"       
    ## [209] "Cnnm1"         "Adrb1"         "Zbtb33"        "Zfp275"       
    ## [213] "Armcx5"        "Piga"

    GOtermsDEGs <- inner_join(GOterms, DGDEGs) %>%
      arrange(GO,gene) %>%
      dplyr::select(gene, GO) %>% 
      group_by(GO) %>%
      summarize(genes = str_c(gene, collapse = ", "))

    ## Joining, by = "gene"

    head(GOtermsDEGs)

    ## # A tibble: 4 x 2
    ##   GO                        genes                                          
    ##   <chr>                     <chr>                                          
    ## 1 1. Response to Stimulus … Abhd2, Adrb1, Ahr, Ankrd27, Apaf1, Arc, Arid5b…
    ## 2 2. Translation (GO:00064… Cpeb4, Eif5                                    
    ## 3 3. Synapse Organization … Amigo2, Arc, Bdnf, Flrt3, Fzd5, Homer1, Lrrtm2…
    ## 4 4. Learning or Memory (G… Adrb1, Arc, Bdnf, Btg2, Egr1, Hmgcr, Jun, Npas…

are genes in GO term significant?
---------------------------------

    totalGOterm <- GOterms %>%
      group_by(GO) %>%
      summarise(totalgenes = n())

    totalSignificantgenesperGO <- inner_join(GOterms, DGDEGs) %>%
      arrange(GO,gene) %>%
      dplyr::select(gene, GO) %>% 
      group_by(GO) %>%
      summarise(significantgenes = n())

    ## Joining, by = "gene"

    GOsummary <- full_join(totalGOterm, totalSignificantgenesperGO) %>%
      mutate(percent = significantgenes / totalgenes * 100) %>%
      mutate(percent = round(percent,2))

    ## Joining, by = "GO"

    GOsummary

    ## # A tibble: 4 x 4
    ##   GO                                   totalgenes significantgenes percent
    ##   <chr>                                     <int>            <int>   <dbl>
    ## 1 1. Response to Stimulus (GO:0050896)       6271               78    1.24
    ## 2 2. Translation (GO:0006412)                 128                2    1.56
    ## 3 3. Synapse Organization (GO:0050808)        465               10    2.15
    ## 4 4. Learning or Memory (GO:0007611)          292               15    5.14

what about random GO terms
--------------------------

FROM
<a href="http://geneontology.org/docs/ontology-documentation/" class="uri">http://geneontology.org/docs/ontology-documentation/</a>
The larger processes, or ‘biological programs’ accomplished by multiple
molecular activities. **Examples of broad biological process terms** are
*DNA repair* or *signal transduction*. Examples of more specific terms
are *pyrimidine nucleobase biosynthetic process* or *glucose
transmembrane transport*. Note that a biological process is not
equivalent to a pathway. At present, the GO does not try to represent
the dynamics or dependencies that would be required to fully describe a
pathway. Examples of broad biological process terms are DNA repair or
signal transduction. Examples of more specific terms are pyrimidine
nucleobase biosynthetic process or glucose transmembrane transport.

Found all those at
<a href="http://www.informatics.jax.org/vocab/gene_ontology" class="uri">http://www.informatics.jax.org/vocab/gene_ontology</a>

*Note: “pyrimidine nucleobase biosynthetic process” was too small, so I
went up to leveles and chose “heterocycle biosynthetic process”
instead.*

    GO_DNArepair <- read.table("../data/goterms/05_DNArepair.txt", sep = "\t", row.names = NULL,  fill=TRUE)
    GO_DNArepair$GO <- "5. DNA repair (GO:0006281)" 

    GO_signaltransduction <- read.table("../data/goterms/06_signaltransduction.txt", sep = "\t", row.names = NULL,  fill=TRUE)

    ## Warning in scan(file = file, what = what, sep = sep, quote = quote, dec =
    ## dec, : EOF within quoted string

    GO_signaltransduction$GO <- "6. Signal transduction (GO:0007165)" 

    GO_heterocyclebiosyntheticprocess <- read.table("../data/goterms/07_heterocyclebiosyntheticprocess.txt", sep = "\t", row.names = NULL,  fill=TRUE)
    GO_heterocyclebiosyntheticprocess$GO <- "7. Heterocycle biosynthetic process (GO:0018130)" 


    GO_glucosetransmembranetransport <- read.table("../data/goterms/08_glucosetransmembranetransport.txt", sep = "\t", row.names = NULL,  fill=TRUE)
    GO_glucosetransmembranetransport$GO <- "8. Glucose transmembrane transport (GO:1904659)" 

    GOterms2 <- rbind(GO_DNArepair, GO_signaltransduction, GO_heterocyclebiosyntheticprocess, GO_glucosetransmembranetransport)
    GOterms2 <- GOterms2 %>%
      dplyr::mutate(gene = str_to_title(MGI.Gene.Marker.ID)) %>%
      dplyr::select(gene, GO) %>% 
      dplyr::distinct(gene, GO) %>% 
     group_by(gene) 
    GOterms2

    ## # A tibble: 6,528 x 2
    ## # Groups:   gene [6,015]
    ##    gene          GO                        
    ##    <chr>         <chr>                     
    ##  1 4930447c04rik 5. DNA repair (GO:0006281)
    ##  2 Abl1          5. DNA repair (GO:0006281)
    ##  3 Abraxas1      5. DNA repair (GO:0006281)
    ##  4 Actl6a        5. DNA repair (GO:0006281)
    ##  5 Actr2         5. DNA repair (GO:0006281)
    ##  6 Actr5         5. DNA repair (GO:0006281)
    ##  7 Actr8         5. DNA repair (GO:0006281)
    ##  8 Adprhl2       5. DNA repair (GO:0006281)
    ##  9 Alkbh1        5. DNA repair (GO:0006281)
    ## 10 Alkbh2        5. DNA repair (GO:0006281)
    ## # … with 6,518 more rows

    totalGOterm2 <- GOterms2 %>%
      group_by(GO) %>%
      summarise(totalgenes = n())

    totalSignificantgenesperGO2 <- inner_join(GOterms2, DGDEGs) %>%
      arrange(GO,gene) %>%
      dplyr::select(gene, GO) %>% 
      group_by(GO) %>%
      summarise(significantgenes = n())

    ## Joining, by = "gene"

    GOsummary2 <- full_join(totalGOterm2, totalSignificantgenesperGO2) %>%
      mutate(percent = significantgenes / totalgenes * 100) %>%
      mutate(percent = round(percent,2))

    ## Joining, by = "GO"

    GOsummary2

    ## # A tibble: 4 x 4
    ##   GO                                    totalgenes significantgenes percent
    ##   <chr>                                      <int>            <int>   <dbl>
    ## 1 5. DNA repair (GO:0006281)                   360                3    0.83
    ## 2 6. Signal transduction (GO:0007165)         4585               64    1.4 
    ## 3 7. Heterocycle biosynthetic process …       1464               29    1.98
    ## 4 8. Glucose transmembrane transport (…        119                5    4.2

    rbind(GOsummary,GOsummary2)

    ## # A tibble: 8 x 4
    ##   GO                                    totalgenes significantgenes percent
    ##   <chr>                                      <int>            <int>   <dbl>
    ## 1 1. Response to Stimulus (GO:0050896)        6271               78    1.24
    ## 2 2. Translation (GO:0006412)                  128                2    1.56
    ## 3 3. Synapse Organization (GO:0050808)         465               10    2.15
    ## 4 4. Learning or Memory (GO:0007611)           292               15    5.14
    ## 5 5. DNA repair (GO:0006281)                   360                3    0.83
    ## 6 6. Signal transduction (GO:0007165)         4585               64    1.4 
    ## 7 7. Heterocycle biosynthetic process …       1464               29    1.98
    ## 8 8. Glucose transmembrane transport (…        119                5    4.2

sanes
-----

    sanesLichtman <- read_csv("../data/05_sanesLichtman.csv")  %>% 
      rename(gene = genes) %>%
      dplyr::mutate(gene = str_to_title(gene))

    ## Parsed with column specification:
    ## cols(
    ##   genes = col_character()
    ## )

    sanesLichtman

    ## # A tibble: 237 x 1
    ##    gene   
    ##    <chr>  
    ##  1 Ache   
    ##  2 Adcy1  
    ##  3 Adra2a 
    ##  4 Adra2b 
    ##  5 Adra2c 
    ##  6 Adrb1  
    ##  7 Adrb2  
    ##  8 Adrb3  
    ##  9 Bdnf   
    ## 10 Cacna1a
    ## # … with 227 more rows

    GOtermssanesLichtman <- inner_join(GOterms, sanesLichtman) %>%
      arrange(GO,gene) %>%
      dplyr::select(gene, GO) %>% 
     group_by(GO) %>%
     summarize(genes = str_c(gene, collapse = ", "))

    ## Joining, by = "gene"

DEGs and Sanes
--------------

    GO_DEGsSanesLichtman <- full_join(GOtermsDEGs, GOtermssanesLichtman, by = "GO") %>%
      rename("Differentially expressed in DG" = genes.x,
             "Molecules associated with LTP"  = genes.y)
    kable(GO_DEGsSanesLichtman)

<table>
<thead>
<tr>
<th style="text-align:left;">
GO
</th>
<th style="text-align:left;">
Differentially expressed in DG
</th>
<th style="text-align:left;">
Molecules associated with LTP
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">

1.  Response to Stimulus
    (<a href="GO:0050896" class="uri">GO:0050896</a>)
    </td>
    <td style="text-align:left;">
    Abhd2, Adrb1, Ahr, Ankrd27, Apaf1, Arc, Arid5b, Arl13b, Arpp21,
    Atf3, B3gnt2, Bach1, Bmt2, Btg2, C2cd4b, Ccnk, Cited2, Fbxw7,
    Fermt2, Flrt3, Fos, Fosb, Fosl2, Foxg1, Foxo1, Fzd4, Fzd5, Gadd45g,
    Gnaz, Gpi1, Gpr19, Hmgcr, Homer1, Hspa1a, Hsph1, Il16, Ing2, Irs1,
    Irs2, Jun, Junb, Jund, Kdm6b, Kitl, Klf2, Klf6, Klkb1, Lbh, Lemd3,
    Lmna, Mc1r, Mest, Myc, Nedd9, Nfil3, Npas4, Nptx2, Nr4a1, Nr4a2,
    Nr4a3, Ppp1r15a, Slc16a1, Slc25a25, Slitrk5, Smad7, Sox9, Srf,
    Srgap1, Stac2, Syt4, Thbs1, Tiparp, Tnip2, Tra2b, Trib1, Tsc22d2,
    Zbtb33, Zfand5
    </td>
    <td style="text-align:left;">
    Adcy1, Adra2a, Adra2b, Adra2c, Adrb1, Adrb2, Adrb3, Cacna1a,
    Cacna1b, Cacna1c, Cacna1d, Cacna1e, Cacna1f, Cacna1s, Calb1, Calm1,
    Calm2, Calm3, Camk1, Camk4, Capn1, Capn10, Capn2, Capn3, Ccr7, Cd47,
    Cdh1, Cdh2, Chrm1, Chrm2, Chrm3, Chrm4, Chrm5, Chrna1, Chrna3,
    Chrna7, Chrnb1, Chrnb2, Chrnb3, Cnga2, Fgf2, Fyn, Gabbr1, Gabra1,
    Gabra2, Gabra3, Gabra5, Gabra6, Gabrb1, Gabrb2, Gabrb3, Gabrr1,
    Gap43, Gfap, Gria1, Gria2, Grin1, Grin2a, Grin2d, Grm1, Grm4, Grm5,
    Grm7, Gucy1a2, Gucy1b2, Gucy2c, Gucy2d, Gucy2e, Gucy2g, Homer1,
    Homer2, Homer3, Htr1a, Htr1b, Htr1f, Htr2a, Htr2b, Htr2c, Htr3a,
    Htr3b, Htr4, Htr5a, Htr5b, Htr6, Htr7, Il1b, Inhba, Itga1, Itga10,
    Itga11, Itga2, Itga2b, Itga3, Itga4, Itga5, Itga6, Itga7, Itga8,
    Itga9, Itgad, Itgae, Itgal, Itgam, Itgav, Itgax, Itgb1, Itgb1bp1,
    Itgb2, Itgb2l, Itgb3, Itgb4, Itgb5, Itgb6, Itgb7, Itgb8, Itgbl1,
    Itpkb, L1cam, Mapk1, Mapk11, Mapk12, Mapk14, Mapk3, Mapk4, Mapk6,
    Mapk7, Mapk8, Mapk9, Mas1, Ncam1, Ngf, Nos1, Nos3, Nrg1, Nrg2, Nrg3,
    Nrgn, Pnoc, Sptbn1, Src, Stx1b, Syp, Th, Thy1, Tnc, Ube3a, Vamp2,
    Vamp3, Vamp4, Vamp8
    </td>
    </tr>
    <tr>
    <td style="text-align:left;">

    1.  Translation (<a href="GO:0006412" class="uri">GO:0006412</a>)
        </td>
        <td style="text-align:left;">
        Cpeb4, Eif5
        </td>
        <td style="text-align:left;">
        NA
        </td>
        </tr>
        <tr>
        <td style="text-align:left;">

        1.  Synapse Organization
            (<a href="GO:0050808" class="uri">GO:0050808</a>)
            </td>
            <td style="text-align:left;">
            Amigo2, Arc, Bdnf, Flrt3, Fzd5, Homer1, Lrrtm2, Npas4,
            Pcdh8, Slitrk5
            </td>
            <td style="text-align:left;">
            Ache, Bdnf, Cacna1a, Cacna1s, Camk1, Cdh1, Cdh2, Chrna1,
            Chrna7, Chrnb1, Chrnb2, Dlg4, Drd1, Efna5, Epha5, Erbb4,
            Fyn, Gabra1, Gabrb2, Gabrb3, Gap43, Grin1, Grin2a, Grm5,
            Homer1, Htr1a, Itga3, Itgam, Itpka, L1cam, Mapk14, Nrg1,
            Nrg2, Ntrk2, Ptn, Rab3a, Syn1, Tnc, Ube3a
            </td>
            </tr>
            <tr>
            <td style="text-align:left;">

            1.  Learning or Memory
                (<a href="GO:0007611" class="uri">GO:0007611</a>)
                </td>
                <td style="text-align:left;">
                Adrb1, Arc, Bdnf, Btg2, Egr1, Hmgcr, Jun, Npas4, Nptx2,
                Pak6, Plk2, Ptgs2, Sgk1, Srf, Syt4
                </td>
                <td style="text-align:left;">
                Adcy1, Adrb1, Adrb2, Bdnf, Cacna1c, Cacna1e, Calb1,
                Camk4, Chrna7, Chrnb2, Cnr1, Creb1, Drd1, Egr1, Gabra5,
                Gria1, Grin1, Grin2a, Grm4, Grm5, Grm7, Htr2a, Htr6,
                Htr7, Il1b, Itga3, Itga5, Itga8, Itgb1, Ncam1, Ngf,
                Ntrk2, Oprl1, Pla2g6, Plcb1, Prkar1b, Prkcz, Ptn, S100b,
                Snap25, Th
                </td>
                </tr>
                </tbody>
                </table>

                DEGs and Sanes and top 10 memory genes
                --------------------------------------

<!-- -->

    gene <- c("Camk2a" ,  "Gria2",  "Igf2" , "Nsf" , "Pick1",
                            "Prkcb" ,  "Prkci",  "Prkcz", "Wwc1" )
    gene <- as.data.frame(gene)
    gene$classicmemgene <- gene$gene

    GOtermsDEGs

    ## # A tibble: 4 x 2
    ##   GO                        genes                                          
    ##   <chr>                     <chr>                                          
    ## 1 1. Response to Stimulus … Abhd2, Adrb1, Ahr, Ankrd27, Apaf1, Arc, Arid5b…
    ## 2 2. Translation (GO:00064… Cpeb4, Eif5                                    
    ## 3 3. Synapse Organization … Amigo2, Arc, Bdnf, Flrt3, Fzd5, Homer1, Lrrtm2…
    ## 4 4. Learning or Memory (G… Adrb1, Arc, Bdnf, Btg2, Egr1, Hmgcr, Jun, Npas…

    candidategenes <- left_join(gene, GOterms) %>% arrange(GO)

    ## Joining, by = "gene"

    ## Warning: Column `gene` joining factor and character vector, coercing into
    ## character vector

    candidategenes

    ##      gene classicmemgene                                   GO
    ## 1  Camk2a         Camk2a 1. Response to Stimulus (GO:0050896)
    ## 2   Gria2          Gria2 1. Response to Stimulus (GO:0050896)
    ## 3    Igf2           Igf2 1. Response to Stimulus (GO:0050896)
    ## 4    Wwc1           Wwc1 1. Response to Stimulus (GO:0050896)
    ## 5   Pick1          Pick1 3. Synapse Organization (GO:0050808)
    ## 6    Igf2           Igf2   4. Learning or Memory (GO:0007611)
    ## 7   Prkcz          Prkcz   4. Learning or Memory (GO:0007611)
    ## 8     Nsf            Nsf                                 <NA>
    ## 9   Prkcb          Prkcb                                 <NA>
    ## 10  Prkci          Prkci                                 <NA>

PC1 correlations
----------------

    corrsPC1sig <- read_csv("../data/04_corrsPC1sig.csv")

    ## Parsed with column specification:
    ## cols(
    ##   gene = col_character(),
    ##   PC1 = col_double()
    ## )

    corrsPC1sig$gene

    ##  [1] "Abra"     "Acan"     "Adamts1"  "Aim1l"    "Amigo2"   "Arc"     
    ##  [7] "Areg"     "Arl4d"    "Arl5b"    "Armcx5"   "Atf3"     "Bdnf"    
    ## [13] "Btg2"     "Chst14"   "Col10a1"  "Cpeb4"    "Dusp8"    "Egr1"    
    ## [19] "Egr4"     "Errfi1"   "Fbxo33"   "Fosl2"    "Frmd6"    "Fzd5"    
    ## [25] "Gm10269"  "Hist1h1d" "Hist1h3i" "Homer1"   "Hoxc4"    "Hspb3"   
    ## [31] "Ighd"     "Kcnk10"   "Lonrf1"   "Mest"     "Naf1"     "Nexn"    
    ## [37] "Nfil3"    "Npas4"    "Nr4a3"    "Pcdh8"    "Peli1"    "Per1"    
    ## [43] "Plk2"     "Ppic"     "Ptgs2"    "Rasd1"    "Rgs2"     "Sgk1"    
    ## [49] "Slc16a1"  "Slc25a25" "Smad7"    "Spty2d1"  "Syt4"     "Tiparp"  
    ## [55] "Tnfrsf23" "Trib1"    "Ubc"      "Zfp804b"

    ## these 58 genes were used for a go analysis using shinygo http://bioinformatics.sdstate.edu/go/
    ## results are stored as 
      #data/05_GO_PC1enrichmentBP.csv
        #data/05_GO_PC1enrichmentCC.csv
        #data/05_GO_PC1enrichmentMF.csv

    BP <- read_csv("../data/05_GO_PC1enrichmentBP.csv", n_max = 5) %>% mutate(Domain = "BP") 

    ## Parsed with column specification:
    ## cols(
    ##   `Enrichment FDR` = col_double(),
    ##   `Genes in list` = col_double(),
    ##   `Total genes` = col_double(),
    ##   `Functional Category` = col_character(),
    ##   Genes = col_character()
    ## )

    CC <- read_csv("../data/05_GO_PC1enrichmentCC.csv", n_max = 5) %>% mutate(Domain = "CC") 

    ## Parsed with column specification:
    ## cols(
    ##   `Enrichment FDR` = col_double(),
    ##   `Genes in list` = col_double(),
    ##   `Total genes` = col_double(),
    ##   `Functional Category` = col_character(),
    ##   Genes = col_character()
    ## )

    MF <- read_csv("../data/05_GO_PC1enrichmentMF.csv", n_max = 5) %>% mutate(Domain = "MF") 

    ## Parsed with column specification:
    ## cols(
    ##   `Enrichment FDR` = col_double(),
    ##   `Genes in list` = col_double(),
    ##   `Total genes` = col_double(),
    ##   `Functional Category` = col_character(),
    ##   Genes = col_character()
    ## )

    GO_PC1 <- rbind(BP, CC, MF) %>% 
      select(Domain, `Functional Category`, `Total genes`, `Genes in list`, Genes) %>%
      rename(gene = Genes) %>%
      dplyr::mutate(gene = str_to_title(gene))

    # alphabetize
    GO_PC1$gene <- GO_PC1$gene %>% str_split(., ' ') %>% 
      lapply(., 'sort') %>%  
      lapply(., 'paste', collapse=' ') %>% 
      unlist(.)

    kable(GO_PC1) 

<table>
<thead>
<tr>
<th style="text-align:left;">
Domain
</th>
<th style="text-align:left;">
Functional Category
</th>
<th style="text-align:right;">
Total genes
</th>
<th style="text-align:right;">
Genes in list
</th>
<th style="text-align:left;">
gene
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
BP
</td>
<td style="text-align:left;">
Memory
</td>
<td style="text-align:right;">
145
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:left;">
Arc Bdnf Egr1 Kcnk10 Npas4 Plk2 Ptgs2 Sgk1 Syt4
</td>
</tr>
<tr>
<td style="text-align:left;">
BP
</td>
<td style="text-align:left;">
Learning or memory
</td>
<td style="text-align:right;">
286
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:left;">
Arc Bdnf Btg2 Egr1 Kcnk10 Npas4 Plk2 Ptgs2 Sgk1 Syt4
</td>
</tr>
<tr>
<td style="text-align:left;">
BP
</td>
<td style="text-align:left;">
Tissue development
</td>
<td style="text-align:right;">
1855
</td>
<td style="text-align:right;">
21
</td>
<td style="text-align:left;">
Acan Arc Areg Atf3 Bdnf Btg2 Col10a1 Egr1 Errfi1 Fosl2 Frmd6 Fzd5 Homer1
Hoxc4 Nr4a3 Pcdh8 Ptgs2 Rgs2 Slc25a25 Smad7 Tiparp
</td>
</tr>
<tr>
<td style="text-align:left;">
BP
</td>
<td style="text-align:left;">
Cognition
</td>
<td style="text-align:right;">
317
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:left;">
Arc Bdnf Btg2 Egr1 Kcnk10 Npas4 Plk2 Ptgs2 Sgk1 Syt4
</td>
</tr>
<tr>
<td style="text-align:left;">
BP
</td>
<td style="text-align:left;">
Behavior
</td>
<td style="text-align:right;">
693
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:left;">
Arc Bdnf Btg2 Egr1 Homer1 Kcnk10 Npas4 Nr4a3 Plk2 Ptgs2 Sgk1 Slc16a1
Syt4
</td>
</tr>
<tr>
<td style="text-align:left;">
CC
</td>
<td style="text-align:left;">
Neuron projection
</td>
<td style="text-align:right;">
1486
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:left;">
Acan Arc Bdnf Cpeb4 Fzd5 Homer1 Nexn Pcdh8 Plk2 Ptgs2 Rgs2 Sgk1 Syt4
</td>
</tr>
<tr>
<td style="text-align:left;">
CC
</td>
<td style="text-align:left;">
Cell junction
</td>
<td style="text-align:right;">
1095
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:left;">
Arc Cpeb4 Frmd6 Fzd5 Homer1 Nexn Pcdh8 Slc16a1 Smad7 Syt4
</td>
</tr>
<tr>
<td style="text-align:left;">
CC
</td>
<td style="text-align:left;">
Dendrite
</td>
<td style="text-align:right;">
736
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:left;">
Arc Bdnf Cpeb4 Fzd5 Homer1 Pcdh8 Plk2 Syt4
</td>
</tr>
<tr>
<td style="text-align:left;">
CC
</td>
<td style="text-align:left;">
Dendritic tree
</td>
<td style="text-align:right;">
738
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:left;">
Arc Bdnf Cpeb4 Fzd5 Homer1 Pcdh8 Plk2 Syt4
</td>
</tr>
<tr>
<td style="text-align:left;">
CC
</td>
<td style="text-align:left;">
Somatodendritic compartment
</td>
<td style="text-align:right;">
1035
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:left;">
Acan Arc Bdnf Cpeb4 Fzd5 Homer1 Pcdh8 Plk2 Syt4
</td>
</tr>
<tr>
<td style="text-align:left;">
MF
</td>
<td style="text-align:left;">
Regulatory region nucleic acid binding
</td>
<td style="text-align:right;">
949
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:left;">
Atf3 Egr1 Egr4 Fosl2 Hoxc4 Nfil3 Npas4 Nr4a3 Per1 Smad7 Tiparp
</td>
</tr>
<tr>
<td style="text-align:left;">
MF
</td>
<td style="text-align:left;">
Transcription regulatory region DNA binding
</td>
<td style="text-align:right;">
946
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:left;">
Atf3 Egr1 Egr4 Fosl2 Hoxc4 Nfil3 Npas4 Nr4a3 Per1 Smad7 Tiparp
</td>
</tr>
<tr>
<td style="text-align:left;">
MF
</td>
<td style="text-align:left;">
RNA polymerase II regulatory region sequence-specific DNA binding
</td>
<td style="text-align:right;">
782
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:left;">
Atf3 Egr1 Egr4 Fosl2 Hoxc4 Nfil3 Npas4 Nr4a3 Per1
</td>
</tr>
<tr>
<td style="text-align:left;">
MF
</td>
<td style="text-align:left;">
DNA-binding transcription factor activity, RNA polymerase II-specific
</td>
<td style="text-align:right;">
738
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:left;">
Atf3 Btg2 Egr1 Egr4 Fosl2 Hoxc4 Nfil3 Npas4 Nr4a3
</td>
</tr>
<tr>
<td style="text-align:left;">
MF
</td>
<td style="text-align:left;">
RNA polymerase II regulatory region DNA binding
</td>
<td style="text-align:right;">
788
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:left;">
Atf3 Egr1 Egr4 Fosl2 Hoxc4 Nfil3 Npas4 Nr4a3 Per1
</td>
</tr>
</tbody>
</table>

    write.csv(GO_DEGsSanesLichtman, "../data/05_GO_DEGsSanesLichtman.csv", row.names = F)
    write.csv(GO_PC1, "../data/05_GO_PC1.csv", row.names = F)
