---
title: "GO Analysis"
output: md_document
---

```{r setup, include=FALSE}
library(tidyverse)
library(knitr)
library(kableExtra)

knitr::opts_chunk$set(echo = TRUE)
```


## Hypothesised GO Terms

```{r}
GO_response <- read.table("../data/goterms/01_responsetostimulus.txt", sep = "\t", row.names = NULL,  fill=TRUE)
GO_response$GO <- "1. Response to Stimulus (GO:0050896)"

GO_translation <- read.table("../data/goterms/02_translation.txt", sep = "\t", row.names = NULL)
GO_translation$GO <- "2. Translation (GO:0006412)"

GO_synapse <- read.table("../data/goterms/03_synapseorganization.txt", sep = "\t", row.names = NULL)
GO_synapse$GO <- "3. Synapse Organization (GO:0050808)"

GO_learningormemory <- read.table("../data/goterms/04_learningormemory.txt", sep = "\t", row.names = NULL)
GO_learningormemory$GO <- "4. Learning or Memory (GO:0007611)"

GOterms <- rbind(GO_learningormemory, GO_response, GO_translation, GO_synapse)
GOterms <- GOterms %>%
  dplyr::mutate(gene = str_to_title(MGI.Gene.Marker.ID)) %>% 
  dplyr::select(gene, GO) %>% 
  dplyr::distinct(gene, GO) %>% 
 group_by(gene) 
GOterms

```

## Upregulated DG DEGS


```{r}
DGDEGs <- read_csv("../data/suppltable-4.csv") %>% 
  filter(tissue == "DG", comparison == "yoked vs. trained")
DGDEGs$gene

GOtermsDEGs <- inner_join(GOterms, DGDEGs) %>%
  arrange(GO,gene) %>%
  dplyr::select(gene, GO) %>% 
  group_by(GO) %>%
  summarize(genes = str_c(gene, collapse = ", "))
head(GOtermsDEGs)
```

## are genes in GO term significant?

```{r}

totalGOterm <- GOterms %>%
  group_by(GO) %>%
  summarise(totalgenes = n())

totalSignificantgenesperGO <- inner_join(GOterms, DGDEGs) %>%
  arrange(GO,gene) %>%
  dplyr::select(gene, GO) %>% 
  group_by(GO) %>%
  summarise(significantgenes = n())

GOsummary <- full_join(totalGOterm, totalSignificantgenesperGO) %>%
  mutate(percent = significantgenes / totalgenes * 100) %>%
  mutate(percent = round(percent,2))
GOsummary

```

## what about random GO terms

FROM http://geneontology.org/docs/ontology-documentation/ 
The larger processes, or ‘biological programs’ accomplished by multiple molecular activities. **Examples of broad biological process terms** are _DNA repair_ or _signal transduction_. Examples of more specific terms are _pyrimidine nucleobase biosynthetic process_ or _glucose transmembrane transport_. Note that a biological process is not equivalent to a pathway. At present, the GO does not try to represent the dynamics or dependencies that would be required to fully describe a pathway. Examples of broad biological process terms are DNA repair or signal transduction. Examples of more specific terms are pyrimidine nucleobase biosynthetic process or glucose transmembrane transport.

Found all those at http://www.informatics.jax.org/vocab/gene_ontology

_Note: "pyrimidine nucleobase biosynthetic process" was too small, so I went up to leveles and chose "heterocycle biosynthetic process" instead._

```{r}


GO_DNArepair <- read.table("../data/goterms/05_DNArepair.txt", sep = "\t", row.names = NULL,  fill=TRUE)
GO_DNArepair$GO <- "5. DNA repair (GO:0006281)" 

GO_signaltransduction <- read.table("../data/goterms/06_signaltransduction.txt", sep = "\t", row.names = NULL,  fill=TRUE)
GO_signaltransduction$GO <- "6. Signal transduction (GO:0007165)" 

GO_heterocyclebiosyntheticprocess <- read.table("../data/goterms/07_heterocyclebiosyntheticprocess.txt", sep = "\t", row.names = NULL,  fill=TRUE)
GO_heterocyclebiosyntheticprocess$GO <- "7. Heterocycle biosynthetic process (GO:0018130)" 


GO_glucosetransmembranetransport <- read.table("../data/goterms/08_glucosetransmembranetransport.txt", sep = "\t", row.names = NULL,  fill=TRUE)
GO_glucosetransmembranetransport$GO <- "8. Glucose transmembrane transport (GO:1904659)" 

GOterms2 <- rbind(GO_DNArepair, GO_signaltransduction, GO_heterocyclebiosyntheticprocess, GO_glucosetransmembranetransport)
GOterms2 <- GOterms2 %>%
  dplyr::mutate(gene = str_to_title(MGI.Gene.Marker.ID)) %>%
  dplyr::select(gene, GO) %>% 
  dplyr::distinct(gene, GO) %>% 
 group_by(gene) 
GOterms2



totalGOterm2 <- GOterms2 %>%
  group_by(GO) %>%
  summarise(totalgenes = n())

totalSignificantgenesperGO2 <- inner_join(GOterms2, DGDEGs) %>%
  arrange(GO,gene) %>%
  dplyr::select(gene, GO) %>% 
  group_by(GO) %>%
  summarise(significantgenes = n())

GOsummary2 <- full_join(totalGOterm2, totalSignificantgenesperGO2) %>%
  mutate(percent = significantgenes / totalgenes * 100) %>%
  mutate(percent = round(percent,2))
GOsummary2

rbind(GOsummary,GOsummary2)
```


## sanes

```{r}
sanesLichtman <- read_csv("../data/05_sanesLichtman.csv")  %>% 
  rename(gene = genes) %>%
  dplyr::mutate(gene = str_to_title(gene))
sanesLichtman

GOtermssanesLichtman <- inner_join(GOterms, sanesLichtman) %>%
  arrange(GO,gene) %>%
  dplyr::select(gene, GO) %>% 
 group_by(GO) %>%
 summarize(genes = str_c(gene, collapse = ", "))

```

## DEGs and Sanes


```{r}
GO_DEGsSanesLichtman <- full_join(GOtermsDEGs, GOtermssanesLichtman, by = "GO") %>%
  rename("Differentially expressed in DG" = genes.x,
         "Molecules associated with LTP"  = genes.y)
kable(GO_DEGsSanesLichtman)

```
## DEGs and Sanes and top 10 memory genes


```{r}
gene <- c("Camk2a" ,  "Gria2",  "Igf2" , "Nsf" , "Pick1",
                        "Prkcb" ,  "Prkci",  "Prkcz", "Wwc1" )
gene <- as.data.frame(gene)
gene$classicmemgene <- gene$gene

GOtermsDEGs
candidategenes <- left_join(gene, GOterms) %>% arrange(GO)
candidategenes
```

## PC1 correlations

```{r}
corrsPC1sig <- read_csv("../data/04_corrsPC1sig.csv")
corrsPC1sig$gene

## these 58 genes were used for a go analysis using shinygo http://bioinformatics.sdstate.edu/go/
## results are stored as 
  #data/05_GO_PC1enrichmentBP.csv
	#data/05_GO_PC1enrichmentCC.csv
	#data/05_GO_PC1enrichmentMF.csv

BP <- read_csv("../data/05_GO_PC1enrichmentBP.csv", n_max = 5) %>% mutate(Domain = "BP") 
CC <- read_csv("../data/05_GO_PC1enrichmentCC.csv", n_max = 5) %>% mutate(Domain = "CC") 
MF <- read_csv("../data/05_GO_PC1enrichmentMF.csv", n_max = 5) %>% mutate(Domain = "MF") 

GO_PC1 <- rbind(BP, CC, MF) %>% 
  select(Domain, `Functional Category`, `Total genes`, `Genes in list`, Genes) %>%
  rename(gene = Genes) %>%
  dplyr::mutate(gene = str_to_title(gene))

# alphabetize
GO_PC1$gene <- GO_PC1$gene %>% str_split(., ' ') %>% 
  lapply(., 'sort') %>%  
  lapply(., 'paste', collapse=' ') %>% 
  unlist(.)

kable(GO_PC1) 
```
```{r write}
write.csv(GO_DEGsSanesLichtman, "../data/05_GO_DEGsSanesLichtman.csv", row.names = F)
write.csv(GO_PC1, "../data/05_GO_PC1.csv", row.names = F)
```