Hypothesised GO Terms
---------------------

    GO_response <- read.table("../data/goterms/01_responsetostimulus.txt", sep = "\t", row.names = NULL,  fill=TRUE)
    GO_response$GO <- "1. Response to Stimulus (GO:0050896)"

    GO_translation <- read.table("../data/goterms/02_translation.txt", sep = "\t", row.names = NULL)
    GO_translation$GO <- "2. Translation (GO:0006412)"

    GO_synapse <- read.table("../data/goterms/03_synapseorganization.txt", sep = "\t", row.names = NULL)
    GO_synapse$GO <- "3. Synapse Organization (GO:0050808)"

    GO_learningormemory <- read.table("../data/goterms/04_learningormemory.txt", sep = "\t", row.names = NULL)
    GO_learningormemory$GO <- "4. Learning or Memory (GO:0007611)"

    GOterms <- rbind(GO_learningormemory, GO_response, GO_translation, GO_synapse)
    GOterms <- GOterms %>%
      dplyr::mutate(gene = str_to_title(MGI.Gene.Marker.ID)) %>% 
      dplyr::select(gene, GO) %>% 
      dplyr::distinct(gene, GO) %>% 
     group_by(gene) 
    GOterms

    ## # A tibble: 7,156 x 2
    ## # Groups:   gene [6,717]
    ##    gene  GO                                
    ##    <chr> <chr>                             
    ##  1 Aaas  4. Learning or Memory (GO:0007611)
    ##  2 Aal   4. Learning or Memory (GO:0007611)
    ##  3 Abca7 4. Learning or Memory (GO:0007611)
    ##  4 Abcc8 4. Learning or Memory (GO:0007611)
    ##  5 Abi2  4. Learning or Memory (GO:0007611)
    ##  6 Abl2  4. Learning or Memory (GO:0007611)
    ##  7 Adam2 4. Learning or Memory (GO:0007611)
    ##  8 Adcy1 4. Learning or Memory (GO:0007611)
    ##  9 Adcy3 4. Learning or Memory (GO:0007611)
    ## 10 Adcy8 4. Learning or Memory (GO:0007611)
    ## # … with 7,146 more rows

Upregulated DG DEGS
-------------------

    DGDEGs <- read_csv("../data/suppltable-4.csv") %>% 
      filter(tissue == "DG", comparison == "yoked vs. trained")

    ## Parsed with column specification:
    ## cols(
    ##   tissue = col_character(),
    ##   gene = col_character(),
    ##   lfc = col_double(),
    ##   padj = col_double(),
    ##   logpadj = col_double(),
    ##   comparison = col_character(),
    ##   direction = col_character()
    ## )

    DGDEGs$gene

    ##   [1] "Hs6st1"        "Pou3f3"        "Zdbf2"         "Fzd5"         
    ##   [5] "Scg2"          "Cul3"          "Irs1"          "Per2"         
    ##   [9] "Btg2"          "Rgs2"          "Ptgs2"         "Rgs4"         
    ##  [13] "Olfml2b"       "Eprs"          "Atf3"          "Fam107b"      
    ##  [17] "Arl5b"         "Otud1"         "Slc25a25"      "Sh2d3c"       
    ##  [21] "Nr4a2"         "Gad1"          "Ctnnd1"        "Gm13889"      
    ##  [25] "Kcna4"         "Bdnf"          "Thbs1"         "Pak6"         
    ##  [29] "Ccdc32"        "Lcmt2"         "Flrt3"         "Tspyl3"       
    ##  [33] "Ctcfl"         "Gmeb2"         "Gyg"           "Foxo1"        
    ##  [37] "Tsc22d2"       "Siah2"         "Tiparp"        "Fbxw7"        
    ##  [41] "Lmna"          "Ciart"         "Slc16a1"       "Prpf38b"      
    ##  [45] "Dnajb4"        "Dnaja1"        "Nr4a3"         "Jun"          
    ##  [49] "Plk3"          "Errfi1"        "Plch2"         "Cyp51"        
    ##  [53] "Cldn12"        "Fosl2"         "Tnip2"         "Rbm47"        
    ##  [57] "Rasl11b"       "Coq2"          "A830010M20Rik" "Mn1"          
    ##  [61] "Ankrd13a"      "Pxn"           "Ubc"           "Nptx2"        
    ##  [65] "Rasl11a"       "Uspl1"         "Hsph1"         "Peg10"        
    ##  [69] "Bmt2"          "Mest"          "Kdm7a"         "Neurod6"      
    ##  [73] "Nap1l5"        "Egr4"          "Cecr6"         "Slc2a3"       
    ##  [77] "Dusp16"        "Gpr19"         "Fosb"          "Erf"          
    ##  [81] "Gpi1"          "Ankrd27"       "Ppp1r15a"      "Spty2d1"      
    ##  [85] "Lysmd4"        "Sv2b"          "Acan"          "Abhd2"        
    ##  [89] "Il16"          "Fzd4"          "Zfp668"        "Dusp8"        
    ##  [93] "Irs2"          "Dusp4"         "Lonrf1"        "Klkb1"        
    ##  [97] "Ing2"          "Mfap3l"        "Naf1"          "Zfp869"       
    ## [101] "Jund"          "Slc5a5"        "Klf2"          "Scoc"         
    ## [105] "Dnajb1"        "Junb"          "Tmem170"       "Cpne7"        
    ## [109] "Mc1r"          "Irf2bp2"       "Senp8"         "C2cd4b"       
    ## [113] "1190002N15Rik" "Arpp21"        "Plagl1"        "Cited2"       
    ## [117] "Heca"          "Sgk1"          "Ranbp2"        "Sowahc"       
    ## [121] "Jmjd1c"        "Arid5b"        "Gnaz"          "Nuak1"        
    ## [125] "Apaf1"         "Dusp6"         "Kitl"          "Phlda1"       
    ## [129] "Dyrk2"         "Lemd3"         "BC048403"      "Srgap1"       
    ## [133] "Pgam2"         "Peli1"         "B3gnt2"        "Cpeb4"        
    ## [137] "Med7"          "Rasd1"         "Per1"          "Kdm6b"        
    ## [141] "Zfp207"        "Dusp14"        "Cwc25"         "Stac2"        
    ## [145] "Arl4d"         "Kcnj2"         "Sox9"          "Lrrc45"       
    ## [149] "Kcnf1"         "Odc1"          "Ahr"           "Arl4a"        
    ## [153] "Foxg1"         "Fbxo33"        "Frmd6"         "Dbpht2"       
    ## [157] "Fos"           "Jdp2"          "Ccnk"          "Eif5"         
    ## [161] "Klf6"          "Eef1e1"        "Nedd9"         "Gadd45g"      
    ## [165] "Nfil3"         "Homer1"        "Hmgcr"         "Plk2"         
    ## [169] "Snx18"         "Ankrd28"       "Fermt2"        "Stmn4"        
    ## [173] "Dpysl2"        "Nefm"          "Egr3"          "Pcdh8"        
    ## [177] "Slitrk5"       "March11"       "Ankrd33b"      "Trib1"        
    ## [181] "Myc"           "Slc45a4"       "Arc"           "Fam118a"      
    ## [185] "Amigo2"        "Nr4a1"         "Igf2bp2"       "Tra2b"        
    ## [189] "Arl13b"        "Zfp654"        "Cxadr"         "Adamts1"      
    ## [193] "Bach1"         "Rgmb"          "Hspa1a"        "Srf"          
    ## [197] "Rfx2"          "Tubb4a"        "Lbh"           "Syt4"         
    ## [201] "Egr1"          "Lrrtm2"        "Pde6a"         "Smad7"        
    ## [205] "Npas4"         "Nxf1"          "Prune2"        "Zfand5"       
    ## [209] "Cnnm1"         "Adrb1"         "Zbtb33"        "Zfp275"       
    ## [213] "Armcx5"        "Piga"

    GOtermsDEGs <- inner_join(GOterms, DGDEGs) %>%
      arrange(GO,gene) %>%
      dplyr::select(gene, GO) %>% 
      group_by(GO) %>%
      summarize(genes = str_c(gene, collapse = ", "))

    ## Joining, by = "gene"

    head(GOtermsDEGs)

    ## # A tibble: 4 x 2
    ##   GO                        genes                                          
    ##   <chr>                     <chr>                                          
    ## 1 1. Response to Stimulus … Abhd2, Adrb1, Ahr, Ankrd27, Apaf1, Arc, Arid5b…
    ## 2 2. Translation (GO:00064… Cpeb4, Eif5                                    
    ## 3 3. Synapse Organization … Amigo2, Arc, Bdnf, Flrt3, Fzd5, Homer1, Lrrtm2…
    ## 4 4. Learning or Memory (G… Adrb1, Arc, Bdnf, Btg2, Egr1, Hmgcr, Jun, Npas…

are genes in GO term significant? what about random GO terms and significantly enriched GO terms
------------------------------------------------------------------------------------------------

Random 1 from
<a href="http://geneontology.org/docs/ontology-documentation/" class="uri">http://geneontology.org/docs/ontology-documentation/</a>
The larger processes, or ‘biological programs’ accomplished by multiple
molecular activities. **Examples of broad biological process terms** are
*DNA repair* or *signal transduction*. Examples of more specific terms
are *pyrimidine nucleobase biosynthetic process* or *glucose
transmembrane transport*. Note that a biological process is not
equivalent to a pathway. At present, the GO does not try to represent
the dynamics or dependencies that would be required to fully describe a
pathway. Examples of broad biological process terms are DNA repair or
signal transduction. Examples of more specific terms are pyrimidine
nucleobase biosynthetic process or glucose transmembrane transport.
Found all those at
<a href="http://www.informatics.jax.org/vocab/gene_ontology" class="uri">http://www.informatics.jax.org/vocab/gene_ontology</a>

Random 2 from
<a href="https://www.biorxiv.org/content/10.1101/2020.01.26.920447v1.full.pdf" class="uri">https://www.biorxiv.org/content/10.1101/2020.01.26.920447v1.full.pdf</a>

*Note: “pyrimidine nucleobase biosynthetic process” was too small, so I
went up to leveles and chose “heterocycle biosynthetic process”
instead.*

Top 4 GO terms from
<a href="http://bioinformatics.sdstate.edu/go/" class="uri">http://bioinformatics.sdstate.edu/go/</a>
identified with 214 DG DEGs.

![](../figures/05_GO/Screen%20Shot%202020-01-30%20at%2011.19.35%20AM.png)

    # https://serialmentor.com/blog/2016/6/13/reading-and-combining-many-tidy-data-files-in-R

    data_path <- "../data/goterms/"   # path to the data
    files <- dir(data_path, pattern = "*.txt") # get file names

    filesdf <- as.data.frame(files)
    filesdf$source <- c(1:17)
    filesdf

    ##                                    files source
    ## 1              01_responsetostimulus.txt      1
    ## 2                     02_translation.txt      2
    ## 3             03_synapseorganization.txt      3
    ## 4                04_learningormemory.txt      4
    ## 5               05_negregcellprocess.txt      5
    ## 6                06_regcellmacbiosyn.txt      6
    ## 7                    07_regmacbiosyn.txt      7
    ## 8              08_cellresponchemstim.txt      8
    ## 9                       09_DNArepair.txt      9
    ## 10             10_signaltransduction.txt     10
    ## 11 11_heterocyclebiosyntheticprocess.txt     11
    ## 12  12_glucosetransmembranetransport.txt     12
    ## 13                 13_responsetocAMP.txt     13
    ## 14               14_inactivaitonMAPK.txt     14
    ## 15                 15_circadianrythm.txt     15
    ## 16               16_responmechanstim.txt     16
    ## 17              17_regtranslRNApoll2.txt     17

    data <- files %>%
      # read in all the files, appending the path before the filename
      map_df(~ read.table(file.path(data_path, .),  sep = "\t", 
                          row.names = NULL,  fill=TRUE), .id = "source") %>%
      mutate(source = as.numeric(source)) %>%
      select(source, MGI.Gene.Marker.ID) %>%
      left_join(filesdf, .)  %>% 
      mutate_at(.vars = "files",  .funs = gsub,
                pattern = ".txt", replacement = "") %>%
      dplyr::rename("GO" = "files",
             "gene" = "MGI.Gene.Marker.ID") %>%
      dplyr::select(GO, gene)

    ## Warning in scan(file = file, what = what, sep = sep, quote = quote, dec =
    ## dec, : EOF within quoted string

    ## Warning in scan(file = file, what = what, sep = sep, quote = quote, dec =
    ## dec, : EOF within quoted string

    ## Warning in scan(file = file, what = what, sep = sep, quote = quote, dec =
    ## dec, : EOF within quoted string

    ## Warning in bind_rows_(x, .id): Unequal factor levels: coercing to character

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): Unequal factor levels: coercing to character

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): Unequal factor levels: coercing to character

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): Unequal factor levels: coercing to character

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): Unequal factor levels: coercing to character

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): Unequal factor levels: coercing to character

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): Unequal factor levels: coercing to character

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): Unequal factor levels: coercing to character

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): Unequal factor levels: coercing to character

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): Unequal factor levels: coercing to character

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): Unequal factor levels: coercing to character

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Warning in bind_rows_(x, .id): binding character and factor vector,
    ## coercing into character vector

    ## Joining, by = "source"

    head(data)

    ##                      GO          gene
    ## 1 01_responsetostimulus 1500002F19Rik
    ## 2 01_responsetostimulus 1600014C10Rik
    ## 3 01_responsetostimulus 1700007K13Rik
    ## 4 01_responsetostimulus 1700007K13Rik
    ## 5 01_responsetostimulus 1700123M08Rik
    ## 6 01_responsetostimulus 1810014B01Rik

    GOtermsDEGs2 <- inner_join(data, DGDEGs) %>%
      arrange(GO,gene) %>%
      dplyr::select(gene, GO) %>% 
      dplyr::distinct(gene, GO) %>% 
      group_by(GO) %>%
      summarize(genes = str_c(gene, collapse = ", "))

    ## Joining, by = "gene"

    head(GOtermsDEGs2)

    ## # A tibble: 6 x 2
    ##   GO                 genes                                                 
    ##   <chr>              <chr>                                                 
    ## 1 01_responsetostim… Abhd2, Adrb1, Ahr, Ankrd27, Apaf1, Arc, Arid5b, Arl13…
    ## 2 02_translation     Cpeb4, Eif5                                           
    ## 3 03_synapseorganiz… Amigo2, Arc, Bdnf, Flrt3, Fzd5, Homer1, Lrrtm2, Npas4…
    ## 4 04_learningormemo… Adrb1, Arc, Bdnf, Btg2, Egr1, Hmgcr, Jun, Npas4, Nptx…
    ## 5 05_negregcellproc… Abhd2, Acan, Ahr, Amigo2, Ankrd13a, Ankrd27, Arc, Ari…
    ## 6 06_regcellmacbios… Ahr, Arid5b, Atf3, Bach1, Bdnf, Btg2, Ccnk, Ciart, Ci…

    totalGOterm3 <- data %>%
      group_by(GO) %>%
      summarise(totalgenes = n())

    totalSignificantgenesperGO3 <- inner_join(data, DGDEGs) %>%
      arrange(GO,gene) %>%
      dplyr::select(gene, GO) %>% 
      dplyr::distinct(gene, GO) %>% 
      group_by(GO) %>%
      summarise(significantgenes = n())

    ## Joining, by = "gene"

    totalSignificantgenesperGO3

    ## # A tibble: 17 x 2
    ##    GO                                significantgenes
    ##    <chr>                                        <int>
    ##  1 01_responsetostimulus                           79
    ##  2 02_translation                                   2
    ##  3 03_synapseorganization                          10
    ##  4 04_learningormemory                             15
    ##  5 05_negregcellprocess                            51
    ##  6 06_regcellmacbiosyn                             28
    ##  7 07_regmacbiosyn                                 28
    ##  8 08_cellresponchemstim                           59
    ##  9 09_DNArepair                                     3
    ## 10 10_signaltransduction                           65
    ## 11 11_heterocyclebiosyntheticprocess               29
    ## 12 12_glucosetransmembranetransport                 5
    ## 13 13_responsetocAMP                                5
    ## 14 14_inactivaitonMAPK                              4
    ## 15 15_circadianrythm                                9
    ## 16 16_responmechanstim                              7
    ## 17 17_regtranslRNApoll2                            17

    GOsummary3 <- full_join(totalGOterm3, totalSignificantgenesperGO3) %>%
      mutate(percent = significantgenes / totalgenes * 100) %>%
      mutate(percent = round(percent,2)) %>%
      mutate(category = ifelse(grepl("01|02|03|04", GO), "hypothesis",
                              ifelse(grepl("05|06|07|08", GO), "ShinyGO",
                                     ifelse(grepl("09|10|11|12", GO), "random", 
                                            ifelse(grepl("13|14|15|16|17", GO), "PANTHER", NA))))) %>%
      select(category, GO, totalgenes, significantgenes, percent)   %>%
      arrange(desc(percent)) %>%
      left_join(.,GOtermsDEGs2)

    ## Joining, by = "GO"

    ## Joining, by = "GO"

    GOsummary3

    ## # A tibble: 17 x 6
    ##    category  GO        totalgenes significantgenes percent genes           
    ##    <chr>     <chr>          <int>            <int>   <dbl> <chr>           
    ##  1 PANTHER   14_inact…         25                4  16     Dusp16, Dusp4, …
    ##  2 PANTHER   13_respo…         68                5   7.35  Ahr, Jun, Junb,…
    ##  3 PANTHER   16_respo…        211                7   3.32  Fos, Jun, Junb,…
    ##  4 random    12_gluco…        159                5   3.14  Irs1, Irs2, Myc…
    ##  5 hypothes… 04_learn…        501               15   2.99  Adrb1, Arc, Bdn…
    ##  6 PANTHER   15_circa…        377                9   2.39  Adrb1, Ahr, Cia…
    ##  7 ShinyGO   08_cellr…       5822               59   1.01  Abhd2, Ahr, Apa…
    ##  8 PANTHER   17_regtr…       1745               17   0.97  Ahr, Arid5b, At…
    ##  9 hypothes… 03_synap…       1129               10   0.89  Amigo2, Arc, Bd…
    ## 10 hypothes… 02_trans…        233                2   0.86  Cpeb4, Eif5     
    ## 11 ShinyGO   05_negre…       6630               51   0.77  Abhd2, Acan, Ah…
    ## 12 random    11_heter…       4108               29   0.71  Ahr, Arid5b, At…
    ## 13 ShinyGO   06_regce…       4496               28   0.62  Ahr, Arid5b, At…
    ## 14 ShinyGO   07_regma…       4663               28   0.6   Ahr, Arid5b, At…
    ## 15 random    10_signa…      13325               65   0.49  Abhd2, Adrb1, A…
    ## 16 hypothes… 01_respo…      26179               79   0.3   Abhd2, Adrb1, A…
    ## 17 random    09_DNAre…       1019                3   0.290 Bach1, Hspa1a, …

    write.csv(GOtermsDEGs2, "../data/05_GOtermsDEGs2.csv", row.names = F)
    write.csv(GOsummary3, "../data/05_GOsummary3.csv", row.names = F)

sanes
-----

    sanesLichtman <- read_csv("../data/05_sanesLichtman.csv")  %>% 
      dplyr::rename("gene" = "genes") %>%
      dplyr::mutate(gene = str_to_title(gene))

    ## Parsed with column specification:
    ## cols(
    ##   genes = col_character()
    ## )

    str(sanesLichtman)

    ## Classes 'spec_tbl_df', 'tbl_df', 'tbl' and 'data.frame': 237 obs. of  1 variable:
    ##  $ gene: chr  "Ache" "Adcy1" "Adra2a" "Adra2b" ...

    GOtermssanesLichtman <- left_join(sanesLichtman, data) %>%
      arrange(GO,gene) %>%
      dplyr::select(gene, GO) %>% 
       dplyr::distinct(gene, GO) %>% 
     group_by(GO) %>%
     summarize(genes = str_c(gene, collapse = ", "))

    ## Joining, by = "gene"

    GOtermssanesLichtman

    ## # A tibble: 15 x 2
    ##    GO                      genes                                           
    ##    <chr>                   <chr>                                           
    ##  1 01_responsetostimulus   Adcy1, Adra2a, Adra2b, Adra2c, Adrb1, Adrb2, Ad…
    ##  2 03_synapseorganization  Ache, Bdnf, Cacna1a, Cacna1s, Camk1, Cdh1, Cdh2…
    ##  3 04_learningormemory     Adcy1, Adrb1, Adrb2, Bdnf, Cacna1c, Cacna1e, Ca…
    ##  4 05_negregcellprocess    Adra2a, Adrb2, Bdnf, Creb1, Dlg4, Drd1, Edn1, E…
    ##  5 06_regcellmacbiosyn     Adcy1, Adrb2, Bdnf, Cacna1d, Camk1, Camk4, Capn…
    ##  6 07_regmacbiosyn         Adcy1, Adrb2, Bdnf, Cacna1d, Camk1, Camk4, Capn…
    ##  7 08_cellresponchemstim   Adra2a, Adrb2, Bdnf, Cacna1a, Cacna1d, Cacna1e,…
    ##  8 09_DNArepair            Prkcg                                           
    ##  9 10_signaltransduction   Adcy1, Adra2a, Adra2b, Adra2c, Adrb1, Adrb2, Ad…
    ## 10 11_heterocyclebiosynth… Adcy1, Adrb2, Bdnf, Cacna1d, Camk1, Camk4, Capn…
    ## 11 12_glucosetransmembran… Capn10, Drd1, Edn1, Erbb4, Il1b, Mapk14         
    ## 12 15_circadianrythm       Adcy1, Adrb1, Chrnb2, Creb1, Drd1, Egr1, Htr7, …
    ## 13 16_responmechanstim     Fyn, Grin2a, Grin2d, Htr2a, Htr7, Itga2, Mapk14…
    ## 14 17_regtranslRNApoll2    Adrb2, Camk1, Creb1, Edn1, Egr1, Egr2, Ube3a    
    ## 15 <NA>                    Calb2, Camk2, Capn11, Capn12, Capn13, Capn15, C…

    GOtermssanesLichtmansummary <- left_join(sanesLichtman, data) %>%
      arrange(GO,gene) %>%
      dplyr::select(gene, GO) %>% 
       dplyr::distinct(gene, GO) %>% 
     group_by(GO) %>%
     summarize(SanesLichtman = n())

    ## Joining, by = "gene"

    GOtermssanesLichtmansummary

    ## # A tibble: 15 x 2
    ##    GO                                SanesLichtman
    ##    <chr>                                     <int>
    ##  1 01_responsetostimulus                       150
    ##  2 03_synapseorganization                       39
    ##  3 04_learningormemory                          41
    ##  4 05_negregcellprocess                         28
    ##  5 06_regcellmacbiosyn                          23
    ##  6 07_regmacbiosyn                              23
    ##  7 08_cellresponchemstim                        93
    ##  8 09_DNArepair                                  1
    ##  9 10_signaltransduction                       154
    ## 10 11_heterocyclebiosyntheticprocess            17
    ## 11 12_glucosetransmembranetransport              6
    ## 12 15_circadianrythm                            13
    ## 13 16_responmechanstim                          12
    ## 14 17_regtranslRNApoll2                          7
    ## 15 <NA>                                         38

    full_join(GOsummary3, GOtermssanesLichtmansummary) %>% 
      mutate(SLpercent = SanesLichtman / totalgenes * 100)  %>% 
      mutate(SLpercent = round(SLpercent, 2))  %>% 
      select(category, GO, totalgenes, SanesLichtman, SLpercent, significantgenes, percent) %>% 
      filter(category != "ShinyGO")  %>%
      drop_na()

    ## Joining, by = "GO"

    ## # A tibble: 10 x 7
    ##    category GO    totalgenes SanesLichtman SLpercent significantgenes
    ##    <chr>    <chr>      <int>         <int>     <dbl>            <int>
    ##  1 PANTHER  16_r…        211            12     5.69                 7
    ##  2 random   12_g…        159             6     3.77                 5
    ##  3 hypothe… 04_l…        501            41     8.18                15
    ##  4 PANTHER  15_c…        377            13     3.45                 9
    ##  5 PANTHER  17_r…       1745             7     0.4                 17
    ##  6 hypothe… 03_s…       1129            39     3.45                10
    ##  7 random   11_h…       4108            17     0.41                29
    ##  8 random   10_s…      13325           154     1.16                65
    ##  9 hypothe… 01_r…      26179           150     0.570               79
    ## 10 random   09_D…       1019             1     0.1                  3
    ## # … with 1 more variable: percent <dbl>

DEGs and Sanes and top 10 memory genes
--------------------------------------

    candidategenes <- c("Camk2a" ,  "Gria2",  "Igf2" , "Nsf" , "Pick1",
                            "Prkcb" ,  "Prkci",  "Prkcz", "Wwc1" ,
              "Fmr1", "Mtor")
    candidategenes <- as.data.frame(candidategenes) %>%
      dplyr::rename("gene" = "candidategenes")

    candidatesGO <- left_join(candidategenes, data, by = "gene")  %>%
      arrange(GO,gene) %>%
      dplyr::select(gene, GO) %>% 
      dplyr::distinct(gene, GO) %>% 
     group_by(GO) %>%
     summarize(genes = str_c(gene, collapse = ", "))

    ## Warning: Column `gene` joining factor and character vector, coercing into
    ## character vector

    candidatesGO

    ## # A tibble: 14 x 2
    ##    GO                           genes                                      
    ##    <chr>                        <chr>                                      
    ##  1 01_responsetostimulus        Camk2a, Fmr1, Gria2, Igf2, Mtor, Wwc1      
    ##  2 02_translation               Fmr1, Mtor                                 
    ##  3 03_synapseorganization       Fmr1, Pick1                                
    ##  4 04_learningormemory          Igf2, Mtor, Prkcz                          
    ##  5 05_negregcellprocess         Fmr1, Prkcb, Prkci, Prkcz                  
    ##  6 06_regcellmacbiosyn          Camk2a, Fmr1, Prkcb, Prkci, Prkcz          
    ##  7 07_regmacbiosyn              Camk2a, Fmr1, Prkcb, Prkci, Prkcz          
    ##  8 08_cellresponchemstim        Camk2a, Fmr1, Igf2, Mtor, Pick1, Prkcb, Pr…
    ##  9 10_signaltransduction        Camk2a, Fmr1, Gria2, Igf2, Mtor, Pick1     
    ## 10 11_heterocyclebiosyntheticp… Camk2a, Wwc1                               
    ## 11 12_glucosetransmembranetran… Prkcb, Prkci                               
    ## 12 15_circadianrythm            Mtor                                       
    ## 13 17_regtranslRNApoll2         Wwc1                                       
    ## 14 <NA>                         Nsf

    AllGOgenelists <- full_join(candidatesGO, GOtermssanesLichtman, by = "GO") %>% 
      full_join(., GOtermsDEGs2, by = "GO") %>% 
      dplyr::rename("candidates" = "genes.x", 
             "SanesLichtman" ="genes.y", 
             "DG DEGs"  = "genes") %>% 
      arrange(GO) %>%
      filter(GO != "NA") %>%
      mutate(category = ifelse(grepl("01|02|03|04", GO), "hypothesis",
                              ifelse(grepl("05|06|07|08", GO), "ShinyGO",
                                     ifelse(grepl("09|10|11|12", GO), "random", 
                                     ifelse(grepl("13|14|15|16|17", GO), "PANTHER", NA))))) %>%
      select(category, everything())
      
    AllGOgenelists 

    ## # A tibble: 17 x 5
    ##    category  GO        candidates      SanesLichtman       `DG DEGs`       
    ##    <chr>     <chr>     <chr>           <chr>               <chr>           
    ##  1 hypothes… 01_respo… Camk2a, Fmr1, … Adcy1, Adra2a, Adr… Abhd2, Adrb1, A…
    ##  2 hypothes… 02_trans… Fmr1, Mtor      <NA>                Cpeb4, Eif5     
    ##  3 hypothes… 03_synap… Fmr1, Pick1     Ache, Bdnf, Cacna1… Amigo2, Arc, Bd…
    ##  4 hypothes… 04_learn… Igf2, Mtor, Pr… Adcy1, Adrb1, Adrb… Adrb1, Arc, Bdn…
    ##  5 ShinyGO   05_negre… Fmr1, Prkcb, P… Adra2a, Adrb2, Bdn… Abhd2, Acan, Ah…
    ##  6 ShinyGO   06_regce… Camk2a, Fmr1, … Adcy1, Adrb2, Bdnf… Ahr, Arid5b, At…
    ##  7 ShinyGO   07_regma… Camk2a, Fmr1, … Adcy1, Adrb2, Bdnf… Ahr, Arid5b, At…
    ##  8 ShinyGO   08_cellr… Camk2a, Fmr1, … Adra2a, Adrb2, Bdn… Abhd2, Ahr, Apa…
    ##  9 random    09_DNAre… <NA>            Prkcg               Bach1, Hspa1a, …
    ## 10 random    10_signa… Camk2a, Fmr1, … Adcy1, Adra2a, Adr… Abhd2, Adrb1, A…
    ## 11 random    11_heter… Camk2a, Wwc1    Adcy1, Adrb2, Bdnf… Ahr, Arid5b, At…
    ## 12 random    12_gluco… Prkcb, Prkci    Capn10, Drd1, Edn1… Irs1, Irs2, Myc…
    ## 13 PANTHER   13_respo… <NA>            <NA>                Ahr, Jun, Junb,…
    ## 14 PANTHER   14_inact… <NA>            <NA>                Dusp16, Dusp4, …
    ## 15 PANTHER   15_circa… Mtor            Adcy1, Adrb1, Chrn… Adrb1, Ahr, Cia…
    ## 16 PANTHER   16_respo… <NA>            Fyn, Grin2a, Grin2… Fos, Jun, Junb,…
    ## 17 PANTHER   17_regtr… Wwc1            Adrb2, Camk1, Creb… Ahr, Arid5b, At…

    write_csv(AllGOgenelists, "../data/05_GO_AllGOgenelists.csv")

PC1 correlations
----------------

    corrsPC1sig <- read_csv("../data/04_corrsPC1sig.csv")

    ## Parsed with column specification:
    ## cols(
    ##   gene = col_character(),
    ##   PC1 = col_double()
    ## )

    corrsPC1sig$gene

    ##  [1] "Abra"     "Acan"     "Adamts1"  "Aim1l"    "Amigo2"   "Arc"     
    ##  [7] "Areg"     "Arl4d"    "Arl5b"    "Armcx5"   "Atf3"     "Bdnf"    
    ## [13] "Btg2"     "Chst14"   "Col10a1"  "Cpeb4"    "Dusp8"    "Egr1"    
    ## [19] "Egr4"     "Errfi1"   "Fbxo33"   "Fosl2"    "Frmd6"    "Fzd5"    
    ## [25] "Gm10269"  "Hist1h1d" "Hist1h3i" "Homer1"   "Hoxc4"    "Hspb3"   
    ## [31] "Ighd"     "Kcnk10"   "Lonrf1"   "Mest"     "Naf1"     "Nexn"    
    ## [37] "Nfil3"    "Npas4"    "Nr4a3"    "Pcdh8"    "Peli1"    "Per1"    
    ## [43] "Plk2"     "Ppic"     "Ptgs2"    "Rasd1"    "Rgs2"     "Sgk1"    
    ## [49] "Slc16a1"  "Slc25a25" "Smad7"    "Spty2d1"  "Syt4"     "Tiparp"  
    ## [55] "Tnfrsf23" "Trib1"    "Ubc"      "Zfp804b"

    ## these 58 genes were used for a go analysis using shinygo http://bioinformatics.sdstate.edu/go/
    ## results are stored as 
      #data/05_GO_PC1enrichmentBP.csv
        #data/05_GO_PC1enrichmentCC.csv
        #data/05_GO_PC1enrichmentMF.csv

    BP <- read_csv("../data/05_GO_PC1enrichmentBP.csv", n_max = 5) %>% mutate(Domain = "BP") 

    ## Parsed with column specification:
    ## cols(
    ##   `Enrichment FDR` = col_double(),
    ##   `Genes in list` = col_double(),
    ##   `Total genes` = col_double(),
    ##   `Functional Category` = col_character(),
    ##   Genes = col_character()
    ## )

    CC <- read_csv("../data/05_GO_PC1enrichmentCC.csv", n_max = 5) %>% mutate(Domain = "CC") 

    ## Parsed with column specification:
    ## cols(
    ##   `Enrichment FDR` = col_double(),
    ##   `Genes in list` = col_double(),
    ##   `Total genes` = col_double(),
    ##   `Functional Category` = col_character(),
    ##   Genes = col_character()
    ## )

    MF <- read_csv("../data/05_GO_PC1enrichmentMF.csv", n_max = 5) %>% mutate(Domain = "MF") 

    ## Parsed with column specification:
    ## cols(
    ##   `Enrichment FDR` = col_double(),
    ##   `Genes in list` = col_double(),
    ##   `Total genes` = col_double(),
    ##   `Functional Category` = col_character(),
    ##   Genes = col_character()
    ## )

    GO_PC1 <- rbind(BP, CC, MF) %>% 
      select(Domain, `Functional Category`, `Total genes`, `Genes in list`, Genes) %>%
      rename(gene = Genes) %>%
      dplyr::mutate(gene = str_to_title(gene))

    # alphabetize
    GO_PC1$gene <- GO_PC1$gene %>% str_split(., ' ') %>% 
      lapply(., 'sort') %>%  
      lapply(., 'paste', collapse=' ') %>% 
      unlist(.)

    kable(GO_PC1) 

<table>
<thead>
<tr>
<th style="text-align:left;">
Domain
</th>
<th style="text-align:left;">
Functional Category
</th>
<th style="text-align:right;">
Total genes
</th>
<th style="text-align:right;">
Genes in list
</th>
<th style="text-align:left;">
gene
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
BP
</td>
<td style="text-align:left;">
Memory
</td>
<td style="text-align:right;">
145
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:left;">
Arc Bdnf Egr1 Kcnk10 Npas4 Plk2 Ptgs2 Sgk1 Syt4
</td>
</tr>
<tr>
<td style="text-align:left;">
BP
</td>
<td style="text-align:left;">
Learning or memory
</td>
<td style="text-align:right;">
286
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:left;">
Arc Bdnf Btg2 Egr1 Kcnk10 Npas4 Plk2 Ptgs2 Sgk1 Syt4
</td>
</tr>
<tr>
<td style="text-align:left;">
BP
</td>
<td style="text-align:left;">
Tissue development
</td>
<td style="text-align:right;">
1855
</td>
<td style="text-align:right;">
21
</td>
<td style="text-align:left;">
Acan Arc Areg Atf3 Bdnf Btg2 Col10a1 Egr1 Errfi1 Fosl2 Frmd6 Fzd5 Homer1
Hoxc4 Nr4a3 Pcdh8 Ptgs2 Rgs2 Slc25a25 Smad7 Tiparp
</td>
</tr>
<tr>
<td style="text-align:left;">
BP
</td>
<td style="text-align:left;">
Cognition
</td>
<td style="text-align:right;">
317
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:left;">
Arc Bdnf Btg2 Egr1 Kcnk10 Npas4 Plk2 Ptgs2 Sgk1 Syt4
</td>
</tr>
<tr>
<td style="text-align:left;">
BP
</td>
<td style="text-align:left;">
Behavior
</td>
<td style="text-align:right;">
693
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:left;">
Arc Bdnf Btg2 Egr1 Homer1 Kcnk10 Npas4 Nr4a3 Plk2 Ptgs2 Sgk1 Slc16a1
Syt4
</td>
</tr>
<tr>
<td style="text-align:left;">
CC
</td>
<td style="text-align:left;">
Neuron projection
</td>
<td style="text-align:right;">
1486
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:left;">
Acan Arc Bdnf Cpeb4 Fzd5 Homer1 Nexn Pcdh8 Plk2 Ptgs2 Rgs2 Sgk1 Syt4
</td>
</tr>
<tr>
<td style="text-align:left;">
CC
</td>
<td style="text-align:left;">
Cell junction
</td>
<td style="text-align:right;">
1095
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:left;">
Arc Cpeb4 Frmd6 Fzd5 Homer1 Nexn Pcdh8 Slc16a1 Smad7 Syt4
</td>
</tr>
<tr>
<td style="text-align:left;">
CC
</td>
<td style="text-align:left;">
Dendrite
</td>
<td style="text-align:right;">
736
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:left;">
Arc Bdnf Cpeb4 Fzd5 Homer1 Pcdh8 Plk2 Syt4
</td>
</tr>
<tr>
<td style="text-align:left;">
CC
</td>
<td style="text-align:left;">
Dendritic tree
</td>
<td style="text-align:right;">
738
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:left;">
Arc Bdnf Cpeb4 Fzd5 Homer1 Pcdh8 Plk2 Syt4
</td>
</tr>
<tr>
<td style="text-align:left;">
CC
</td>
<td style="text-align:left;">
Somatodendritic compartment
</td>
<td style="text-align:right;">
1035
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:left;">
Acan Arc Bdnf Cpeb4 Fzd5 Homer1 Pcdh8 Plk2 Syt4
</td>
</tr>
<tr>
<td style="text-align:left;">
MF
</td>
<td style="text-align:left;">
Regulatory region nucleic acid binding
</td>
<td style="text-align:right;">
949
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:left;">
Atf3 Egr1 Egr4 Fosl2 Hoxc4 Nfil3 Npas4 Nr4a3 Per1 Smad7 Tiparp
</td>
</tr>
<tr>
<td style="text-align:left;">
MF
</td>
<td style="text-align:left;">
Transcription regulatory region DNA binding
</td>
<td style="text-align:right;">
946
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:left;">
Atf3 Egr1 Egr4 Fosl2 Hoxc4 Nfil3 Npas4 Nr4a3 Per1 Smad7 Tiparp
</td>
</tr>
<tr>
<td style="text-align:left;">
MF
</td>
<td style="text-align:left;">
RNA polymerase II regulatory region sequence-specific DNA binding
</td>
<td style="text-align:right;">
782
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:left;">
Atf3 Egr1 Egr4 Fosl2 Hoxc4 Nfil3 Npas4 Nr4a3 Per1
</td>
</tr>
<tr>
<td style="text-align:left;">
MF
</td>
<td style="text-align:left;">
DNA-binding transcription factor activity, RNA polymerase II-specific
</td>
<td style="text-align:right;">
738
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:left;">
Atf3 Btg2 Egr1 Egr4 Fosl2 Hoxc4 Nfil3 Npas4 Nr4a3
</td>
</tr>
<tr>
<td style="text-align:left;">
MF
</td>
<td style="text-align:left;">
RNA polymerase II regulatory region DNA binding
</td>
<td style="text-align:right;">
788
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:left;">
Atf3 Egr1 Egr4 Fosl2 Hoxc4 Nfil3 Npas4 Nr4a3 Per1
</td>
</tr>
</tbody>
</table>

for GOrilla
-----------

    backgroundgeneset <- read_csv("../data/00_geneids.csv") %>% 
      select(gene) %>%
      distinct(gene) %>%
      arrange(gene)

    ## Parsed with column specification:
    ## cols(
    ##   id = col_character(),
    ##   ENSMUST = col_character(),
    ##   ENSMUSG = col_character(),
    ##   OTTMUSG = col_character(),
    ##   OTTMUST = col_character(),
    ##   transcript = col_character(),
    ##   gene = col_character(),
    ##   length = col_double(),
    ##   structure1 = col_character(),
    ##   structure2 = col_character(),
    ##   structure3 = col_character(),
    ##   transcript_lenght = col_character()
    ## )

    head(backgroundgeneset,20)

    ## # A tibble: 20 x 1
    ##    gene         
    ##    <chr>        
    ##  1 0610007P14Rik
    ##  2 0610009B22Rik
    ##  3 0610009L18Rik
    ##  4 0610009O20Rik
    ##  5 0610010F05Rik
    ##  6 0610010K14Rik
    ##  7 0610012G03Rik
    ##  8 0610030E20Rik
    ##  9 0610037L13Rik
    ## 10 0610040J01Rik
    ## 11 1110002E22Rik
    ## 12 1110004E09Rik
    ## 13 1110004F10Rik
    ## 14 1110008F13Rik
    ## 15 1110008L16Rik
    ## 16 1110008P14Rik
    ## 17 1110012L19Rik
    ## 18 1110017D15Rik
    ## 19 1110025L11Rik
    ## 20 1110032A03Rik

    write.csv(backgroundgeneset, "../data/05_GO_backgroundgeneset.csv", row.names = F)

    write.csv(GO_PC1, "../data/05_GO_PC1.csv", row.names = F)
