---
title: "GO Analysis"
output: md_document
---

```{r setup, include=FALSE}
library(tidyverse)
library(knitr)
library(kableExtra)
library(stringr)

knitr::opts_chunk$set(echo = TRUE, message = F, error = F, warning = F)
```


## Hypothesised and random GO Terms


Do DEGs fall into one of four hypothesized GO catagoreis? Do they do this more often that 4 randomly selected GO terms that are similar in size to the hypothesized?

GO terms and genes collected from http://www.informatics.jax.org/vocab/gene_ontology

```{r readfiles}
# list of random and hypothesized GO terms and associated gens 
# https://serialmentor.com/blog/2016/6/13/reading-and-combining-many-tidy-data-files-in-R

data_path <- "../data/goterms/"   # path to the data
files <- dir(data_path, pattern = "*.txt") # get file names

filesdf <- as.data.frame(files)
filesdf$source <- c(1:8)
filesdf

GOterms <- files %>%
  # read in all the files, appending the path before the filename
  map_df(~ read.table(file.path(data_path, .),  sep = "\t", 
                      row.names = NULL,  fill=TRUE), .id = "source") %>%
  mutate(source = as.numeric(source)) %>%
  select(source, MGI.Gene.Marker.ID) %>%
  left_join(filesdf, .)  %>% 
  mutate_at(.vars = "files",  .funs = gsub,
            pattern = ".txt", replacement = "") %>%
  dplyr::rename("GO" = "files",
         "gene" = "MGI.Gene.Marker.ID") %>%
  dplyr::select(GO, gene)  %>%
  dplyr::distinct(GO, gene) 
head(GOterms)

GOtermCounts <- GOterms %>%
  group_by(GO) %>%
  summarize("total" = n())
head(GOtermCounts)



# DEGs
DGDEGs <- read_csv("../data/suppltable-4.csv") %>% 
  filter(tissue == "DG", comparison == "yoked vs. trained") %>%
  select(gene) 
head(DGDEGs)

# candidate genes
candidategenes <- c("Camk2a" ,  "Gria2",  "Igf2" , "Nsf" , "Pick1",
                        "Prkcb" ,  "Prkci",  "Prkcz", "Wwc1" , "Fmr1", "Mtor")
candidategenes <- as.data.frame(candidategenes) %>%
  dplyr::rename("gene" = "candidategenes")
head(candidategenes)

# Sanes & Lichtman genes
sanesLichtman <- read_csv("../data/05_sanesLichtman.csv")  %>% 
  dplyr::rename("gene" = "genes") %>%
  dplyr::mutate(gene = str_to_title(gene))  
head(sanesLichtman)

joinGOnGenes <- function(whichgenes){
  GOtermsgenes <- inner_join(GOterms, whichgenes) %>%
    arrange(GO,gene) %>%
    dplyr::select(gene, GO) %>% 
    group_by(GO) %>%
    summarize(genes = str_c(gene, collapse = ", ")) 
  return(GOtermsgenes)
}

GO_DEGs <- joinGOnGenes(DGDEGs)
GO_candidates <- joinGOnGenes(candidategenes)
GO_sanesLichtman <- joinGOnGenes(sanesLichtman)

table2 <- left_join(GO_DEGs, GO_candidates, by = "GO") %>%
  left_join(., GO_sanesLichtman, by = "GO") %>% 
  rename("DG_DEGs" = "genes.x",
        "candidates"  = "genes.y", 
        "Sanes_Lichtman" = "genes") 
table2$GO <- factor(table2$GO)
table2 <- table2 %>%
  mutate(GO = fct_recode(GO,
                         "Response to stimulus (GO:0050896)" = "01_responsetostimulus",
                          "Translation (GO:0006412)"   =   "02_translation" ,
                    "Synapse organization (GO:0050808)" =    "03_synapseorganization" ,
                        "Learning or memory (GO:0007611)" =   "04_learningormemory" ))  %>%
  filter(!GO %in% c("05_random_436_GO0050673", "06_random_1513_GO0022414",
                  "07_random_283_GO0048511" ,  "08_random_2720_GO0051704" ) ) %>%
  rename("GO terms" = "GO",
         "DG DEGs" = "DG_DEGs",
         "Candidate genes" = "candidates",
         "LTP genes" = "Sanes_Lichtman")
table2

DGDEGs$candidategene <- DGDEGs$gene

Allgogenes <- full_join(GOterms, DGDEGs, by = "gene") %>%
    arrange(GO, candidategene,gene) %>%
    dplyr::select( GO, gene)  %>%
  mutate(category = ifelse(grepl("random", GO), "random", "hypothesis")) %>%
  drop_na() %>%
  group_by(GO) %>%
    summarize(total = n()) 
Allgogenes

DEGGOgenes <- full_join(GOterms, DGDEGs, by = "gene") %>%
    arrange(GO, candidategene, gene) %>%
    dplyr::select( GO, gene, candidategene)  %>%
  mutate(category = ifelse(grepl("random", GO), "random", "hypothesis")) %>%
  drop_na() %>%
  group_by(GO) %>%
    summarize(DEGs = n(),
              genes = str_c(gene, collapse = ", ")) 
DEGGOgenes

propALLgoDEGs <- inner_join(Allgogenes, DEGGOgenes) %>%  
  select(-genes) %>%  
   mutate(prop = DEGs / total * 100)
propALLgoDEGs


ALLgoDEGs <- inner_join(Allgogenes, DEGGOgenes) %>%  
  mutate(category = ifelse(grepl("random", GO), "random", "hypothesis")) %>%
   group_by(category) %>%
    summarize(totals = sum(total),
              DEGss = sum(DEGs)) %>%
  mutate(prop = DEGss / totals * 100)
ALLgoDEGs

res <- prop.test(x = c(106, 62), n = c(7156, 4373), alternative = "greater")
res

res <- prop.test(x = c(15, 62), n = c(292, 4373), alternative = "greater") #LM
# Printing the results
res 

res <- prop.test(x = c(10, 62), n = c(465, 4373), alternative = "greater") #LM
# Printing the results
res 


```




## PC1 correlations

```{r}
## these 58 genes that are differentially expressed in DG and correlated with PC1 were use to query GO http://bioinformatics.sdstate.edu/go/
## results are stored as 
  #data/05_GO_PC1enrichmentBP.csv
	#data/05_GO_PC1enrichmentCC.csv
	#data/05_GO_PC1enrichmentMF.csv

BP <- read_csv("../data/05_GO_PC1enrichmentBP.csv", n_max = 5) %>% mutate(Domain = "BP") 
CC <- read_csv("../data/05_GO_PC1enrichmentCC.csv", n_max = 5) %>% mutate(Domain = "CC") 
MF <- read_csv("../data/05_GO_PC1enrichmentMF.csv", n_max = 5) %>% mutate(Domain = "MF") 

table3 <- rbind(BP, CC, MF) %>% 
  select(Domain, `Functional Category`,  Genes) %>%
  rename(gene = Genes) %>%
  dplyr::mutate(gene = str_to_title(gene))

# alphabetize
table3$gene <- table3$gene %>% str_split(., ' ') %>% 
  lapply(., 'sort') %>%  
  lapply(., 'paste', collapse=' ') %>% 
  unlist(.)

head(table3) 
```


```{r write}
write.csv(table2, "../data/table-2.csv", row.names = F)
write.csv(table3, "../data/table-3.csv", row.names = F)
```